--- fltk-1.1.7/fltk-config.in.config	2005-12-07 14:13:12.000000000 -0600
+++ fltk-1.1.7/fltk-config.in	2006-12-11 23:33:40.000000000 -0600
@@ -52,12 +52,6 @@
 # post-process command (only needed for MacOS)
 POSTBUILD="@POSTBUILD@"
 
-# flags for C++ compiler:
-CFLAGS="@CFLAGS@"
-CXXFLAGS="@CXXFLAGS@"
-LDFLAGS="@LDFLAGS@"
-LDLIBS="@LIBS@"
-
 # Check for local invocation, and update paths accordingly...
 if test -f "$selfdir/FL/Fl_Window.H"; then
 	bindir="$selfdir/fluid"
