diff -Naupr xmame-0.88-cvs.orig/src/unix/video-drivers/gltool.c xmame-0.88-cvs/src/unix/video-drivers/gltool.c
--- xmame-0.88-cvs.orig/src/unix/video-drivers/gltool.c	2004-10-19 07:34:21.000000000 +0200
+++ xmame-0.88-cvs/src/unix/video-drivers/gltool.c	2004-11-08 09:23:43.574152808 +0100
@@ -26,7 +26,7 @@
 	#include <string.h>
 	#include "glxtool.h"
 
-	#define GLXLIB_NAME "libglut.so"
+	#define GLXLIB_NAME "libglut.so.1"
 #endif
 
 #ifdef _WIN32_
diff -Naupr xmame-0.88-cvs.orig/src/unix/video-drivers/xgl.c xmame-0.88-cvs/src/unix/video-drivers/xgl.c
--- xmame-0.88-cvs.orig/src/unix/video-drivers/xgl.c	2004-10-29 16:36:27.000000000 +0200
+++ xmame-0.88-cvs/src/unix/video-drivers/xgl.c	2004-11-08 09:23:07.692607632 +0100
@@ -80,8 +80,8 @@ struct rc_option xgl_opts[] = {
 
 int xgl_init(void)
 {
-  const char *libGLName = "libGL.so";
-  const char *libGLUName = "libGLU.so";
+  const char *libGLName = "libGL.so.1";
+  const char *libGLUName = "libGLU.so.1";
   
   if (!loadGLLibrary(libGLName, libGLUName))
   {
