--- dhcp-3.0/Makefile.sopwith	Thu Dec 20 19:28:07 2001
+++ dhcp-3.0/Makefile	Thu Dec 20 19:28:30 2001
@@ -43,7 +43,7 @@
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make all); \
+	   (cd work.$$sysname; $(MAKE) all); \
 	fi
 
 install:
@@ -51,7 +51,7 @@
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make install); \
+	   (cd work.$$sysname; $(MAKE) install); \
 	fi
 
 depend:
@@ -59,7 +59,7 @@
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make depend); \
+	   (cd work.$$sysname; $(MAKE) depend); \
 	fi
 
 clean:
@@ -91,6 +91,6 @@
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make links); \
+	   (cd work.$$sysname; $(MAKE) links); \
 	fi
 
