--- dhcp-3.0pl1/client/scripts/linux.sopwith	2002-08-13 07:16:38.000000000 -0400
+++ dhcp-3.0pl1/client/scripts/linux	2002-08-13 07:27:28.000000000 -0400
@@ -126,16 +126,10 @@
   
 if [ x$reason = xBOUND ] || [ x$reason = xRENEW ] || \
    [ x$reason = xREBIND ] || [ x$reason = xREBOOT ]; then
-  current_hostname=`hostname`
-
-  if need_hostname; then
-    if [ x$current_hostname = x ] || \
-       [ x$current_hostname = x$old_host_name ]; then
-      if [ x$current_hostname = x ] || \
-         [ x$new_host_name != x$old_host_name ]; then
-        hostname $new_host_name
-      fi
-    fi
+  if [ -n "$new_host_name" ]; then
+	if need_hostname || [ "`hostname`" != "$new_host_name" ]; then
+	        hostname $new_host_name
+	fi
   fi
     
   if [ x$old_ip_address != x ] && [ x$alias_ip_address != x ] && \
--- dhcp-3.0pl1/client/scripts/linux.sopwith	2002-08-15 06:41:50.000000000 -0400
+++ dhcp-3.0pl1/client/scripts/linux	2002-08-15 06:42:37.000000000 -0400
@@ -126,6 +126,11 @@
   
 if [ x$reason = xBOUND ] || [ x$reason = xRENEW ] || \
    [ x$reason = xREBIND ] || [ x$reason = xREBOOT ]; then
+
+  if [ -n "$new_nis_domain" ]; then
+	domainname "$new_nis_domain"
+  fi
+
   if [ -n "$new_host_name" ]; then
 	if need_hostname || [ "`hostname`" != "$new_host_name" ]; then
 	        hostname $new_host_name
