--- dhcp-3.0pl1/minires/ns_name.c~	2000-02-02 02:28:14.000000000 -0500
+++ dhcp-3.0pl1/minires/ns_name.c	2003-01-08 04:25:23.000000000 -0500
@@ -255,6 +255,11 @@
 	dn = dst;
 	eom = dst + dstsiz;
 
+	if (dn >= eom) {
+ 		errno = EMSGSIZE;
+ 		return (-1);
+ 	}
+
 	while ((n = *cp++) != 0) {
 		if ((n & NS_CMPRSFLGS) != 0) {
 			/* Some kind of compression pointer. */
--- dhcp-3.0pl1/minires/ns_samedomain.c~	2001-01-16 17:33:09.000000000 -0500
+++ dhcp-3.0pl1/minires/ns_samedomain.c	2003-01-08 04:26:57.000000000 -0500
@@ -168,7 +168,7 @@
 	size_t n = strlen(src);
 
 	if (n + sizeof "." > dstsize) {
-		ISC_R_NOSPACE;
+		return ISC_R_NOSPACE;
 	}
 	strcpy(dst, src);
 	while (n > 0 && dst[n - 1] == '.')		/* Ends in "." */
--- dhcp-3.0pl1/minires/ns_sign.c~	2001-05-17 16:47:33.000000000 -0400
+++ dhcp-3.0pl1/minires/ns_sign.c	2003-01-08 04:28:08.000000000 -0500
@@ -87,7 +87,7 @@
 
 	dst_init();
 	if (msg == NULL || msglen == NULL || sig == NULL || siglen == NULL)
-		ISC_R_INVALIDARG;
+		return ISC_R_INVALIDARG;
 
 	/* Name. */
 	if (key != NULL && error != ns_r_badsig && error != ns_r_badkey)
@@ -118,7 +118,7 @@
 	else
 		n = dn_comp("", cp, (unsigned)(eob - cp), NULL, NULL);
 	if (n < 0)
-		ISC_R_NOSPACE;
+		return ISC_R_NOSPACE;
 	alg = cp;
 	cp += n;
 	
@@ -190,7 +190,7 @@
 		n = dst_sign_data(SIG_MODE_FINAL, key, &ctx, NULL, 0,
 				  sig, *siglen);
 		if (n < 0)
-			ISC_R_BADKEY;
+			return ISC_R_BADKEY;
 		*siglen = n;
 	} else
 		*siglen = 0;
--- dhcp-3.0pl1/minires/res_findzonecut.c~	2001-05-17 16:47:35.000000000 -0400
+++ dhcp-3.0pl1/minires/res_findzonecut.c	2003-01-08 04:28:40.000000000 -0500
@@ -355,7 +355,7 @@
 		while (*dname != '.') {
 			if (*dname == '\\')
 				if (*++dname == '\0') {
-					ISC_R_NOSPACE;
+					return ISC_R_NOSPACE;
 				}
 			dname++;
 		}
