diff -Naupr MPlayer-0.90pre9.orig/mplayer.c MPlayer-0.90pre9/mplayer.c
--- MPlayer-0.90pre9.orig/mplayer.c	Thu Oct 24 00:07:11 2002
+++ MPlayer-0.90pre9/mplayer.c	Fri Oct 25 17:32:50 2002
@@ -539,9 +539,7 @@ int gui_no_filename=0;
       gCpuCaps.hasMMX,gCpuCaps.hasMMX2,
       gCpuCaps.has3DNow, gCpuCaps.has3DNowExt,
       gCpuCaps.hasSSE, gCpuCaps.hasSSE2);
-#ifdef RUNTIME_CPUDETECT
-  mp_msg(MSGT_CPLAYER,MSGL_INFO, MSGTR_CompiledWithRuntimeDetection);
-#else
+#ifndef RUNTIME_CPUDETECT
   mp_msg(MSGT_CPLAYER,MSGL_INFO, MSGTR_CompiledWithCPUExtensions);
 #ifdef HAVE_MMX
   mp_msg(MSGT_CPLAYER,MSGL_INFO," MMX");
