diff -Naupr MPlayer-20051211.orig/configure MPlayer-20051211/configure
--- MPlayer-20051211.orig/configure	2005-12-11 19:19:05.000000000 +0100
+++ MPlayer-20051211/configure	2005-12-12 09:20:15.000000000 +0100
@@ -6962,7 +6962,7 @@ echores "$_joystick"
 echocheck "lirc"
 if test "$_lirc" = auto ; then
   _lirc=no
-  if test -c /dev/lirc -o -c /dev/lirc/0 ; then
+  if /bin/true ; then
       cat > $TMPC <<EOF
 #include <lirc/lirc_client.h>
 int main(void) { return 0; }
