diff -Naupr MPlayer-1.0pre6a.orig/subreader.c MPlayer-1.0pre6a/subreader.c
--- MPlayer-1.0pre6a.orig/subreader.c	2004-10-18 22:56:11.000000000 +0200
+++ MPlayer-1.0pre6a/subreader.c	2005-01-03 12:47:20.994640088 +0100
@@ -33,7 +33,7 @@ char *sub_cp=NULL;
 #include <fribidi/fribidi.h>
 char *fribidi_charset = NULL;   ///character set that will be passed to FriBiDi
 int flip_hebrew = 1;            ///flip subtitles using fribidi
-int fribidi_flip_commas = 0;    ///flip comma when fribidi is used
+int fribidi_flip_commas = 1;    ///flip comma when fribidi is used
 #endif
 
 extern char* dvdsub_lang;
