diff -Naupr MPlayer-1.0pre6a.orig/configure MPlayer-1.0pre6a/configure
--- MPlayer-1.0pre6a.orig/configure	2004-12-23 16:36:00.000000000 +0100
+++ MPlayer-1.0pre6a/configure	2005-01-13 22:56:35.381351040 +0100
@@ -5995,8 +5995,8 @@ echocheck "Video 4 Linux TV interface"
 if test "$_tv_v4l" = auto ; then
  _tv_v4l=no
  if test "$_tv" = yes && linux ; then
-  for I in /dev/video /dev/video? ; do
-    if test -c $I ; then
+  for I in /dev/video ; do
+    if /bin/true ; then
       cat > $TMPC <<EOF
 #include <stdlib.h>
 #include <linux/videodev.h>
@@ -6022,8 +6022,8 @@ echocheck "Video 4 Linux 2 TV interface"
 if test "$_tv_v4l2" = auto ; then
  _tv_v4l2=no
  if test "$_tv" = yes && linux ; then
-  for I in /dev/video /dev/video? ; do
-    if test -c $I ; then
+  for I in /dev/video ; do
+    if /bin/true ; then
       _tv_v4l2=yes
       break
     fi
@@ -6477,7 +6477,7 @@ echo Checking for joystick ... "$_joysti
 echocheck "lirc"
 if test "$_lirc" = auto ; then
   _lirc=no
-  if test -c /dev/lirc -o -c /dev/lirc/0 ; then
+  if /bin/true ; then
       cat > $TMPC <<EOF
 #include <lirc/lirc_client.h>
 int main(void) { return 0; }
