--- lbzip2/Makefile.oldgcc	2009-01-06 05:48:49.000000000 +0100
+++ lbzip2/Makefile	2009-01-19 09:32:39.000000000 +0100
@@ -8,8 +8,8 @@
 
 
 
-LDFLAGS=-s $$($(SHELL) lfs.sh LDFLAGS)
-LIBS=-l pthread -l bz2 $$($(SHELL) lfs.sh LIBS)
+#LDFLAGS=-s $$($(SHELL) lfs.sh LDFLAGS)
+LIBS=-lpthread -lbz2 $$($(SHELL) lfs.sh LIBS)
 
 lbzip2: main.o lbzip2.o lbunzip2.o lbunzip2_single.o lacos_rbtree.o
 	$(CC) -o lbzip2 $(LDFLAGS) main.o lbzip2.o lbunzip2.o \
