--- hostap-driver-0.1.2/driver/modules/hostap_compat.h.orig	2003-10-11 12:11:56.000000000 -0500
+++ hostap-driver-0.1.2/driver/modules/hostap_compat.h	2003-11-17 15:36:03.000000000 -0600
@@ -175,7 +175,7 @@
 #endif /* WIRELESS_EXT > 12 */
 
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,4,23))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,4,22))
 struct net_device;
 static inline void free_netdev(struct net_device *dev)
 {
