--- Makefile.orig	2003-08-03 13:50:09.000000000 -0500
+++ Makefile	2003-10-30 17:25:32.000000000 -0600
@@ -17,7 +17,9 @@
 CC=gcc
 CFLAGS=-O2 -D__KERNEL__ -DMODULE -Wall -g -c $(EXTRA_CFLAGS)
 
-include $(KERNEL_PATH)/.config
+KERNEL_CONFIG=$(KERNEL_PATH)/.config
+
+include $(KERNEL_CONFIG)
 
 INCLUDES=-I$(KERNEL_PATH)/include
 ifdef PCMCIA_PATH
@@ -217,12 +219,14 @@
 	@if [ -r $(MODPATH)/kernel/drivers/net/wireless/hostap_crypt.ko ]; then \
 		echo "You may need to remove old $(MODPATH)/kernel/drivers/net/wireless/hostap_crypt.ko"; \
 	fi
+ifndef DESTDIR
 	@if [ -f "/boot/System.map-$(KERNELRELEASE)" ] || [ "$(CURRENTKV)" = "$(KERNELRELEASE)" ]; then \
 		echo "*** Running depmod..."; \
 		/sbin/depmod -ae; \
 	else \
 		echo "*** I won't run depmod because I can't find an appropriate System.map file or you are not running the same kernel as the one you are building against. Please run it manually when the conditions are correct..."; \
 	fi
+endif
 
 install_hostap: $(MSRC)/hostap.o
 	@echo "Installing hostap.o to $(MODPATH_PCI)"
@@ -258,13 +262,17 @@
 	@echo "Installing $(MODULE_PLX) to $(MODPATH_PLX)"
 	mkdir -p $(MODPATH_PLX)
 	cp -f $(MSRC)/$(MODULE_PLX) $(MODPATH_PLX)
+ifndef DESTDIR
 	/sbin/depmod -ae
+endif
 
 install_pci: install_check install_crypt install_hostap
 	@echo "Installing $(MODULE_PCI) to $(MODPATH_PCI)"
 	mkdir -p $(MODPATH_PCI)
 	cp -f $(MSRC)/$(MODULE_PCI) $(MODPATH_PCI)
+ifndef DESTDIR
 	/sbin/depmod -ae
+endif
 
 uninstall:
 	find $(MODPATH) -name "hostap*" | xargs rm -f
