--- hostap-driver-0.1.2/driver/modules/hostap_config.h.orig	2003-11-02 21:40:26.000000000 -0600
+++ hostap-driver-0.1.2/driver/modules/hostap_config.h	2003-11-17 16:31:34.000000000 -0600
@@ -47,6 +47,7 @@
 #define PRISM2_DOWNLOAD_SUPPORT
 #endif
 
+#define PRISM2_DOWNLOAD_SUPPORT
 #ifdef PRISM2_DOWNLOAD_SUPPORT
 /* Allow writing firmware images into flash, i.e., to non-volatile storage.
  * Before you enable this option, you should make absolutely sure that you are
@@ -54,7 +55,7 @@
  * In addition, please note that it is possible to kill your card with
  * non-volatile download if you are using incorrect image. This feature has not
  * been fully tested, so please be careful with it. */
-/* #define PRISM2_NON_VOLATILE_DOWNLOAD */
+#define PRISM2_NON_VOLATILE_DOWNLOAD
 #endif /* PRISM2_DOWNLOAD_SUPPORT */
 
 /* Include wireless extensions sub-ioctl support even if wireless extensions
