--- Makefile.orig	2006-02-07 14:09:57.486623848 +0100
+++ Makefile	2006-02-07 14:14:25.815831608 +0100
@@ -3,16 +3,18 @@
 # Please uncomment the appropriate settings
 
 CC ?= gcc
-CFLAGS += -Wall -DHAVE_GETOPT_LONG -DUSE_SSL
-#CFLAGS += -DSETPROCTITLE
+
+CFLAGS = -Wall
+CFLAGS += -DHAVE_GETOPT_LONG -DUSE_SSL -DSETPROCTITLE
 LDFLAGS += -lssl
-BINDIR = /usr/local/bin
-INSTALLPATH = $(DESTDIR)/$(BINDIR)
-MANPATH = /usr/share/man/man1
-INSTALLMANPATH = $(DESTDIR)/$(MANPATH)
 
+PREFIX =/usr/local
+BINDIR = $(PREFIX)/bin
+DATADIR = $(PREFIX)/share
+MANDIR = $(DATADIR)/man
+
+DESTDIR = 
 
-PROGNAME = proxytunnel
 OBJ = proxytunnel.o	\
 	base64.o	\
 	setproctitle.o	\
@@ -24,12 +26,11 @@
 	ntlm.o
 
 proxytunnel: $(OBJ)
-	$(CC) -o $(PROGNAME) $(LDFLAGS) $(OBJ)
+	$(CC) -o proxytunnel $(CFLAGS) $(LDFLAGS) $(OBJ)
 
 clean:		
-	@rm -f $(PROGNAME) $(OBJ)
+	@rm -f proxytunnel $(OBJ)
 
 install:
-		mkdir -p $(INSTALLPATH) $(INSTALLMANPATH)
-		install -m755 $(PROGNAME) $(INSTALLPATH)/$(PROGNAME)
-		install -m644 debian/$(PROGNAME).1 $(INSTALLMANPATH)/$(PROGNAME).1
+		install -D -m755 proxytunnel $(DESTDIR)$(BINDIR)/proxytunnel
+		install -D -m644 debian/proxytunnel.1 $(DESTDIR)$(MANDIR)/man1/proxytunnel.1
