--- memcached.c.orig	2005-03-07 03:04:00.665169976 +0100
+++ memcached.c	2005-03-07 03:04:35.741837512 +0100
@@ -1145,8 +1145,13 @@
 
 void delete_handler(int fd, short which, void *arg) {
     struct timeval t;
+    static int initialized=0;
 
-    evtimer_del(&deleteevent);
+    if (initialized==1) {
+        evtimer_del(&deleteevent);
+    } else {
+        initialized=1;
+    }
     evtimer_set(&deleteevent, delete_handler, 0);
     t.tv_sec = 5; t.tv_usec=0;
     evtimer_add(&deleteevent, &t);
