--- xpdf-3.01/xpdf-chinese-traditional/add-to-xpdfrc.orig	2004-07-27 17:28:07.000000000 -0500
+++ xpdf-3.01/xpdf-chinese-traditional/add-to-xpdfrc	2006-09-25 12:58:50.000000000 -0500
@@ -1,8 +1,8 @@
 #----- begin Chinese Traditional support package (2004-jul-27)
-cidToUnicode	Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
-unicodeMap	Big5		/usr/local/share/xpdf/chinese-traditional/Big5.unicodeMap
-unicodeMap	Big5ascii	/usr/local/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
-cMapDir		Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-traditional/CMap
+cidToUnicode	Adobe-CNS1	/usr/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
+unicodeMap	Big5		/usr/share/xpdf/chinese-traditional/Big5.unicodeMap
+unicodeMap	Big5ascii	/usr/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
+cMapDir		Adobe-CNS1	/usr/share/xpdf/chinese-traditional/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-traditional/CMap
 #displayCIDFontTT	Adobe-CNS1	/usr/..../bkai00mp.ttf
 #----- end Chinese Traditional support package
--- xpdf-3.01/xpdf-korean/add-to-xpdfrc.orig	2005-07-07 13:05:05.000000000 -0500
+++ xpdf-3.01/xpdf-korean/add-to-xpdfrc	2006-09-25 12:56:15.000000000 -0500
@@ -1,8 +1,8 @@
 #----- begin Korean support package (2005-jul-07)
-cidToUnicode	Adobe-Korea1	/usr/local/share/xpdf/korean/Adobe-Korea1.cidToUnicode
-unicodeMap	ISO-2022-KR	/usr/local/share/xpdf/korean/ISO-2022-KR.unicodeMap
-cMapDir		Adobe-Korea1	/usr/local/share/xpdf/korean/CMap
-toUnicodeDir			/usr/local/share/xpdf/korean/CMap
+cidToUnicode	Adobe-Korea1	/usr/share/xpdf/korean/Adobe-Korea1.cidToUnicode
+unicodeMap	ISO-2022-KR	/usr/share/xpdf/korean/ISO-2022-KR.unicodeMap
+cMapDir		Adobe-Korea1	/usr/share/xpdf/korean/CMap
+toUnicodeDir			/usr/share/xpdf/korean/CMap
 #displayCIDFontTT	Adobe-Korea1	/usr/..../batang.ttf"
 #displayCIDFontTT	Unidocs-Korea1	/usr/..../batang.ttf"
 #----- end Korean support package
--- xpdf-3.01/xpdf-chinese-simplified/add-to-xpdfrc.orig	2004-07-27 17:28:08.000000000 -0500
+++ xpdf-3.01/xpdf-chinese-simplified/add-to-xpdfrc	2006-09-25 12:57:18.000000000 -0500
@@ -1,9 +1,9 @@
 #----- begin Chinese Simplified support package (2004-jul-27)
-cidToUnicode	Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
-unicodeMap	ISO-2022-CN	/usr/local/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
-unicodeMap	EUC-CN		/usr/local/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
-unicodeMap	GBK		/usr/local/share/xpdf/chinese-simplified/GBK.unicodeMap
-cMapDir		Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-simplified/CMap
+cidToUnicode	Adobe-GB1	/usr/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
+unicodeMap	ISO-2022-CN	/usr/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
+unicodeMap	EUC-CN		/usr/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
+unicodeMap	GBK		/usr/share/xpdf/chinese-simplified/GBK.unicodeMap
+cMapDir		Adobe-GB1	/usr/share/xpdf/chinese-simplified/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-simplified/CMap
 #displayCIDFontTT	Adobe-GB1	/usr/..../gkai00mp.ttf
 #----- end Chinese Simplified support package
--- xpdf-3.01/xpdf-japanese/add-to-xpdfrc.orig	2004-07-27 17:28:07.000000000 -0500
+++ xpdf-3.01/xpdf-japanese/add-to-xpdfrc	2006-09-25 12:58:10.000000000 -0500
@@ -1,9 +1,9 @@
 #----- begin Japanese support package (2004-jul-27)
-cidToUnicode	Adobe-Japan1	/usr/local/share/xpdf/japanese/Adobe-Japan1.cidToUnicode
-unicodeMap	ISO-2022-JP	/usr/local/share/xpdf/japanese/ISO-2022-JP.unicodeMap
-unicodeMap	EUC-JP		/usr/local/share/xpdf/japanese/EUC-JP.unicodeMap
-unicodeMap	Shift-JIS	/usr/local/share/xpdf/japanese/Shift-JIS.unicodeMap
-cMapDir		Adobe-Japan1	/usr/local/share/xpdf/japanese/CMap
-toUnicodeDir			/usr/local/share/xpdf/japanese/CMap
+cidToUnicode	Adobe-Japan1	/usr/share/xpdf/japanese/Adobe-Japan1.cidToUnicode
+unicodeMap	ISO-2022-JP	/usr/share/xpdf/japanese/ISO-2022-JP.unicodeMap
+unicodeMap	EUC-JP		/usr/share/xpdf/japanese/EUC-JP.unicodeMap
+unicodeMap	Shift-JIS	/usr/share/xpdf/japanese/Shift-JIS.unicodeMap
+cMapDir		Adobe-Japan1	/usr/share/xpdf/japanese/CMap
+toUnicodeDir			/usr/share/xpdf/japanese/CMap
 #displayCIDFontTT	Adobe-Japan1	/usr/..../kochi-mincho.ttf
 #----- end Japanese support package
--- xpdf-3.01/doc/xpdfrc.5.orig	2005-08-17 00:34:30.000000000 -0500
+++ xpdf-3.01/doc/xpdfrc.5	2006-09-25 12:54:53.000000000 -0500
@@ -7,7 +7,7 @@ All of the Xpdf tools read a single conf
 .I .xpdfrc
 file in your home directory, it will be read.  Otherwise, a
 system-wide configuration file will be read from
-.IR /usr/local/etc/xpdfrc ,
+.IR /etc/xpdfrc ,
 if it exists.  (This is its default location; depending on build
 options, it may be placed elsewhere.)  On Win32 systems, the
 .I xpdfrc
@@ -459,7 +459,7 @@ urlCommand      "netscape -remote 'openU
 .fi
 .SH FILES
 .TP
-.B /usr/local/etc/xpdfrc
+.B /etc/xpdfrc
 This is the default location for the system-wide configuration file.
 Depending on build options, it may be placed elsewhere.
 .TP
--- xpdf-3.01/doc/sample-xpdfrc.orig	2005-08-17 00:34:30.000000000 -0500
+++ xpdf-3.01/doc/sample-xpdfrc	2006-09-25 12:54:53.000000000 -0500
@@ -56,7 +56,7 @@
 
 # Set the default PostScript file or command.
 
-#psFile			"|lpr -Pmyprinter"
+psFile			"|lpr"
 
 # Set the default PostScript paper size -- this can be letter, legal,
 # A4, or A3.  You can also specify a paper size as width and height
@@ -88,4 +88,11 @@
 # Set the command used to run a web browser when a URL hyperlink is
 # clicked.
 
+urlCommand "htmlview '%s'"
 #urlCommand	"netscape -remote 'openURL(%s)'"
+
+# CJK
+include /etc/xpdf/add-to-xpdfrc.japanese
+include /etc/xpdf/add-to-xpdfrc.korean
+include /etc/xpdf/add-to-xpdfrc.chinese-simplified
+include /etc/xpdf/add-to-xpdfrc.chinese-traditional
--- xpdf-3.01/doc/xpdf.1.orig	2005-08-17 00:34:30.000000000 -0500
+++ xpdf-3.01/doc/xpdf.1	2006-09-25 12:54:53.000000000 -0500
@@ -41,7 +41,7 @@ xpdf
 .SH CONFIGURATION FILE
 Xpdf reads a configuration file at startup.  It first tries to find
 the user's private config file, ~/.xpdfrc.  If that doesn't exist, it
-looks for a system-wide config file, typically /usr/local/etc/xpdfrc
+looks for a system-wide config file, typically /etc/xpdfrc
 (but this location can be changed when xpdf is built).  See the
 .BR xpdfrc (5)
 man page for details.
--- xpdf-3.01/doc/pdfimages.1.orig	2005-08-17 00:34:30.000000000 -0500
+++ xpdf-3.01/doc/pdfimages.1	2006-09-25 12:54:53.000000000 -0500
@@ -29,7 +29,7 @@ color inversion, etc. done by the PDF co
 Pdfimages reads a configuration file at startup.  It first tries to
 find the user's private config file, ~/.xpdfrc.  If that doesn't
 exist, it looks for a system-wide config file, typically
-/usr/local/etc/xpdfrc (but this location can be changed when pdfimages
+/etc/xpdfrc (but this location can be changed when pdfimages
 is built).  See the
 .BR xpdfrc (5)
 man page for details.
--- xpdf-3.01/xpdf-chinese-traditional/README.orig	2006-09-25 13:33:34.000000000 -0500
+++ xpdf-3.01/xpdf-chinese-traditional/README	2006-09-25 13:37:33.000000000 -0500
@@ -19,13 +19,13 @@ Contents:
 
 Place all of these files in a directory, typically:
 
-    Unix - /usr/local/share/xpdf/chinese-traditional
+    Unix - /usr/share/xpdf/chinese-traditional
     Win32 - C:\Program Files\xpdf\chinese-traditional
 
 Add the contents of the "add-to-xpdfrc" file to your system-wide
 xpdfrc config file, which is typically:
 
-    Unix - /usr/local/etc/xpdfrc
+    Unix - /etc/xpdfrc
     Win32 - C:\Program Files\xpdf\xpdfrc
 
 Alternatively, on Unix systems you can add these lines to your
--- xpdf-3.01/xpdf-korean/README.orig	2006-09-25 13:34:27.000000000 -0500
+++ xpdf-3.01/xpdf-korean/README	2006-09-25 13:38:26.000000000 -0500
@@ -18,13 +18,13 @@ Contents:
 
 Place all of these files in a directory, typically:
 
-    Unix - /usr/local/share/xpdf/korean
+    Unix - /usr/share/xpdf/korean
     Win32 - C:\Program Files\xpdf\korean
 
 Add the contents of the "add-to-xpdfrc" file to your system-wide
 xpdfrc config file, which is typically:
 
-    Unix - /usr/local/etc/xpdfrc
+    Unix - /etc/xpdfrc
     Win32 - C:\Program Files\Xpdf\xpdfrc
 
 Alternatively, on Unix systems you can add these lines to your
--- xpdf-3.01/xpdf-chinese-simplified/README.orig	2006-09-25 13:33:18.000000000 -0500
+++ xpdf-3.01/xpdf-chinese-simplified/README	2006-09-25 13:37:45.000000000 -0500
@@ -20,13 +20,13 @@ Contents:
 
 Place all of these files in a directory, typically:
 
-    Unix - /usr/local/share/xpdf/chinese-simplified
+    Unix - /usr/share/xpdf/chinese-simplified
     Win32 - C:\Program Files\xpdf\chinese-simplified
 
 Add the contents of the "add-to-xpdfrc" file to your system-wide
 xpdfrc config file, which is typically:
 
-    Unix - /usr/local/etc/xpdfrc
+    Unix - /etc/xpdfrc
     Win32 - C:\Program Files\xpdf\xpdfrc
 
 Alternatively, on Unix systems you can add these lines to your
--- xpdf-3.01/xpdf-japanese/README.orig	2006-09-25 13:33:47.000000000 -0500
+++ xpdf-3.01/xpdf-japanese/README	2006-09-25 13:38:12.000000000 -0500
@@ -20,13 +20,13 @@ Contents:
 
 Place all of these files in a directory, typically:
 
-    Unix - /usr/local/share/xpdf/japanese
+    Unix - /usr/share/xpdf/japanese
     Win32 - C:\Program Files\xpdf\japanese
 
 Add the contents of the "add-to-xpdfrc" file to your system-wide
 xpdfrc config file, which is typically:
 
-    Unix - /usr/local/etc/xpdfrc
+    Unix - /etc/xpdfrc
     Win32 - C:\Program Files\xpdf\xpdfrc
 
 Alternatively, on Unix systems you can add these lines to your
--- xpdf-3.01/xpdf-thai/README.BAD	2006-09-25 15:42:22.000000000 -0500
+++ xpdf-3.01/xpdf-thai/README	2006-09-25 15:42:33.000000000 -0500
@@ -18,13 +18,13 @@ Contents:
 
 Place all of these files in a directory, typically:
 
-    Unix - /usr/local/share/xpdf/thai
+    Unix - /usr/share/xpdf/thai
     Win32 - C:\Program Files\xpdf\thai
 
 Add the contents of the "add-to-xpdfrc" file to your system-wide
 xpdfrc config file, which is typically:
 
-    Unix - /usr/local/etc/xpdfrc
+    Unix - /etc/xpdfrc
     Win32 - C:\Program Files\xpdf\xpdfrc
 
 Alternatively, on Unix systems you can add these lines to your
--- xpdf-3.01/xpdf-thai/add-to-xpdfrc.BAD	2006-09-25 15:43:35.000000000 -0500
+++ xpdf-3.01/xpdf-thai/add-to-xpdfrc	2006-09-25 15:43:44.000000000 -0500
@@ -1,4 +1,4 @@
 #----- begin Thai support package (2002-jan-16)
-nameToUnicode			/usr/local/share/xpdf/thai/Thai.nameToUnicode
-unicodeMap	TIS-620		/usr/local/share/xpdf/thai/TIS-620.unicodeMap
+nameToUnicode			/usr/share/xpdf/thai/Thai.nameToUnicode
+unicodeMap	TIS-620		/usr/share/xpdf/thai/TIS-620.unicodeMap
 #----- end Thai support package
--- xpdf-3.01/xpdf-cyrillic/README.BAD	2006-09-25 15:42:48.000000000 -0500
+++ xpdf-3.01/xpdf-cyrillic/README	2006-09-25 15:42:59.000000000 -0500
@@ -18,13 +18,13 @@ Contents:
 
 Place all of these files in a directory, typically:
 
-    Unix - /usr/local/share/xpdf/cyrillic
+    Unix - /usr/share/xpdf/cyrillic
     Win32 - C:\Program Files\xpdf\cyrillic
 
 Add the contents of the "add-to-xpdfrc" file to your system-wide
 xpdfrc config file, which is typically:
 
-    Unix - /usr/local/etc/xpdfrc
+    Unix - /etc/xpdfrc
     Win32 - C:\Program Files\xpdf\xpdfrc
 
 Alternatively, on Unix systems you can add these lines to your
--- xpdf-3.01/xpdf-cyrillic/add-to-xpdfrc.BAD	2006-09-25 15:43:28.000000000 -0500
+++ xpdf-3.01/xpdf-cyrillic/add-to-xpdfrc	2006-09-25 15:43:11.000000000 -0500
@@ -1,4 +1,4 @@
 #----- begin Cyrillic support package (2003-jun-28)
-nameToUnicode			/usr/local/share/xpdf/cyrillic/Bulgarian.nameToUnicode
-unicodeMap	KOI8-R		/usr/local/share/xpdf/cyrillic/KOI8-R.unicodeMap
+nameToUnicode			/usr/share/xpdf/cyrillic/Bulgarian.nameToUnicode
+unicodeMap	KOI8-R		/usr/share/xpdf/cyrillic/KOI8-R.unicodeMap
 #----- end Cyrillic support package
