diff -up xpdf-3.02/goo/gmem.cc.BAD xpdf-3.02/goo/gmem.cc
--- xpdf-3.02/goo/gmem.cc.BAD	2007-08-01 17:20:00.000000000 -0500
+++ xpdf-3.02/goo/gmem.cc	2007-08-01 17:22:04.000000000 -0500
@@ -47,9 +47,9 @@ static int gMemInUse = 0;
 
 #endif /* DEBUG_MEM */
 
-void *gmalloc(int size) GMEM_EXCEP {
+void *gmalloc(size_t size) GMEM_EXCEP {
 #ifdef DEBUG_MEM
-  int size1;
+  size_t size1;
   char *mem;
   GMemHdr *hdr;
   void *data;
@@ -106,11 +106,11 @@ void *gmalloc(int size) GMEM_EXCEP {
 #endif
 }
 
-void *grealloc(void *p, int size) GMEM_EXCEP {
+void *grealloc(void *p, size_t size) GMEM_EXCEP {
 #ifdef DEBUG_MEM
   GMemHdr *hdr;
   void *q;
-  int oldSize;
+  size_t oldSize;
 
   if (size <= 0) {
     if (p) {
@@ -195,7 +195,7 @@ void *greallocn(void *p, int nObjs, int 
 
 void gfree(void *p) {
 #ifdef DEBUG_MEM
-  int size;
+  size_t size;
   GMemHdr *hdr;
   unsigned long *trl, *clr;
 
diff -up xpdf-3.02/goo/gmem.h.BAD xpdf-3.02/goo/gmem.h
--- xpdf-3.02/goo/gmem.h.BAD	2007-08-01 17:23:11.000000000 -0500
+++ xpdf-3.02/goo/gmem.h	2007-08-01 17:23:45.000000000 -0500
@@ -36,13 +36,13 @@ extern "C" {
  * Same as malloc, but prints error message and exits if malloc()
  * returns NULL.
  */
-extern void *gmalloc(int size) GMEM_EXCEP;
+extern void *gmalloc(size_t size) GMEM_EXCEP;
 
 /*
  * Same as realloc, but prints error message and exits if realloc()
  * returns NULL.  If <p> is NULL, calls malloc instead of realloc().
  */
-extern void *grealloc(void *p, int size) GMEM_EXCEP;
+extern void *grealloc(void *p, size_t size) GMEM_EXCEP;
 
 /*
  * These are similar to gmalloc and grealloc, but take an object count
