diff -Naupr gaim-0.75.orig/gaim.desktop gaim-0.75/gaim.desktop
--- gaim-0.75.orig/gaim.desktop	2003-11-29 22:24:33.000000000 +0100
+++ gaim-0.75/gaim.desktop	2004-01-11 01:56:49.000000000 +0100
@@ -1,16 +1,16 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=Gaim
-Comment=Multi-protocol Instant Messaging Client
-Comment[it]=Client Multi-Protocollo per Messaggi Immediati
-Comment[es]=Cliente de mensajería instantánea multiprotocolo
-Comment[fr]=Client de messagerie instantanée multiprotocole
-Comment[de]=Multi-Protokoll Instant Messenger Client
-Comment[ko]=다중 프로토콜 메신저
-Comment[pt_BR]=Cliente de mensagens instantâneas multi-protocolo
+Name=Instant Messenger
+Comment=Multi-protocol Instant Messaging Client (Gaim)
+Comment[it]=Client Multi-Protocollo per Messaggi Immediati (Gaim)
+Comment[es]=Cliente de mensajería instantánea multiprotocolo (Gaim)
+Comment[fr]=Client de messagerie instantanée multiprotocole (Gaim)
+Comment[de]=Multi-Protokoll Instant Messenger Client (Gaim)
+Comment[ko]=다중 프로토콜 메신저 (Gaim)
+Comment[pt_BR]=Cliente de mensagens instantâneas multi-protocolo (Gaim)
 Exec=gaim
 Icon=gaim.png
 Terminal=false
 Type=Application
-Categories=Application;Network;
+Categories=Application;Network;X-Red-Hat-Base;
 StartupNotify=true
