--- gst-plugins-0.8.6/configure.ac.~1~	2004-11-25 07:41:47.000000000 -0800
+++ gst-plugins-0.8.6/configure.ac	2005-01-30 19:14:23.468903934 -0800
@@ -952,8 +952,14 @@
 dnl **** Free AAC Decoder (FAAD) ****
 translit(dnm, m, l) AM_CONDITIONAL(USE_FAAD, true)
 GST_CHECK_FEATURE(FAAD, [AAC decoder plug-in], faad, [
-  HAVE_FAAD="yes"
-  GST_CHECK_LIBHEADER(FAAD, faad, faacDecOpen, -lm, faad.h, FAAD_LIBS="-lfaad -lm", HAVE_FAAD="no")
+  HAVE_FAAD="no"
+  HAVE_FAAD1="yes"
+  HAVE_FAAD2="yes"
+  GST_CHECK_LIBHEADER(FAAD, faad, faacDecOpen, -lm, faad.h, FAAD_LIBS="-lfaad -lm", HAVE_FAAD1="no")
+  GST_CHECK_LIBHEADER(FAAD, faad, NeAACDecOpen, -lm, faad.h, FAAD_LIBS="-lfaad -lm", HAVE_FAAD2="no")
+  if test $HAVE_FAAD1 = "yes" -o $HAVE_FAAD2 = "yes"; then
+	HAVE_FAAD="yes"
+  fi;
   if test $HAVE_FAAD = "yes"; then
     AC_MSG_CHECKING([Checking for FAAD >= 2])
     AC_TRY_COMPILE([
