diff -Nur pdflib-4.0.3.orig/Makefile.in pdflib-4.0.3/Makefile.in
--- pdflib-4.0.3.orig/Makefile.in	Mon Mar 25 15:13:34 2002
+++ pdflib-4.0.3/Makefile.in	Wed Jul 24 20:09:08 2002
@@ -114,11 +114,11 @@
 # --------------------------- install ---------------------------------
 
 install: pdflib_install @BINDINGS_INSTALL@
-	if test ! -d $(bindir) ; then 	\
-	    mkdir -p $(bindir);		\
-	    chmod 755 $(bindir);	\
+	if test ! -d $(DESTDIR)$(bindir) ; then 	\
+	    mkdir -p $(DESTDIR)$(bindir);		\
+	    chmod 755 $(DESTDIR)$(bindir);	\
 	fi
-	$(INSTALL) pdflib-config $(bindir)
+	$(INSTALL) pdflib-config $(DESTDIR)$(bindir)
 
 pdflib_install:
 	-cd pdflib && $(MAKE) install
diff -Nur pdflib-4.0.3.orig/bind/java/Makefile.in pdflib-4.0.3/bind/java/Makefile.in
--- pdflib-4.0.3.orig/bind/java/Makefile.in	Mon Jan 28 13:57:53 2002
+++ pdflib-4.0.3/bind/java/Makefile.in	Wed Jul 24 20:13:35 2002
@@ -93,12 +93,13 @@
 	    -dlopen $(SWIG_LIB) java quickreference-ibm
 
 install: $(SWIG_LIB) pdflib.jar
+	$(INSTALL) -d $(DESTDIR)$(LANG_LIBDIR)
 	@-if test -f .libs/libpdf_java.jnilib; then		\
 	    $(LIBTOOL) --mode=install $(INSTALL_DATA)		\
-		.libs/libpdf_java.jnilib $(LANG_LIBDIR);	\
+		.libs/libpdf_java.jnilib $(DESTDIR)$(LANG_LIBDIR);	\
 	else							\
 	    $(LIBTOOL) --mode=install $(INSTALL_DATA)		\
-		$(SWIG_LIB) $(LANG_LIBDIR);			\
+		$(SWIG_LIB) $(DESTDIR)$(LANG_LIBDIR);			\
 	fi
 	@echo Copy pdflib.jar and javadoc to an appropriate directory,
 	@echo and include pdflib.jar in your CLASSPATH.
@@ -106,10 +107,10 @@
 uninstall:
 	@-if test -f .libs/libpdf_java.jnilib; then		\
 	    $(LIBTOOL) --mode=uninstall $(RM)			\
-		$(LANG_LIBDIR)/libpdf_java.jnilib;		\
+		$(DESTDIR)$(LANG_LIBDIR)/libpdf_java.jnilib;		\
 	else							\
 	    $(LIBTOOL) --mode=uninstall $(RM)			\
-		$(LANG_LIBDIR)/$(SWIG_LIB);			\
+		$(DESTDIR)$(LANG_LIBDIR)/$(SWIG_LIB);			\
 	fi
 
 clean:
diff -Nur pdflib-4.0.3.orig/bind/perl/Makefile.in pdflib-4.0.3/bind/perl/Makefile.in
--- pdflib-4.0.3.orig/bind/perl/Makefile.in	Fri Oct 12 10:54:48 2001
+++ pdflib-4.0.3/bind/perl/Makefile.in	Wed Jul 24 20:13:15 2002
@@ -60,13 +60,14 @@
 	-$(PERLBIN) quickreference.pl
 
 install: $(SWIG_LIB)
+	$(INSTALL) -d $(DESTDIR)$(LANG_LIBDIR)
 	$(LIBTOOL) --mode=install \
-		$(INSTALL_DATA) $(SWIG_LIB) $(LANG_LIBDIR)
-	$(INSTALL_DATA) pdflib_pl.pm $(LANG_LIBDIR)
+		$(INSTALL_DATA) $(SWIG_LIB) $(DESTDIR)$(LANG_LIBDIR)
+	$(INSTALL_DATA) pdflib_pl.pm $(DESTDIR)$(LANG_LIBDIR)
 
 uninstall:
 	$(LIBTOOL) --mode=uninstall \
-		$(RM) $(LANG_LIBDIR)/$(SWIG_LIB) $(LANG_LIBDIR)/pdflib_pl.pm 
+		$(RM) $(LANG_LIBDIR)/$(SWIG_LIB) $(DESTDIR)$(LANG_LIBDIR)/pdflib_pl.pm 
 
 clean:
 	-$(RM) *.$(OBJ) *.lo *.la .libs hello.pdf pdfclock.pdf	\
diff -Nur pdflib-4.0.3.orig/bind/python/Makefile.in pdflib-4.0.3/bind/python/Makefile.in
--- pdflib-4.0.3.orig/bind/python/Makefile.in	Fri Oct 12 10:54:48 2001
+++ pdflib-4.0.3/bind/python/Makefile.in	Wed Jul 24 20:14:21 2002
@@ -62,18 +62,19 @@
 	-$(LIBTOOL) --mode=execute -dlopen $(SWIG_LIB) python personalize.py
 
 install: $(SWIG_LIB)
+	$(INSTALL) -d $(DESTDIR)$(LANG_LIBDIR)
 	@-if test -f .libs/libpdflib_py.dylib; then		\
-	    $(INSTALL_DATA) .libs/pdflib_py.so $(LANG_LIBDIR);	\
+	    $(INSTALL_DATA) .libs/pdflib_py.so $(DESTDIR)$(LANG_LIBDIR);	\
 	else							\
 	    $(LIBTOOL) --mode=install $(INSTALL_DATA) 		\
-		$(SWIG_LIB) $(LANG_LIBDIR);			\
+		$(SWIG_LIB) $(DESTDIR)$(LANG_LIBDIR);			\
 	fi
 
 uninstall:
 	@-if test -f .libs/libpdflib_py.dylib; then			 \
-	    $(LIBTOOL) --mode=uninstall $(RM) $(LANG_LIBDIR)/pdflib_py.so\
+	    $(LIBTOOL) --mode=uninstall $(RM) $(DESTDIR)$(LANG_LIBDIR)/pdflib_py.so\
 	else 								 \
-	    $(LIBTOOL) --mode=uninstall $(RM) $(LANG_LIBDIR)/$(SWIG_LIB) \
+	    $(LIBTOOL) --mode=uninstall $(RM) $(DESTDIR)$(LANG_LIBDIR)/$(SWIG_LIB) \
 	fi
 
 clean:
diff -Nur pdflib-4.0.3.orig/bind/tcl/Makefile.in pdflib-4.0.3/bind/tcl/Makefile.in
--- pdflib-4.0.3.orig/bind/tcl/Makefile.in	Fri Oct 12 10:54:48 2001
+++ pdflib-4.0.3/bind/tcl/Makefile.in	Wed Jul 24 20:15:34 2002
@@ -63,29 +63,29 @@
 	-$(LIBTOOL) --mode=execute -dlopen $(SWIG_LIB) $(TCLBIN) personalize.tcl
 
 install: $(SWIG_LIB)
-	if test ! -d $(LANG_LIBDIR) ; then 	\
-	    mkdir -p $(LANG_LIBDIR);		\
-	    chmod 755 $(LANG_LIBDIR);		\
+	if test ! -d $(DESTDIR)$(LANG_LIBDIR) ; then 	\
+	    mkdir -p $(DESTDIR)$(LANG_LIBDIR);		\
+	    chmod 755 $(DESTDIR)$(LANG_LIBDIR);		\
 	fi
-	$(INSTALL_DATA) pkgIndex.tcl $(LANG_LIBDIR)
+	$(INSTALL_DATA) pkgIndex.tcl $(DESTDIR)$(LANG_LIBDIR)
 	@-if test -f .libs/libpdflib_tcl.dylib; then			\
-	    $(INSTALL_DATA) .libs/pdflib_tcl.dylib $(LANG_LIBDIR);	\
+	    $(INSTALL_DATA) .libs/pdflib_tcl.dylib $(DESTDIR)$(LANG_LIBDIR);	\
 	else								\
 	    $(LIBTOOL) --mode=install $(INSTALL_DATA)			\
-		$(SWIG_LIB) $(LANG_LIBDIR);				\
+		$(SWIG_LIB) $(DESTDIR)$(LANG_LIBDIR);				\
 	fi
 
 uninstall:
 	@-if test -f .libs/libpdflib_tcl.dylib; then			\
 	    $(LIBTOOL) --mode=uninstall $(RM)				\
-		$(LANG_LIBDIR)/pdflib_tcl.dylib				\
-		$(LANG_LIBDIR)/pkgIndex.tcl;				\
+		$(DESTDIR)$(LANG_LIBDIR)/pdflib_tcl.dylib				\
+		$(DESTDIR)$(LANG_LIBDIR)/pkgIndex.tcl;				\
 	else								\
 	    $(LIBTOOL) --mode=uninstall $(RM)				\
-		$(LANG_LIBDIR)/$(SWIG_LIB)				\
-		$(LANG_LIBDIR)/pkgIndex.tcl;				\
+		$(DESTDIR)$(LANG_LIBDIR)/$(SWIG_LIB)				\
+		$(DESTDIR)$(LANG_LIBDIR)/pkgIndex.tcl;				\
 	fi
-	$(RM) $(LANG_LIBDIR)
+	$(RM) $(DESTDIR)$(LANG_LIBDIR)
 
 clean:
 	-$(RM) *.$(OBJ) *.lo *.la .libs hello.pdf image.pdf pdfclock.pdf personalize.pdf quickreference.pdf
diff -Nur pdflib-4.0.3.orig/pdflib/Makefile.in pdflib-4.0.3/pdflib/Makefile.in
--- pdflib-4.0.3.orig/pdflib/Makefile.in	Fri Oct 12 10:54:48 2001
+++ pdflib-4.0.3/pdflib/Makefile.in	Wed Jul 24 20:18:23 2002
@@ -107,28 +107,28 @@
 	-$(RM) *.$(OBJ) *.lo *.la *.bak .libs
 
 install: dummy
-	if test ! -d $(libdir) ; then 	\
-	    mkdir -p $(libdir);		\
-	    chmod 755 $(libdir);	\
+	if test ! -d $(DESTDIR)$(libdir) ; then 	\
+	    mkdir -p $(DESTDIR)$(libdir);		\
+	    chmod 755 $(DESTDIR)$(libdir);	\
 	fi
-	if test ! -d $(includedir) ; then 	\
-	    mkdir -p $(includedir);		\
-	    chmod 755 $(includedir);		\
+	if test ! -d $(DESTDIR)$(includedir) ; then 	\
+	    mkdir -p $(DESTDIR)$(includedir);		\
+	    chmod 755 $(DESTDIR)$(includedir);		\
 	fi
-	$(INSTALL_DATA) pdflib.h $(includedir)
+	$(INSTALL_DATA) pdflib.h $(DESTDIR)$(includedir)
 	@-if test "$(WITH_SHARED)" = "no"; then	\
 	    cp .libs/libpdf.la .libs/libpdf.lai;\
 	fi
-	$(LIBTOOL) $(INSTALL_DATA) $(LIBPDF).la $(libdir);
+	$(LIBTOOL) $(INSTALL_DATA) $(LIBPDF).la $(DESTDIR)$(libdir);
 	@-if test "$(WITH_SHARED)" = "yes"; then	\
-	    $(LIBTOOL) -n --finish $(libdir);\
+	    $(LIBTOOL) -n --finish $(DESTDIR)$(libdir);\
 	else\
 	    rm -f .libs/libpdf.lai;\
 	fi
 
 uninstall: dummy
-	$(RM) $(includedir)/pdflib.h
-	-$(LIBTOOL) --mode=uninstall $(RM) $(libdir)/$(LIBPDF).la
+	$(RM) $(DESTDIR)$(includedir)/pdflib.h
+	-$(LIBTOOL) --mode=uninstall $(RM) $(DESTDIR)$(libdir)/$(LIBPDF).la
 
 depend: dummy
 	cp Makefile.in Makefile.in.bak
