diff --git a/CRT.c b/CRT.c
index 86771ed..3ff7e62 100644
--- a/CRT.c
+++ b/CRT.c
@@ -25,6 +25,7 @@ in the source distribution for its full text.
 #define COLORSCHEME_BLACKONWHITE2 3
 #define COLORSCHEME_MIDNIGHT 4
 #define COLORSCHEME_BLACKNIGHT 5
+#define COLORSCHEME_BLUEWEB 6
 
 #define Black COLOR_BLACK
 #define Red COLOR_RED
@@ -207,7 +208,66 @@ void CRT_setColors(int colorScheme) {
             init_pair((7-i)*8+j, i, (j==0?-1:j));
    }
 
-   if (colorScheme == COLORSCHEME_MONOCHROME) {
+   if (colorScheme == COLORSCHEME_BLUEWEB) {
+      CRT_colors[RESET_COLOR] = A_NORMAL;
+      CRT_colors[DEFAULT_COLOR] = A_NORMAL;
+      CRT_colors[FUNCTION_BAR] = ColorPair(White,Blue);
+      CRT_colors[FUNCTION_KEY] = A_REVERSE;
+      CRT_colors[PANEL_HEADER_FOCUS] = A_BOLD | ColorPair(White,Blue);
+      CRT_colors[PANEL_HEADER_UNFOCUS] = ColorPair(White,Blue);
+      CRT_colors[PANEL_HIGHLIGHT_FOCUS] = ColorPair(Blue,White);
+      CRT_colors[PANEL_HIGHLIGHT_UNFOCUS] = A_BOLD;
+      CRT_colors[FAILED_SEARCH] = A_REVERSE | A_BOLD;
+      CRT_colors[UPTIME] = A_BOLD;
+      CRT_colors[BATTERY] = A_BOLD;
+      CRT_colors[LARGE_NUMBER] = A_BOLD;
+      CRT_colors[METER_TEXT] = A_NORMAL;
+      CRT_colors[METER_VALUE] = A_BOLD;
+      CRT_colors[LED_COLOR] = A_NORMAL;
+      CRT_colors[TASKS_RUNNING] = A_BOLD;
+      CRT_colors[PROCESS] = A_NORMAL;
+      CRT_colors[PROCESS_SHADOW] = A_DIM;
+      CRT_colors[PROCESS_TAG] = A_BOLD;
+      CRT_colors[PROCESS_MEGABYTES] = A_BOLD;
+      CRT_colors[PROCESS_BASENAME] = A_BOLD;
+      CRT_colors[PROCESS_TREE] = A_BOLD;
+      CRT_colors[PROCESS_R_STATE] = A_BOLD;
+      CRT_colors[PROCESS_HIGH_PRIORITY] = A_BOLD;
+      CRT_colors[PROCESS_LOW_PRIORITY] = A_DIM;
+      CRT_colors[PROCESS_THREAD] = A_BOLD;
+      CRT_colors[PROCESS_THREAD_BASENAME] = A_REVERSE;
+      CRT_colors[BAR_BORDER] = A_BOLD;
+      CRT_colors[BAR_SHADOW] = A_DIM;
+      CRT_colors[SWAP] = A_BOLD;
+      CRT_colors[GRAPH_1] = A_BOLD;
+      CRT_colors[GRAPH_2] = A_BOLD;
+      CRT_colors[GRAPH_3] = A_BOLD;
+      CRT_colors[GRAPH_4] = A_NORMAL;
+      CRT_colors[GRAPH_5] = A_NORMAL;
+      CRT_colors[GRAPH_6] = A_NORMAL;
+      CRT_colors[GRAPH_7] = A_DIM;
+      CRT_colors[GRAPH_8] = A_DIM;
+      CRT_colors[GRAPH_9] = A_DIM;
+      CRT_colors[MEMORY_USED] = A_BOLD;
+      CRT_colors[MEMORY_BUFFERS] = A_NORMAL;
+      CRT_colors[MEMORY_CACHE] = A_NORMAL;
+      CRT_colors[LOAD_AVERAGE_FIFTEEN] = A_DIM;
+      CRT_colors[LOAD_AVERAGE_FIVE] = A_NORMAL;
+      CRT_colors[LOAD_AVERAGE_ONE] = A_BOLD;
+      CRT_colors[LOAD] = A_BOLD;
+      CRT_colors[HELP_BOLD] = A_BOLD;
+      CRT_colors[CPU_NICE] = A_BOLD;
+      CRT_colors[CPU_NORMAL] = A_BOLD;
+      CRT_colors[CPU_KERNEL] = A_BOLD;
+      CRT_colors[CLOCK] = A_BOLD;
+      CRT_colors[CHECK_BOX] = A_BOLD;
+      CRT_colors[CHECK_MARK] = A_NORMAL;
+      CRT_colors[CHECK_TEXT] = A_NORMAL;
+      CRT_colors[CPU_IOWAIT] = A_BOLD;
+      CRT_colors[CPU_IRQ] = A_BOLD;
+      CRT_colors[CPU_SOFTIRQ] = A_BOLD;
+      CRT_colors[HOSTNAME] = A_BOLD;
+   } else if (colorScheme == COLORSCHEME_MONOCHROME) {
       CRT_colors[RESET_COLOR] = A_NORMAL;
       CRT_colors[DEFAULT_COLOR] = A_NORMAL;
       CRT_colors[FUNCTION_BAR] = A_REVERSE;
diff --git a/CRT.h b/CRT.h
index 02ed5c4..56d7484 100644
--- a/CRT.h
+++ b/CRT.h
@@ -28,6 +28,7 @@ in the source distribution for its full text.
 #define COLORSCHEME_BLACKONWHITE2 3
 #define COLORSCHEME_MIDNIGHT 4
 #define COLORSCHEME_BLACKNIGHT 5
+#define COLORSCHEME_BLUEWEB 6
 
 #define Black COLOR_BLACK
 #define Red COLOR_RED
diff --git a/ColorsPanel.c b/ColorsPanel.c
index cdc0166..1737ea3 100644
--- a/ColorsPanel.c
+++ b/ColorsPanel.c
@@ -34,6 +34,7 @@ static char* ColorSchemes[] = {
    "Light Terminal",
    "MC",
    "Black Night",
+   "BlueWeb",
    NULL
 };
 
diff --git a/Settings.c b/Settings.c
index f062986..8263a59 100644
--- a/Settings.c
+++ b/Settings.c
@@ -118,7 +118,7 @@ static bool Settings_read(Settings* this, char* fileName) {
       } else if (String_eq(option[0], "color_scheme")) {
          this->colorScheme = atoi(option[1]);
          if (this->colorScheme < 0) this->colorScheme = 0;
-         if (this->colorScheme > 5) this->colorScheme = 5;
+         if (this->colorScheme > 6) this->colorScheme = 6;
       } else if (String_eq(option[0], "left_meters")) {
          Settings_readMeters(this, option[1], LEFT_HEADER);
          readMeters = true;