diff -Nur livecd.orig/imgcreate/live.py livecd/imgcreate/live.py
--- livecd.orig/imgcreate/live.py	2009-08-18 09:54:35.000000000 -0400
+++ livecd/imgcreate/live.py	2009-08-18 14:16:35.000000000 -0400
@@ -68,7 +68,7 @@
 
         self.__isodir = None
 
-        self.__modules = ["=ata", "sym53c8xx", "aic7xxx", "=usb", "=firewire", "=mmc"]
+        self.__modules = ["sym53c8xx", "aic7xxx", "=usb", "=firewire", "=mmc"]
         self.__modules.extend(kickstart.get_modules(self.ks))
 
     #
@@ -218,6 +218,7 @@
 
         f.write('MODULES+="squashfs ext3 ext2 vfat msdos "\n')
         f.write('MODULES+="sr_mod sd_mod ide-cd "\n')
+        f.write('MODULES+="libata ahci ata_piix "\n')
 
         for module in self.__modules:
             if module == "=usb":
diff -Nur livecd.orig/tools/mayflower livecd/tools/mayflower
--- livecd.orig/tools/mayflower	2009-08-18 09:54:35.000000000 -0400
+++ livecd/tools/mayflower	2009-08-18 14:14:30.000000000 -0400
@@ -107,15 +107,7 @@
     char=$(echo $m | cut -c1)
     if [ $char = '=' ]; then
 	NAME=$(echo $m | cut -c2-)
-	if [ "$NAME" = "ata" ]; then
-	    if [ -f /lib/modules/$KERNEL/modules.libata ]; then
-		MODS="$MODS $(cat /lib/modules/$KERNEL/modules.libata |sed -e 's/.ko//')"
-	    else
-		MODS="$MODS $(cat /lib/modules/$KERNEL/modules.block |egrep '(ata|ahci)' |sed -e 's/.ko//')"
-	    fi
-	else
-	    MODS="$MODS $(cat /lib/modules/$KERNEL/modules.$NAME |sed -e 's/.ko//')"
-	fi
+	MODS="$MODS $(cat /lib/modules/$KERNEL/modules.$NAME |sed -e 's/.ko//')"
     else
 	MODS="$MODS $m"
     fi
