--- Mail-SpamAssassin-2.60/spamd/redhat-rc-script.sh~	2003-06-26 16:56:47.000000000 +0200
+++ Mail-SpamAssassin-2.60/spamd/redhat-rc-script.sh	2003-09-24 20:37:10.000000000 +0200
@@ -2,7 +2,7 @@
 #
 # spamassassin This script starts and stops the spamd daemon
 #
-# chkconfig: 2345 80 30
+# chkconfig: - 80 20
 #
 # description: spamd is a daemon process which uses SpamAssassin to check
 #              email messages for SPAM.  It is normally called by spamc
