--- Text-Markdown-1.000031/Makefile.PL	2010-03-21 00:05:43.000000000 +0100
+++ Text-Markdown-1.000031/Makefile.PL.orig	2011-02-08 12:44:30.000000000 +0100
@@ -24,14 +24,15 @@
 if ($Module::Install::AUTHOR) {
     system("pod2text lib/Text/Markdown.pm > README")
         and die $!;
-    require Module::Install::CheckConflicts;
+#    require Module::Install::CheckConflicts;
 }
 
-check_conflicts(
-    'Text::MultiMarkdown' => '1.0.27',
-);
-
-install_script 'script/Markdown.pl';
+# we'll handle this stuff in the RPM, thanks
+# check_conflicts(
+#     'Text::MultiMarkdown' => '1.0.27',
+# );
+# 
+#install_script 'script/Markdown.pl';
 
 WriteAll;
 
