diff -Naur libtheora-1.0alpha3/include/theora/Makefile.am libtheora-1.0alpha3.patched/include/theora/Makefile.am
--- libtheora-1.0alpha3/include/theora/Makefile.am	2002-09-23 05:02:07.000000000 +0200
+++ libtheora-1.0alpha3.patched/include/theora/Makefile.am	2004-03-29 16:20:56.000000000 +0200
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-includedir = $(prefix)/include/theora
+theoraincludedir = $(includedir)/theora
 
-include_HEADERS = theora.h
+theorainclude_HEADERS = theora.h
 
diff -Naur libtheora-1.0alpha3/include/theora/Makefile.in libtheora-1.0alpha3.patched/include/theora/Makefile.in
--- libtheora-1.0alpha3/include/theora/Makefile.in	2004-03-20 06:43:24.000000000 +0100
+++ libtheora-1.0alpha3.patched/include/theora/Makefile.in	2004-03-29 16:29:24.000000000 +0200
@@ -128,7 +128,7 @@
 host_os = @host_os@
 host_vendor = @host_vendor@
 
-includedir = $(prefix)/include/theora
+includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
 libdir = @libdir@
@@ -147,18 +147,19 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 
+theoraincludedir = $(includedir)/theora
 AUTOMAKE_OPTIONS = foreign
 
-include_HEADERS = theora.h
+theorainclude_HEADERS = theora.h
 subdir = include/theora
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 DIST_SOURCES =
-HEADERS = $(include_HEADERS)
+HEADERS = $(theorainclude_HEADERS)
 
-DIST_COMMON = $(include_HEADERS) $(srcdir)/Makefile.in Makefile.am
+DIST_COMMON = $(theorainclude_HEADERS) $(srcdir)/Makefile.in Makefile.am
 all: all-am
 
 .SUFFIXES:
@@ -177,23 +178,23 @@
 distclean-libtool:
 	-rm -f libtool
 uninstall-info-am:
-includeHEADERS_INSTALL = $(INSTALL_HEADER)
-install-includeHEADERS: $(include_HEADERS)
+theoraincludeHEADERS_INSTALL = $(INSTALL_HEADER)
+install-theoraincludeHEADERS: $(theorainclude_HEADERS)
 	@$(NORMAL_INSTALL)
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	@list='$(include_HEADERS)'; for p in $$list; do \
+	$(mkinstalldirs) $(DESTDIR)$(theoraincludedir)
+	@list='$(theorainclude_HEADERS)'; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  f="`echo $$p | sed -e 's|^.*/||'`"; \
-	  echo " $(includeHEADERS_INSTALL) $$d$$p $(DESTDIR)$(includedir)/$$f"; \
-	  $(includeHEADERS_INSTALL) $$d$$p $(DESTDIR)$(includedir)/$$f; \
+	  echo " $(theoraincludeHEADERS_INSTALL) $$d$$p $(DESTDIR)$(theoraincludedir)/$$f"; \
+	  $(theoraincludeHEADERS_INSTALL) $$d$$p $(DESTDIR)$(theoraincludedir)/$$f; \
 	done
 
-uninstall-includeHEADERS:
+uninstall-theoraincludeHEADERS:
 	@$(NORMAL_UNINSTALL)
-	@list='$(include_HEADERS)'; for p in $$list; do \
+	@list='$(theorainclude_HEADERS)'; for p in $$list; do \
 	  f="`echo $$p | sed -e 's|^.*/||'`"; \
-	  echo " rm -f $(DESTDIR)$(includedir)/$$f"; \
-	  rm -f $(DESTDIR)$(includedir)/$$f; \
+	  echo " rm -f $(DESTDIR)$(theoraincludedir)/$$f"; \
+	  rm -f $(DESTDIR)$(theoraincludedir)/$$f; \
 	done
 
 ETAGS = etags
@@ -286,7 +287,7 @@
 all-am: Makefile $(HEADERS)
 
 installdirs:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
+	$(mkinstalldirs) $(DESTDIR)$(theoraincludedir)
 install: install-am
 install-exec: install-exec-am
 install-data: install-data-am
@@ -328,7 +329,7 @@
 
 info-am:
 
-install-data-am: install-includeHEADERS
+install-data-am: install-theoraincludeHEADERS
 
 install-exec-am:
 
@@ -354,18 +355,18 @@
 
 ps-am:
 
-uninstall-am: uninstall-includeHEADERS uninstall-info-am
+uninstall-am: uninstall-theoraincludeHEADERS uninstall-info-am
 
 .PHONY: CTAGS GTAGS all all-am check check-am clean clean-generic \
 	clean-libtool ctags distclean distclean-generic \
 	distclean-libtool distclean-tags distdir dvi dvi-am info \
 	info-am install install-am install-data install-data-am \
-	install-exec install-exec-am install-includeHEADERS \
+	install-exec install-exec-am install-theoraincludeHEADERS \
 	install-info install-info-am install-man install-strip \
 	installcheck installcheck-am installdirs maintainer-clean \
 	maintainer-clean-generic mostlyclean mostlyclean-generic \
 	mostlyclean-libtool pdf pdf-am ps ps-am tags uninstall \
-	uninstall-am uninstall-includeHEADERS uninstall-info-am
+	uninstall-am uninstall-theoraincludeHEADERS uninstall-info-am
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
