diff -Naupr grip-3.1.7.orig/src/grip.c grip-3.1.7/src/grip.c
--- grip-3.1.7.orig/src/grip.c	2004-03-22 18:02:30.000000000 +0100
+++ grip-3.1.7/src/grip.c	2004-03-23 15:08:04.251836120 +0100
@@ -805,7 +805,7 @@ static void DoLoadConfig(GripInfo *ginfo
   ginfo->using_builtin_cdp=FALSE;
 #endif
   ginfo->in_rip_thread=FALSE;
-  strcpy(ginfo->ripfileformat,"~/mp3/%A/%d/%n.wav");
+  strcpy(ginfo->ripfileformat,"~/ogg/%A/%d/%n.wav");
 #ifdef __linux__
   FindExeInPath("cdparanoia", ginfo->ripexename, sizeof(ginfo->ripexename));
   strcpy(ginfo->ripcmdline,"-d %c %t:[.%s]-%t:[.%e] %w");
@@ -829,9 +829,9 @@ static void DoLoadConfig(GripInfo *ginfo
   *ginfo->wav_filter_cmd='\0';
   *ginfo->disc_filter_cmd='\0';
   ginfo->selected_encoder=1;
-  strcpy(ginfo->mp3cmdline,"-h -b %b %w %m");
-  FindExeInPath("lame", ginfo->mp3exename, sizeof(ginfo->mp3exename));
-  strcpy(ginfo->mp3fileformat,"~/mp3/%A/%d/%n.%x");
+  strcpy(ginfo->mp3cmdline,"-o %m -a %a -l %d -t %n -N %t -G %G -d %y -q 4 %w");
+  FindExeInPath("oggenc", ginfo->mp3exename, sizeof(ginfo->mp3exename));
+  strcpy(ginfo->mp3fileformat,"~/ogg/%A/%d/%n.%x");
   strcpy(ginfo->mp3extension,"mp3");
   ginfo->mp3nice=0;
   *ginfo->mp3_filter_cmd='\0';
@@ -839,7 +839,7 @@ static void DoLoadConfig(GripInfo *ginfo
   ginfo->add_to_db=FALSE;
   ginfo->add_m3u=TRUE;
   ginfo->rel_m3u=TRUE;
-  strcpy(ginfo->m3ufileformat,"~/mp3/%A-%d.m3u");
+  strcpy(ginfo->m3ufileformat,"~/ogg/%A-%d.m3u");
   ginfo->kbits_per_sec=128;
   ginfo->edit_num_cpu=1;
   ginfo->doid3=TRUE;
diff -Naupr grip-3.1.7.orig/src/gripcfg.c grip-3.1.7/src/gripcfg.c
--- grip-3.1.7.orig/src/gripcfg.c	2004-03-12 17:59:38.000000000 +0100
+++ grip-3.1.7/src/gripcfg.c	2004-03-23 15:11:09.065740136 +0100
@@ -46,16 +46,16 @@ static Ripper ripper_defaults[]={
   {"",""}
 };
 
-static MP3Encoder encoder_defaults[]={{"bladeenc","-%b -QUIT %w %m","mp3"},
+static MP3Encoder encoder_defaults[]={{"oggenc",
+				       "-o %m -a %a -l %d -t %n -N %t -G %G -d %y -q 4 %w",
+				       "ogg"},
 				      {"lame","-h -b %b %w %m","mp3"},
 				      {"l3enc","-br %b %w %m","mp3"},
 				      {"xingmp3enc","-B %b -Q %w","mp3"},
 				      {"mp3encode","-p 2 -l 3 -b %b %w %m",
 				       "mp3"},
 				      {"gogo","-b %b %w %m","mp3"},
-				      {"oggenc",
-				       "-o %m -a %a -l %d -t %n -b %b %w -N %t -G %g -d %y",
-				       "ogg"},
+				      {"bladeenc","-%b -QUIT %w %m","mp3"},
 				      {"flac","-V -o %m %w","flac"},
 				      {"other",""},
 				      {"",""}
