diff -urN razor-agents-1.20-orig/etc/razor.conf razor-agents-1.20/etc/razor.conf
--- razor-agents-1.20-orig/etc/razor.conf	Wed Dec 31 19:00:00 1969
+++ razor-agents-1.20/etc/razor.conf	Fri May  3 12:44:25 2002
@@ -0,0 +1,53 @@
+#########################################################################
+# Copyright @2002 - Vipul Ved Prakash, All rights reserved.
+#
+# Project     : Razor
+# Component   : etc/razor.conf
+# Author      : Michael McLagan <Michael.McLagan@linux.org>
+# Creation    : 03-May-2002 11:02
+# Description : razor.conf is a configuration file read by Razor::Config
+#               and used by razor-report, razor-check and razor-discover. 
+#               The format is "attribute = value" pairs, one pair per line. 
+#               Lines that begin with `#' are treated as comments.
+#
+#               Installed as /etc/mail/razor/razor.conf in RedHat RPM.  
+#
+#               See razor.conf(5) for a complete list of attributes.
+#               
+# Current Revision:
+#
+# $Source: /cvsroot/rap/rpms/razor-agents/razor-agents-redhat.patch,v $
+# $Revision: 1.1 $
+# $Author: dag- $
+# $Date: 2004/02/24 23:49:34 $
+#
+# Revision History:
+# 
+# $Log: razor-agents-redhat.patch,v $
+# Revision 1.1  2004/02/24 23:49:34  dag-
+# Initial commit.
+#
+##########################################################################
+
+# The default zone is razor.vipul.net
+# razorzone=razor.vipul.net
+
+# A list of available servers in the zone, see rediscovery_wait for frequency.
+# This file needs to be writable by the process using Razor if discovery is
+# permitted
+listfile=/etc/mail/razor/zone
+
+# Turned off by default
+# debug=0
+
+# The frequency between attempts to find the nearest server in the
+# razor zone.  The default is 2 days, the value is in seconds.
+# rediscovery_wait=172800
+
+# To stop the discovery process, set this to 1.
+# turn_off_discovery=0
+
+# A list of addresses & hashes to accept without checking servers
+# See razor-whitelist(5) manpage for the format
+# whitelist=/etc/mail/razor/whitelist
+
+# To turn off automatic mailing list ignoring, set this to 1
+# turn_off_ignorelist=0
+
diff -urN razor-agents-1.20-orig/etc/whitelist razor-agents-1.20/etc/whitelist
--- razor-agents-1.20-orig/etc/whitelist	Wed Dec 31 19:00:00 1969
+++ razor-agents-1.20/etc/whitelist	Fri May  3 12:43:05 2002
@@ -0,0 +1,39 @@
+#########################################################################
+# Copyright @2002 - Vipul Ved Prakash, All rights reserved.
+#
+# Project     : Razor
+# Component   : etc/whitelist
+# Author      : Michael McLagan <Michael.McLagan@linux.org>
+# Creation    : 03-May-2002 12:26
+# Description : Razor whitelist file.  Contains addresses and SHA1
+#               hashes of email messages that should not be considered
+#               spam messages.  Installed as /etc/mail/razor/whitelist
+#               and referenced in the default razor.conf file in the
+#               same directory.
+#
+#               See razor-whitelist(5) manpage for more information.
+#
+# Current Revision:
+#
+# $Source: /cvsroot/rap/rpms/razor-agents/razor-agents-redhat.patch,v $
+# $Revision: 1.1 $
+# $Author: dag- $
+# $Date: 2004/02/24 23:49:34 $
+#
+# Revision History:
+# 
+# $Log: razor-agents-redhat.patch,v $
+# Revision 1.1  2004/02/24 23:49:34  dag-
+# Initial commit.
+#
+##########################################################################
+
+# Here are some samples taken from the man page.  You should customize
+# this for your installation!  One per line, regex allowed.
+
+# to bugtraq@securityfocus.com
+# cc a.list@somewhere.net
+# from .*@mydomain.com
+
+# A message with body 'test'.  You can generate these by piping the message
+# body thru sha1sum found in the textutils package
+#
+# sha1 75f8bcc2357366bbfa9c6ab0b6e5648ed0cf7083
+
diff -urN razor-agents-1.20-orig/lib/Razor/Config.pm razor-agents-1.20/lib/Razor/Config.pm
--- razor-agents-1.20-orig/lib/Razor/Config.pm	Sat Feb 23 06:13:24 2002
+++ razor-agents-1.20/lib/Razor/Config.pm	Fri May  3 12:25:47 2002
@@ -99,7 +99,7 @@
 sub findconf { 
     my ($name) = @_;
     my $conf;
-    my  (@CPATHS) = ("$ENV{HOME}/etc/$name", "$ENV{HOME}/$name", "/etc/$name");
+    my  (@CPATHS) = ("$ENV{HOME}/etc/$name", "$ENV{HOME}/$name", "/etc/mail/razor/$name", "/etc/mail/$name", "/etc/$name");
     for (@CPATHS) { if (-e $_) { $conf = $_; last } }
     return $conf if $conf;
     return;
