--- razor-agents-1.20/Makefile.PL.makefile	Wed Dec 26 02:15:57 2001
+++ razor-agents-1.20/Makefile.PL	Wed Aug 14 00:07:46 2002
@@ -26,9 +26,102 @@
                        'Digest::SHA1'   => 0,  
                        'Mail::Internet' => 0, 
                      },
-    MAN3PODS      => { 
-        "man/razor.conf.pod"      => 'razor.conf',
-        "man/razor-whitelist.pod" => 'razor-whitelist.pod', 
-    },
-
 );
+
+
+##########################################################################
+# 
+# Author      : Michael McLagan <Michael.McLagan@linux.org>
+# Creation    : 03-May-2002 09:35
+# Description : The following code modifies the MakeMaker output
+#               to incorporate the building of man5 pages for the
+#               configuration files required by the Razor classes.
+#
+#               It is based on hacking up the output of ExtUtils::MakeMaker
+#               version 5.45 (came with perl 5.6.1) and will need to be
+#               edited should the relevant sections be changed in future
+#               versions.
+# Changes     : 13-Aug-2002 Ramiro Morales <rm-rpms@gmx.net>
+#               Modify so it's more independent of the Perl version:
+#               Dont hardwire the Perl version in the paths to perl.h,
+#               use PERL_INC instead; make the MY package replacements
+#               smarter.
+#
+##########################################################################
+
+sub MY::constants
+{
+  package MY;
+  my $self = shift;
+  my $inherited = $self->SUPER::constants(@_);
+
+  my $pods = "\nMAN5PODS = man/razor.conf.pod \\\n" .
+             "           man/razor-whitelist.pod";
+
+  $inherited =~ s/^(MAN3EXT = 3)/$1$pods/gm;
+
+  my $man5 = "INST_MAN5DIR = blib/man5\n" . 
+             'INSTALLMAN5DIR = $(PREFIX)/share/man/man5' . "\n" . 
+             "MAN5EXT = 5\n";
+
+  $inherited =~ s/^(PERM_RW = )/$man5$1/gm;
+
+  $inherited;
+}
+
+sub MY::top_targets 
+{
+  package MY;
+  my $self = shift;
+  my $inherited = $self->SUPER::top_targets(@_);
+
+  my $man5 = 'config :: $(INST_MAN5DIR)/.exists' . "\n" . 
+             '	@$(NOOP)' . "\n\n\n" . 
+             '$(INST_MAN5DIR)/.exists :: $(PERL_INC)/perl.h' . "\n" . 
+             '	@$(MKPATH) $(INST_MAN5DIR)' . "\n" . 
+             '	@$(EQUALIZE_TIMESTAMP) $(PERL_INC)/perl.h $(INST_MAN5DIR)/.exists' . "\n" . 
+             '	-@$(CHMOD) $(PERM_RWX) $(INST_MAN5DIR)' . "\n\n";
+
+  $inherited =~ s/^(help:)/$man5$1/gm;
+
+  $inherited;
+}
+
+sub MY::manifypods
+{
+  package MY;
+  my $self = shift;
+  my $inherited = $self->SUPER::manifypods(@_);
+
+  my $deps = " \\\n" . 
+             "	man/razor.conf.pod \\\n" . 
+             "	man/razor-whitelist.pod";
+ 
+  $inherited =~ s/(\.pm)$/$1$deps/gm;
+
+  my $files = ' \\' . "\n" . 
+              '	man/razor.conf.pod \\' . "\n" . 
+              '	$(INST_MAN5DIR)/razor.conf.$(MAN5EXT) \\' . "\n" .
+              '	man/razor-whitelist.pod \\' . "\n" . 
+              '	$(INST_MAN5DIR)/razor-whitelist.$(MAN5EXT)';
+
+  $inherited =~ s/(\$\(MAN[13]EXT\))$/$1$files/gm;
+
+  $inherited;
+
+}
+
+sub MY::install 
+{
+  package MY;
+  my $self = shift;
+  my $inherited = $self->SUPER::install(@_);
+
+  my $man5 = " \\\n" . 
+             '		$(INST_MAN5DIR) $(INSTALLMAN5DIR)';
+ 
+  $inherited =~ s/(\$\(INSTALLMAN3DIR\))$/$1$man5/gm;
+
+  $inherited;
+}
+
