--- rdesktop-1.6.0/xclip.c.orig	2008-09-04 08:56:44.000000000 +0200
+++ rdesktop-1.6.0/xclip.c	2008-09-04 08:57:19.000000000 +0200
@@ -156,7 +156,7 @@
 utf16_lf2crlf(uint8 * data, uint32 * size)
 {
 	uint8 *result;
-	uint16 *inptr, *outptr;
+	uint16 *inptr, *outptr, uvalue_previous;
 	RD_BOOL swap_endianess;
 
 	/* Worst case: Every char is LF */
@@ -170,7 +170,7 @@
 	/* Check for a reversed BOM */
 	swap_endianess = (*inptr == 0xfffe);
 
-	uint16 uvalue_previous = 0;	/* Kept so we'll avoid translating CR-LF to CR-CR-LF */
+	uvalue_previous = 0;	/* Kept so we'll avoid translating CR-LF to CR-CR-LF */
 	while ((uint8 *) inptr < data + *size)
 	{
 		uint16 uvalue = *inptr;
