diff -Naupr libmpeg3-1.6.orig/Makefile libmpeg3-1.6/Makefile
--- libmpeg3-1.6.orig/Makefile	2005-08-28 05:21:13.000000000 +0200
+++ libmpeg3-1.6/Makefile	2006-01-19 17:33:15.000000000 +0100
@@ -9,6 +9,10 @@ ifeq ("$(PREFIX)", "")
 PREFIX=/usr
 endif
 
+ifeq ("$(LIBDIR)", "")
+LIBDIR=$(PREFIX)/lib
+endif
+
 ifeq ($(origin CFLAGS), environment)
 HAVE_CFLAGS := y
 else
@@ -16,36 +20,13 @@ HAVE_CFLAGS := n
 endif
 
 
+ifeq ("$(OBJDIR)", "")
 OBJDIR := $(shell uname --machine)
-
-
-
-
-ifeq ($(OBJDIR), alpha)
-  USE_MMX = 0
-  ifneq ($(HAVE_CFLAGS), y)
-    CFLAGS := -O4 -arch ev67 -ieee -accept c99_keywords -gcc_messages
-  endif
-endif
-
-ifeq ($(OBJDIR), i686)
-  USE_MMX = 1
-  ifneq ($(HAVE_CFLAGS), y)
-    CFLAGS := -O2 -fomit-frame-pointer -falign-loops=2 -falign-jumps=2 -falign-functions=2 -I/usr/local/include
-  endif
-  CFLAGS += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-endif
-
-ifeq ($(OBJDIR), x86_64)
-  ifneq ($(HAVE_CFLAGS), y)
-    CFLAGS := -O2 -fomit-frame-pointer -I/usr/local/include
-  endif
-  CFLAGS += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-
-
 endif
 
 
+CFLAGS += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
+
 
 ifeq ($(USE_CSS), 1)
   CFLAGS += -DHAVE_CSS
@@ -188,9 +169,12 @@ $(OBJDIR)/mpeg2qt: $(OUTPUT)
 		-ldl
 
 install: 
-	cp $(UTILS) $(PREFIX)/bin
-#	cp $(OUTPUT) $(PREFIX)/lib
-#	cp libmpeg3.h mpeg3private.h $(PREFIX)/include
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	cp $(UTILS) $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(LIBDIR)
+	cp $(OUTPUT) $(DESTDIR)$(LIBDIR)
+	mkdir -p $(DESTDIR)$(PREFIX)/include
+	cp libmpeg3.h mpeg3private.h $(DESTDIR)$(PREFIX)/include
 
 clean:
 	rm -rf $(OBJDIR)
