diff -ur t1lib-5.1.0/type1afm/Makefile.in t1lib-5.1.0.new/type1afm/Makefile.in
--- t1lib-5.1.0/type1afm/Makefile.in	2004-12-12 18:45:12.000000000 -0800
+++ t1lib-5.1.0.new/type1afm/Makefile.in	2005-09-27 09:42:50.000000000 -0700
@@ -89,7 +89,7 @@
 install: dummy
 	$(MKINSTALLDIRS) $(DESTDIR)$(bindir)
 	$(LIBTOOL) --mode=install \
-		$(INSTALL_PROGRAM) type1afm $(bindir)/type1afm
+		$(INSTALL_PROGRAM) type1afm $(DESTDIR)$(bindir)/type1afm
 
 
 uninstall: dummy
