diff -Naupr bmpx-0.20.0.orig/src/Makefile.am bmpx-0.20.0/src/Makefile.am
--- bmpx-0.20.0.orig/src/Makefile.am	2006-07-09 02:11:03.000000000 +0200
+++ bmpx-0.20.0/src/Makefile.am	2006-07-10 09:39:35.000000000 +0200
@@ -294,5 +294,5 @@ CLEANFILES = \
 DISTCLEANFILES = $(CLEANFILES)
 
 install-exec-hook: 
-	$(LN_S) -f $(DESTDIR)$(bindir)/beep-media-player-2$(EXEEXT) \
+	$(LN_S) -f beep-media-player-2$(EXEEXT) \
        		   $(DESTDIR)$(bindir)/bmp2$(EXEEXT)
diff -Naupr bmpx-0.20.0.orig/src/Makefile.in bmpx-0.20.0/src/Makefile.in
--- bmpx-0.20.0.orig/src/Makefile.in	2006-07-09 02:42:02.000000000 +0200
+++ bmpx-0.20.0/src/Makefile.in	2006-07-10 09:39:44.000000000 +0200
@@ -1727,7 +1727,7 @@ bmp-marshalers.h: bmp-marshalers.list
 @ENABLE_TESTING_CODE_TRUE@	@rm -f service_core_glue.h.temp
 
 install-exec-hook: 
-	$(LN_S) -f $(DESTDIR)$(bindir)/beep-media-player-2$(EXEEXT) \
+	$(LN_S) -f beep-media-player-2$(EXEEXT) \
        		   $(DESTDIR)$(bindir)/bmp2$(EXEEXT)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
