diff -Naupr bmpx-0.30.3.orig/src/Makefile.am bmpx-0.30.3/src/Makefile.am
--- bmpx-0.30.3.orig/src/Makefile.am	2006-09-25 21:23:02.000000000 +0200
+++ bmpx-0.30.3/src/Makefile.am	2006-09-26 11:25:54.000000000 +0200
@@ -315,5 +315,5 @@ CLEANFILES = \
 DISTCLEANFILES = $(CLEANFILES)
 
 install-exec-hook: 
-	$(LN_S) -f $(DESTDIR)$(bindir)/beep-media-player-2$(EXEEXT) \
+	$(LN_S) -f beep-media-player-2$(EXEEXT) \
 		$(DESTDIR)$(bindir)/bmp2$(EXEEXT)
diff -Naupr bmpx-0.30.3.orig/src/Makefile.in bmpx-0.30.3/src/Makefile.in
--- bmpx-0.30.3.orig/src/Makefile.in	2006-09-25 21:59:42.000000000 +0200
+++ bmpx-0.30.3/src/Makefile.in	2006-09-26 11:33:15.000000000 +0200
@@ -1752,7 +1752,7 @@ service_core_glue.h: service-core.xml
 	@rm -f service_core_glue.h.temp.2
 
 install-exec-hook: 
-	$(LN_S) -f $(DESTDIR)$(bindir)/beep-media-player-2$(EXEEXT) \
+	$(LN_S) -f beep-media-player-2$(EXEEXT) \
 		$(DESTDIR)$(bindir)/bmp2$(EXEEXT)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
