diff -crB spamass-milter-0.3.1.orig/spamass-milter.cpp spamass-milter-0.3.1/spamass-milter.cpp
*** spamass-milter-0.3.1.orig/spamass-milter.cpp        2006-03-24 08:41:36.000000000 +1100
--- spamass-milter-0.3.1/spamass-milter.cpp     2010-02-12 03:59:16.000000000 +1100 ***************
*** 775,780 ****
--- 775,784 ----
    SpamAssassin* assassin;
    struct context *sctx = (struct context *)smfi_getpriv(ctx);
    char *queueid;
+   if (smfi_getsymval (ctx, "{auth_type}") != NULL)
+   {
+     return SMFIS_ACCEPT;
+   }

    if (sctx == NULL)
    {
diff -crB spamass-milter-0.3.1.orig/spamass-milter.cpp spamass-milter-0.3.1/spamass-milter.cpp
*** spamass-milter-0.3.1.orig/spamass-milter.cpp	2006-03-24 08:41:36.000000000 +1100
--- spamass-milter-0.3.1/spamass-milter.cpp	2010-05-07 13:18:54.000000000 +1000
***************
*** 776,781 ****
--- 776,788 ----
    struct context *sctx = (struct context *)smfi_getpriv(ctx);
    char *queueid;
  
+   // spamass-milter-0.3.1-smtp-auth-bypass.patch start.
+   if (smfi_getsymval (ctx, "{auth_type}") != NULL)
+   {
+     return SMFIS_ACCEPT;
+   }
+   // spamass-milter-0.3.1-smtp-auth-bypass.patch end.
+ 
    if (sctx == NULL)
    {
      debug(D_ALWAYS, "smfi_getpriv failed!");
