--- dosemu-1.1.4/src/arch/linux/Makefile.main.orig	Wed Dec 18 03:36:44 2002
+++ dosemu-1.1.4/src/arch/linux/Makefile.main	Fri Dec 27 12:56:27 2002
@@ -8,7 +8,7 @@
 # You should do a "make" to compile and a "make install" as root to
 # install DOSEMU.
 
-INSTALL = install -c -o root
+INSTALL = install -c
 TMP = ../tmp
 COM = ../src/plugin/commands
 SYS = ../commands
@@ -273,11 +273,11 @@
 install:
 	rm -rf $(TMP)
 	mkdir -p -m 0755 $(TMP)/dosemu/freedos/dosemu
-	if [ ! -d $(dosemudir) ]; then \
-	  $(INSTALL) -d $(dosemudir); \
+	if [ ! -d $(INSTROOT)$(dosemudir) ]; then \
+	  $(INSTALL) -d $(INSTROOT)$(dosemudir); \
 	fi
-	if [ ! -d $(syshdimagedir) ]; then \
-	  $(INSTALL) -d $(syshdimagedir); \
+	if [ ! -d $(INSTROOT)$(syshdimagedir) ]; then \
+	  $(INSTALL) -d $(INSTROOT)$(syshdimagedir); \
 	fi
 	for i in aspi.sys cdrom.sys dumpconf.sys ems.sys emufs.sys fossil.com isemu.com mgarrot.com; do \
 	  $(INSTALL) -m 0644 $(SYS)/$$i $(TMP)/dosemu/freedos/dosemu; \
@@ -286,93 +286,92 @@
 	for i in `(cd $(COM); for i in *.com; do if [ -L $$i ]; then echo -n " $$i";fi;done)`; do \
 	  ln -sf generic.com $(TMP)/dosemu/freedos/dosemu/$$i; \
 	done
-	cp -a $(TMP)/dosemu/freedos $(dosemudir)
+	cp -a $(TMP)/dosemu/freedos $(INSTROOT)$(dosemudir)
 	cd $(TMP)/dosemu; \
 	for i in `find freedos -type f` ; do \
 	  ln -snf $(dosemudir)/$$i $$i; \
 	done
 	tar -cf- -C $(TMP) dosemu/freedos | gzip -9 > $(TMP)/dosemu/dosemu-bin.tgz
-	$(INSTALL) -m 0644 $(TMP)/dosemu/dosemu-bin.tgz $(dosemudir)
+	$(INSTALL) -m 0644 $(TMP)/dosemu/dosemu-bin.tgz $(INSTROOT)$(dosemudir)
 	rm -rf $(TMP)
 	mkdir -p -m 0755 $(TMP)/dosemu/freedos/dosemu
 	if [ -f ../$(fdtarball) ]; then \
-	  tar -C $(dosemudir)/.. -xpzf ../dosemu-freedos-bin.tgz; \
+	  tar -C $(INSTROOT)$(dosemudir)/.. -xpzf ../dosemu-freedos-bin.tgz; \
 	  tar -C $(TMP) -xpzf ../dosemu-freedos-bin.tgz; \
 	  cd $(TMP)/dosemu; \
-	  rm -f FDchange.log $(dosemudir)/FDchange.log; \
-	  rm -f README.bindist $(dosemudir)/README.bindist; \
+	  rm -f FDchange.log $(INSTROOT)$(dosemudir)/FDchange.log; \
+	  rm -f README.bindist $(INSTROOT)$(dosemudir)/README.bindist; \
 	  for i in `find freedos -type f | grep -v autoexec.bat | grep -v config.sys` ; do \
-	    ln -snf $(dosemudir)/$$i $$i; \
+	    ln -snf $(INSTROOT)$(dosemudir)/$$i $$i; \
 	  done; \
-	  $(INSTALL) $(dosemudir)/freedos/config.sys freedos; \
-	  $(INSTALL) $(dosemudir)/freedos/autoexec.bat freedos; \
-	  cd .. ; tar -cf- dosemu |gzip -9 >$(dosemudir)/dosemu-freedos-bin.tgz; \
+	  $(INSTALL) $(INSTROOT)$(dosemudir)/freedos/config.sys freedos; \
+	  $(INSTALL) $(INSTROOT)$(dosemudir)/freedos/autoexec.bat freedos; \
+	  cd .. ; tar -cf- dosemu |gzip -9 >$(INSTROOT)$(dosemudir)/dosemu-freedos-bin.tgz; \
 	fi
 	sed -e "s|SYSTEM_INSTALL_PATH=NOT_SYSTEM_WIDE|SYSTEM_INSTALL_PATH=$(dosemudir)|" \
 	     -e "s|SYSTEM_CONF_PATH=NOT_SYSTEM_WIDE|SYSTEM_CONF_PATH=$(sysconfdir)|" \
 	     -e "s|SYSTEM_XFONTS_PATH=NOT_SYSTEM_WIDE|SYSTEM_XFONTS_PATH=$(x11fontdir)|" \
 	              ../dist/dosemu > $(TMP)/dosemu/dosemu
-	if [ ! -d $(sysconfdir) ]; then \
-	  $(INSTALL) -d $(sysconfdir); \
+	if [ ! -d $(INSTROOT)$(sysconfdir) ]; then \
+	  $(INSTALL) -d $(INSTROOT)$(sysconfdir); \
 	fi
-	if [ ! -f $(sysconfdir)/dosemu.conf ]; then \
-	  $(INSTALL) -m 0644 ../etc/dosemu.conf ../etc/dosemu.users.easy ../etc/dosemu.users.secure $(sysconfdir); \
-	  $(INSTALL) -m 0644 ../etc/global.conf $(sysconfdir)/global.conf.example; \
+	if [ ! -f $(INSTROOT)$(sysconfdir)/dosemu.conf ]; then \
+	  $(INSTALL) -m 0644 ../etc/dosemu.conf ../etc/dosemu.users.easy ../etc/dosemu.users.secure $(INSTROOT)$(sysconfdir); \
+	  $(INSTALL) -m 0644 ../etc/global.conf $(INSTROOT)$(sysconfdir)/global.conf.example; \
 	fi
-	if [ ! -d $(dosemudir)/keymap ]; then \
-	  $(INSTALL) -d $(dosemudir)/keymap; \
-	  $(INSTALL) -m 0644 ../etc/keymap/* $(dosemudir)/keymap; \
+	if [ ! -d $(INSTROOT)$(dosemudir)/keymap ]; then \
+	  $(INSTALL) -d $(INSTROOT)$(dosemudir)/keymap; \
+	  $(INSTALL) -m 0644 ../etc/keymap/* $(INSTROOT)$(dosemudir)/keymap; \
 	fi
-	$(INSTALL) -m 0755 ../bin/$(DOSBIN) $(bindir)
-	$(INSTALL) -m 0755 $(TMP)/dosemu/dosemu $(bindir)
-	$(INSTALL) -m 0644 ../bin/dosemu.map $(dosemudir)/dosemu-$(THISVERSION).map
-	$(INSTALL) -m 0755 ../bin/mkfatimage $(bindir)
-	$(INSTALL) -m 0755 ../bin/mkfatimage16 $(bindir)
+	$(INSTALL) -m 0755 ../bin/$(DOSBIN) $(INSTROOT)$(bindir)
+	$(INSTALL) -m 0755 $(TMP)/dosemu/dosemu $(INSTROOT)$(bindir)
+	$(INSTALL) -m 0644 ../bin/dosemu.map $(INSTROOT)$(dosemudir)/dosemu-$(THISVERSION).map
+	$(INSTALL) -m 0755 ../bin/mkfatimage $(INSTROOT)$(bindir)
+	$(INSTALL) -m 0755 ../bin/mkfatimage16 $(INSTROOT)$(bindir)
 ifdef USE_SBEMU
-	$(INSTALL) -m 0755 ../bin/midid $(bindir)
+	$(INSTALL) -m 0755 ../bin/midid $(INSTROOT)$(bindir)
 endif
 ifdef  USE_MHPDBG
-	$(INSTALL) -m 0755 ../bin/dosdebug $(bindir)
+	$(INSTALL) -m 0755 ../bin/dosdebug $(INSTROOT)$(bindir)
 endif
-	if [ -f $(bindir)/xdosemu ]; then \
-		$(INSTALL) -m 0700 $(bindir)/xdosemu /tmp; \
+	if [ -f $(INSTROOT)$(bindir)/xdosemu ]; then \
+		$(INSTALL) -m 0700 $(INSTROOT)$(bindir)/xdosemu /tmp; \
 		rm -f $(bindir)/xdosemu; \
 	fi
-	if [ ! -d $(docdir) ]; then \
-	  $(INSTALL) -d $(docdir); \
+	if [ ! -d $(INSTROOT)$(docdir) ]; then \
+	  $(INSTALL) -d $(INSTROOT)$(docdir); \
 	fi
-	$(INSTALL) -m 0644 ../README.bindist $(docdir)
+	$(INSTALL) -m 0644 ../README.bindist $(INSTROOT)$(docdir)
 	$(INSTALL) -m 0644 ../doc/{README,README-tech,dosemu-HOWTO,sound-usage}.txt \
-	   ../doc/announce $(docdir)
-	if [ ! -d $(mandir)/man1 ]; then \
-	  $(INSTALL) -d $(mandir)/man1; \
+	   ../doc/announce $(INSTROOT)$(docdir)
+	if [ ! -d $(INSTROOT)$(mandir)/man1 ]; then \
+	  $(INSTALL) -d $(INSTROOT)$(mandir)/man1; \
 	fi
-	$(INSTALL) -m 0644 ../man/{mkfatimage16,dosemu,dosemu.bin,dosdebug}.1 \
-	   $(mandir)/man1
-	ln -sf dosemu.1 $(mandir)/man1/xdosemu.1
+	$(INSTALL) -m 0644 ../man/*.1 \
+	   $(INSTROOT)$(mandir)/man1
 ifdef X_SUPPORT
-	if [ ! -e $(prefix)/xdosemu ]; then ln -s dosemu $(bindir)/xdosemu; fi
-	if [ ! -d $(x11fontdir) ]; then \
-	  $(INSTALL) -d $(x11fontdir); \
+	if [ ! -e $(INSTROOT)$(prefix)/xdosemu ]; then ln -s dosemu $(INSTROOT)$(bindir)/xdosemu; fi
+	if [ ! -d $(INSTROOT)$(x11fontdir) ]; then \
+	  $(INSTALL) -d $(INSTROOT)$(x11fontdir); \
 	fi
-	if [ -w $(x11fontdir) ] && \
-	    [ -d $(X11fontdir) ] && \
-	    [ ! -e $(x11fontdir)/vga.pcf ] && \
-	    [ ! -e $(x11fontdir)/vga.pcf.Z ] && \
-	    [ ! -e $(x11fontdir)/vga.pcf.gz ]; then \
+	if [ -w $(INSTROOT)$(x11fontdir) ] && \
+	    [ -d $(INSTROOT)$(X11fontdir) ] && \
+	    [ ! -e $(INSTROOT)$(x11fontdir)/vga.pcf ] && \
+	    [ ! -e $(INSTROOT)$(x11fontdir)/vga.pcf.Z ] && \
+	    [ ! -e $(INSTROOT)$(x11fontdir)/vga.pcf.gz ]; then \
 		echo "-> Main DOSEMU files installation done. Installing the X PC fonts..."; \
-		$(INSTALL) -m 0644 ../etc/vga.pcf $(x11fontdir); \
-		gzip -9 $(x11fontdir)/vga.pcf; \
-		bdftopcf ../etc/vga11x19.bdf | gzip -c -9 > $(x11fontdir)/vga11x19.pcf.gz; \
-		bdftopcf ../etc/vga-cp866.bdf | gzip -c -9 > $(x11fontdir)/vga-cp866.pcf.gz; \
-		bdftopcf ../etc/vga10x20-cp866.bdf | gzip -c -9 > $(x11fontdir)/vga10x20-cp866.pcf.gz; \
-		chmod 0644 $(x11fontdir)/vga11x19.pcf.gz $(x11fontdir)/vga-cp866.pcf.gz $(x11fontdir)/vga10x20-cp866.pcf.gz; \
-		if [ ! -f $(x11fontdir)/fonts.alias ]; then \
-		  $(INSTALL) -m 0644 ../etc/dosemu.alias $(x11fontdir)/fonts.alias; \
-		elif [ grep ^vga $(x11fontdir)/fonts.alias ]; then \
-	          cat ../etc/dosemu.alias >> $(x11fontdir)/fonts.alias; \
+		$(INSTALL) -m 0644 ../etc/vga.pcf $(INSTROOT)$(x11fontdir); \
+		gzip -9 $(INSTROOT)$(x11fontdir)/vga.pcf; \
+		bdftopcf ../etc/vga11x19.bdf | gzip -c -9 > $(INSTROOT)$(x11fontdir)/vga11x19.pcf.gz; \
+		bdftopcf ../etc/vga-cp866.bdf | gzip -c -9 > $(INSTROOT)$(x11fontdir)/vga-cp866.pcf.gz; \
+		bdftopcf ../etc/vga10x20-cp866.bdf | gzip -c -9 > $(INSTROOT)$(x11fontdir)/vga10x20-cp866.pcf.gz; \
+		chmod 0644 $(INSTROOT)$(x11fontdir)/vga11x19.pcf.gz $(INSTROOT)$(x11fontdir)/vga-cp866.pcf.gz $(INSTROOT)$(x11fontdir)/vga10x20-cp866.pcf.gz; \
+		if [ ! -f $(INSTROOT)$(x11fontdir)/fonts.alias ]; then \
+		  $(INSTALL) -m 0644 ../etc/dosemu.alias $(INSTROOT)$(x11fontdir)/fonts.alias; \
+		elif [ grep ^vga $(INSTROOT)$(x11fontdir)/fonts.alias ]; then \
+	          cat ../etc/dosemu.alias >> $(INSTROOT)$(x11fontdir)/fonts.alias; \
 		fi; \
-		cd $(x11fontdir); \
+		cd $(INSTROOT)$(x11fontdir); \
 		mkfontdir; \
 	fi;
 endif
