diff -Naupr Video-DVDRip-0.97.6.orig/lib/Video/DVDRip/Config.pm Video-DVDRip-0.97.6/lib/Video/DVDRip/Config.pm
--- Video-DVDRip-0.97.6.orig/lib/Video/DVDRip/Config.pm	2006-01-03 20:37:38.000000000 +0100
+++ Video-DVDRip-0.97.6/lib/Video/DVDRip/Config.pm	2006-03-23 13:13:45.000000000 +0100
@@ -307,7 +307,7 @@ my %CONFIG_PARAMETER = (
 	workaround_nptl_bugs => {
 		label	=> __"Workaround transcode NPTL bugs",
 		type	=> 'switch',
-		value	=> 1,
+		value	=> 0,
 	},
 	nptl_ld_assume_kernel => {
 		label   => __"Set LD_ASSUME_KERNEL to",
