Index: kerberosgss.c
===================================================================
--- src/kerberosgss.c	(revision 1735)
+++ src/kerberosgss.c	(working copy)
@@ -188,7 +188,7 @@
 									  &state->context,
 									  state->server_name,
 									  GSS_C_NO_OID,
-									  GSS_C_MUTUAL_FLAG | GSS_C_SEQUENCE_FLAG,
+									  GSS_C_MUTUAL_FLAG | GSS_C_SEQUENCE_FLAG | GSS_C_DELEG_FLAG,
 									  0,
 									  GSS_C_NO_CHANNEL_BINDINGS,
 									  &input_token,
