diff -Naupr quake3-1.34-rc1_SVN792.orig/code/tools/asm/Makefile quake3-1.34-rc1_SVN792/code/tools/asm/Makefile
--- quake3-1.34-rc1_SVN792.orig/code/tools/asm/Makefile	2006-04-23 22:59:06.000000000 +0200
+++ quake3-1.34-rc1_SVN792/code/tools/asm/Makefile	2006-05-29 13:45:17.000000000 +0200
@@ -37,7 +37,7 @@ clean:
 	rm -f q3asm *~ *.o
 
 install: default
-	$(INSTALL) -s -m 0755 q3asm$(BINEXT) ../
+	$(INSTALL) -m 0755 q3asm$(BINEXT) ../
 
 uninstall:
 	rm -f ../q3asm$(BINEXT)
diff -Naupr quake3-1.34-rc1_SVN792.orig/code/tools/lcc/Makefile quake3-1.34-rc1_SVN792/code/tools/lcc/Makefile
--- quake3-1.34-rc1_SVN792.orig/code/tools/lcc/Makefile	2006-04-23 22:59:06.000000000 +0200
+++ quake3-1.34-rc1_SVN792/code/tools/lcc/Makefile	2006-05-29 13:45:42.000000000 +0200
@@ -140,9 +140,9 @@ $(BD)etc/%$(O): etc/%.c
 
 
 install: q3lcc q3cpp q3rcc
-	$(INSTALL) -s -m 0755 $(BD)q3lcc$(E) ../
-	$(INSTALL) -s -m 0755 $(BD)q3cpp$(E) ../
-	$(INSTALL) -s -m 0755 $(BD)q3rcc$(E) ../
+	$(INSTALL) -m 0755 $(BD)q3lcc$(E) ../
+	$(INSTALL) -m 0755 $(BD)q3cpp$(E) ../
+	$(INSTALL) -m 0755 $(BD)q3rcc$(E) ../
 
 uninstall:
 	-$(RM) ../q3lcc$(E)
diff -Naupr quake3-1.34-rc1_SVN792.orig/Makefile quake3-1.34-rc1_SVN792/Makefile
--- quake3-1.34-rc1_SVN792.orig/Makefile	2006-05-29 05:02:54.000000000 +0200
+++ quake3-1.34-rc1_SVN792/Makefile	2006-05-29 13:48:25.000000000 +0200
@@ -173,7 +173,7 @@ ifeq ($(PLATFORM),linux)
   endif
   endif
 
-  BASE_CFLAGS = -Wall -fno-strict-aliasing -Wimplicit -Wstrict-prototypes -pipe
+  BASE_CFLAGS = -Wall -fno-strict-aliasing -Wimplicit -Wstrict-prototypes -pipe $(OPTFLAGS)
 
   ifeq ($(USE_OPENAL),1)
     BASE_CFLAGS += -DUSE_OPENAL=1
@@ -1675,33 +1675,33 @@ copyfiles: build_release
 	-$(MKDIR) -p -m 0755 $(COPYDIR)/missionpack
 
 ifneq ($(BUILD_CLIENT),0)
-	$(INSTALL) -s -m 0755 $(BR)/ioquake3.$(ARCH)$(BINEXT) $(COPYDIR)/ioquake3.$(ARCH)$(BINEXT)
+	$(INSTALL) -m 0755 $(BR)/ioquake3.$(ARCH)$(BINEXT) $(COPYDIR)/ioquake3
 endif
 
 # Don't copy the SMP until it's working together with SDL.
 #ifneq ($(BUILD_CLIENT_SMP),0)
-#	$(INSTALL) -s -m 0755 $(BR)/ioquake3-smp.$(ARCH)$(BINEXT) $(COPYDIR)/ioquake3-smp.$(ARCH)$(BINEXT)
+#	$(INSTALL) -m 0755 $(BR)/ioquake3-smp.$(ARCH)$(BINEXT) $(COPYDIR)/ioquake3-smp.$(ARCH)$(BINEXT)
 #endif
 
 ifneq ($(BUILD_SERVER),0)
 	@if [ -f $(BR)/ioq3ded.$(ARCH)$(BINEXT) ]; then \
-		$(INSTALL) -s -m 0755 $(BR)/ioq3ded.$(ARCH)$(BINEXT) $(COPYDIR)/ioq3ded.$(ARCH)$(BINEXT); \
+		$(INSTALL) -m 0755 $(BR)/ioq3ded.$(ARCH)$(BINEXT) $(COPYDIR)/ioq3ded; \
 	fi
 endif
 
 ifneq ($(BUILD_GAME_SO),0)	
-	$(INSTALL) -s -m 0755 $(BR)/baseq3/cgame$(ARCH).$(SHLIBEXT) \
+	$(INSTALL) -m 0755 $(BR)/baseq3/cgame$(ARCH).$(SHLIBEXT) \
 					$(COPYDIR)/baseq3/.
-	$(INSTALL) -s -m 0755 $(BR)/baseq3/qagame$(ARCH).$(SHLIBEXT) \
+	$(INSTALL) -m 0755 $(BR)/baseq3/qagame$(ARCH).$(SHLIBEXT) \
 					$(COPYDIR)/baseq3/.
-	$(INSTALL) -s -m 0755 $(BR)/baseq3/ui$(ARCH).$(SHLIBEXT) \
+	$(INSTALL) -m 0755 $(BR)/baseq3/ui$(ARCH).$(SHLIBEXT) \
 					$(COPYDIR)/baseq3/.
 	-$(MKDIR) -p -m 0755 $(COPYDIR)/missionpack
-	$(INSTALL) -s -m 0755 $(BR)/missionpack/cgame$(ARCH).$(SHLIBEXT) \
+	$(INSTALL) -m 0755 $(BR)/missionpack/cgame$(ARCH).$(SHLIBEXT) \
 					$(COPYDIR)/missionpack/.
-	$(INSTALL) -s -m 0755 $(BR)/missionpack/qagame$(ARCH).$(SHLIBEXT) \
+	$(INSTALL) -m 0755 $(BR)/missionpack/qagame$(ARCH).$(SHLIBEXT) \
 					$(COPYDIR)/missionpack/.
-	$(INSTALL) -s -m 0755 $(BR)/missionpack/ui$(ARCH).$(SHLIBEXT) \
+	$(INSTALL) -m 0755 $(BR)/missionpack/ui$(ARCH).$(SHLIBEXT) \
 					$(COPYDIR)/missionpack/.
 endif
 
