diff -Naur Mail-Karmasphere-Client-2.18-orig/Makefile.PL Mail-Karmasphere-Client-2.18/Makefile.PL
--- Mail-Karmasphere-Client-2.18-orig/Makefile.PL	2008-04-03 21:28:32.000000000 -0400
+++ Mail-Karmasphere-Client-2.18/Makefile.PL	2009-10-29 11:25:32.000000000 -0400
@@ -10,10 +10,11 @@
 karma-publish|y|Karmasphere feed publisher command
 QUERIES
 	my ($program, $default, $description) = split /\|/, $_;
-	if (prompt("Do you want to install '$program', the $description?", 
-				$default) =~ /^y/) {
+    ## when building an RPM, do not prompt
+	# if (prompt("Do you want to install '$program', the $description?", 
+	# 			$default) =~ /^y/) {
 		push(@programs, $program);
-	}
+	# }
 }
 
 
