diff -Naupr ipw3945-1.1.0.orig/ipw3945.h ipw3945-1.1.0/ipw3945.h
--- ipw3945-1.1.0.orig/ipw3945.h	2006-07-14 20:52:16.000000000 +0200
+++ ipw3945-1.1.0/ipw3945.h	2006-10-09 14:15:26.000000000 +0200
@@ -29,7 +29,6 @@
 
 #include <linux/module.h>
 #include <linux/moduleparam.h>
-#include <linux/config.h>
 #include <linux/init.h>
 
 #include <linux/version.h>
diff -Naupr ipw3945-1.1.0.orig/Makefile ipw3945-1.1.0/Makefile
--- ipw3945-1.1.0.orig/Makefile	2006-07-14 17:34:55.000000000 +0200
+++ ipw3945-1.1.0/Makefile	2006-10-09 14:15:11.000000000 +0200
@@ -90,7 +90,7 @@ IEEE80211_VER := $(shell $(DIR)/snapshot
 IEEE80211_API := $(shell sed -ne "s:\#define.*IEEE80211_API_VERSION[^0-9]*\([0-9]*\).*:\1:p" $(IEEE80211_PATH)net/ieee80211.h)
 
 ifeq ($(IEEE80211_API),)
-	IEEE80211_API := $(shell [[ "$(IEEE80211_VER)" == 1.1.14* ]] && \
+	IEEE80211_API := $(shell [[ "$(IEEE80211_VER)" == *1.1.13* ]] && \
 		{ echo "2" ; } || \
 		{ ver=$(IEEE80211_VER); ver=$${ver\#*.} ; echo $${ver%.*} ; })
 	EXTRA_CFLAGS += -DIEEE80211_API_VERSION=$(IEEE80211_API)
