--- Makefile.orig	2003-12-04 17:04:57.000000000 +0100
+++ Makefile	2003-12-04 17:07:48.000000000 +0100
@@ -15,25 +15,25 @@
 DIR_SRC:=tpctl-$(VER_RELEASE)/
 FILE_TARBALL:=tpctl_$(VER_RELEASE).tar.gz
 
-install: uninstall install_libraries install_binaries install_man
+install: install_libraries install_binaries install_man
 
 install_libraries:
 	if [ ! -d $(DEST)$(PATH_LIB) ]; then mkdir -p -m 755 $(DEST)$(PATH_LIB) ; fi
-	install -m 755 -o 0 -g 0 lib/libsmapidev.so.$(VER_SMAPIDEV) $(DEST)$(PATH_LIB)
+	install -m 755 lib/libsmapidev.so.$(VER_SMAPIDEV) $(DEST)$(PATH_LIB)
 	ln -sf libsmapidev.so.$(VER_SMAPIDEV) $(DEST)$(PATH_LIB)libsmapidev.so.2
 
 install_binaries:
 	if [ ! -d $(DEST)$(PATH_BIN) ]; then mkdir -p -m 755 $(DEST)$(PATH_BIN) ; fi
-	install -m 755 -o 0 -g 0 tpctl/tpctl $(DEST)$(PATH_BIN)tpctl
-	install -m 755 -o 0 -g 0 ntpctl/ntpctl $(DEST)$(PATH_BIN)ntpctl
-	if [ -e tpctlir/tpctlir ]; then install -m 755 -o 0 -g 0 tpctlir/tpctlir $(DEST)$(PATH_SBIN)tpctlir ; fi
+	install -m 755 tpctl/tpctl $(DEST)$(PATH_BIN)tpctl
+	install -m 755 ntpctl/ntpctl $(DEST)$(PATH_BIN)ntpctl
+	if [ -e tpctlir/tpctlir ]; then install -m 755 tpctlir/tpctlir $(DEST)$(PATH_SBIN)tpctlir ; fi
 	if [ ! -d $(DEST)$(PATH_SBIN) ]; then mkdir -p -m 755 $(DEST)$(PATH_SBIN) ; fi
-	install -m 755 -o 0 -g 0 apmiser/apmiser $(DEST)$(PATH_SBIN)apmiser
+	install -m 755 apmiser/apmiser $(DEST)$(PATH_SBIN)apmiser
 
 install_man:
 	if [ ! -d $(DEST)$(PATH_MAN)man1 ]; then mkdir -p -m 755 $(DEST)$(PATH_MAN)man1 ; fi
-	install -m 644 -o 0 -g 0 man/tpctl.1 $(DEST)$(PATH_MAN)man1/tpctl.1
-	install -m 644 -o 0 -g 0 man/ntpctl.1 $(DEST)$(PATH_MAN)man1/ntpctl.1
+	install -m 644 man/tpctl.1 $(DEST)$(PATH_MAN)man1/tpctl.1
+	install -m 644 man/ntpctl.1 $(DEST)$(PATH_MAN)man1/ntpctl.1
 
 uninstall: uninstall_libraries uninstall_binaries uninstall_man
 
--- lib/Makefile.rpm	2002-11-30 16:09:34.000000000 +0100
+++ lib/Makefile	2003-12-04 17:03:37.000000000 +0100
@@ -3,7 +3,7 @@
 
 VER_SMAPIDEV:=2.0
 
-CFLAGS:=-Wall -Wcast-align -Wstrict-prototypes -O2
+CFLAGS:=-Wall -Wcast-align -Wstrict-prototypes -O2 $(RPM_OPT_FLAGS)
 #-DDEBUG_REGS
 #-DDEBUG_STRUCT_SIZES -DDEBUG_FILEOPS -DDEBUG_IOPARM
 
--- ntpctl/Makefile.rpm	2003-01-18 16:34:23.000000000 +0100
+++ ntpctl/Makefile	2003-12-04 17:03:37.000000000 +0100
@@ -1,7 +1,7 @@
 
 all: ntpctl
 
-CFLAGS:=-Wall -Wcast-align -Wstrict-prototypes -O2
+CFLAGS:=-Wall -Wcast-align -Wstrict-prototypes -O2 $(RPM_OPT_FLAGS)
 
 ALLCFLAGS:=$(CFLAGS) -L../lib -I../include -DLINUX
 
--- tpctl/Makefile.rpm	2002-11-30 16:10:01.000000000 +0100
+++ tpctl/Makefile	2003-12-04 17:03:37.000000000 +0100
@@ -1,7 +1,7 @@
 
 all: tpctl
 
-CFLAGS:=-Wall -Wcast-align -Wstrict-prototypes -Wno-uninitialized -O2
+CFLAGS:=-Wall -Wcast-align -Wstrict-prototypes -Wno-uninitialized -O2 $(RPM_OPT_FLAGS)
 #-DTP570_KLUDGE
 #-DALLOW_POWEROFF
 #-DDEBUG_REGS -DDEBUG_STRUCT_SIZES -DDEBUG_FILEOPS -DDEBUG_IOPARM
