--- pam_shield-0.9.3/Makefile.orig	2010-09-01 16:44:33.000000000 +0200
+++ pam_shield-0.9.3/Makefile	2010-09-01 16:51:13.000000000 +0200
@@ -2,10 +2,16 @@
 #	pam_shield	WJ106
 #
 
-bindir=/usr/sbin
-confdir=/etc/security
-pamdir=/lib/security
-crondir=/etc/cron.daily
+prefix=/usr
+bindir=$(prefix)/sbin
+sysconfdir=/etc
+confdir=$(sysconfdir)/security
+libdir=/lib
+pamdir=$(libdir)/security
+crondir=$(sysconfdir)/cron.daily
+localstatedir=/var
+
+DESTDIR=
 
 CC = gcc
 LD = ld
@@ -43,19 +49,21 @@
 	$(CC) -M pam_shield.c shield_purge.c > .depend
 
 install: all
-	$(INSTALL) -s -o root -g root -m 644 pam_shield.so ${pamdir}
-	$(INSTALL) -o root -g root -m 755 -T pam_shield.cron ${crondir}/pam-shield
-	$(INSTALL) -o root -g root -m 755 shield-trigger.sh ${bindir}
-	$(INSTALL) -s -o root -g root -m 755 shield-purge ${bindir}
-	$(INSTALL) -o root -g root -m 644 shield.conf ${confdir}
-	$(MKDIR) -p -m 700 /var/lib/pam_shield
+	$(INSTALL) -Dp -m0644 pam_shield.so $(DESTDIR)$(pamdir)/pam_shield.so
+	$(INSTALL) -Dp -m0755 pam_shield.cron $(DESTDIR)$(crondir)/pam-shield
+	$(INSTALL) -Dp -m0755 shield-trigger.sh $(DESTDIR)$(bindir)/shield-trigger
+	$(INSTALL) -Dp -m0755 shield-trigger-iptables.sh $(DESTDIR)$(bindir)/shield-trigger-iptables
+	$(INSTALL) -Dp -m0755 shield-purge $(DESTDIR)$(bindir)/shield-purge
+	$(INSTALL) -Dp -m0644 shield.conf $(DESTDIR)$(confdir)/shield.conf
+	$(INSTALL) -dp -m0700 $(DESTDIR)$(localstatedir)/lib/pam_shield/
 
 uninstall:
-	$(RM) ${pamdir}/pam_shield.so
-	$(RM) ${crondir}/pam-shield
-	$(RM) ${bindir}/shield-trigger.sh
-	$(RM) ${bindir}/shield-purge
-	$(RM) ${confdir}/shield.conf
-	$(RM) -r /var/lib/pam_shield
+	$(RM) $(pamdir)/pam_shield.so
+	$(RM) $(crondir)/pam-shield
+	$(RM) $(bindir)/shield-trigger
+	$(RM) $(bindir)/shield-trigger-iptables
+	$(RM) $(bindir)/shield-purge
+	$(RM) $(confdir)/shield.conf
+	$(RM) -r $(localstatedir)/lib/pam_shield/
 
 # EOB
