--- pam_shield-0.9.4/Makefile.orig	2010-09-04 21:31:28.000000000 +0200
+++ pam_shield-0.9.4/Makefile	2010-09-04 21:34:23.000000000 +0200
@@ -2,14 +2,14 @@
 #	pam_shield	WJ106
 #
 
-# for 32-bit systems:
-#pamdir = /lib/security
-# for 64-bit systems:
-pamdir = /lib64/security
-
-bindir = /usr/sbin
-confdir = /etc/security
-crondir = /etc/cron.daily
+prefix=/usr
+bindir=$(prefix)/sbin
+sysconfdir=/etc
+confdir=$(sysconfdir)/security
+libdir=/lib
+pamdir=$(libdir)/security
+crondir=$(sysconfdir)/cron.daily
+localstatedir=/var
 
 CC = gcc
 LD = ld
@@ -47,21 +47,19 @@
 	$(CC) -M pam_shield.c shield_purge.c > .depend
 
 install: all
-	$(INSTALL) -s -o root -g root -m 644 pam_shield.so ${pamdir}
-	$(INSTALL) -o root -g root -m 755 -T pam_shield.cron ${crondir}/pam-shield
-	$(INSTALL) -o root -g root -m 755 shield-trigger ${bindir}
-	$(INSTALL) -s -o root -g root -m 755 shield-purge ${bindir}
-	if ! test -e ${confdir}/shield.conf; then \
-	$(INSTALL) -o root -g root -m 644 shield.conf ${confdir} ; \
-	fi
-	$(MKDIR) -p -m 700 /var/lib/pam_shield
+	$(INSTALL) -Dp -m0644 pam_shield.so $(DESTDIR)$(pamdir)/pam_shield.so
+	$(INSTALL) -Dp -m0755 pam_shield.cron $(DESTDIR)$(crondir)/pam-shield
+	$(INSTALL) -Dp -m0755 shield-trigger $(DESTDIR)$(bindir)/shield-trigger
+	$(INSTALL) -Dp -m0755 shield-purge $(DESTDIR)$(bindir)/shield-purge
+	$(INSTALL) -Dp -m0644 shield.conf $(DESTDIR)$(confdir)/shield.conf
+	$(INSTALL) -dp -m0700 $(DESTDIR)$(localstatedir)/lib/pam_shield/
 
 uninstall:
-	$(RM) ${pamdir}/pam_shield.so
-	$(RM) ${crondir}/pam-shield
-	$(RM) ${bindir}/shield-trigger
-	$(RM) ${bindir}/shield-purge
-	$(RM) ${confdir}/shield.conf
+	$(RM) $(pamdir)/pam_shield.so
+	$(RM) $(crondir)/pam-shield
+	$(RM) $(bindir)/shield-trigger
+	$(RM) $(bindir)/shield-purge
+	$(RM) $(confdir)/shield.conf
 	$(RM) -r /var/lib/pam_shield
 
 # EOB
