--- Makefile.old	2002-10-02 16:53:13.000000000 -0400
+++ Makefile	2003-10-23 15:01:37.000000000 -0400
@@ -23,7 +23,7 @@
 pdatadir=${proot}/share/pixmaps
 pdocdir=${proot}/share/doc/${name}-${version}
 
-SUBDIRS=i s lang
+SUBDIRS=docs icons skins lang plugins
 transcript=uninstall.sh
 
 # For integration with the current desktop
@@ -74,7 +74,7 @@
 		  if [ ! -d $$myfile ]; then \
 		    srcfile=$$subdir/$$myfile; \
 		    dstfile=$$mydir; \
-		    cp $$srcfile $$dstfile; \
+		    cp -a $$srcfile $$dstfile; \
 		    echo "rm -f $$dstfile/$$myfile" >> .$(transcript); \
 		  fi; \
 		done; \
@@ -88,11 +88,11 @@
 		cp $$corefile $(libdir); \
 		echo "rm -f $$myfile" >> .$(transcript); \
 		done; \
-		cp i/$(icon) $(datadir)/$(icon); \
+		#cp i/$(icon) $(datadir)/$(icon); \
 		echo "rm -f $(datadir)/$(icon)" >> .$(transcript); \
 		cp $(desklink) $(wmapps); \
 		echo "rm -f $(wmapps)/$(desklink)" >> .$(transcript)
-#		ln -sf $(libdir)/i/$(icon) $(datadir)/$(icon)
+#		#ln -sf $(libdir)/i/$(icon) $(datadir)/$(icon)
 	@cp .$(transcript) $(libdir)/$(transcript)
 #	Now we install the executable in the binary directory
 #	@ln -sf $(libdir)/${name} $(bindir)/${name}
