--- ./lib/python/App/dtml/zope_quick_start.dtml.pythonwarning	2005-10-25 15:23:22.000000000 +0200
+++ ./lib/python/App/dtml/zope_quick_start.dtml	2005-10-25 15:35:55.000000000 +0200
@@ -134,5 +134,10 @@
 
 </ul>
 
+<p style="color: red; font-weight: bold">
+WARNING: you are running Zope 2.8.X with python 2.4.X, this is not a supported
+combination. Don't file bugreports or ask for support on zope.org.
+</p>
+
 </body>
 </html>
