diff -ruN magicpoint-1.10a.orig/contrib/xwintoppm/dsimple.c magicpoint-1.10a/contrib/xwintoppm/dsimple.c
--- magicpoint-1.10a.orig/contrib/xwintoppm/dsimple.c	2004-06-21 22:05:39.870925000 +0900
+++ magicpoint-1.10a/contrib/xwintoppm/dsimple.c	2004-06-21 22:05:32.567891488 +0900
@@ -35,6 +35,7 @@
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 #include <stdio.h>
+#include <stdarg.h>
 /*
  * Other_stuff.h: Definitions of routines in other_stuff.
  *
@@ -55,6 +56,7 @@
 void blip();
 Window Window_With_Name();
 void Fatal_Error();
+void outl(char *msg, ...);
 
 /*
  * Just_display: A group of routines designed to make the writting of simple
@@ -496,12 +498,16 @@
  */
 /* VARARGS1 */
 void
-outl(msg, arg0,arg1,arg2,arg3,arg4,arg5,arg6)
-     char *msg;
-     char *arg0, *arg1, *arg2, *arg3, *arg4, *arg5, *arg6;
+outl(char *msg, ...)
 {
+	va_list args;
+
 	fflush(stdout);
-	fprintf(stderr, msg, arg0, arg1, arg2, arg3, arg4, arg5, arg6);
+
+	va_start(args, msg);
+	vfprintf(stderr, msg, args);
+	va_end(args);
+
 	fprintf(stderr, "\n");
 	fflush(stderr);
 }
diff -ruN magicpoint-1.10a.orig/contrib/xwintoppm/xwintoppm.c magicpoint-1.10a/contrib/xwintoppm/xwintoppm.c
--- magicpoint-1.10a.orig/contrib/xwintoppm/xwintoppm.c	2004-06-21 22:05:39.871924000 +0900
+++ magicpoint-1.10a/contrib/xwintoppm/xwintoppm.c	2004-06-21 22:05:32.568891356 +0900
@@ -138,6 +138,7 @@
 
 extern int (*_XErrorFunction)();
 extern int _XDefaultError();
+void Window_Dump(Window window, FILE *out);
 
 static long parse_long (s)
     char *s;
