diff -up subversion-1.6.12/subversion/bindings/swig/python/tests/run_all.py.disable-client-test subversion-1.6.12/subversion/bindings/swig/python/tests/run_all.py
--- subversion-1.6.12/subversion/bindings/swig/python/tests/run_all.py.disable-client-test	2010-07-23 09:30:31.951996475 -0400
+++ subversion-1.6.12/subversion/bindings/swig/python/tests/run_all.py	2010-07-23 09:30:46.953930404 -0400
@@ -9,7 +9,6 @@ def suite():
   suite = unittest.TestSuite()
   suite.addTest(core.suite())
   suite.addTest(mergeinfo.suite())
-  suite.addTest(client.suite())
   suite.addTest(delta.suite())
   suite.addTest(pool.suite())
   suite.addTest(ra.suite())
