--- k3b-0.12.2/libk3b/jobs/k3bcdcopyjob.cpp.statfs	2005-07-01 15:31:42.000000000 +0200
+++ k3b-0.12.2/libk3b/jobs/k3bcdcopyjob.cpp	2005-07-13 14:42:08.000000000 +0200
@@ -309,7 +309,7 @@
 	  imageSpaceNeeded += (*it).length().mode1Bytes();
       }
 
-      unsigned long avail, size;
+      unsigned long long avail, size;
       QString pathToTest = m_tempPath.left( m_tempPath.findRev( '/' ) );
       if( !K3b::kbFreeOnFs( pathToTest, size, avail ) ) {
 	emit infoMessage( i18n("Unable to determine free space in temporary directory '%1'.").arg(pathToTest), ERROR );
--- k3b-0.12.2/libk3b/jobs/k3bdvdcopyjob.cpp.statfs	2005-07-01 15:31:42.000000000 +0200
+++ k3b-0.12.2/libk3b/jobs/k3bdvdcopyjob.cpp	2005-07-13 14:42:08.000000000 +0200
@@ -288,7 +288,7 @@
       // check free temp space
       //
       KIO::filesize_t imageSpaceNeeded = (KIO::filesize_t)(d->lastSector.lba()+1)*2048;
-      unsigned long avail, size;
+      unsigned long long avail, size;
       QString pathToTest = m_imagePath.left( m_imagePath.findRev( '/' ) );
       if( !K3b::kbFreeOnFs( pathToTest, size, avail ) ) {
 	emit infoMessage( i18n("Unable to determine free space in temporary directory '%1'.").arg(pathToTest), ERROR );
--- k3b-0.12.2/libk3b/core/k3bglobals.h.statfs	2005-07-01 15:31:13.000000000 +0200
+++ k3b-0.12.2/libk3b/core/k3bglobals.h	2005-07-13 14:42:08.000000000 +0200
@@ -86,7 +86,7 @@
    * We use this since we encountered problems with the KDE version.
    * @returns true on success.
    */
-  bool kbFreeOnFs( const QString& path, unsigned long& size, unsigned long& avail );
+  bool kbFreeOnFs( const QString& path, unsigned long long& size, unsigned long long& avail );
 
   /**
    * Cut a filename preserving the extension
--- k3b-0.12.2/libk3b/core/k3bglobals.cpp.statfs	2005-07-01 15:31:13.000000000 +0200
+++ k3b-0.12.2/libk3b/core/k3bglobals.cpp	2005-07-13 14:42:08.000000000 +0200
@@ -231,15 +231,15 @@
 }
 
 
-bool K3b::kbFreeOnFs( const QString& path, unsigned long& size, unsigned long& avail )
+bool K3b::kbFreeOnFs( const QString& path, unsigned long long& size, unsigned long long& avail )
 {
   struct statfs fs;
 
   if( ::statfs( QFile::encodeName(path), &fs ) == 0 ) {
-    unsigned long kBfak = fs.f_bsize/1024;
+    // unsigned long kBfak = fs.f_bsize/1024;
 
-    size = fs.f_blocks*kBfak;
-    avail = fs.f_bavail*kBfak;
+    size = (fs.f_blocks*static_cast<unsigned long long>( fs.f_bsize )) / 1024;
+    avail = (fs.f_bavail*static_cast<unsigned long long>( fs.f_bsize )) / 1024;
 
     return true;
   }
--- k3b-0.12.2/src/k3bstatusbarmanager.cpp.statfs	2005-07-01 15:31:10.000000000 +0200
+++ k3b-0.12.2/src/k3bstatusbarmanager.cpp	2005-07-13 14:42:08.000000000 +0200
@@ -89,7 +89,7 @@
   if( !QFile::exists( path ) )
     path.truncate( path.findRev('/') );
 
-  unsigned long size, avail;
+  unsigned long long size, avail;
   if( K3b::kbFreeOnFs( path, size, avail ) )
     slotFreeTempSpace( path, size, 0, avail );
   else
--- k3b-0.12.2/src/rip/k3bvideocdrippingdialog.h.statfs	2005-07-01 15:31:09.000000000 +0200
+++ k3b-0.12.2/src/rip/k3bvideocdrippingdialog.h	2005-07-13 14:42:08.000000000 +0200
@@ -66,7 +66,7 @@
   void slotStartClicked();
 
   void slotUpdateFreeSpace();
-  void slotFreeSpace(const QString&, unsigned long, unsigned long, unsigned long);
+  void slotFreeSpace(const QString&, unsigned long long, unsigned long long, unsigned long long);
 
 };
 
--- k3b-0.12.2/src/rip/k3bvideocdrippingdialog.cpp.statfs	2005-07-01 15:31:09.000000000 +0200
+++ k3b-0.12.2/src/rip/k3bvideocdrippingdialog.cpp	2005-07-13 14:42:08.000000000 +0200
@@ -188,9 +188,9 @@
 }
 
 void K3bVideoCdRippingDialog::slotFreeSpace(const QString&,
-						  unsigned long,
-						  unsigned long,
-						  unsigned long kbAvail)
+						  unsigned long long,
+						  unsigned long long,
+						  unsigned long long kbAvail)
 {
     m_labelFreeSpace->setText( KIO::convertSizeFromKB(kbAvail) );
 
@@ -212,7 +212,7 @@
     if( !QFile::exists( path ) )
         path.truncate( path.findRev('/') );
 
-    unsigned long size, avail;
+    unsigned long long size, avail;
     if( K3b::kbFreeOnFs( path, size, avail ) )
         slotFreeSpace( path, size, 0, avail );
     else
--- k3b-0.12.2/src/rip/k3baudioconvertingoptionwidget.cpp.statfs	2005-07-13 14:56:18.000000000 +0200
+++ k3b-0.12.2/src/rip/k3baudioconvertingoptionwidget.cpp	2005-07-13 14:56:45.000000000 +0200
@@ -141,7 +141,7 @@
   if( !QFile::exists( path ) )
     path.truncate( path.findRev('/') );
 
-  unsigned long size, avail;
+  unsigned long long size, avail;
   if( K3b::kbFreeOnFs( path, size, avail ) ) {
     m_labelFreeSpace->setText( KIO::convertSizeFromKB(avail) );
     if( avail < d->neededSize/1024 )
--- k3b-0.12.2/src/rip/k3bdvdripperwidget.cpp.statfs	2005-07-01 15:31:09.000000000 +0200
+++ k3b-0.12.2/src/rip/k3bdvdripperwidget.cpp	2005-07-13 14:42:08.000000000 +0200
@@ -233,12 +233,17 @@
     }
     struct statfs fs;
     ::statfs( QFile::encodeName( tmp ), &fs );
-    unsigned int kBfak = (unsigned int)(fs.f_bsize/1024);
-    slotFreeTempSpace( tmp, fs.f_blocks*kBfak, (fs.f_blocks-fs.f_bfree)*kBfak, fs.f_bavail*kBfak );
+    unsigned long long size, used, avail;
+    size = (fs.f_blocks * static_cast<unsigned long long> (fs.f_bsize)) / 1024;
+    used = ((fs.f_blocks-fs.f_bfree) * static_cast<unsigned long long> (fs.f_bsize)) / 1024;
+    avail = (fs.f_bavail * static_cast<unsigned long long> (fs.f_bsize)) / 1024;
+    slotFreeTempSpace( tmp, size, used, avail );
+    // unsigned int kBfak = (unsigned int)(fs.f_bsize/1024);
+    // slotFreeTempSpace( tmp, fs.f_blocks*kBfak, (fs.f_blocks-fs.f_bfree)*kBfak, fs.f_bavail*kBfak );
 }
 
-void K3bDvdRipperWidget::slotFreeTempSpace( const QString &, unsigned long kBSize,
-        unsigned long kBUsed, unsigned long kBAvail ){
+void K3bDvdRipperWidget::slotFreeTempSpace( const QString &, unsigned long long kBSize,
+        unsigned long long kBUsed, unsigned long long kBAvail ){
     m_fillDisplay->setKbSize( kBSize );
     m_fillDisplay->setKbAvailable( kBAvail );
     m_fillDisplay->setKbUsed( kBUsed );
--- k3b-0.12.2/src/rip/k3bdvdripperwidget.h.statfs	2005-07-01 15:31:09.000000000 +0200
+++ k3b-0.12.2/src/rip/k3bdvdripperwidget.h	2005-07-13 14:42:08.000000000 +0200
@@ -99,8 +99,8 @@
   void rip();
   void slotFindStaticDir();
   void slotSetDependDirs( const QString& );
-  void slotFreeTempSpace( const QString & mountPoint, unsigned long kBSize,
-			 unsigned long kBUsed, unsigned long kBAvail );
+  void slotFreeTempSpace( const QString & mountPoint, unsigned long long kBSize,
+			 unsigned long long kBUsed, unsigned long long kBAvail );
   void slotParseError( KProcess *p, char *text, int len );
   void slotCheckOpenEncoding(int);
   void slotOpenEncoding( bool );
--- k3b-0.12.2/src/k3btempdirselectionwidget.cpp.statfs	2005-07-01 15:31:10.000000000 +0200
+++ k3b-0.12.2/src/k3btempdirselectionwidget.cpp	2005-07-13 14:42:08.000000000 +0200
@@ -109,7 +109,7 @@
   if( !QFile::exists( path ) )
     path.truncate( path.findRev('/') );
 
-  unsigned long size, avail;
+  unsigned long long size, avail;
   if( K3b::kbFreeOnFs( path, size, avail ) )
     slotFreeTempSpace( path, size, 0, avail );
   else
