diff -ur wsdlpull-1.14-orig/src/wsdlparser/WsdlInvoker.h wsdlpull-1.14/src/wsdlparser/WsdlInvoker.h
--- wsdlpull-1.14-orig/src/wsdlparser/WsdlInvoker.h	2007-07-22 09:25:21.000000000 +0200
+++ wsdlpull-1.14/src/wsdlparser/WsdlInvoker.h	2007-08-06 14:21:57.000000000 +0200
@@ -204,7 +204,7 @@
   // set credentials for the service
   void setCredentials(const std::string & user, const std::string & pass);
   //ouput the soap message without invoking the service
-  std::string WsdlInvoker::getSoapMessage();
+  std::string getSoapMessage();
 
   void setProxy(const std::string & host,int  port=80);
   //enable logging
Only in wsdlpull-1.14/src/wsdlparser: WsdlInvoker.h~
diff -ur wsdlpull-1.14-orig/src/xmlpull/XmlPullParser.h wsdlpull-1.14/src/xmlpull/XmlPullParser.h
--- wsdlpull-1.14-orig/src/xmlpull/XmlPullParser.h	2007-05-25 21:25:25.000000000 +0200
+++ wsdlpull-1.14/src/xmlpull/XmlPullParser.h	2007-08-06 14:05:30.000000000 +0200
@@ -110,7 +110,7 @@
   int next ();
   int nextToken ();
   int nextTag ();
-  void XmlPullParser::prevTag();
+  void prevTag();
 
   //----------------------------------------------------------------------
   // utility methods to make XML parsing easier ...
Only in wsdlpull-1.14/src/xmlpull: XmlPullParser.h~
