diff -ur wsdlpull-1.12-orig/src/schemaparser/TypeContainer.h wsdlpull-1.12/src/schemaparser/TypeContainer.h
--- wsdlpull-1.12-orig/src/schemaparser/TypeContainer.h	2006-11-23 12:27:53.000000000 +0100
+++ wsdlpull-1.12/src/schemaparser/TypeContainer.h	2006-11-26 18:49:18.000000000 +0100
@@ -45,7 +45,7 @@
 {
  public:
   TypeContainer(int  typeId,const SchemaParser * sp);
-  TypeContainer::TypeContainer(ContentModel* cm,
+  TypeContainer(ContentModel* cm,
 			       const SchemaParser * sp);
   ~TypeContainer();
   TypeContainer *getAttributeContainer(std::string attName, 
Only in wsdlpull-1.12/src/schemaparser: TypeContainer.h~
diff -ur wsdlpull-1.12-orig/src/wsdlparser/WsdlParser.h wsdlpull-1.12/src/wsdlparser/WsdlParser.h
--- wsdlpull-1.12-orig/src/wsdlparser/WsdlParser.h	2006-06-09 15:23:26.000000000 +0200
+++ wsdlpull-1.12/src/wsdlparser/WsdlParser.h	2006-11-26 19:00:23.000000000 +0100
@@ -228,7 +228,7 @@
    */
 
   void
-    WsdlParser::getSchemaParsers(std::vector<SchemaParser* >::iterator & from,
+    getSchemaParsers(std::vector<SchemaParser* >::iterator & from,
 				 std::vector<SchemaParser* >::iterator & to) ;
   
   /**
Only in wsdlpull-1.12/src/wsdlparser: WsdlParser.h~
