--- ./gettext.xs.gege	2003-03-04 15:05:18.000000000 +0100
+++ ./gettext.xs	2003-03-04 15:07:20.000000000 +0100
@@ -26,6 +26,32 @@
     return 0;
 }
 
+#include <langinfo.h>
+#include <iconv.h>
+
+/* bad: duplicated from gi/perl-install/c/stuff.xs in http://cvs.mandrakesoft.com/cgi-bin/cvsweb.cgi/ */
+SV * iconv_(char* s, char* from_charset, char* to_charset) {
+  iconv_t cd = iconv_open(to_charset, from_charset);
+  char* retval = s;
+  if (cd != (iconv_t) (-1)) {
+      size_t s_len = strlen(retval);
+      /* the maximum expansion when converting happens when converting
+	 tscii to utf-8; each tscii char can become up to 4 unicode chars
+	 and each one of those unicode chars can be 3 bytes long */
+      char *buf = alloca(4 * 3 * s_len);
+      {
+	  char *ptr = buf;
+	  size_t ptr_len = 4 * 3 * s_len;
+	  if ((iconv(cd, &s, &s_len, &ptr, &ptr_len)) != (size_t) (-1)) {
+	      *ptr = 0;
+	      retval = buf;
+	  }
+      }
+      iconv_close(cd);
+  }
+  return newSVpv(retval, 0);
+}
+
 MODULE = Locale::gettext	PACKAGE = Locale::gettext
 
 double
@@ -56,3 +79,13 @@
 bindtextdomain(domain, dirname)
 	char *		domain
 	char *		dirname
+
+SV *
+iconv(s, from_charset, to_charset)
+  char *s
+  SV *from_charset
+  char *to_charset
+  CODE:
+  RETVAL = iconv_(s, from_charset == &PL_sv_undef ? nl_langinfo(CODESET) : SvPV_nolen(from_charset), to_charset);
+  OUTPUT:
+  RETVAL
