--- sendmail-8.12.7/sendmail/recipient.c
+++ sendmail-8.12.7/sendmail/recipient.c
@@ -1146,7 +1146,7 @@
 
 	*fuzzyp = false;
 
-#if HESIOD
+#if disableHESIOD
 	/* DEC Hesiod getpwnam accepts numeric strings -- short circuit it */
 	for (p = name; *p != '\0'; p++)
 		if (!isascii(*p) || !isdigit(*p))
--- sendmail-8.12.7/libsm/mbdb.c
+++ sendmail-8.12.7/libsm/mbdb.c
@@ -315,7 +315,7 @@
 {
 	struct passwd *pw;
 
-#ifdef HESIOD
+#ifdef disableHESIOD
 	/* DEC Hesiod getpwnam accepts numeric strings -- short circuit it */
 	{
 		char *p;
