--- sendmail-8.12.5/cf/cf/submit.mc
+++ sendmail-8.12.5/cf/cf/submit.mc
@@ -15,12 +15,16 @@
 #
 
 divert(0)dnl
-VERSIONID(`$Id: sendmail-8.12.5-pid.patch,v 1.1 2004/02/24 23:49:35 dag- Exp $')
+include(`/usr/share/sendmail-cf/m4/cf.m4')
+VERSIONID(`linux setup for Red Hat Linux')dnl
 define(`confCF_VERSION', `Submit')dnl
 define(`__OSTYPE__',`')dnl dirty hack to keep proto.m4 from complaining
 define(`_USE_DECNET_SYNTAX_', `1')dnl support DECnet
 define(`confTIME_ZONE', `USE_TZ')dnl
 define(`confDONT_INIT_GROUPS', `True')dnl
+define(`confPID_FILE', `/var/run/sm-client.pid')dnl
+dnl define(`confDIRECT_SUBMISSION_MODIFIERS',`C')
+FEATURE(`use_ct_file')dnl
 dnl
 dnl If you use IPv6 only, change [127.0.0.1] to [IPv6:::1]
 FEATURE(`msp', `[127.0.0.1]')dnl
