Index: src/afsd/afs.rc.linux
===================================================================
RCS file: /cvs/openafs/src/afsd/afs.rc.linux,v
retrieving revision 1.4
diff -u -r1.4 afs.rc.linux
--- src/afsd/afs.rc.linux	2001/10/10 03:16:54	1.4
+++ src/afsd/afs.rc.linux	2001/10/13 14:42:31
@@ -92,6 +92,19 @@
 # LIBAFS manually.
 choose_client() {
 
+        # Use the associated script to determine the module name
+	if test -f /usr/vice/etc/afsmodname && \
+	    test -f /usr/vice/etc/modload/SymTable ; then
+	    
+            LIBAFS=`/usr/vice/etc/afsmodname`
+	    if test "x$LIBAFS" = "x" ; then
+		echo Did not find matching module in SymTable
+	    else
+		echo Found $LIBAFS from SymTable... Loading...
+		return 0
+	    fi
+	fi
+
 	# Use the second field of the uname -v output instead of just
 	# doing a match on the whole thing to protect against matching
 	# a timezone named SMP -- I don't know of one, but let's be
