--- src/webdav.c.orig	2007-05-29 21:19:51.000000000 +0200
+++ src/webdav.c	2008-03-14 04:48:54.748583296 +0100
@@ -380,7 +380,7 @@
 
     if (ret == 0) {
         initialized = 1;
-        if (!caps.dav_class1) {
+        if (!caps.dav_class1 && !caps.dav_class2) {
             if (have_terminal) {
                 error(EXIT_FAILURE, 0,
                       _("mounting failed; the server does not support WebDAV"));
