--- quicktime/Makefile~	Tue May  7 07:17:29 2002
+++ quicktime/Makefile	Tue Jul  9 19:11:02 2002
@@ -1,5 +1,5 @@
 OBJDIR := $(shell uname --machine)
-USE_MMX = 0
+USE_MMX = 1
 CC = gcc
 HAVE_CFLAGS := $(shell if [ -n "$CFLAGS" ]; then echo y; fi )
 #ENCORE_DIR := encore2/src
@@ -26,11 +26,12 @@
   CFLAGS = -O3 -fmessage-length=0 -funroll-all-loops -fomit-frame-pointer -malign-loops=2 -malign-jumps=2 -malign-functions=2 -I/usr/local/include
 endif
 
+endif
 
+CFLAGS = -O3 -fmessage-length=0 -funroll-all-loops -fomit-frame-pointer -malign-loops=2 -malign-jumps=2 -malign-functions=2 -I/usr/local/include
 
 
 
-endif
 
 $(shell if ! test -d $(OBJDIR) \; then mkdir -p $(OBJDIR) \; fi )
 
