--- configure.orig	Tue Nov 12 14:01:12 2002
+++ configure	Tue Nov 12 14:02:48 2002
@@ -369,7 +369,7 @@ if test -z "$_target" ; then
 else
   system_name=`echo $_target | cut -d '-' -f 2`
   case "`echo $system_name | tr A-Z a-z`" in
-    linux) system_name=Linux ;;
+    linux|redhat) system_name=Linux ;;
     freebsd) system_name=FreeBSD ;;
     netbsd) system_name=NetBSD ;;
     bsd/os) system_name=BSD/OS ;;
