Only in rosetta-0.01/IO: .depends-io
diff -u -r rosetta-0.01/IO/DictHandler.cc rosetta-0.01.patched/IO/DictHandler.cc
--- rosetta-0.01/IO/DictHandler.cc	2002-12-21 13:30:26.000000000 +0100
+++ rosetta-0.01.patched/IO/DictHandler.cc	2003-01-03 14:13:17.000000000 +0100
@@ -1,9 +1,12 @@
 #include "DictHandler.h"
 
+#include <iostream>
 #include <qregexp.h>
 
 #include "Structure/CharData.h"
 
+using namespace std;
+
 DictHandler::DictHandler(Dictionary& aDict)
 	: theDict(aDict), State(Neutral)
 {
Only in rosetta-0.01/IO: DictHandler.cc.rh8
Only in rosetta-0.01/Structure: .depends-struct
Only in rosetta-0.01/UI: .depends-ui
Only in rosetta-0.01/UI: MainWindow.moc.cc
Only in rosetta-0.01/UI: TranslationText.moc.cc
Only in rosetta-0.01/UI: TranslationView.moc.cc
