diff -Naupr libpri-1.0.9.orig/Makefile libpri-1.0.9/Makefile
--- libpri-1.0.9.orig/Makefile	2005-05-11 05:37:29.000000000 +0200
+++ libpri-1.0.9/Makefile	2005-08-23 14:24:23.000000000 +0200
@@ -36,7 +36,6 @@ STATIC_LIBRARY=libpri.a
 DYNAMIC_LIBRARY=libpri.so.1.0
 STATIC_OBJS=pri.o q921.o prisched.o q931.o
 DYNAMIC_OBJS=pri.lo q921.lo prisched.lo q931.lo
-CFLAGS=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes -g $(ALERTING) $(LIBPRI_COUNTERS)
 INSTALL_PREFIX=
 ifeq (${OSARCH},Linux)
 LDCONFIG_FLAGS=-n
@@ -69,7 +68,6 @@ install: $(STATIC_LIBRARY) $(DYNAMIC_LIB
 	if [ -x /usr/sbin/sestatus ] && ( /usr/sbin/sestatus | grep "SELinux status:" | grep -q "enabled"); then  restorecon -v $(INSTALL_PREFIX)/$(INSTALL_BASE)/lib/$(DYNAMIC_LIBRARY); fi
 	( cd $(INSTALL_PREFIX)/usr/lib ; ln -sf libpri.so.1 libpri.so )
 	install -m 644 $(STATIC_LIBRARY) $(INSTALL_PREFIX)/usr/lib
-	/sbin/ldconfig
 
 uninstall:
 	@echo "Removing Libpri"
