Index: flumotion/wizard/save.py
===================================================================
--- flumotion/wizard/save.py	(revision 1055)
+++ flumotion/wizard/save.py	(working copy)
@@ -142,6 +142,13 @@
     def getAudioSource(self, video_source):
         options = self.wizard.get_step_options('Source')
         source = options['audio']
+        # If we selected firewire and have selected video
+        # and the selected video is Firewire,
+        #   return the source
+        if (source == AudioDevice.Firewire and video_source and
+            options['video'] == VideoDevice.Firewire):
+            return video_source
+        
         props = {}
         
         audio_step = self.wizard['Audio Source']
