--- fortune/fortune.c.remove-offensive-option	Thu Aug 22 04:06:19 2002
+++ fortune/fortune.c	Thu Aug 22 04:09:21 2002
@@ -232,7 +232,7 @@
 #ifndef	NO_REGEX
     (void) fprintf(stderr, "i");
 #endif /* NO_REGEX */
-    (void) fprintf(stderr, "losw]");
+    (void) fprintf(stderr, "lsw]");
 #ifndef	NO_REGEX
     (void) fprintf(stderr, " [-m pattern]");
 #endif /* NO_REGEX */
@@ -709,9 +709,9 @@
     ignore_case = FALSE;
 
 #ifdef DEBUG
-    while ((ch = getopt(argc, argv, "aDefilm:n:osvw")) != EOF)
+    while ((ch = getopt(argc, argv, "aDefilm:n:svw")) != EOF)
 #else
-    while ((ch = getopt(argc, argv, "aefilm:n:osvw")) != EOF)
+    while ((ch = getopt(argc, argv, "aefilm:n:svw")) != EOF)
 #endif /* DEBUG */
 	switch (ch)
 	  {
@@ -736,9 +736,6 @@
 	  case 'n':
 	      SLEN = atoi(optarg);
 	      break;
-	  case 'o':		/* offensive ones only */
-	      Offend++;
-	      break;
 	  case 's':		/* short ones only */
 	      Short_only++;
 	      Long_only = FALSE;
--- fortune/fortune-man.part1.remove-offensive-option	Thu Aug 22 03:57:14 2002
+++ fortune/fortune-man.part1	Thu Aug 22 04:00:42 2002
@@ -58,15 +58,12 @@
 When
 .B fortune
 is run with no arguments it prints out a random epigram. Epigrams are
-divided into several categories, where each category is sub-divided
-into those which are potentially offensive and those which are not.
+divided into several categories.
 .SS Options
 The options are as follows:
 .TP
 .B -a
-Choose from all lists of maxims, both offensive and not.  (See the
-.B -o 
-option for more information on offensive fortunes.)
+Choose from all lists of maxims.
 .TP
 .B -e
 Consider all fortune files to be of equal size (see discussion below
@@ -111,28 +108,6 @@
 ask for short fortunes, or too long and ask for long ones, fortune goes
 into a never-ending thrash loop.
 .TP
-.B -o
-Choose only from potentially offensive aphorisms.
-.sp
-.B Please, please, please request a potentially
-.B offensive fortune if and only if
-.B you believe, deep in your heart,
-.B that you are willing to be
-.B offended. (And that you'll just quit
-.BR using " -o " rather
-.B than give us grief about it,
-.B okay?)
-.sp
-.RS
-\&... let us keep in mind the basic governing philosophy of The
-Brotherhood, as handsomely summarized in these words: we believe in
-healthy, hearty laughter -- at the expense of the whole human race, if
-needs be.  Needs be.
-.RS
---H. Allen Smith, "Rude Jokes"
-.RE
-.RE
-.TP
 .B -s
 Short apothegms only.  See
 .B -n
@@ -203,30 +178,6 @@
 .RI "50% " funny " 50% " not-funny
 .sp
 .RE
-This fortune also supports the BSD method of appending ``-o'' to
-database names to specify offensive fortunes.  However this is
-.B not
-how fortune stores them: offensive fortunes are stored in a seperate
-directory without the ``-o'' infix.  A plain name (i.e., not a path to a
-file or directory) that ends in ``-o'' will be assumed to be an
-offensive database, and will have its suffix stripped off and be
-searched in the offensive directory (even if the neither of the
-.IR -a " or " -o
-options were specified).  This feature is not only for
-backwards-compatibility, but also to allow users to distinguish between
-inoffensive and offensive databases of the same name.
-.PP
-For example, assuming there is a database named
-.I definitions
-in both the inoffensive and potentially offensive collections, then the
-following command will select an inoffensive definition 90% of the time,
-and a potentially offensive definition for the remaining 10%:
-.RS
-.sp
-.B fortune
-90%
-.I definitions definitions-o
-.RE
 .SH FILES
 Note: these are the defaults as defined at compile time.
 .PP
