--- MIME-tools-5.411/lib/MIME/Parser.pm~	Sun Nov 12 07:55:11 2000
+++ MIME-tools-5.411/lib/MIME/Parser.pm	Sat Apr 27 17:48:08 2002
@@ -592,6 +592,7 @@
 #
 # I<Instance method.>
 # Process and return the next header.
+# Return undef if, instead of a header, the encapsulation boundary is found.
 # Fatal exception on failure.
 #
 sub process_header {
@@ -612,6 +613,10 @@
     foreach (@headlines) { s/[\r\n]+\Z/\n/ }  ### fold
 
     ### How did we do?
+    if ($hdr_rdr->eos_type eq 'DELIM') {
+	$self->whine("bogus part, without CRLF before body");
+	return;
+    }
     ($hdr_rdr->eos_type eq 'DONE') or
 	$self->error("unexpected end of header\n");
 
@@ -983,7 +988,17 @@
 
     ### Parse and add the header:
     my $head = $self->process_header($in, $rdr);
-    $ent->head($head);   
+    if (not defined $head) {
+	$self->debug("bogus empty part");
+	$head = $self->interface('HEAD_CLASS')->new;
+	$head->mime_type('text/plain; charset=US-ASCII');
+	$ent->head($head);
+	$ent->bodyhandle($self->new_body_for($head));
+	$ent->bodyhandle->open("w")->close;
+	$self->results->level(-1);
+	return $ent;
+    }
+    $ent->head($head);
 
     ### Tweak the content-type based on context from our parent...
     ### For example, multipart/digest messages default to type message/rfc822:

