--- Makefile.in.orig	2003-03-08 00:40:16.000000000 +0200
+++ Makefile.in	2003-03-23 11:30:16.000000000 +0200
@@ -36,7 +36,7 @@
 
 CAT=cat
 CD=cd
-MKDIR=mkdir
+MKDIR=mkdir -p
 BUILDTYPE=build
 RM=rm -rf
 SH=sh
@@ -58,10 +58,10 @@
 	$(CC) -c regex.c -DHAVE_STRING_H -DSTDC_HEADERS -DREGEX_MALLOC
 
 install: lft lft.8
-	test -d $(DESTDIR)/. || mkdir $(DESTDIR)
-	$(INSTALL) -o root -m 4755 lft $(DESTDIR)/lft
-	test -d $(MANDIR)/. || mkdir $(MANDIR)
-	$(INSTALL) -o root -m 444 lft.8 $(MANDIR)/lft.8
+	test -d $(DESTDIR)/. || $(MKDIR) $(DESTDIR)
+	$(INSTALL) lft $(DESTDIR)/lft
+	test -d $(MANDIR)/. || $(MKDIR) $(MANDIR)
+	$(INSTALL) lft.8 $(MANDIR)/lft.8
 
 clean:
 	$(RM) *.o core* lft *~
