Index: src/rss-config-factory.c
===================================================================
--- src/rss-config-factory.c	(revision 334)
+++ src/rss-config-factory.c	(working copy)
@@ -1747,7 +1747,9 @@
 	gtk_tree_view_column_set_sort_column_id (column, 2);
 	gtk_tree_view_set_search_column (GTK_TREE_VIEW (treeview),
                                                    2);
-	gtk_tree_view_set_tooltip_column (treeview, 1);
+#if GTK_VERSION >= 2012000
+        gtk_tree_view_set_tooltip_column (treeview, 1);
+#endif
 
 	if (rf->hr != NULL)
         	g_hash_table_foreach(rf->hrname, construct_list, store);
