--- py/mock/backend.py.orig	2011-03-07 15:05:14.000000000 +0100
+++ py/mock/backend.py	2011-03-07 15:05:24.000000000 +0100
@@ -412,7 +412,7 @@
 
         # if hosted on EL{4,5} create /dev/tty node in chroot
         if mock.util.cmpKernelEVR(kver, '2.6.19') <= 0:
-            os.mknod(self.makeChrootPath("dev/tty", stat.S_IFCHR|0666, os.makedev(5, 0)))
+            os.mknod(self.makeChrootPath("dev/tty"), stat.S_IFCHR|0666, os.makedev(5, 0))
             getLog().debug("created /dev/tty node device node")
 
         # symlink /dev/fd in the chroot for everything except RHEL4
