--- ./topuser.c	2005-02-21 15:13:22.000000000 -0300
+++ /opt2/sarg/sarg-2.0.4/topuser.c	2005-02-26 22:17:02.419565736 -0300
@@ -65,11 +65,6 @@
     exit(1);
    }
 
-   if((fp_top1=fopen(top1,"w"))==NULL) {
-      fprintf(stderr, "SARG: (topuser) %s: %s\n",text[45],top1);
-      exit(1);
-   }
-
    if((fp_top2=fopen(top2,"w"))==NULL) {
     fprintf(stderr, "SARG: (topuser) %s: %s\n",text[45],top2);
     exit(1);
@@ -174,7 +169,6 @@
    if(strcmp(TopuserSortOrder,"normal") == 0)
       order[0]='\0';
 
-
    sprintf(csort,"export LC_ALL=C;sort -n -T %s %s -k %s -o %s %s", TempDir, order, sfield, top1, top2);
    system(csort);
 
@@ -661,8 +655,6 @@
    fclose(fp_top1);
    unlink(top1);
 
-   fclose(fp_top3);
-
    if((fp_ou=fopen(tusr,"w"))==NULL) {
       fprintf(stderr, "SARG: (topuser) %s: %s\n",text[45],tusr);
       exit(1);
@@ -672,7 +664,7 @@
    fputs(preg,fp_ou);
 
    fputs("</body>\n</html>\n",fp_top3);
-
+   fclose(fp_top3);
    fclose(fp_ou);
 
    return;

