--- atitvout.c~	2002-09-16 22:37:08.000000000 +0300
+++ atitvout.c	2002-12-25 20:42:26.000000000 +0200
@@ -118,7 +118,8 @@
     } else {
         if (strncmp(oem, "ATI MACH64", 10) == 0)
             radeon = 0;
-        else if (strncmp(oem, "ATI RAGE128", 11) == 0)
+        else if ((strncmp(oem, "ATI RAGE128", 11) == 0) ||
+                 (strncmp(oem, "ATI RV200",    9) == 0))
             radeon = 1;
         /* Note: Also it's possible to have here 'RG6' for first radeons */
         else if (strstr(oem, "RADEON") != 0)
