diff -up live/config.linux.shared live/config.linux
--- live/config.linux.shared	2010-01-16 04:04:54.000000000 +0100
+++ live/config.linux	2010-01-16 17:16:37.479611105 +0100
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
+COMPILE_OPTS =		$(INCLUDES) -I. $(CFLAGS) -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
@@ -9,9 +9,9 @@ OBJ =			o
 LINK =			c++ -o
 LINK_OPTS =		-L.
 CONSOLE_LINK_OPTS =	$(LINK_OPTS)
-LIBRARY_LINK =		ar cr 
-LIBRARY_LINK_OPTS =	
-LIB_SUFFIX =			a
+LIBRARY_LINK =		c++ -o
+LIBRARY_LINK_OPTS =	$(LINK_OPTS) -shared -Wl,-soname,$@.$(SO_VERSION) $(LIBADDS)
+LIB_SUFFIX =			so
 LIBS_FOR_CONSOLE_APPLICATION =
 LIBS_FOR_GUI_APPLICATION =
 EXE =
diff -up live/config.linux.static.shared live/config.linux.static
--- live/config.linux.static.shared	2010-01-16 17:16:50.562862926 +0100
+++ live/config.linux.static	2010-01-16 17:16:57.271866612 +0100
@@ -1,4 +1,4 @@
-COMPILE_OPTS =		$(INCLUDES) -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
+COMPILE_OPTS =		$(INCLUDES) -I. $(CFLAGS) -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
diff -up live/Makefile.tail.shared live/Makefile.tail
--- live/Makefile.tail.shared	2010-01-16 04:04:54.000000000 +0100
+++ live/Makefile.tail	2010-01-16 17:16:37.479611105 +0100
@@ -10,12 +10,12 @@ TESTPROGS_DIR = testProgs
 MEDIA_SERVER_DIR = mediaServer
 
 all:
-	cd $(LIVEMEDIA_DIR) ; $(MAKE)
-	cd $(GROUPSOCK_DIR) ; $(MAKE)
-	cd $(USAGE_ENVIRONMENT_DIR) ; $(MAKE)
-	cd $(BASIC_USAGE_ENVIRONMENT_DIR) ; $(MAKE)
-	cd $(TESTPROGS_DIR) ; $(MAKE)
-	cd $(MEDIA_SERVER_DIR) ; $(MAKE)
+	cd $(LIVEMEDIA_DIR) ; C_FLAGS="$(C_FLAGS)" $(MAKE)
+	cd $(GROUPSOCK_DIR) ; C_FLAGS="$(C_FLAGS)" $(MAKE)
+	cd $(USAGE_ENVIRONMENT_DIR) ; C_FLAGS="$(C_FLAGS)" $(MAKE)
+	cd $(BASIC_USAGE_ENVIRONMENT_DIR) ; C_FLAGS="$(C_FLAGS)" $(MAKE)
+	cd $(TESTPROGS_DIR) ; C_FLAGS="$(C_FLAGS)" $(MAKE)
+	cd $(MEDIA_SERVER_DIR) ; C_FLAGS="$(C_FLAGS)" $(MAKE)
 
 clean:
 	cd $(LIVEMEDIA_DIR) ; $(MAKE) clean
