--- fritz/src/tools.h.xchg	2005-07-07 00:00:00.000000000 +0200
+++ fritz/src/tools.h	2007-03-29 01:48:00.000000000 +0200
@@ -71,15 +71,6 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-static inline unsigned long atomic_xchg (
-	volatile atomic_t *	v, 
-	unsigned		value
-) {
-	return __xchg (value, &v->counter, sizeof (unsigned));
-} /* atomic_xchg */
-
-/*---------------------------------------------------------------------------*\
-\*---------------------------------------------------------------------------*/
 #ifndef NDEBUG
 extern unsigned hallocated (void);
 extern int hvalid (void *);
