--- beecrypt-3.0.1/python/Makefile.am.rh	2003-06-19 05:38:10.000000000 -0400
+++ beecrypt-3.0.1/python/Makefile.am	2003-06-30 15:05:45.000000000 -0400
@@ -19,21 +19,16 @@
 
 noinst_HEADERS = mpw-py.h rng-py.h
 
-mylibs= $(top_builddir)/.libs/libbeecrypt.so
+mylibs= $(top_builddir)/libbeecrypt.la
 
 LDADD =
 
 pythondir = $(pylibdir)/site-packages
-python_PROGRAMS = _bc.so
+python_LTLIBRARIES = _bc.la
 
-_bc_so_SOURCES =
-_bc_so_LDFLAGS = $(mylibs) $(LIBS) -shared -Wl,-soname,_bc.so
-
-noinst_LTLIBRARIES = libbc.la
-libbc_la_SOURCES = _bc-py.c mpw-py.c rng-py.c
-
-_bc.so$(EXEEXT): $(libbc_la_OBJECTS)
-	$(LINK) -o $@ $(libbc_la_OBJECTS) $(_bc_so_LDFLAGS)
+_bc_la_SOURCES = _bc-py.c mpw-py.c rng-py.c
+_bc_la_LDFLAGS = -avoid-version -module
+_bc_la_LIBADD = $(mylibs)
 
 splint_srcs = _bc-py.c $(libbc_la_sources)
 
