--- bind-9.1.0b1/bin/named/include/named/globals.h.varrun	Wed Dec 20 23:22:36 2000
+++ bind-9.1.0b1/bin/named/include/named/globals.h	Wed Dec 20 23:23:00 2000
@@ -98,9 +98,9 @@
 EXTERN isc_boolean_t		ns_g_logstderr		INIT(ISC_FALSE);
 
 EXTERN const char *		ns_g_defaultpidfile 	INIT(NS_LOCALSTATEDIR
-							     "/run/named.pid");
+							     "/run/named/named.pid");
 EXTERN const char *		lwresd_g_defaultpidfile INIT(NS_LOCALSTATEDIR
-							     "/run/lwresd.pid");
+							     "/run/named/lwresd.pid");
 EXTERN const char *		ns_g_username		INIT(NULL);
 
 /*
