--- bind-9.1.3rc1/lib/dns/master.c.ttl	Wed May 23 19:50:13 2001
+++ bind-9.1.3rc1/lib/dns/master.c	Fri May 25 14:49:41 2001
@@ -929,11 +929,12 @@
 				 */
 				if (!ctx->ttl_known &&
 				    !ctx->default_ttl_known) {
-					(*callbacks->error)(callbacks,
-					    "%s: %s:%lu: no TTL specified",
+					UNEXPECTED_ERROR(__FILE__, __LINE__,
+					    "%s: %s:%lu: no TTL specified. THIS
+					    ZONE WILL NO LONGER WORK IN FUTURE VERSIONS. Add a TTL.",
 					    "dns_master_load", source, line);
-					result = DNS_R_NOTTL;
-					goto insist_and_cleanup;
+					ctx->ttl = 86400;
+					ctx->ttl_known = ISC_TRUE;
 				} else if (ctx->default_ttl_known) {
 					ctx->ttl = ctx->default_ttl;
 				}
@@ -1193,12 +1194,12 @@
 			/*
 			 * BIND 4 / 8 'USE_SOA_MINIMUM' could be set here.
 			 */
-			(*callbacks->error)(callbacks,
-					    "%s: %s:%lu: no TTL specified",
+			UNEXPECTED_ERROR(__FILE__, __LINE__,
+					    "%s: %s:%lu: no TTL specified. THIS ZONE WILL NO LONGER WORK IN FUTURE VERSIONS. Add a TTL.",
 					    "dns_master_load",
 					    source, line);
-			result = DNS_R_NOTTL;
-			goto insist_and_cleanup;
+			ctx->ttl = 86400;
+			ctx->ttl_known = ISC_TRUE;
 		} else if (ctx->default_ttl_known) {
 			ctx->ttl = ctx->default_ttl;
 		} else if (ctx->warn_1035) {
