diff -Naur --exclude='*.ui' smplayer-0.6.8-orig/src/basegui.cpp smplayer-0.6.8/src/basegui.cpp
--- smplayer-0.6.8-orig/src/basegui.cpp	2009-07-15 03:08:26.000000000 +0400
+++ smplayer-0.6.8/src/basegui.cpp	2009-07-31 21:01:20.000000000 +0400
@@ -38,6 +38,9 @@
 #include <QDesktopServices>
 #include <QInputDialog>
 
+// ZYV - need to include this for Qt4.2
+#include <QTextStream>
+
 #include <cmath>
 
 #include "mplayerwindow.h"
diff -Naur --exclude='*.ui' smplayer-0.6.8-orig/src/findsubtitles/filedownloader/filedownloader.cpp smplayer-0.6.8/src/findsubtitles/filedownloader/filedownloader.cpp
--- smplayer-0.6.8-orig/src/findsubtitles/filedownloader/filedownloader.cpp	2009-07-12 14:08:27.000000000 +0400
+++ smplayer-0.6.8/src/findsubtitles/filedownloader/filedownloader.cpp	2009-07-31 21:16:48.000000000 +0400
@@ -47,15 +47,22 @@
 
 void FileDownloader::setProxy(QNetworkProxy proxy) {
 	http->abort();
-	http->setProxy(proxy);
+
+	// ZYV - proxy trick
+//	http->setProxy(proxy);
+	http->setProxy(proxy.hostName(), proxy.port(), proxy.user(), proxy.password());
 
 	qDebug("FileDownloader::setProxy: host: '%s' port: %d type: %d",
            proxy.hostName().toUtf8().constData(), proxy.port(), proxy.type());
 }
 
 void FileDownloader::download(QUrl url) {
-	QHttp::ConnectionMode mode = url.scheme().toLower() == "https" ? QHttp::ConnectionModeHttps : QHttp::ConnectionModeHttp;
-	http->setHost(url.host(), mode, url.port() == -1 ? 0 : url.port());
+
+	// ZYV - This is only supported in Qt4.3 and later
+//	QHttp::ConnectionMode mode = url.scheme().toLower() == "https" ? QHttp::ConnectionModeHttps : QHttp::ConnectionModeHttp;
+//	http->setHost(url.host(), mode, url.port() == -1 ? 0 : url.port());
+
+	http->setHost(url.host(), url.port() == -1 ? 0 : url.port());
     
 	if (!url.userName().isEmpty())
 		http->setUser(url.userName(), url.password());
diff -Naur --exclude='*.ui' smplayer-0.6.8-orig/src/findsubtitles/findsubtitlesconfigdialog.cpp smplayer-0.6.8/src/findsubtitles/findsubtitlesconfigdialog.cpp
--- smplayer-0.6.8-orig/src/findsubtitles/findsubtitlesconfigdialog.cpp	2009-01-02 01:41:28.000000000 +0300
+++ smplayer-0.6.8/src/findsubtitles/findsubtitlesconfigdialog.cpp	2009-07-31 21:20:05.000000000 +0400
@@ -24,8 +24,18 @@
 {
 	setupUi(this);
 
-	proxy_type_combo->addItem( tr("Http"), QNetworkProxy::HttpProxy);
-	proxy_type_combo->addItem( tr("Socks5"), QNetworkProxy::Socks5Proxy);
+	// ZYV - HTTP Proxy type is only supported in Qt4.3+
+
+	// Anyway, there are no calls to setApplicationProxy() anywhere in the code.
+	// These proxy settings are only used with QHttp. Which in Qt4.2 only supports 
+	// HTTP proxy :-) So what we are going to do is just to ignore Socks5Proxy and 
+	// use these settings for HTTP proxy in QHttp instead. Sorry, it's by no means 
+	// an elegant solution, but supporting Socks5 would require way too many changes 
+	// in the upstream code. Better solutions are welcome.
+
+//	proxy_type_combo->addItem( tr("Http"), QNetworkProxy::HttpProxy);
+//	proxy_type_combo->addItem( tr("Socks5"), QNetworkProxy::Socks5Proxy);
+	proxy_type_combo->addItem( tr("Http"), QNetworkProxy::Socks5Proxy);
 
 	use_proxy_check->setWhatsThis( tr("Enable/disable the use of the proxy.") );
 	proxy_hostname_edit->setWhatsThis( tr("The host name of the proxy.") );
diff -Naur --exclude='*.ui' smplayer-0.6.8-orig/src/findsubtitles/findsubtitleswindow.cpp smplayer-0.6.8/src/findsubtitles/findsubtitleswindow.cpp
--- smplayer-0.6.8-orig/src/findsubtitles/findsubtitleswindow.cpp	2009-07-12 03:30:24.000000000 +0400
+++ smplayer-0.6.8/src/findsubtitles/findsubtitleswindow.cpp	2009-07-31 23:53:03.000000000 +0400
@@ -147,8 +147,13 @@
 	language_filter->setCurrentIndex(0);
 
 	// Proxy
+
+	// ZYV - HTTP Proxy is only supported in Qt4.3+ - make Socks5 default
+	// (this is actually fake, becuase these settings are used for HTTP)
+
 	use_proxy = false;
-	proxy_type = QNetworkProxy::HttpProxy;
+//	proxy_type = QNetworkProxy::HttpProxy;
+	proxy_type = QNetworkProxy::Socks5Proxy;
 	proxy_host = "";
 	proxy_port = 0;
 	proxy_username = "";
@@ -169,7 +174,10 @@
 
 void FindSubtitlesWindow::setProxy(QNetworkProxy proxy) {
 	downloader->abort();
-	downloader->setProxy(proxy);
+
+	// ZYV - Different calling conventions for Qt < 4.3
+//	downloader->setProxy(proxy);
+	downloader->setProxy(proxy.hostName(), proxy.port(), proxy.user(), proxy.password());
 
 #ifdef DOWNLOAD_SUBS
 	file_downloader->setProxy(proxy);
diff -Naur --exclude='*.ui' smplayer-0.6.8-orig/src/videopreview/videopreview.cpp smplayer-0.6.8/src/videopreview/videopreview.cpp
--- smplayer-0.6.8-orig/src/videopreview/videopreview.cpp	2009-03-18 13:34:40.000000000 +0300
+++ smplayer-0.6.8/src/videopreview/videopreview.cpp	2009-07-31 23:54:28.000000000 +0400
@@ -324,7 +324,11 @@
 	}
 
 	if (run.thumbnail_width == 0) {
-		int spacing = grid_layout->horizontalSpacing() * (prop.n_cols-1);
+
+		// ZYV - no horizontalSpacing in Qt4.2
+//		int spacing = grid_layout->horizontalSpacing() * (prop.n_cols-1);
+		int spacing = grid_layout->spacing() * (prop.n_cols-1);
+
 		if (spacing < 0) spacing = 0;
 		qDebug("VideoPreview::addPicture: spacing: %d", spacing);
 		run.thumbnail_width = (prop.max_width - spacing) / prop.n_cols;
