diff -uNr stellarium-0.8.0-orig/src/s_gui.h stellarium-0.8.0/src/s_gui.h
--- stellarium-0.8.0-orig/src/s_gui.h	2006-04-29 17:13:47.000000000 +0200
+++ stellarium-0.8.0/src/s_gui.h	2006-05-06 23:31:25.000000000 +0200
@@ -510,7 +510,7 @@
 		
 	private:
 		callback<void> onChangeCallback;
-		void ScrollBar::adjustSize(void);
+		void adjustSize(void);
 		Button scrollBt;
 		bool vertical;
 		unsigned int scrollOffset, scrollSize;
@@ -837,7 +837,7 @@
 	class City 
 	{
 	public:
-		City::City(const string& _name = "", const string& _state = "", const string& _country = "", 
+		City(const string& _name = "", const string& _state = "", const string& _country = "", 
 			double _longitude = 0.f, double _latitude = 0.f, float zone = 0, int _showatzoom = 0, int _altitude = 0);
 		void addCity(const string& _name = "", const string& _state = "", const string& _country = "", 
 			double _longitude = 0.f, double _latitude = 0.f, float zone = 0, int _showatzoom = 0, int _altitude = 0);
