diff -Naupr audacity-src-1.3.0b-beta.orig/configure audacity-src-1.3.0b-beta/configure
--- audacity-src-1.3.0b-beta.orig/configure	2005-11-27 17:47:27.000000000 +0100
+++ audacity-src-1.3.0b-beta/configure	2005-11-27 17:47:27.000000000 +0100
@@ -5103,23 +5103,23 @@ fi
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo "$as_me:$LINENO: checking for samplerate >= 0.15.0" >&5
-echo $ECHO_N "checking for samplerate >= 0.15.0... $ECHO_C" >&6
+        echo "$as_me:$LINENO: checking for samplerate >= 0.0.15" >&5
+echo $ECHO_N "checking for samplerate >= 0.0.15... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "samplerate >= 0.15.0" ; then
+        if $PKG_CONFIG --exists "samplerate >= 0.0.15" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking SAMPLERATE_CFLAGS" >&5
 echo $ECHO_N "checking SAMPLERATE_CFLAGS... $ECHO_C" >&6
-            SAMPLERATE_CFLAGS=`$PKG_CONFIG --cflags "samplerate >= 0.15.0"`
+            SAMPLERATE_CFLAGS=`$PKG_CONFIG --cflags "samplerate >= 0.0.15"`
             echo "$as_me:$LINENO: result: $SAMPLERATE_CFLAGS" >&5
 echo "${ECHO_T}$SAMPLERATE_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking SAMPLERATE_LIBS" >&5
 echo $ECHO_N "checking SAMPLERATE_LIBS... $ECHO_C" >&6
-            SAMPLERATE_LIBS=`$PKG_CONFIG --libs "samplerate >= 0.15.0"`
+            SAMPLERATE_LIBS=`$PKG_CONFIG --libs "samplerate >= 0.0.15"`
             echo "$as_me:$LINENO: result: $SAMPLERATE_LIBS" >&5
 echo "${ECHO_T}$SAMPLERATE_LIBS" >&6
         else
@@ -5127,7 +5127,7 @@ echo "${ECHO_T}$SAMPLERATE_LIBS" >&6
             SAMPLERATE_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            SAMPLERATE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "samplerate >= 0.15.0"`
+            SAMPLERATE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "samplerate >= 0.0.15"`
 
         fi
 
