Index: lib/Video/DVDRip/JobPlanner.pm
===================================================================
RCS file: /home/cvsroot/dvdrip/lib/Video/DVDRip/JobPlanner.pm,v
retrieving revision 1.14.2.2
diff -u -r1.14.2.2 JobPlanner.pm
--- lib/Video/DVDRip/JobPlanner.pm	24 Mar 2007 11:01:07 -0000	1.14.2.2
+++ lib/Video/DVDRip/JobPlanner.pm	1 Apr 2007 14:13:43 -0000
@@ -498,7 +498,7 @@
                     );
                 }
             }
-            if ( $self->exists("ffmpeg") and $buffer =~ /frame=\s*1\s*q=/ ) {
+            if ( $self->exists("ffmpeg") and $buffer =~ /frame=\s*1.*?q\s*=/ ) {
                 $got_frame_with_ffmpeg = 1;
             }
         },
