--- gpm-1.20.1/configure.in.math	Tue Jun 24 10:52:44 2003
+++ gpm-1.20.1/configure.in	Tue Jun 24 10:53:54 2003
@@ -128,7 +128,7 @@
 AC_SUBST(lispdir)
 
 CPPFLAGS='-I$(srcdir) $(DEFS) -include headers/config.h -Wall -DSYSCONFDIR="\"$(sysconfdir)\"" -DSBINDIR="\"$(sbindir)\""'
-LDFLAGS='-L$(srcdir)'
+LDFLAGS='-lm -L$(srcdir)'
 
 dnl AC_DEFINE_UNQUOTED(SYSCONFDIR,"$sysconfdir")
 dnl AC_DEFINE_UNQUOTED(SBINDIR,"$sbindir")
