--- pine4.21/pico/osdep/term.cap.fixhome	Tue Mar  7 16:12:39 2000
+++ pine4.21/pico/osdep/term.cap	Tue Mar  7 16:13:30 2000
@@ -133,8 +133,8 @@
      */
     kpinsert("\033[4J", KEY_PGUP, 1);
     kpinsert("\033[3J", KEY_PGDN, 1);
-    kpinsert("\033[2J", KEY_HOME, 1);
-    kpinsert("\033[N",  KEY_END, 1);
+    kpinsert("\033[1~", KEY_HOME, 1);
+    kpinsert("\033[4~", KEY_END, 1);
 
     /* 
      * ANSI mode.
--- pine4.21/pico/osdep/term.inf.fixhome	Tue Mar  7 16:13:09 2000
+++ pine4.21/pico/osdep/term.inf	Tue Mar  7 16:13:20 2000
@@ -137,8 +137,8 @@
      */
     kpinsert("\033[4J", KEY_PGUP, 1);
     kpinsert("\033[3J", KEY_PGDN, 1);
-    kpinsert("\033[2J", KEY_HOME, 1);
-    kpinsert("\033[N",  KEY_END, 1);
+    kpinsert("\033[1~", KEY_HOME, 1);
+    kpinsert("\033[4~", KEY_END, 1);
 
     /* 
      * ANSI mode.
