diff -uNr pine4.21.orig/pine/osdep/os-lnx.h pine4.21/pine/osdep/os-lnx.h
--- pine4.21.orig/pine/osdep/os-lnx.h	Thu Sep  7 15:41:48 2000
+++ pine4.21/pine/osdep/os-lnx.h	Thu Sep  7 15:57:17 2000
@@ -223,7 +223,7 @@
 
 
 /*--------- Program employed by users to change their password ---------*/
-#define	PASSWD_PROG	"/bin/passwd"
+#define	PASSWD_PROG	"/usr/bin/passwd"
 
 
 /*-------------- A couple constants used to size arrays ----------------*/
