Index: gst/asfdemux/gstasfdemux.c
===================================================================
RCS file: /cvs/gstreamer/gst-plugins-ugly/gst/asfdemux/gstasfdemux.c,v
retrieving revision 1.95
retrieving revision 1.97
diff -u -r1.95 -r1.97
--- gst/asfdemux/gstasfdemux.c	17 Feb 2006 17:58:44 -0000	1.95
+++ gst/asfdemux/gstasfdemux.c	27 Mar 2006 10:39:03 -0000	1.97
@@ -828,6 +828,7 @@
   stream->frag_offset = 0;
   stream->sequence = 0;
   stream->delay = 0;
+  stream->first_pts = GST_CLOCK_TIME_NONE;
   stream->last_pts = GST_CLOCK_TIME_NONE;
   stream->fps_known = !is_video;        /* bit hacky for audio */
   stream->is_video = is_video;
@@ -1812,7 +1813,10 @@
   if (segment_info->frag_offset == 0) {
     /* new packet */
     stream->sequence = segment_info->sequence;
-    demux->pts = segment_info->frag_timestamp - demux->preroll;
+    if (!GST_CLOCK_TIME_IS_VALID (stream->first_pts))
+      stream->first_pts = segment_info->frag_timestamp - demux->preroll;
+    demux->pts =
+        segment_info->frag_timestamp - demux->preroll - stream->first_pts;
 
     /*
        if (stream->is_video) {
@@ -1871,7 +1875,8 @@
 #endif
         stream->frag_offset = 0;
       }
-      demux->pts = segment_info->frag_timestamp - demux->preroll;
+      demux->pts =
+          segment_info->frag_timestamp - demux->preroll - stream->first_pts;
 
       /*
          if (stream->is_video) {
@@ -2203,14 +2208,14 @@
 
   packet_info.multiple = ((flags & 0x01) == 0x01);
 
+  packet_length =
+      gst_asf_demux_get_var_length ((flags >> 5) & 0x03, p_data, p_size);
+
   sequence = gst_asf_demux_get_var_length ((flags >> 1) & 0x03, p_data, p_size);
 
   packet_info.padsize =
       gst_asf_demux_get_var_length ((flags >> 3) & 0x03, p_data, p_size);
 
-  packet_length =
-      gst_asf_demux_get_var_length ((flags >> 5) & 0x03, p_data, p_size);
-
   if (packet_length == 0)
     packet_length = demux->packet_size;
 
Index: gst/asfdemux/gstasfdemux.h
===================================================================
RCS file: /cvs/gstreamer/gst-plugins-ugly/gst/asfdemux/gstasfdemux.h,v
retrieving revision 1.18
retrieving revision 1.19
diff -u -r1.18 -r1.19
--- gst/asfdemux/gstasfdemux.h	15 Feb 2006 15:48:07 -0000	1.18
+++ gst/asfdemux/gstasfdemux.h	27 Mar 2006 10:39:03 -0000	1.19
@@ -47,6 +47,7 @@
   guint32     frag_offset;
   guint32     sequence;
   guint64     delay;
+  guint64     first_pts;
   guint64     last_pts;
   GstBuffer  *payload;
 
