diff -ru openal.orig/linux/Makefile.in openal/linux/Makefile.in
--- openal.orig/linux/Makefile.in	2002-09-06 00:54:36.000000000 +0200
+++ openal/linux/Makefile.in	2002-10-29 18:23:31.000000000 +0100
@@ -52,8 +52,8 @@
 install_minor_libs:
 	$(INSTALL) -d -m 755 $(DESTLIB)
 	$(INSTALL) -m 755 src/libopenal.$(SO_EXT).$(VERSION) $(DESTLIB)
-	$(LN_S) -f $(DESTLIB)/libopenal.$(SO_EXT).$(VERSION) $(DESTLIB)/libopenal.$(SO_EXT).$(MAJOR_VERSION)
-	$(LN_S) -f $(DESTLIB)/libopenal.$(SO_EXT).$(VERSION) $(DESTLIB)/libopenal.so
+	$(LN_S) -f libopenal.$(SO_EXT).$(VERSION) $(DESTLIB)/libopenal.$(SO_EXT).$(MAJOR_VERSION)
+	$(LN_S) -f libopenal.$(SO_EXT).$(VERSION) $(DESTLIB)/libopenal.so
 
 install_major_lib_only:
 	$(INSTALL) -d -m 755 $(DESTLIB)
