diff -up facter-1.5.9/lib/facter/hardwareisa.rb.sl facter-1.5.9/lib/facter/hardwareisa.rb
--- facter-1.5.9/lib/facter/hardwareisa.rb.sl	2011-05-20 12:06:13.000000000 -0600
+++ facter-1.5.9/lib/facter/hardwareisa.rb	2011-05-26 09:10:20.234403286 -0600
@@ -1,4 +1,4 @@
 Facter.add(:hardwareisa) do
     setcode 'uname -p', '/bin/sh'
-    confine :operatingsystem => %w{Solaris Linux Fedora RedHat CentOS SuSE SLES Debian Ubuntu Gentoo FreeBSD OpenBSD NetBSD OEL OVS GNU/kFreeBSD}
+    confine :operatingsystem => %w{Solaris Linux Fedora RedHat CentOS Scientific SuSE SLES Debian Ubuntu Gentoo FreeBSD OpenBSD NetBSD OEL OVS GNU/kFreeBSD}
 end
diff -up facter-1.5.9/lib/facter/lsbmajdistrelease.rb.sl facter-1.5.9/lib/facter/lsbmajdistrelease.rb
--- facter-1.5.9/lib/facter/lsbmajdistrelease.rb.sl	2011-05-20 12:06:13.000000000 -0600
+++ facter-1.5.9/lib/facter/lsbmajdistrelease.rb	2011-05-26 09:10:27.690134946 -0600
@@ -3,7 +3,7 @@
 require 'facter'
 
 Facter.add("lsbmajdistrelease") do
-    confine :operatingsystem => %w{Linux Fedora RedHat CentOS SuSE SLES Debian Ubuntu Gentoo OEL OVS GNU/kFreeBSD}
+    confine :operatingsystem => %w{Linux Fedora RedHat CentOS Scientific SuSE SLES Debian Ubuntu Gentoo OEL OVS GNU/kFreeBSD}
     setcode do
         if /(\d*)\./i =~ Facter.value(:lsbdistrelease)
             result=$1
diff -up facter-1.5.9/lib/facter/macaddress.rb.sl facter-1.5.9/lib/facter/macaddress.rb
--- facter-1.5.9/lib/facter/macaddress.rb.sl	2011-05-20 12:06:13.000000000 -0600
+++ facter-1.5.9/lib/facter/macaddress.rb	2011-05-26 09:10:35.028870874 -0600
@@ -1,7 +1,7 @@
 require 'facter/util/macaddress'
 
 Facter.add(:macaddress) do
-    confine :operatingsystem => %w{Solaris Linux Fedora RedHat CentOS SuSE SLES Debian Gentoo Ubuntu OEL OVS GNU/kFreeBSD}
+    confine :operatingsystem => %w{Solaris Linux Fedora RedHat CentOS Scientific SuSE SLES Debian Gentoo Ubuntu OEL OVS GNU/kFreeBSD}
     setcode do
         ether = []
         output = %x{/sbin/ifconfig -a}
diff -up facter-1.5.9/lib/facter/operatingsystemrelease.rb.sl facter-1.5.9/lib/facter/operatingsystemrelease.rb
--- facter-1.5.9/lib/facter/operatingsystemrelease.rb.sl	2011-05-20 12:06:13.000000000 -0600
+++ facter-1.5.9/lib/facter/operatingsystemrelease.rb	2011-05-26 09:11:08.932650813 -0600
@@ -1,8 +1,8 @@
 Facter.add(:operatingsystemrelease) do
-    confine :operatingsystem => %w{CentOS Fedora oel ovs RedHat MeeGo}
+    confine :operatingsystem => %w{CentOS Fedora oel ovs RedHat MeeGo Scientific}
     setcode do
         case Facter.value(:operatingsystem)
-        when "CentOS", "RedHat"
+        when "CentOS", "RedHat", "Scientific"
             releasefile = "/etc/redhat-release"
         when "Fedora"
             releasefile = "/etc/fedora-release"
diff -up facter-1.5.9/lib/facter/uniqueid.rb.sl facter-1.5.9/lib/facter/uniqueid.rb
--- facter-1.5.9/lib/facter/uniqueid.rb.sl	2011-04-22 14:06:55.000000000 -0600
+++ facter-1.5.9/lib/facter/uniqueid.rb	2011-05-26 09:11:17.418345424 -0600
@@ -1,4 +1,4 @@
 Facter.add(:uniqueid) do
     setcode 'hostid', '/bin/sh'
-    confine :operatingsystem => %w{Solaris Linux Fedora RedHat CentOS SuSE SLES Debian Ubuntu Gentoo AIX OEL OVS GNU/kFreeBSD}
+    confine :operatingsystem => %w{Solaris Linux Fedora RedHat CentOS Scientific SuSE SLES Debian Ubuntu Gentoo AIX OEL OVS GNU/kFreeBSD}
 end
diff -up facter-1.5.9/spec/unit/operatingsystemrelease_spec.rb.sl facter-1.5.9/spec/unit/operatingsystemrelease_spec.rb
--- facter-1.5.9/spec/unit/operatingsystemrelease_spec.rb.sl	2011-04-22 14:06:55.000000000 -0600
+++ facter-1.5.9/spec/unit/operatingsystemrelease_spec.rb	2011-05-26 09:11:39.517550163 -0600
@@ -15,14 +15,15 @@ describe "Operating System Release fact"
     end
 
     test_cases = {
-        "CentOS"    => "/etc/redhat-release",
-        "RedHat"    => "/etc/redhat-release",
-        "Fedora"    => "/etc/fedora-release",
-        "MeeGo"     => "/etc/meego-release",
-        "OEL"       => "/etc/enterprise-release",
-        "oel"       => "/etc/enterprise-release",
-        "OVS"       => "/etc/ovs-release",
-        "ovs"       => "/etc/ovs-release",
+        "CentOS"     => "/etc/redhat-release",
+        "RedHat"     => "/etc/redhat-release",
+        "Scientific" => "/etc/redhat-release",
+        "Fedora"     => "/etc/fedora-release",
+        "MeeGo"      => "/etc/meego-release",
+        "OEL"        => "/etc/enterprise-release",
+        "oel"        => "/etc/enterprise-release",
+        "OVS"        => "/etc/ovs-release",
+        "ovs"        => "/etc/ovs-release",
     }
 
     test_cases.each do |system, file|
