diff -Naupr libcaca-0.9.orig/doc/Makefile.am libcaca-0.9/doc/Makefile.am
--- libcaca-0.9.orig/doc/Makefile.am	2004-02-02 04:00:20.000000000 +0100
+++ libcaca-0.9/doc/Makefile.am	2004-11-03 13:54:17.583171192 +0100
@@ -44,8 +44,6 @@ if BUILD_DOCUMENTATION
 	$(mkinstalldirs) $(DESTDIR)$(datadir)/doc/libcaca-dev/html
 	cp `find html -name '*.html' -o -name '*.gif' -o -name '*.png'` \
 	  $(DESTDIR)$(datadir)/doc/libcaca-dev/html
-	$(mkinstalldirs) $(DESTDIR)$(mandir)/man3
-	cp man/man3/*.3caca $(DESTDIR)$(mandir)/man3
 	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
 	rm -f $(DESTDIR)$(mandir)/man1/cacaball.1
 	ln -s cacademo.1 $(DESTDIR)$(mandir)/man1/cacaball.1
diff -Naupr libcaca-0.9.orig/doc/Makefile.in libcaca-0.9/doc/Makefile.in
--- libcaca-0.9.orig/doc/Makefile.in	2004-02-02 04:00:44.000000000 +0100
+++ libcaca-0.9/doc/Makefile.in	2004-11-03 13:54:10.763207984 +0100
@@ -404,8 +404,6 @@ install-data-local:
 @BUILD_DOCUMENTATION_TRUE@	$(mkinstalldirs) $(DESTDIR)$(datadir)/doc/libcaca-dev/html
 @BUILD_DOCUMENTATION_TRUE@	cp `find html -name '*.html' -o -name '*.gif' -o -name '*.png'` \
 @BUILD_DOCUMENTATION_TRUE@	  $(DESTDIR)$(datadir)/doc/libcaca-dev/html
-@BUILD_DOCUMENTATION_TRUE@	$(mkinstalldirs) $(DESTDIR)$(mandir)/man3
-@BUILD_DOCUMENTATION_TRUE@	cp man/man3/*.3caca $(DESTDIR)$(mandir)/man3
 @BUILD_DOCUMENTATION_TRUE@	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
 @BUILD_DOCUMENTATION_TRUE@	rm -f $(DESTDIR)$(mandir)/man1/cacaball.1
 @BUILD_DOCUMENTATION_TRUE@	ln -s cacademo.1 $(DESTDIR)$(mandir)/man1/cacaball.1
