--- pinentry-0.7.2/doc/pinentry.texi~	2005-01-27 09:58:03.000000000 +0200
+++ pinentry-0.7.2/doc/pinentry.texi	2005-10-18 21:03:51.000000000 +0300
@@ -62,7 +62,7 @@
 @ifnottex
 @dircategory GNU Utilities
 @direntry
-* pinentry::               Ask securely for a passphrase or PIN.
+* pinentry: (pinentry).    Ask securely for a passphrase or PIN.
 @end direntry
 This file documents the use and the internals of the @pinentry{}.
 
--- pinentry-0.7.2/doc/pinentry.info~	2005-01-27 10:06:25.000000000 +0200
+++ pinentry-0.7.2/doc/pinentry.info	2005-10-18 21:04:11.000000000 +0300
@@ -3,7 +3,7 @@
 
 INFO-DIR-SECTION GNU Utilities
 START-INFO-DIR-ENTRY
-* pinentry::               Ask securely for a passphrase or PIN.
+* pinentry: (pinentry).    Ask securely for a passphrase or PIN.
 END-INFO-DIR-ENTRY
    This file documents the use and the internals of the PINENTRY.
 
