--- xine-ui-0.99.3/src/xitk/mediamark.c.orig	2005-04-21 21:12:34.000000000 +0800
+++ xine-ui-0.99.3/src/xitk/mediamark.c	2005-04-21 21:15:31.000000000 +0800
@@ -3121,11 +3121,11 @@ static void mmkeditor_apply(xitk_widget_
     
     mrl = atoa(xitk_inputtext_get_text(mmkeditor->mrl));
     if(mrl && (!strlen(mrl)))
-      xine_strdupa((char *) mrl, (*mmkeditor->mmk)->mrl);
+      xine_strdupa(mrl, (*mmkeditor->mmk)->mrl);
     
     ident = atoa(xitk_inputtext_get_text(mmkeditor->ident));
     if(ident && (!strlen(ident)))
-      xine_strdupa((char *) ident, mrl);
+      xine_strdupa(ident, mrl);
 
     sub = xitk_inputtext_get_text(mmkeditor->sub);
     if(sub && (!strlen(sub)))
