--- m4/_xine.m4~	2003-12-24 15:04:09.000000000 +0200
+++ m4/_xine.m4	2005-07-10 21:43:53.000000000 +0300
@@ -37,11 +37,4 @@
 
-        if test x"$LIRC_PREFIX" != "x"; then
-           lirc_libprefix="$LIRC_PREFIX/lib"
-  	   LIRC_INCLUDE="-I$LIRC_PREFIX/include"
-        fi
-        for llirc in $lirc_libprefix /lib /usr/lib /usr/local/lib; do
-          AC_CHECK_FILE(["$llirc/liblirc_client.a"],
-             [LIRC_LIBS="$llirc/liblirc_client.a"]
-             AC_DEFINE([HAVE_LIRC],,[Define this if you have LIRC (liblir_client) installed]),,)
-        done
+        AC_DEFINE([HAVE_LIRC],,[Define this if you have LIRC (liblirc_client) installed])
+        LIRC_LIBS=-llirc_client
      else
