diff -ur xine-ui-0.99.3/src/xitk/help.c xine-ui-0.99.3-uifixups/src/xitk/help.c
--- xine-ui-0.99.3/src/xitk/help.c	2004-12-15 13:10:44.000000000 -0800
+++ xine-ui-0.99.3-uifixups/src/xitk/help.c	2005-07-07 15:44:39.000000000 -0700
@@ -392,7 +392,7 @@
   xitk_list_append_content ((XITK_WIDGET_LIST_LIST(help->widget_list)),
     (help->tabs = 
      xitk_noskin_tabs_create(help->widget_list, 
-			     &tab, 20, 24, WINDOW_WIDTH - 40, tabsfontname)));
+			     &tab, 20, 24, WINDOW_WIDTH - 40, 28, tabsfontname)));
   
   xitk_enable_and_show_widget(help->tabs);
 
diff -ur xine-ui-0.99.3/src/xitk/setup.c xine-ui-0.99.3-uifixups/src/xitk/setup.c
--- xine-ui-0.99.3/src/xitk/setup.c	2004-12-15 13:10:46.000000000 -0800
+++ xine-ui-0.99.3-uifixups/src/xitk/setup.c	2005-07-12 17:49:15.000000000 -0700
@@ -56,7 +56,7 @@
     int                  lbearing, rbearing, width, ascent, descent;                            \
     xitk_font_t         *fs;                                                                    \
                                                                                                 \
-    image = xitk_image_create_image(gGui->imlib_data, FRAME_WIDTH + 1, FRAME_HEIGHT + 1);       \
+    image = xitk_image_create_image(gGui->imlib_data, FRAME_WIDTH + 1, FRAME_HEIGHT + 2);       \
                                                                                                 \
     fs = xitk_font_load_font(gGui->display, boldfontname);                                      \
     xitk_font_set_font(fs, (XITK_WIDGET_LIST_GC(setup->widget_list)));                          \
@@ -1006,7 +1006,7 @@
   tab.userdata          = NULL;
   xitk_list_append_content ((XITK_WIDGET_LIST_LIST(setup->widget_list)),
     (setup->tabs = 
-     xitk_noskin_tabs_create(setup->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, tabsfontname)));
+     xitk_noskin_tabs_create(setup->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, 28, tabsfontname)));
   
   xitk_enable_and_show_widget(setup->tabs);
 
@@ -1047,7 +1047,6 @@
 
   if(rpos != setup->first_displayed) {
     setup->first_displayed = rpos;
-    setup_clear_tab();
     setup_paint_widgets();
   }
 }
diff -ur xine-ui-0.99.3/src/xitk/viewlog.c xine-ui-0.99.3-uifixups/src/xitk/viewlog.c
--- xine-ui-0.99.3/src/xitk/viewlog.c	2004-12-15 13:10:46.000000000 -0800
+++ xine-ui-0.99.3-uifixups/src/xitk/viewlog.c	2005-07-07 15:44:39.000000000 -0700
@@ -319,7 +319,7 @@
   tab.userdata          = NULL;
   xitk_list_append_content ((XITK_WIDGET_LIST_LIST(viewlog->widget_list)),
     (viewlog->tabs = 
-     xitk_noskin_tabs_create(viewlog->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, tabsfontname)));
+     xitk_noskin_tabs_create(viewlog->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, 28, tabsfontname)));
   xitk_enable_and_show_widget(viewlog->tabs);
 
   bg = xitk_image_create_xitk_pixmap(gGui->imlib_data, WINDOW_WIDTH, WINDOW_HEIGHT);
diff -ur xine-ui-0.99.3/src/xitk/xine-toolkit/tabs.c xine-ui-0.99.3-uifixups/src/xitk/xine-toolkit/tabs.c
--- xine-ui-0.99.3/src/xitk/xine-toolkit/tabs.c	2004-04-13 12:55:50.000000000 -0700
+++ xine-ui-0.99.3-uifixups/src/xitk/xine-toolkit/tabs.c	2005-07-07 15:44:39.000000000 -0700
@@ -323,7 +323,8 @@
  */
 xitk_widget_t *xitk_noskin_tabs_create(xitk_widget_list_t *wl,
 				       xitk_tabs_widget_t *t, 
-				       int x, int y, int width, char *fontname) {
+				       int x, int y, int width, int height,
+				       char *fontname) {
   xitk_widget_t         *mywidget;
   tabs_private_data_t   *private_data;
   
@@ -367,7 +368,10 @@
     XITK_WIDGET_INIT(&lb, t->imlibdata);
     XITK_WIDGET_INIT(&b, t->imlibdata);
 
-    private_data->bheight = fheight * 2;
+    if (height == 0)
+      private_data->bheight = fheight * 2;
+    else
+      private_data->bheight = height;
 
     for(i = 0; i < t->num_entries; i++) {
 
diff -ur xine-ui-0.99.3/src/xitk/xine-toolkit/test.c xine-ui-0.99.3-uifixups/src/xitk/xine-toolkit/test.c
--- xine-ui-0.99.3/src/xitk/xine-toolkit/test.c	2004-07-25 17:43:47.000000000 -0700
+++ xine-ui-0.99.3-uifixups/src/xitk/xine-toolkit/test.c	2005-07-07 15:44:39.000000000 -0700
@@ -509,7 +509,7 @@
   t.userdata          = NULL;
   xitk_list_append_content (XITK_WIDGET_LIST_LIST(test->widget_list),
 		    (test->tabs = 
-		     xitk_noskin_tabs_create(test->widget_list, &t, x, y, w, fontname)));
+		     xitk_noskin_tabs_create(test->widget_list, &t, x, y, w, 0, fontname)));
   xitk_enable_and_show_widget(test->tabs);
 
 }
diff -ur xine-ui-0.99.3/src/xitk/xine-toolkit/xitk.h xine-ui-0.99.3-uifixups/src/xitk/xine-toolkit/xitk.h
--- xine-ui-0.99.3/src/xitk/xine-toolkit/xitk.h	2004-12-16 13:35:30.000000000 -0800
+++ xine-ui-0.99.3-uifixups/src/xitk/xine-toolkit/xitk.h	2005-07-07 15:44:39.000000000 -0700
@@ -2351,7 +2351,8 @@
 
 xitk_widget_t *xitk_noskin_tabs_create(xitk_widget_list_t *wl,
 				       xitk_tabs_widget_t *t, 
-				       int x, int y, int width, char *fontname);
+				       int x, int y, int width, int height,
+				       char *fontname);
 int xitk_tabs_get_current_selected(xitk_widget_t *w);
 char *xitk_tabs_get_current_tab_selected(xitk_widget_t *w);
 void xitk_tabs_set_current_selected(xitk_widget_t *w, int select);

