--- src/augtool.c.orig	2008-07-20 01:46:54.025728456 +0200
+++ src/augtool.c	2008-07-20 01:47:11.610055232 +0200
@@ -37,7 +37,7 @@
     const char *help;
 };
 
-static const struct command const commands[];
+static const struct command commands[];
 
 static augeas *aug = NULL;
 static const char *const progname = "augtool";
@@ -298,7 +298,7 @@
     return cmd;
 }
 
-static const struct command const commands[] = {
+static const struct command commands[] = {
     { "ls",  1, 1, cmd_ls, "ls <PATH>",
       "List the direct children of PATH"
     },
