--- xmms.spec.orig	2004-10-13 21:12:42.000000000 +0200
+++ xmms.spec	2004-11-04 00:31:03.769122304 +0100
@@ -4,12 +4,12 @@
 Summary: A media player for X which resembles Winamp.
 Name: xmms
 Version: 1.2.10
-Release: 9
+Release: 9.1
 Epoch: 1
 License: GPL
 Group: Applications/Multimedia
 URL: http://www.xmms.org/
-Source: http://www.xmms.org/files/1.2.x/%{name}-%{version}.patched.tar.bz2
+Source: http://www.xmms.org/files/1.2.x/%{name}-%{version}.tar.bz2
 Source4: arts_output-%{artsplugin_ver}.tar.gz
 Source5: xmms.req
 Source6: xmms.xpm
@@ -36,6 +36,7 @@ BuildRequires: alsa-lib-devel glib2-deve
 PreReq: desktop-file-utils >= 0.9
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root
 Obsoletes: x11amp0.7-1-1 x11amp xmms-esd xmms-gl xmms-mikmod xmms-gnome
+Obsoletes: xmms-alsa alsa-xmms
 
 Conflicts: arts < 1.2.0-1.5
 
@@ -57,6 +58,15 @@ Requires: %{name} = %{epoch}:%{version} 
 The static libraries and header files needed for building plug-ins for
 the Xmms multimedia player.
 
+%package mp3
+Summary: XMMS plugin for mp3 playback.
+Group: Applications/Multimedia
+Requires: %{name} = %{epoch}:%{version}
+
+%description mp3
+This is the mp3 plugin for XMMS that was removed from Red Hat Linux because
+the patented mp3 format itself is theoretically GPL incompatible.
+
 %package skins
 Summary: Skins for the xmms multimedia player.
 Group: Applications/Multimedia
@@ -76,7 +86,7 @@ skins were obtained from http://www.xmms
 # Change the default skin
 %patch3 -p1 -b .default-skin
 # Don't build MP3 support, support bits for MP3 placeholder
-%patch4 -p1 -b .nomp3
+#patch4 -p1 -b .nomp3
 %if %{arts_plugin}
 # Link arts dynamically and detect its presence for choosing output plugin
 %patch5 -p1 -b .arts
@@ -110,8 +120,8 @@ make
 cd ..
 %endif
 
-gcc -fPIC $RPM_OPT_FLAGS -shared -Wl,-soname -Wl,librh_mp3.so -o librh_mp3.so \
-     %{SOURCE8} -I. `gtk-config --cflags gtk`
+#gcc -fPIC $RPM_OPT_FLAGS -shared -Wl,-soname -Wl,librh_mp3.so -o librh_mp3.so \
+#     %{SOURCE8} -I. `gtk-config --cflags gtk`
 
 %install
 rm -rf %{buildroot}
@@ -125,7 +135,7 @@ make install DESTDIR=%{buildroot}
 cd ..
 %endif
 
-install -m 755 librh_mp3.so %{buildroot}%{_libdir}/xmms/Input
+#install -m 755 librh_mp3.so %{buildroot}%{_libdir}/xmms/Input
 
 mkdir -p %{buildroot}%{_datadir}/xmms/Skins
 pushd %{buildroot}%{_datadir}/xmms/Skins
@@ -169,7 +179,12 @@ rm -rf %{buildroot}
 %dir %{_libdir}/xmms
 %{_libdir}/xmms/Effect
 %{_libdir}/xmms/General
-%{_libdir}/xmms/Input
+%dir %{_libdir}/xmms/Input
+%{_libdir}/xmms/Input/libcdaudio.so
+%{_libdir}/xmms/Input/libmikmod.so
+%{_libdir}/xmms/Input/libtonegen.so
+%{_libdir}/xmms/Input/libvorbis.so
+%{_libdir}/xmms/Input/libwav.so
 %{_libdir}/xmms/Output
 %{_libdir}/xmms/Visualization
 %{_datadir}/applications/*
@@ -188,11 +203,23 @@ rm -rf %{buildroot}
 %{_libdir}/lib*.a
 %{_libdir}/lib*.so
 
+%files mp3
+%defattr(-,root,root)
+%dir %{_libdir}/xmms/Input
+%{_libdir}/xmms/Input/libmpg123.so
+
 %files skins
 %defattr(-,root,root)
 %{_datadir}/xmms/Skins
 
 %changelog
+* Thu Nov  4 2004 Matthias Saou <http://freshrpms.net/> 1:1.2.10-9.1
+- Put back pristine sources.
+- Added the usual mp3 sub-package.
+- Removed nomp3 patch and commented out rh_mp3 plugin build/install.
+- Added xmms-alsa obsoletes.
+- No longer fix the missing gtk+-devel dep of the devel package, yeah!
+
 * Wed Oct 13 2004 Colin Walters <walters@redhat.com> 1:1.2.10-9
 - Correct update-desktop-database correction for postun
 
