--- mozilla/modules/libpref/src/unix/unix.js.foo	Thu Sep 11 09:42:30 2003
+++ mozilla/modules/libpref/src/unix/unix.js	Fri Oct 31 19:27:40 2003
@@ -380,8 +380,10 @@
 pref("print.postscript.nativefont.ar",             "");
 pref("print.postscript.nativefont.el",             "");
 pref("print.postscript.nativefont.he",             "");
-pref("print.postscript.nativefont.ja",             "");
-pref("print.postscript.nativefont.ko",             "");
+pref("print.postscript.nativecode.ja",             "euc-jp");
+pref("print.postscript.nativefont.ja",             "Ryumin-Light-EUC-H");
+pref("print.postscript.nativecode.ko",             "euc-kr");
+pref("print.postscript.nativefont.ko",             "Baekmuk-Gulim-KSC-EUC-H");
 pref("print.postscript.nativefont.th",             "");
 pref("print.postscript.nativefont.tr",             "");
 pref("print.postscript.nativefont.x-baltic",       "");
@@ -390,8 +392,10 @@
 pref("print.postscript.nativefont.x-unicode",      "");
 pref("print.postscript.nativefont.x-user-def",     "");
 pref("print.postscript.nativefont.x-western",      "");
-pref("print.postscript.nativefont.zh-CN",          "");
-pref("print.postscript.nativefont.zh-TW",          "");
+pref("print.postscript.nativecode.zh-CN", "gb18030");
+pref("print.postscript.nativefont.zh-CN", "MSungGBK-Light-GBK2K-H");
+pref("print.postscript.nativecode.zh-TW", "big5");
+pref("print.postscript.nativefont.zh-TW", "ShanHeiSun-Light-B5-H");
 pref("print.postscript.nativefont.zh-HK",          "");
 
 // For the download dialog
