--- siege-2.65/doc/Makefile.in.orig	2007-07-02 03:21:12.000000000 -0300
+++ siege-2.65/doc/Makefile.in	2007-07-02 03:22:55.000000000 -0300
@@ -512,16 +512,16 @@
 	else \
 	  $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC); \
 	fi 
-	@if test -f $(URLSTXT); then \
-	  if cmp -s $(srcdir)/siegerc $(URLSTXT); then echo ""; \
-	    else \
-	    echo "WARNING: File $(URLSTXT) already exists."; \
-	    echo "         It was NOT replaced with this installation."; \
-	  fi; \
-	else \
-	  $(mkinstalldirs) $(sysconfdir); \
-	  $(INSTALL_DATA) $(srcdir)/urls.txt $(URLSTXT); \
-	fi 
+	#@if test -f $(URLSTXT); then \
+	#  if cmp -s $(srcdir)/siegerc $(URLSTXT); then echo ""; \
+	#    else \
+	#    echo "WARNING: File $(URLSTXT) already exists."; \
+	#    echo "         It was NOT replaced with this installation."; \
+	#  fi; \
+	#else \
+	#  $(mkinstalldirs) $(sysconfdir); \
+	#  $(INSTALL_DATA) $(srcdir)/urls.txt $(URLSTXT); \
+	#fi 
 
 uninstall:
 	rm -f $(SIEGERC)
