diff -ur madwifi-ng-r1708-20060907.orig/ath/if_ath_ahb.c madwifi-ng-r1708-20060907/ath/if_ath_ahb.c
--- madwifi-ng-r1708-20060907.orig/ath/if_ath_ahb.c	2006-05-21 21:39:55.000000000 -0700
+++ madwifi-ng-r1708-20060907/ath/if_ath_ahb.c	2006-09-07 08:33:31.000000000 -0700
@@ -10,7 +10,6 @@
 #define	EXPORT_SYMTAB
 #endif
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/ath/if_ath.c madwifi-ng-r1708-20060907/ath/if_ath.c
--- madwifi-ng-r1708-20060907.orig/ath/if_ath.c	2006-09-02 21:39:21.000000000 -0700
+++ madwifi-ng-r1708-20060907/ath/if_ath.c	2006-09-07 08:33:31.000000000 -0700
@@ -44,7 +44,6 @@
  */
 #include "opt_ah.h"
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/ath/if_ath_pci.c madwifi-ng-r1708-20060907/ath/if_ath_pci.c
--- madwifi-ng-r1708-20060907.orig/ath/if_ath_pci.c	2006-07-21 01:00:32.000000000 -0700
+++ madwifi-ng-r1708-20060907/ath/if_ath_pci.c	2006-09-07 08:33:31.000000000 -0700
@@ -42,7 +42,6 @@
 #define	EXPORT_SYMTAB
 #endif
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/ath_rate/amrr/amrr.c madwifi-ng-r1708-20060907/ath_rate/amrr/amrr.c
--- madwifi-ng-r1708-20060907.orig/ath_rate/amrr/amrr.c	2006-07-04 03:23:35.000000000 -0700
+++ madwifi-ng-r1708-20060907/ath_rate/amrr/amrr.c	2006-09-07 08:33:31.000000000 -0700
@@ -43,7 +43,6 @@
  * "IEEE 802.11 Rate Adaptation: A Practical Approach" by
  *    Mathieu Lacage, Hossein Manshaei, Thierry Turletti
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/ath_rate/onoe/onoe.c madwifi-ng-r1708-20060907/ath_rate/onoe/onoe.c
--- madwifi-ng-r1708-20060907.orig/ath_rate/onoe/onoe.c	2006-07-04 03:23:35.000000000 -0700
+++ madwifi-ng-r1708-20060907/ath_rate/onoe/onoe.c	2006-09-07 08:33:31.000000000 -0700
@@ -39,7 +39,6 @@
 /*
  * Atsushi Onoe's rate control algorithm.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/ath_rate/sample/sample.c madwifi-ng-r1708-20060907/ath_rate/sample/sample.c
--- madwifi-ng-r1708-20060907.orig/ath_rate/sample/sample.c	2006-07-04 03:23:35.000000000 -0700
+++ madwifi-ng-r1708-20060907/ath_rate/sample/sample.c	2006-09-07 08:33:31.000000000 -0700
@@ -41,7 +41,6 @@
  * John Bicket's SampleRate control algorithm.
  */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/hal/linux/ah_osdep.c madwifi-ng-r1708-20060907/hal/linux/ah_osdep.c
--- madwifi-ng-r1708-20060907.orig/hal/linux/ah_osdep.c	2006-05-19 03:25:45.000000000 -0700
+++ madwifi-ng-r1708-20060907/hal/linux/ah_osdep.c	2006-09-07 08:33:31.000000000 -0700
@@ -41,7 +41,6 @@
 #define	EXPORT_SYMTAB
 #endif
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_acl.c madwifi-ng-r1708-20060907/net80211/ieee80211_acl.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_acl.c	2006-05-31 14:05:08.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_acl.c	2006-09-07 08:33:31.000000000 -0700
@@ -45,7 +45,6 @@
  * and if found the frame is either accepted (ACL_POLICY_ALLOW)
  * or rejected (ACL_POLICY_DENT).
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_beacon.c madwifi-ng-r1708-20060907/net80211/ieee80211_beacon.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_beacon.c	2006-05-19 13:42:19.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_beacon.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 beacon handling routines
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211.c madwifi-ng-r1708-20060907/net80211/ieee80211.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211.c	2006-07-04 03:22:11.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 generic handler
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto.c madwifi-ng-r1708-20060907/net80211/ieee80211_crypto.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto.c	2006-02-01 12:07:11.000000000 -0800
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_crypto.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 generic crypto support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kmod.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_ccmp.c madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_ccmp.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_ccmp.c	2006-06-13 07:09:51.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_ccmp.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
  * AP driver. The code is used with the consent of the author and
  * it's license is included below.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_none.c madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_none.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_none.c	2006-02-01 12:07:11.000000000 -0800
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_none.c	2006-09-07 08:34:29.000000000 -0700
@@ -34,7 +34,6 @@
 /*
  * IEEE 802.11 NULL crypto support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_tkip.c madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_tkip.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_tkip.c	2006-02-01 12:07:11.000000000 -0800
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_tkip.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
  * AP driver. The code is used with the consent of the author and
  * it's license is included below.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_wep.c madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_wep.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_crypto_wep.c	2006-02-06 12:20:57.000000000 -0800
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_crypto_wep.c	2006-09-07 08:33:30.000000000 -0700
@@ -34,7 +34,6 @@
 /*
  * IEEE 802.11 WEP crypto support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_input.c madwifi-ng-r1708-20060907/net80211/ieee80211_input.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_input.c	2006-07-05 20:23:08.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_input.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 input handling.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_linux.c madwifi-ng-r1708-20060907/net80211/ieee80211_linux.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_linux.c	2006-07-21 01:59:10.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_linux.c	2006-09-07 08:34:29.000000000 -0700
@@ -33,7 +33,6 @@
 /*
  * IEEE 802.11 support (Linux-specific code)
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kmod.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_monitor.c madwifi-ng-r1708-20060907/net80211/ieee80211_monitor.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_monitor.c	2006-09-05 19:24:34.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_monitor.c	2006-09-07 08:34:29.000000000 -0700
@@ -34,7 +34,6 @@
 /*
  * IEEE 802.11 monitor mode 
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kmod.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_node.c madwifi-ng-r1708-20060907/net80211/ieee80211_node.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_node.c	2006-06-13 01:50:37.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_node.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 node handling support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_output.c madwifi-ng-r1708-20060907/net80211/ieee80211_output.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_output.c	2006-06-09 19:17:05.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_output.c	2006-09-07 08:33:31.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 output handling.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_power.c madwifi-ng-r1708-20060907/net80211/ieee80211_power.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_power.c	2006-06-09 19:17:05.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_power.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 power save support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_proto.c madwifi-ng-r1708-20060907/net80211/ieee80211_proto.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_proto.c	2006-08-08 07:24:15.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_proto.c	2006-09-07 08:34:29.000000000 -0700
@@ -38,7 +38,6 @@
 /*
  * IEEE 802.11 protocol support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/kmod.h>
 #include <linux/module.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_scan_ap.c madwifi-ng-r1708-20060907/net80211/ieee80211_scan_ap.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_scan_ap.c	2006-08-08 05:24:37.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_scan_ap.c	2006-09-07 08:34:29.000000000 -0700
@@ -37,7 +37,6 @@
 /*
  * IEEE 802.11 ap scanning support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_scan.c madwifi-ng-r1708-20060907/net80211/ieee80211_scan.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_scan.c	2006-08-08 05:24:37.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_scan.c	2006-09-07 08:34:29.000000000 -0700
@@ -37,7 +37,6 @@
 /*
  * IEEE 802.11 scanning support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_scan_sta.c madwifi-ng-r1708-20060907/net80211/ieee80211_scan_sta.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_scan_sta.c	2006-08-08 05:24:37.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_scan_sta.c	2006-09-07 08:34:29.000000000 -0700
@@ -37,7 +37,6 @@
 /*
  * IEEE 802.11 station scanning support.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/skbuff.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_wireless.c madwifi-ng-r1708-20060907/net80211/ieee80211_wireless.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_wireless.c	2006-08-10 22:35:30.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_wireless.c	2006-09-07 08:34:29.000000000 -0700
@@ -39,7 +39,6 @@
 /*
  * Wireless extensions support for 802.11 common code.
  */
-#include <linux/config.h>
 
 #ifdef CONFIG_NET_WIRELESS
 #include <linux/version.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/ieee80211_xauth.c madwifi-ng-r1708-20060907/net80211/ieee80211_xauth.c
--- madwifi-ng-r1708-20060907.orig/net80211/ieee80211_xauth.c	2006-02-01 12:07:11.000000000 -0800
+++ madwifi-ng-r1708-20060907/net80211/ieee80211_xauth.c	2006-09-07 08:34:29.000000000 -0700
@@ -46,7 +46,6 @@
  * of the available callbacks--the user mode authenticator process works
  * entirely from messages about stations joining and leaving.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/net80211/if_media.c madwifi-ng-r1708-20060907/net80211/if_media.c
--- madwifi-ng-r1708-20060907.orig/net80211/if_media.c	2006-05-31 14:05:08.000000000 -0700
+++ madwifi-ng-r1708-20060907/net80211/if_media.c	2006-09-07 08:33:40.000000000 -0700
@@ -49,7 +49,6 @@
 #define	EXPORT_SYMTAB
 #endif
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
diff -ur madwifi-ng-r1708-20060907.orig/regression/ccmp/test_ccmp.c madwifi-ng-r1708-20060907/regression/ccmp/test_ccmp.c
--- madwifi-ng-r1708-20060907.orig/regression/ccmp/test_ccmp.c	2006-02-01 12:07:11.000000000 -0800
+++ madwifi-ng-r1708-20060907/regression/ccmp/test_ccmp.c	2006-09-07 08:34:29.000000000 -0700
@@ -44,7 +44,6 @@
  * you want; e.g. insmod ccmp_test tests=7 will run only test mpdu's
  * 1, 2, and 3.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/regression/tkip/test_tkip.c madwifi-ng-r1708-20060907/regression/tkip/test_tkip.c
--- madwifi-ng-r1708-20060907.orig/regression/tkip/test_tkip.c	2006-02-01 12:07:11.000000000 -0800
+++ madwifi-ng-r1708-20060907/regression/tkip/test_tkip.c	2006-09-07 08:34:26.000000000 -0700
@@ -34,7 +34,6 @@
 /*
  * TKIP test module.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff -ur madwifi-ng-r1708-20060907.orig/regression/wep/test_wep.c madwifi-ng-r1708-20060907/regression/wep/test_wep.c
--- madwifi-ng-r1708-20060907.orig/regression/wep/test_wep.c	2006-02-01 12:07:11.000000000 -0800
+++ madwifi-ng-r1708-20060907/regression/wep/test_wep.c	2006-09-07 08:34:29.000000000 -0700
@@ -44,7 +44,6 @@
  * you want; e.g. insmod wep_test tests=7 will run only test mpdu's
  * 1, 2, and 3.
  */
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
