--- Makefile.in.orig	2004-02-29 10:11:15.000000000 +0100
+++ Makefile.in	2004-02-29 10:11:56.000000000 +0100
@@ -80,7 +80,7 @@
 # Object files for the SQLite library.
 #
 LIBOBJ = attach.lo auth.lo btree.lo build.lo copy.lo date.lo \
-         delete.lo expr.lo func.lo hash.lo insert.lo \
+         delete.lo encode.lo expr.lo func.lo hash.lo insert.lo \
          main.lo opcodes.lo os.lo pager.lo parse.lo pragma.lo \
          printf.lo random.lo select.lo table.lo tokenize.lo \
          update.lo util.lo vacuum.lo vdbe.lo vdbeaux.lo \
@@ -103,6 +103,7 @@
   $(TOP)/src/copy.c \
   $(TOP)/src/date.c \
   $(TOP)/src/delete.c \
+  $(TOP)/src/encode.c \
   $(TOP)/src/expr.c \
   $(TOP)/src/func.c \
   $(TOP)/src/hash.c \
@@ -300,6 +301,9 @@
 expr.lo:	$(TOP)/src/expr.c $(HDR)
 	$(LIBTOOL) $(TCC) -c $(TOP)/src/expr.c
 
+encode.lo:	$(TOP)/src/encode.c $(HDR)
+	$(LIBTOOL) $(TCC) -c $(TOP)/src/encode.c
+
 func.lo:	$(TOP)/src/func.c $(HDR)
 	$(LIBTOOL) $(TCC) -c $(TOP)/src/func.c
 
