--- mysqlcc-0.9.4-src/src/CApplication.cpp.orig	2003-12-14 01:10:43.000000000 +0200
+++ mysqlcc-0.9.4-src/src/CApplication.cpp	2004-01-16 06:29:57.945206138 +0200
@@ -109,8 +109,8 @@
   mydir += "/";
 #else
   //Directory where mysqlcc.exe (or mysqlcc) is.
-  QString mydir = QDir::currentDirPath() + "/";
-  //QString mydir = "/usr/share/mysqlcc/";  This needs to be done with a PREFIX define ..
+  //QString mydir = QDir::currentDirPath() + "/";
+  QString mydir = "/usr/share/mysqlcc/";  //This needs to be done with a PREFIX define ..
 #endif
 
   //Translations path
