diff -Naupr snes9x-1.5-src.orig/bsx.h snes9x-1.5-src/bsx.h
--- snes9x-1.5-src.orig/bsx.h	2006-06-30 15:40:56.000000000 +0200
+++ snes9x-1.5-src/bsx.h	2006-10-17 17:38:47.000000000 +0200
@@ -164,7 +164,7 @@ struct SBSX
 	uint8	test2192[32];
 };
 
-extern struct SBSX	BSX;
+extern "C" struct SBSX	BSX;
 
 uint8 S9xGetBSX(uint32);
 void S9xSetBSX(uint8, uint32);
diff -Naupr snes9x-1.5-src.orig/dsp1.h snes9x-1.5-src/dsp1.h
--- snes9x-1.5-src.orig/dsp1.h	2006-06-30 15:40:56.000000000 +0200
+++ snes9x-1.5-src/dsp1.h	2006-10-17 17:38:31.000000000 +0200
@@ -180,6 +180,6 @@ uint8 S9xGetDSP (uint16 Address);
 void S9xSetDSP (uint8 Byte, uint16 Address);
 END_EXTERN_C
 
-extern struct SDSP1 DSP1;
+extern "C" struct SDSP1 DSP1;
 
 #endif
diff -Naupr snes9x-1.5-src.orig/ppu.h snes9x-1.5-src/ppu.h
--- snes9x-1.5-src.orig/ppu.h	2006-06-30 15:40:56.000000000 +0200
+++ snes9x-1.5-src/ppu.h	2006-10-17 17:39:13.000000000 +0200
@@ -409,9 +409,9 @@ typedef struct{
 	uint8 _5A22;
 } SnesModel;
 
-extern SnesModel* Model;
-extern SnesModel M1SNES;
-extern SnesModel M2SNES;
+extern "C" SnesModel* Model;
+extern "C" SnesModel M1SNES;
+extern "C" SnesModel M2SNES;
 
 #define MAX_5C77_VERSION 0x01
 #define MAX_5C78_VERSION 0x03
