--- snes9x/snes9x.cpp~	2004-12-31 00:15:47.000000000 +0200
+++ snes9x/snes9x.cpp	2005-01-01 18:20:52.699687431 +0200
@@ -107,14 +107,15 @@
     S9xMessage (S9X_INFO, S9X_USAGE, "Where <options> can be:\n");
     
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--cycles or -h <num>       Percentage of CPU cycles to execute every scan line (default 90)\n");
+-cycles or -h <num>       Percentage of CPU cycles to execute every scan line\n\
+                          (default: 90)\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--frameskip or -f <num>    Screen update frame skip rate (default 2)\n");
+-frameskip or -f <num>    Screen update frame skip rate (default: 2)\n");
     S9xExtraUsage ();
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--hirom or -fh or -hr      Force Hi-ROM memory map, useful for hacked ROM images.\n");
+-hirom or -fh or -hr      Force Hi-ROM memory map, useful for hacked ROM images\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--lorom or -fl or -lr      Force Lo-ROM memory map, useful for hacked ROM images.\n");
+-lorom or -fl or -lr      Force Lo-ROM memory map, useful for hacked ROM images\n");
     //FIXME: -old bsolete?
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -old or -o                Enable old-style SNES joypad emulation\n");
@@ -122,13 +123,13 @@
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -noold or -no             Disbale old-style SNES joypad emulation\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--soundskip or -ss <num>   Sound CPU skip-waiting method, 0 - 3 (default 0)\n");
+-soundskip or -ss <num>   Sound CPU skip-waiting method, 0 - 3 (default: 0)\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -sound or -s              Enable digital sound output (default: enabled)\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -nosound or -ns           Disable digital sound output\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--soundquality or -r <num> Sound sample playback rate/quality, 0-7 (default 4)\n");
+-soundquality or -r <num> Sound sample playback rate/quality, 0-7 (default: 4)\n");
 
 #ifdef __sgi
 /* BS: changed the sample rate values to match the IRIX options */
@@ -148,9 +149,10 @@
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -mono                     Enable mono sound (default: mono sound)\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--buffersize or -B         Sound playback buffer size (default auto for playback rate)\n");
+-buffersize or -B         Sound playback buffer size (default: auto for\n\
+                          playback rate)\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--nospeedhacks or -N       Disable some internal speed ups that break a few  ROMs\n");
+-nospeedhacks or -N       Disable some internal speed ups that break a few ROMs\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -speedhacks or -SH        Enable some internal speed ups that break a few ROMs\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
@@ -162,9 +164,9 @@
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -pal or -p                Fool ROM into thinking that this is a PAL SNES system\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--ntsc or -n               Fool ROM into thinking that this is a NTCS SNES system\n");
+-ntsc or -n               Fool ROM into thinking that this is a NTSC SNES system\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--interleaved or -i        ROM image is in interleaved format.\n");
+-interleaved or -i        ROM image is in interleaved format\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -interleaved2 or -i2      ROM image is in interleaved 2 format\n");
 	    S9xMessage (S9X_INFO, S9X_USAGE, "\
@@ -174,7 +176,8 @@
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -hdma or -NH              Enable H-DMA emulation (default: enabled)\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--layering or -L           Swap some background priority levels - helps some games\n");
+-layering or -L           Swap some background priority levels - helps some\n\
+                          games\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -graphicwindows           Enable graphic window effects (default: enabled)\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
@@ -197,9 +200,11 @@
 #ifdef JOYSTICK_SUPPORT
 #ifdef __linux
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--joydevX /dev/jsY         Use joystick device /dev/jsY for emulation of gamepad X\n");
+-joydevX /dev/jsY         Use joystick device /dev/jsY for emulation of\n\
+                          gamepad X\n");
     S9xMessage (S9X_INFO, S9X_USAGE, "\
--joymapX 0 1 2 3 4 5 6 7  Joystick buttons which should be assigned to gamepad X - A B X Y TL TR Start and Select\n");
+-joymapX 0 1 2 3 4 5 6 7  Joystick buttons which should be assigned to\n\
+                          gamepad X - A B X Y TL TR Start and Select\n");
 #else
     S9xMessage (S9X_INFO, S9X_USAGE, "\
 -four or -4               Single standard PC joystick has four buttons\n");
