diff -Naupr acx-20070101.orig/common.c acx-20070101/common.c
--- acx-20070101.orig/common.c	2007-01-10 22:15:27.000000000 +0100
+++ acx-20070101/common.c	2007-08-22 12:55:39.000000000 +0200
@@ -2754,7 +2754,7 @@ acx_l_rxmonitor(acx_device_t *adev, cons
 	skb->dev = adev->ndev;
 	skb->dev->last_rx = jiffies;
 
-	skb->mac.raw = skb->data;
+	// skb->mac.raw = skb->data;
 	skb->ip_summed = CHECKSUM_NONE;
 	skb->pkt_type = PACKET_OTHERHOST;
 	skb->protocol = htons(ETH_P_80211_RAW);
