diff -Naupr drivers.orig/net/wireless/tiacx/common.c drivers/net/wireless/tiacx/common.c
--- drivers.orig/net/wireless/tiacx/common.c	2006-06-29 15:33:30.000000000 +0200
+++ drivers/net/wireless/tiacx/common.c	2006-10-09 14:24:34.000000000 +0200
@@ -30,7 +30,6 @@
 ** ---------------------------------------------------------------------
 */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
diff -Naupr drivers.orig/net/wireless/tiacx/ioctl.c drivers/net/wireless/tiacx/ioctl.c
--- drivers.orig/net/wireless/tiacx/ioctl.c	2006-06-29 15:33:30.000000000 +0200
+++ drivers/net/wireless/tiacx/ioctl.c	2006-10-09 14:24:34.000000000 +0200
@@ -30,7 +30,6 @@
 ** ---------------------------------------------------------------------
 */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/types.h>
diff -Naupr drivers.orig/net/wireless/tiacx/Makefile drivers/net/wireless/tiacx/Makefile
--- drivers.orig/net/wireless/tiacx/Makefile	2006-06-29 15:33:04.000000000 +0200
+++ drivers/net/wireless/tiacx/Makefile	2006-10-09 14:27:54.000000000 +0200
@@ -2,5 +2,13 @@ acx-common-y := wlan.o ioctl.o common.o
 acx-usb-y := usb.o
 acx-pci-y := pci.o
 
+ifndef CONFIG_ACX_PCI
+CONFIG_ACX_PCI=m
+endif
+
+ifndef CONFIG_ACX_USB
+CONFIG_ACX_USB=m
+endif
+
 obj-$(CONFIG_ACX_PCI) += acx-common.o acx-pci.o
 obj-$(CONFIG_ACX_USB) += acx-common.o acx-usb.o
diff -Naupr drivers.orig/net/wireless/tiacx/pci.c drivers/net/wireless/tiacx/pci.c
--- drivers.orig/net/wireless/tiacx/pci.c	2006-06-29 15:33:30.000000000 +0200
+++ drivers/net/wireless/tiacx/pci.c	2006-10-09 14:24:34.000000000 +0200
@@ -31,7 +31,6 @@
 */
 #define ACX_PCI 1
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/compiler.h> /* required for Lx 2.6.8 ?? */
 #include <linux/kernel.h>
@@ -51,6 +50,7 @@
 #include <linux/pm.h>
 #include <linux/vmalloc.h>
 #include <linux/dma-mapping.h>
+#include <linux/utsrelease.h>
 
 #include "acx.h"
 
diff -Naupr drivers.orig/net/wireless/tiacx/usb.c drivers/net/wireless/tiacx/usb.c
--- drivers.orig/net/wireless/tiacx/usb.c	2006-06-29 15:33:30.000000000 +0200
+++ drivers/net/wireless/tiacx/usb.c	2006-10-09 14:24:34.000000000 +0200
@@ -45,7 +45,6 @@
 #define ACX_USB 1
 
 #include <linux/version.h>
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/module.h>
 #include <linux/moduleparam.h>
diff -Naupr drivers.orig/net/wireless/tiacx/wlan.c drivers/net/wireless/tiacx/wlan.c
--- drivers.orig/net/wireless/tiacx/wlan.c	2006-06-29 15:33:04.000000000 +0200
+++ drivers/net/wireless/tiacx/wlan.c	2006-10-09 14:24:34.000000000 +0200
@@ -37,7 +37,6 @@
 ** http://www.linux-wlan.com
 */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/types.h>
 #include <linux/if_arp.h>
