--- transcode-1.0.2.orig/configure	2005-11-06 07:04:59.000000000 +0100
+++ transcode-1.0.2/configure	2005-11-06 07:04:59.000000000 +0100
@@ -28107,7 +28107,12 @@ main()
     printf("install ffmpeg 0.4.9-pre1 or newer, or a cvs version after 20040703");
     return(1);
   }
+  /* 3344640 == ((51<<16)+(9<<8)+0) == 51.9.0 */
+#ifdef LIBAVCODEC_BUILD >= 3344640
+  printf("VER=%s\n", AV_STRINGIFY(LIBAVCODEC_VERSION));
+#else
   printf("VER=%s\n", FFMPEG_VERSION);
+#endif /* LIBAVCODEC_BUILD >= 51.9.0 check */
   printf("BUILD=%d\n", LIBAVCODEC_BUILD);
   return(0);
 }
--- transcode-1.0.2.orig/configure.in	2005-10-30 05:57:45.000000000 +0100
+++ transcode-1.0.2/configure.in	2005-10-30 05:57:45.000000000 +0100
@@ -628,7 +628,12 @@ main()
     printf("install ffmpeg 0.4.9-pre1 or newer, or a cvs version after 20040703");
     return(1);
   }
+  /* 3344640 == ((51<<16)+(9<<8)+0) == 51.9.0 */
+#ifdef LIBAVCODEC_BUILD >= 3344640
+  printf("VER=%s\n", AV_STRINGIFY(LIBAVCODEC_VERSION));
+#else
   printf("VER=%s\n", FFMPEG_VERSION);
+#endif /* LIBAVCODEC_BUILD >= 51.9.0 check */
   printf("BUILD=%d\n", LIBAVCODEC_BUILD);
   return(0);
 }
