diff -u3 -rN transcode-0.6.13/libac3/bitstream.c transcode-0.6.13-patched/libac3/bitstream.c
--- transcode-0.6.13/libac3/bitstream.c	2004-10-17 23:56:50.000000000 +0100
+++ transcode-0.6.13-patched/libac3/bitstream.c	2004-10-28 19:37:06.497339637 +0100
@@ -97,7 +97,7 @@
 static inline void
 bitstream_fill_current()
 {
-	current_word = *(buffer_start)++;
+	current_word = *((uint_32*)buffer_start)++;
 	current_word = swab32(current_word);
 }
 
