--- export/export_mov.c.orig	Thu Jan  8 09:18:28 2004
+++ export/export_mov.c	Sat Jul 10 00:53:52 2004
@@ -176,6 +176,7 @@
         rawVideo = 1;    
     }
     else{
+        int divx_bitrate;
         /* check if we can encode with this codec */    
         if (!quicktime_supported_video(qtfile, 0)){
         fprintf(stderr,"[%s] qt video codec '%s' unsupported\n",
@@ -218,7 +219,7 @@
 
         /* set codec parameters */
         // if max bitrate > bitrate  use difference for bitrate tolerance
-        int divx_bitrate = vob->divxbitrate * 1000; // tc uses kb        
+        divx_bitrate = vob->divxbitrate * 1000; // tc uses kb        
         if (vob->video_max_bitrate > vob->divxbitrate) div3_bitrate_tolerance = (vob->video_max_bitrate - vob->divxbitrate) * 1000; //kb again
 
         /* set codec parameters */
