--- Makefile.common.in.orig	2007-01-22 11:08:24.000000000 +0100
+++ Makefile.common.in	2007-01-22 11:09:31.000000000 +0100
@@ -60,9 +60,9 @@
 
 # ------------------- defines -------------------------
 
-INSTALL_BIN = echo installing $$file to $(bindir);$(INSTALL_PROGRAM) $$file $(bindir)/`echo $$file|sed 's/$(EXEEXT)$$//'|sed '$(transform)'|sed 's/$$/$(EXEEXT)/'`
+INSTALL_BIN = echo installing $$file to $(DESTDIR)$(bindir);$(INSTALL_PROGRAM) $$file $(DESTDIR)$(bindir)/`echo $$file|sed 's/$(EXEEXT)$$//'|sed '$(transform)'|sed 's/$$/$(EXEEXT)/'`
 UNINSTALL_BIN = ff=`echo $$file|sed 's/$(EXEEXT)$$//'|sed '$(transform)'|sed 's/$$/$(EXEEXT)/'`;echo rm -f $(bindir)/$$ff;rm -f $(bindir)/$$ff
-INSTALL_MAN1 = ff=$(srcdir)/$$file.1; inst=`echo $$file | sed '$(transform)'`.1; echo "$(INSTALL_DATA) $$ff $(man1dir)/$$inst"; $(INSTALL_DATA) $$ff $(man1dir)/$$inst
+INSTALL_MAN1 = ff=$(srcdir)/$$file.1; inst=`echo $$file | sed '$(transform)'`.1; echo "$(INSTALL_DATA) $$ff $(DESTDIR)$(man1dir)/$$inst"; $(INSTALL_DATA) $$ff $(DESTDIR)$(man1dir)/$$inst
 UNINSTALL_MAN1 = ff=$(srcdir)/$$file.1; inst=`echo $$file | sed '$(transform)'`.1; echo "rm -f $(man1dir)/$$inst"; rm -f $(man1dir)/$$inst
 
 #%.o: %.c
--- pdf2swf/fonts/Makefile.in.orig	2007-01-22 11:35:09.000000000 +0100
+++ pdf2swf/fonts/Makefile.in	2007-01-22 11:35:40.000000000 +0100
@@ -8,8 +8,8 @@
 fonts = n021003l.pfb n021023l.pfb n021004l.pfb n021024l.pfb n019003l.pfb n019023l.pfb n019004l.pfb n019024l.pfb n022003l.pfb n022023l.pfb n022004l.pfb n022024l.pfb s050000l.pfb d050000l.pfb n021003l.afm n021023l.afm n021004l.afm n021024l.afm n019003l.afm n019023l.afm n019004l.afm n019024l.afm n022003l.afm n022023l.afm n022004l.afm n022024l.afm s050000l.afm d050000l.afm
 
 install:
-	$(mkinstalldirs) $(pkgdatadir)/fonts
-	@echo installing fonts in $(pkgdatadir)/fonts;for file in $(fonts);do $(INSTALL_DATA) $$file $(pkgdatadir)/fonts/$$file;done
+	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)/fonts
+	@echo installing fonts in $(pkgdatadir)/fonts;for file in $(fonts);do $(INSTALL_DATA) $$file $(DESTDIR)$(pkgdatadir)/fonts/$$file;done
 	
 uninstall:
 	@for file in $(fonts);do $(UNINSTALL_DATA) $(pkgdatadir)/fonts/$$file;done
--- swfs/Makefile.in.orig	2007-01-22 11:39:19.000000000 +0100
+++ swfs/Makefile.in	2007-01-22 11:41:08.000000000 +0100
@@ -34,17 +34,17 @@
 	./PreLoaderTemplate$(E) || true
 
 install:
-	$(mkinstalldirs) $(pkgdatadir)
-	$(mkinstalldirs) $(pkgdatadir)/swfs
-	$(INSTALL_DATA) ./simple_viewer.swf $(pkgdatadir)/swfs/simple_viewer.swf
-	$(INSTALL_DATA) ./keyboard_viewer.swf $(pkgdatadir)/swfs/keyboard_viewer.swf
-	$(INSTALL_DATA) ./PreLoaderTemplate.swf $(pkgdatadir)/swfs/PreLoaderTemplate.swf
-	$(INSTALL_DATA) ./tessel_loader.swf $(pkgdatadir)/swfs/tessel_loader.swf
-	$(INSTALL_DATA) ./swft_loader.swf $(pkgdatadir)/swfs/swft_loader.swf
-	test -f $(pkgdatadir)/swfs/default_viewer.swf || \
-	    $(LN_S) $(pkgdatadir)/swfs/simple_viewer.swf $(pkgdatadir)/swfs/default_viewer.swf;
-	test -f $(pkgdatadir)/swfs/default_loader.swf || \
-	    $(LN_S) $(pkgdatadir)/swfs/tessel_loader.swf $(pkgdatadir)/swfs/default_loader.swf;
+	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)
+	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)/swfs
+	$(INSTALL_DATA) ./simple_viewer.swf $(DESTDIR)$(pkgdatadir)/swfs/simple_viewer.swf
+	$(INSTALL_DATA) ./keyboard_viewer.swf $(DESTDIR)$(pkgdatadir)/swfs/keyboard_viewer.swf
+	$(INSTALL_DATA) ./PreLoaderTemplate.swf $(DESTDIR)$(pkgdatadir)/swfs/PreLoaderTemplate.swf
+	$(INSTALL_DATA) ./tessel_loader.swf $(DESTDIR)$(pkgdatadir)/swfs/tessel_loader.swf
+	$(INSTALL_DATA) ./swft_loader.swf $(DESTDIR)$(pkgdatadir)/swfs/swft_loader.swf
+	test -f $(DESTDIR)$(pkgdatadir)/swfs/default_viewer.swf || \
+	    $(LN_S) simple_viewer.swf $(DESTDIR)$(pkgdatadir)/swfs/default_viewer.swf;
+	test -f $(DESTDIR)$(pkgdatadir)/swfs/default_loader.swf || \
+	    $(LN_S) tessel_loader.swf $(DESTDIR)$(pkgdatadir)/swfs/default_loader.swf;
 		
 uninstall:
 	$(UNINSTALL_DATA) $(pkgdatadir)/swfs/simple_viewer.swf
