--- fortune-mod-9708/Makefile.mike	Thu Aug 28 15:36:15 1997
+++ fortune-mod-9708/Makefile	Tue Jun  6 17:57:18 2000
@@ -3,27 +3,27 @@
 #
 
 # Where does the fortune program go?
-FORTDIR=/usr/local/games
+FORTDIR=/usr/games
 # Where do the data files (fortunes, or cookies) go?
-COOKIEDIR=/usr/local/share/games/fortunes
+COOKIEDIR=/usr/share/games/fortunes
 # Offensive ones?
 OCOOKIEDIR=$(COOKIEDIR)/off
 # The ones with html tags?
 WCOOKIEDIR=$(COOKIEDIR)/html
 # Where do strfile and unstr go?
-BINDIR=/usr/local/bin
+BINDIR=/usr/sbin
 # What is the proper mode for strfile and unstr? 755= everyone, 700= root only
 BINMODE=0755
 #BINMODE=0700
 # Where do the man pages for strfile and unstr go?
-BINMANDIR=/usr/local/man/man1
+BINMANDIR=/usr/man/man1
 # What is their proper extension?
 BINMANEXT=1
 # And the same for the fortune man page
-FORTMANDIR=/usr/local/man/man6
+FORTMANDIR=/usr/man/man6
 FORTMANEXT=6
 # Do we want to install the offensive files? (0 no, 1 yes)
-OFFENSIVE=1
+OFFENSIVE=0
 # Do we want to install files with html tags? (0 no, 1 yes)
 # (Note: These files are not yet available)
 WEB=0
@@ -55,8 +55,8 @@
 REGEXLIBS=
 
 DEFINES=-DFORTDIR="\"$(COOKIEDIR)\"" -DOFFDIR="\"$(OCOOKIEDIR)\""
-CFLAGS=-O2 $(DEFINES) -Wall -fomit-frame-pointer -pipe
-LDFLAGS=-s
+CFLAGS=$(RPM_OPT_FLAGS) $(DEFINES)
+LDFLAGS=
 
 # The above flags are used by default; the debug flags are used when make
 # is called with a debug target, such as 'make debug'
