--- mozilla/security/coreconf/Linux.mk.foo	2003-10-01 13:06:30.000000000 -0400
+++ mozilla/security/coreconf/Linux.mk	2003-10-01 13:07:22.000000000 -0400
@@ -53,7 +53,11 @@
 ifeq ($(OS_TEST),m68k)
 	OS_REL_CFLAGS	= -DLINUX1_2 -D_XOPEN_SOURCE
 	CPU_ARCH	= m68k
-else		
+else
+ifeq ($(OS_TEST),ppc64)
+	OS_REL_CFLAGS	= -DLINUX1_2 -D_XOPEN_SOURCE
+	CPU_ARCH	= ppc64
+else
 ifeq ($(OS_TEST),ppc)
 	OS_REL_CFLAGS	= -DLINUX1_2 -D_XOPEN_SOURCE
 	CPU_ARCH	= ppc
@@ -112,6 +116,7 @@
 endif
 endif
 endif
+endif
 
 
 LIBC_TAG		= _glibc
