--- mozilla/nsprpub/pr/src/misc/Makefile.in.foo	Wed Jun 11 15:42:57 2003
+++ mozilla/nsprpub/pr/src/misc/Makefile.in	Wed Jun 11 15:42:49 2003
@@ -82,6 +82,14 @@
 
 include $(topsrcdir)/config/rules.mk
 
+NONOPT_CFLAGS=$(filter-out -O%, $(CFLAGS))
+
+ifeq ($(OS_ARCH),Linux)
+$(OBJDIR)/prdtoa.$(OBJ_SUFFIX): prdtoa.c
+	@$(MAKE_OBJDIR)
+	$(CC) -o $@ -c $(NONOPT_CFLAGS) -ffloat-store $<
+endif
+
 #
 # Generate prerr.h, prerr.c, and prerr.properties from prerr.et.
 #
