diff -Naupr oxine-0.6.orig/data/skins/Makefile.am oxine-0.6/data/skins/Makefile.am
--- oxine-0.6.orig/data/skins/Makefile.am	2006-10-18 12:57:38.000000000 +0200
+++ oxine-0.6/data/skins/Makefile.am	2006-12-18 16:34:59.000000000 +0100
@@ -1,7 +1,7 @@
 
 SUBDIRS			= oxinetic greyraven
 
-DEFAULT_SKIN	= @OXINE_SKINDIR@/default
+DEFAULT_SKIN	= $(DESTDIR)@OXINE_SKINDIR@/default
 
 install-data-local:
 	test -L $(DEFAULT_SKIN) || \
diff -Naupr oxine-0.6.orig/data/skins/Makefile.in oxine-0.6/data/skins/Makefile.in
--- oxine-0.6.orig/data/skins/Makefile.in	2006-11-03 18:23:53.000000000 +0100
+++ oxine-0.6/data/skins/Makefile.in	2006-12-18 16:37:07.000000000 +0100
@@ -211,7 +211,7 @@ target_alias = @target_alias@
 umount_path = @umount_path@
 xmllint_path = @xmllint_path@
 SUBDIRS = oxinetic greyraven
-DEFAULT_SKIN = @OXINE_SKINDIR@/default
+DEFAULT_SKIN = $(DESTDIR)@OXINE_SKINDIR@/default
 all: all-recursive
 
 .SUFFIXES:
