--- Makefile.orig	2004-09-20 18:40:10.000000000 +0200
+++ Makefile	2004-11-26 11:37:15.515804160 +0100
@@ -1,38 +1,40 @@
 prefix = /usr
 libdir = $(prefix)/lib
-includedir = $(prefix)/include/musepack
-installopts = -g root -o root -m 755
-VERSION=1.0.2
+includedir = $(prefix)/include
+VERSION = 1.0.2
+DESTDIR = 
 
 CC = g++
-CPPFLAGS  = -Wall -fPIC -DPIC -O2 -march=i686 -pipe -DPOSIX -D_REENTRANT
+CPPFLAGS = -O2
+MUSEPACK_CPPFLAGS = $(CPPFLAGS) -Wall -fPIC -DPIC -DPOSIX -D_REENTRANT
 
 SOURCES := $(wildcard *.cpp)
 HEADERS := $(wildcard *.h)
 OBJS := $(patsubst %.cpp, %.o, $(SOURCES))
 
 libmusepack.so.${VERSION}: ${OBJS}
-	${CC} -shared -Wl,-soname,libmusepack.so.1 ${CPPFLAGS} ${OBJS} -o libmusepack.so.${VERSION}
+	${CC} -shared -Wl,-soname,libmusepack.so.1 ${MUSEPACK_CPPFLAGS} ${OBJS} -o libmusepack.so.${VERSION}
 
 ${OBJS}: %.o: %.cpp ${HEADERS}
-	${CC} ${CPPFLAGS} -c $*.cpp
+	${CC} ${MUSEPACK_CPPFLAGS} -c $*.cpp
 
 sample: 
-	${CC} ${CPPFLAGS} -c sample.cpp
-	${CC} -shared -Wl,-soname,sample ${CPPFLAGS} sample.o -o sample
+	${CC} ${MUSEPACK_CPPFLAGS} -c sample.cpp
+	${CC} -shared -Wl,-soname,sample ${MUSEPACK_CPPFLAGS} sample.o -o sample
 clean:
 	rm -f ${OBJS}
 	rm -f libmusepack.so.${VERSION}
 	
-install: uninstall libmusepack.so.${VERSION}
-	install ${installopts} ./libmusepack.so.${VERSION} ${libdir}/.
-	/sbin/ldconfig ${libdir}/libmusepack.so.${VERSION}
-	ln -sf ${libdir}/libmusepack.so.1 ${libdir}/libmusepack.so
-	install ${installopts} -d ${includedir}
-	install ${installopts} ${HEADERS} ${includedir}
+install:
+	install -m 0755 -d ${DESTDIR}${libdir}
+	install -m 0755 ./libmusepack.so.${VERSION} ${DESTDIR}${libdir}/.
+	ln -sf libmusepack.so.${VERSION} ${DESTDIR}${libdir}/libmusepack.so.1
+	ln -sf libmusepack.so.1 ${DESTDIR}${libdir}/libmusepack.so
+	install -m 0755 -d ${DESTDIR}${includedir}/musepack
+	install -m 0644 ${HEADERS} ${DESTDIR}${includedir}/musepack/
 
 uninstall:
 	rm -rf ${libdir}/libmusepack.*
-	rm -rf ${includedir}
+	rm -rf ${includedir}/musepack
   		
 #END OF FILE
