diff -Naupr tmp.orig/usr/src/nv/Makefile.kbuild tmp/usr/src/nv/Makefile.kbuild
--- usr/src/nv/Makefile.kbuild.orig	2007-02-27 08:59:25.000000000 +0100
+++ usr/src/nv/Makefile.kbuild	2007-03-15 13:48:42.000000000 +0100
@@ -292,7 +292,7 @@ xen-sanity-check:
 # the "package-install" target below.
 #
 
-module: cc-version-check xen-sanity-check rivafb-sanity-check nvidiafb-sanity-check
+module: cc-version-check rivafb-sanity-check nvidiafb-sanity-check
 	@if [ -z "$(PATCHLEVEL)" ]; then \
 	  echo "failed to determine PATCHLEVEL!"; \
 	  exit 1; \
