diff -ur aldo-0.6.5/include/dialog.h aldo-0.6.5-new/include/dialog.h
--- aldo-0.6.5/include/dialog.h	2003-10-02 19:23:55.000000000 +0200
+++ aldo-0.6.5-new/include/dialog.h	2005-06-24 18:43:28.150026184 +0200
@@ -24,8 +24,10 @@
  *                                                                         *
  ***************************************************************************/
 
+
 #include <string>
 #include <vector>
+#include <iostream>
 
 #pragma interface
 
@@ -121,7 +123,7 @@
     unsigned int choice = 0;
     do
     {
-	std::cout<<endl<<m_prompt;
+	std::cout<<std::endl<<m_prompt;
 
 	std::string trash;
 	std::cin>>trash;
diff -ur aldo-0.6.5/libmenus/dialog.h aldo-0.6.5-new/libmenus/dialog.h
--- aldo-0.6.5/libmenus/dialog.h	2003-10-02 19:23:55.000000000 +0200
+++ aldo-0.6.5-new/libmenus/dialog.h	2005-06-24 18:43:28.150026184 +0200
@@ -24,8 +24,10 @@
  *                                                                         *
  ***************************************************************************/
 
+
 #include <string>
 #include <vector>
+#include <iostream>
 
 #pragma interface
 
@@ -121,7 +123,7 @@
     unsigned int choice = 0;
     do
     {
-	std::cout<<endl<<m_prompt;
+	std::cout<<std::endl<<m_prompt;
 
 	std::string trash;
 	std::cin>>trash;
