--- synaptic-0.36.1/common/rpackagelister.cc.sectver	2003-05-02 13:56:43.000000000 +0300
+++ synaptic-0.36.1/common/rpackagelister.cc	2003-05-02 14:27:29.000000000 +0300
@@ -409,8 +409,10 @@
 	}
 
 	// gather list of sections
-	if (I.Section()) {
-	    sectionSet.insert(I.Section());
+	// use the section of candidate version
+	pkgCache::VerIterator V = (*deps)[I].CandidateVerIter(*deps);
+	if (V.Section()) {
+	    sectionSet.insert(V.Section());
 	} else {
 	    cerr << _("Package ") << I.Name() << _(" has no section?!") << endl;
 	}
