diff -Naupr cinelerra-2.0.orig/cinelerra/ffmpeg.h cinelerra-2.0/cinelerra/ffmpeg.h
--- cinelerra-2.0.orig/cinelerra/ffmpeg.h	2004-09-07 10:42:43.000000000 +0200
+++ cinelerra-2.0/cinelerra/ffmpeg.h	2005-10-04 18:18:34.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef FFMPEG_H
 #define FFMPEG_H
 
-#include <ffmpeg/avcodec.h>
+#include <ffmpeg/libavcodec/avcodec.h>
 
 #include "asset.h"
 #include "guicast.h"
diff -Naupr cinelerra-2.0.orig/cinelerra/fileac3.h cinelerra-2.0/cinelerra/fileac3.h
--- cinelerra-2.0.orig/cinelerra/fileac3.h	2004-05-31 21:23:27.000000000 +0200
+++ cinelerra-2.0/cinelerra/fileac3.h	2005-10-04 18:20:12.000000000 +0200
@@ -2,7 +2,7 @@
 #define FILEAC3_H
 
 
-#include <ffmpeg/avcodec.h>
+#include <ffmpeg/libavcodec/avcodec.h>
 #include "filebase.h"
 #include <stdio.h>
 
