--- libscompat/issetugid.c.orig	2010-01-12 15:01:00.000000000 +0100
+++ libscompat/issetugid.c	2010-01-12 15:03:39.000000000 +0100
@@ -47,20 +47,23 @@
  *
  */
 
-#if HAVE_LIBC_ENABLE_SECURE
-extern int __libc_enable_secure;
-#endif /* HAVE_LIBC_ENABLE_SECURE */
+//#if HAVE_LIBC_ENABLE_SECURE
+//extern int __libc_enable_secure;
+//#endif /* HAVE_LIBC_ENABLE_SECURE */
 
 #include "issetugid.h"
 
 int
 issetugid(void)
 {
-#if HAVE_LIBC_ENABLE_SECURE
-   if (__libc_enable_secure)
-      return 1;
-   else
-      return 0;
-#endif /* HAVE_LIBC_ENABLE_SECURE */
-   return 1; /* don't know, better safe than sorry. */
+//#if HAVE_LIBC_ENABLE_SECURE
+//   if (__libc_enable_secure)
+//      return 1;
+//   else
+//      return 0;
+//#endif /* HAVE_LIBC_ENABLE_SECURE */
+//   return 1; /* don't know, better safe than sorry. */
+    if (getuid() != geteuid()) return 1;
+    if (getgid() != getegid()) return 1;
+    return 0;
 }
