diff -up uudeview-0.5.20/tcl/Makefile.in.shared uudeview-0.5.20/tcl/Makefile.in
--- uudeview-0.5.20/tcl/Makefile.in.shared	2001-06-04 18:26:57.000000000 +0200
+++ uudeview-0.5.20/tcl/Makefile.in	2012-03-28 15:07:05.875796084 +0200
@@ -95,8 +95,8 @@ new:		clean
 	rm -f uuwish
 	make all
 
-uuwish:		$(OBJ) ../uulib/libuu.a
-	$(CC) -o $@ $(OBJ) -L../uulib -luu $(LIBS)
+uuwish:		$(OBJ) ../uulib/libuu.la
+	$(CC) -o $@ $(OBJ) -L../uulib/.libs -luu $(LIBS)
 
 xdeview:
 	chmod 755 $@
diff -up uudeview-0.5.20/unix/Makefile.in.shared uudeview-0.5.20/unix/Makefile.in
--- uudeview-0.5.20/unix/Makefile.in.shared	2001-06-04 18:26:58.000000000 +0200
+++ uudeview-0.5.20/unix/Makefile.in	2012-03-28 15:07:05.874678652 +0200
@@ -114,11 +114,11 @@ new:		clean
 	rm -f uudeview uuenview
 	make all
 
-uudeview:	$(UUDEVIEW_OBJ) ../uulib/libuu.a
-	$(CC) -o $@ $(UUDEVIEW_OBJ) -L../uulib -luu $(LIBS)
+uudeview:	$(UUDEVIEW_OBJ) ../uulib/libuu.la
+	$(CC) -o $@ $(UUDEVIEW_OBJ) -L../uulib/.libs -luu $(LIBS)
 
-uuenview:	$(UUENVIEW_OBJ) ../uulib/libuu.a
-	$(CC) -o $@ $(UUENVIEW_OBJ) -L../uulib -luu $(LIBS)
+uuenview:	$(UUENVIEW_OBJ) ../uulib/libuu.la
+	$(CC) -o $@ $(UUENVIEW_OBJ) -L../uulib/.libs -luu $(LIBS)
 
 .c.o:
 	$(CC) -c $(CFLAGS) -I../uulib $(VDEF) $<
diff -up uudeview-0.5.20/uulib/Makefile.in.shared uudeview-0.5.20/uulib/Makefile.in
--- uudeview-0.5.20/uulib/Makefile.in.shared	2002-03-11 10:15:46.000000000 +0100
+++ uudeview-0.5.20/uulib/Makefile.in	2012-03-28 15:11:38.946670413 +0200
@@ -40,9 +40,13 @@ PATCH	=	@PATCH@
 VDEF	=	-DVERSION=\"$(VERSION)\" -DPATCH=\"$(PATCH)\"
 #
 
+DESTDIR =
+libdir = @libdir@
+includedir = @includedir@
+
 UULIB_SOURCE	=	uulib.c uucheck.c uunconc.c uuutil.c uuencode.c \
 			uuscan.c uustring.c fptools.c crc32.c
-UULIB_OBJ	=	${UULIB_SOURCE:.c=.o}
+UULIB_OBJ	=	${UULIB_SOURCE:.c=.lo}
 
 #
 # make stuff
@@ -51,11 +55,11 @@ UULIB_OBJ	=	${UULIB_SOURCE:.c=.o}
 .SUFFIXES:
 .SUFFIXES: .c .o
 
-all:		libuu.a
+all:		libuu.la
 
 clean:
 	rm -f [Xx]deview gif2gfp
-	rm -f *.o *.a *.so core *~ TAGS
+	rm -f *.o *.a *.so core *~ TAGS *.lo *.la
 
 distclean:	clean
 	rm -f config.status config.cache config.log Makefile config.h
@@ -64,25 +68,28 @@ distclean:	clean
 realclean:	distclean
 
 new:		clean
-	rm -f libuu.a
+	rm -f libuu.la
 	$(MAKE) all
 
-libuu.a:	$(UULIB_OBJ)
-	rm -f $@
-	ar r $@ $(UULIB_OBJ)
-	-$(RANLIB) $@
-
-.c.o:
-	$(CC) -c $(CFLAGS) $(VDEF) $<
-
-uuencode.o:	uuencode.c uudeview.h uuint.h uustring.h fptools.h config.h
-uulib.o:	uulib.c uudeview.h uuint.h uustring.h fptools.h config.h
-uunconc.o:	uunconc.c uudeview.h uuint.h uustring.h fptools.h config.h
-uucheck.o:	uucheck.c uudeview.h uuint.h uustring.h fptools.h config.h
-uuutil.o:	uuutil.c  uudeview.h uuint.h uustring.h fptools.h config.h
-uuscan.o:	uuutil.c  uudeview.h uuint.h uustring.h fptools.h config.h
-uustring.o:	uustring.c uudeview.h uuint.h uustring.h config.h
-fptools.o:	fptools.c fptools.h config.h
+libuu.la:	$(UULIB_OBJ)
+	libtool --mode=link $(CC) -o libuu.la $(LDFLAGS) $(UULIB_OBJ) -rpath /usr/lib
+
+%.lo : %.c
+	libtool --mode=compile $(CC) -c $(CFLAGS) $(VDEF) $<
+
+install:	libuu.la
+	install -d $(DESTDIR)$(libdir) $(DESTDIR)$(includedir)
+	libtool --mode=install install libuu.la $(DESTDIR)$(libdir)
+	install uudeview.h fptools.h $(DESTDIR)$(includedir)
+
+uuencode.lo:	uuencode.c uudeview.h uuint.h uustring.h fptools.h config.h
+uulib.lo:	uulib.c uudeview.h uuint.h uustring.h fptools.h config.h
+uunconc.lo:	uunconc.c uudeview.h uuint.h uustring.h fptools.h config.h
+uucheck.lo:	uucheck.c uudeview.h uuint.h uustring.h fptools.h config.h
+uuutil.lo:	uuutil.c  uudeview.h uuint.h uustring.h fptools.h config.h
+uuscan.lo:	uuutil.c  uudeview.h uuint.h uustring.h fptools.h config.h
+uustring.lo:	uustring.c uudeview.h uuint.h uustring.h config.h
+fptools.lo:	fptools.c fptools.h config.h
 
 uustring.h:	uustring.c
 	awk -f uustring.awk < uustring.c > uustring.h
