diff -Naupr libquicktime-0.9.4.orig/plugins/Makefile.am libquicktime-0.9.4/plugins/Makefile.am
--- libquicktime-0.9.4.orig/plugins/Makefile.am	2004-07-01 00:27:41.000000000 +0200
+++ libquicktime-0.9.4/plugins/Makefile.am	2005-04-22 01:33:22.000000000 +0200
@@ -2,8 +2,7 @@
 
 base_subdirs = audiocodec \
 videocodec \
-opendivx \
-rtjpeg
+opendivx
 
 if HAVE_LIBJPEG
 jpeg_subdirs = mjpeg
diff -Naupr libquicktime-0.9.4.orig/plugins/Makefile.in libquicktime-0.9.4/plugins/Makefile.in
--- libquicktime-0.9.4.orig/plugins/Makefile.in	2005-01-12 23:17:38.000000000 +0100
+++ libquicktime-0.9.4/plugins/Makefile.in	2005-04-22 01:33:50.000000000 +0200
@@ -58,7 +58,7 @@ RECURSIVE_TARGETS = all-recursive check-
 	uninstall-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = audiocodec videocodec opendivx rtjpeg mjpeg dv png \
+DIST_SUBDIRS = audiocodec videocodec opendivx mjpeg dv png \
 	vorbis ffmpeg lame
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
@@ -240,8 +240,7 @@ sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 base_subdirs = audiocodec \
 videocodec \
-opendivx \
-rtjpeg
+opendivx
 
 @HAVE_LIBJPEG_FALSE@jpeg_subdirs = 
 @HAVE_LIBJPEG_TRUE@jpeg_subdirs = mjpeg
diff -Naupr libquicktime-0.9.4.orig/quicktime1394/libdv.h libquicktime-0.9.4/quicktime1394/libdv.h
--- libquicktime-0.9.4.orig/quicktime1394/libdv.h	2002-04-15 08:03:13.000000000 +0200
+++ libquicktime-0.9.4/quicktime1394/libdv.h	2005-04-21 18:49:04.000000000 +0200
@@ -16,7 +16,7 @@ extern "C" {
 #define DV_PAL 1
 
 #ifndef u_int64_t
-#define u_int64_t unsigned long long
+#define u_int64_t long long
 #endif
 
 #include <libdv/dv.h>
