diff -Naupr lirc-0.8.1pre2.orig/configure.in lirc-0.8.1pre2/configure.in
--- lirc-0.8.1pre2.orig/configure.in	2006-10-14 08:19:17.000000000 +0200
+++ lirc-0.8.1pre2/configure.in	2006-10-18 01:36:19.000000000 +0200
@@ -1129,14 +1129,12 @@ elif test "$lirc_driver" = "all"; then
 	lirc_atiusb \
 	lirc_bt829 \
 	lirc_cmdir \
-	lirc_gpio \
 	lirc_i2c \
 	lirc_igorplugusb \
 	lirc_imon \
 	lirc_it87 \
 	lirc_mceusb \
 	lirc_mceusb2 \
-	lirc_parallel \
 	lirc_sasem \
 	lirc_serial \
 	lirc_sir \
