--- apt-pkg/rpm/rpmhandler.cc.orig	2003-04-16 17:10:50.000000000 +0200
+++ apt-pkg/rpm/rpmhandler.cc	2003-04-16 17:12:03.000000000 +0200
@@ -307,6 +307,9 @@
       _error->Error(_("could not open RPM database"));
       return;
    }
+   rpmtsSetVSFlags(Handler, rpmVSFlags(rpmtsVSFlags(Handler) | 
+                   _RPMVSF_NODIGESTS | _RPMVSF_NOSIGNATURES));
+
 #else
    const char *RootDir = NULL;
    if (!Dir.empty())
--- tools/gensrclist.cc.orig	2003-04-16 17:13:18.000000000 +0200
+++ tools/gensrclist.cc	2003-04-16 17:13:24.000000000 +0200
@@ -261,6 +261,7 @@
 #ifdef HAVE_RPM41
    rpmts ts = rpmtsCreate();
    rpmReadConfigFiles(NULL, NULL);
+   rpmtsSetVSFlags(ts, rpmVSFlags(rpmtsVSFlags(ts) | _RPMVSF_NODIGESTS | _RPMVSF_NOSIGNATURES));
 #else
    Header sigs;
 #endif   
