diff -Naupr ffmpeg-20051207.orig/libavcodec/Makefile ffmpeg-20051207/libavcodec/Makefile
--- ffmpeg-20051207.orig/libavcodec/Makefile	2005-11-30 02:40:50.000000000 +0100
+++ ffmpeg-20051207/libavcodec/Makefile	2005-12-07 21:13:38.000000000 +0100
@@ -249,7 +249,9 @@ ifeq ($(CONFIG_AC3),yes)
 OBJS+= a52dec.o
 
 # using builtin liba52 or runtime linked liba52.so.0
-ifneq ($(CONFIG_A52BIN),yes)
+ifeq ($(CONFIG_A52BIN),yes)
+EXTRALIBS += -la52
+else
 OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
 	liba52/imdct.o  liba52/parse.o liba52/crc.o liba52/resample.o
 endif
