diff -Naupr ffmpeg-20061215.orig/configure ffmpeg-20061215/configure
--- ffmpeg-20061215.orig/configure	2006-12-13 21:42:11.000000000 +0100
+++ ffmpeg-20061215/configure	2006-12-18 11:41:24.000000000 +0100
@@ -1456,7 +1456,7 @@ fi
 
 # these are off by default, so fail if requested and not available
 enabled dts     && require libdts dts.h dts_init -ldts -lm
-enabled libgsm  && require libgsm gsm.h gsm_create -lgsm
+enabled libgsm  && require libgsm gsm/gsm.h gsm_create -lgsm
 enabled mp3lame && require LAME lame/lame.h lame_init -lmp3lame -lm
 enabled libvorbis && require libvorbis vorbis/vorbisenc.h vorbis_info_init -lvorbis -lvorbisenc -logg
 enabled libogg  && require libogg ogg/ogg.h ogg_sync_init -logg
diff -Naupr ffmpeg-20061215.orig/libavcodec/libgsm.c ffmpeg-20061215/libavcodec/libgsm.c
--- ffmpeg-20061215.orig/libavcodec/libgsm.c	2006-10-07 17:30:46.000000000 +0200
+++ ffmpeg-20061215/libavcodec/libgsm.c	2006-12-18 11:41:47.000000000 +0100
@@ -25,7 +25,7 @@
  */
 
 #include "avcodec.h"
-#include <gsm.h>
+#include <gsm/gsm.h>
 
 // gsm.h miss some essential constants
 #define GSM_BLOCK_SIZE 33
