diff -Naupr ffmpeg-20070530.orig/configure ffmpeg-20070530/configure
--- ffmpeg-20070530.orig/configure	2007-05-27 18:31:02.000000000 +0200
+++ ffmpeg-20070530/configure	2007-05-30 20:58:15.000000000 +0200
@@ -1624,7 +1624,7 @@ enabled_any libamr_nb libamr_wb && enabl
 enabled liba52     && require liba52 a52dec/a52.h a52_init -la52
 enabled libamr_nb  && require libamrnb amrnb/interf_dec.h Speech_Decode_Frame_init -lamrnb -lm
 enabled libamr_wb  && require libamrwb amrwb/dec_if.h D_IF_init -lamrwb -lm
-enabled libgsm     && require libgsm gsm.h gsm_create -lgsm
+enabled libgsm     && require libgsm gsm/gsm.h gsm_create -lgsm
 enabled libmp3lame && require LAME lame/lame.h lame_init -lmp3lame -lm
 enabled libtheora  && require libtheora theora/theora.h theora_info_init -ltheora -logg
 enabled libvorbis  && require libvorbis vorbis/vorbisenc.h vorbis_info_init -lvorbis -lvorbisenc -logg
diff -Naupr ffmpeg-20070530.orig/libavcodec/libgsm.c ffmpeg-20070530/libavcodec/libgsm.c
--- ffmpeg-20070530.orig/libavcodec/libgsm.c	2007-02-20 12:09:47.000000000 +0100
+++ ffmpeg-20070530/libavcodec/libgsm.c	2007-05-30 20:58:55.000000000 +0200
@@ -28,7 +28,7 @@
 // The idiosyncrasies of GSM-in-WAV are explained at http://kbs.cs.tu-berlin.de/~jutta/toast.html
 
 #include "avcodec.h"
-#include <gsm.h>
+#include <gsm/gsm.h>
 
 // gsm.h miss some essential constants
 #define GSM_BLOCK_SIZE 33
