--- libavcodec/libpostproc/postprocess_template.c.gcc4	2005-02-27 09:56:26.000000000 +0100
+++ libavcodec/libpostproc/postprocess_template.c	2005-04-28 20:52:39.000000000 +0200
@@ -2898,6 +2898,7 @@
 
 		src+= step; // src points to begin of the 8x8 Block
 
+		uint64_t tmp = dc_mask & eq_mask;
 		asm volatile(
 		"movq %4, %%mm6					\n\t"
 		"pcmpeqb %%mm5, %%mm5				\n\t"
@@ -2930,7 +2931,7 @@
 		" js 1b						\n\t"
 
 		: "+r"(offset), "+r"(temp_sums)
-		: "r" ((long)step), "r"(src - offset), "m"(dc_mask & eq_mask)
+		: "r" ((long)step), "r"(src - offset), "m"(tmp)
 		);
 	}else
 		src+= step; // src points to begin of the 8x8 Block
