--- libavcodec/Makefile~	2005-04-20 12:40:04.000000000 +0300
+++ libavcodec/Makefile	2005-04-29 20:26:08.000000000 +0300
@@ -73,7 +73,9 @@
 OBJS+= a52dec.o
 
 # using builtin liba52 or runtime linked liba52.so.0
-ifneq ($(CONFIG_A52BIN),yes)
+ifeq ($(CONFIG_A52BIN),yes)
+EXTRALIBS += -la52
+else
 OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
 	liba52/imdct.o  liba52/parse.o liba52/crc.o liba52/resample.o
 endif
