diff -Naupr ffmpeg-20051207.orig/libavcodec/libgsm.c ffmpeg-20051207/libavcodec/libgsm.c
--- ffmpeg-20051207.orig/libavcodec/libgsm.c	2005-05-23 01:39:11.000000000 +0200
+++ ffmpeg-20051207/libavcodec/libgsm.c	2005-12-08 00:56:37.000000000 +0100
@@ -23,7 +23,7 @@
  */
 
 #include "avcodec.h"
-#include <gsm.h>
+#include <gsm/gsm.h>
 
 // gsm.h miss some essential constants
 #define GSM_BLOCK_SIZE 33
