diff -Naru ffmpeg-0.4.9-pre1.orig/libavcodec/Makefile ffmpeg-0.4.9-pre1/libavcodec/Makefile
--- ffmpeg-0.4.9-pre1.orig/libavcodec/Makefile	2004-06-18 16:11:15.000000000 +0300
+++ ffmpeg-0.4.9-pre1/libavcodec/Makefile	2004-08-31 21:35:15.922873172 +0300
@@ -89,7 +89,7 @@
 
 ifeq ($(CONFIG_PP),yes)
 ifeq ($(SHARED_PP),yes)
-EXTRALIBS += -lpostproc
+EXTRALIBS += -Llibpostproc -lpostproc
 else
 # LIBS += libpostproc/libpostproc.a ... should be fixed
 OBJS += libpostproc/postprocess.o
diff -Naru ffmpeg-0.4.9-pre1.orig/Makefile ffmpeg-0.4.9-pre1/Makefile
--- ffmpeg-0.4.9-pre1.orig/Makefile	2004-07-08 14:02:04.000000000 +0300
+++ ffmpeg-0.4.9-pre1/Makefile	2004-08-31 21:34:43.092347004 +0300
@@ -77,6 +77,9 @@
 OBJS = ffmpeg.o ffserver.o cmdutils.o ffplay.o
 SRCS = $(OBJS:.o=.c) $(ASM_OBJS:.o=.s)
 FFLIBS = -L./libavformat -lavformat -L./libavcodec -lavcodec
+ifeq ($(SHARED_PP),yes)
+FFLIBS += -L./libavcodec/libpostproc -lpostproc
+endif
 
 all: lib $(PROG) $(PROGTEST) $(VHOOK) $(QTFASTSTART) $(DOC)
 
