--- libavcodec/x264.c	2006-08-30 23:18:17.000000000 +0200
+++ x264.c	2006-10-24 13:21:09.000000000 +0200
@@ -144,7 +146,7 @@ X264_init(AVCodecContext *avctx)
     else{
         if(avctx->crf){
             x4->params.rc.i_rc_method = X264_RC_CRF;
-            x4->params.rc.i_rf_constant = avctx->crf;
+            x4->params.rc.f_rf_constant = avctx->crf;
         }else if(avctx->cqp > -1){
             x4->params.rc.i_rc_method = X264_RC_CQP;
             x4->params.rc.i_qp_constant = avctx->cqp;
