diff -Naupr ffmpeg-0.4.9-pre1.orig/libavcodec/i386/dsputil_mmx.c ffmpeg-0.4.9-pre1/libavcodec/i386/dsputil_mmx.c
--- ffmpeg-0.4.9-pre1.orig/libavcodec/i386/dsputil_mmx.c	2004-06-08 04:13:44.000000000 +0200
+++ ffmpeg-0.4.9-pre1/libavcodec/i386/dsputil_mmx.c	2004-10-29 16:12:45.849678584 +0200
@@ -57,7 +57,7 @@ static const uint64_t ff_pb_FC attribute
     "pcmpeqd %%" #regd ", %%" #regd " \n\t"\
     "paddb %%" #regd ", %%" #regd " \n\t" ::)
 
-#ifndef PIC
+#ifndef __PIC__
 #define MOVQ_BONE(regd)  __asm __volatile ("movq %0, %%" #regd " \n\t" ::"m"(mm_bone))
 #define MOVQ_WTWO(regd)  __asm __volatile ("movq %0, %%" #regd " \n\t" ::"m"(mm_wtwo))
 #else
diff -Naupr ffmpeg-0.4.9-pre1.orig/libavcodec/i386/dsputil_mmx_rnd.h ffmpeg-0.4.9-pre1/libavcodec/i386/dsputil_mmx_rnd.h
--- ffmpeg-0.4.9-pre1.orig/libavcodec/i386/dsputil_mmx_rnd.h	2004-05-18 19:09:46.000000000 +0200
+++ ffmpeg-0.4.9-pre1/libavcodec/i386/dsputil_mmx_rnd.h	2004-10-29 16:13:06.325565776 +0200
@@ -96,7 +96,7 @@ static void DEF(put, pixels8_l2)(uint8_t
 	"addl	%5, %3			\n\t"
 	"subl	$4, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
         :"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
         :"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -191,7 +191,7 @@ static void DEF(put, pixels16_l2)(uint8_
 	"addl	$32, %2			\n\t"
 	"subl	$2, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
diff -Naupr ffmpeg-0.4.9-pre1.orig/libavcodec/msmpeg4.c ffmpeg-0.4.9-pre1/libavcodec/msmpeg4.c
--- ffmpeg-0.4.9-pre1.orig/libavcodec/msmpeg4.c	2004-05-25 20:07:24.000000000 +0200
+++ ffmpeg-0.4.9-pre1/libavcodec/msmpeg4.c	2004-10-29 16:13:30.652867464 +0200
@@ -716,7 +716,7 @@ static inline int msmpeg4_pred_dc(MpegEn
        necessitate to modify mpegvideo.c. The problem comes from the
        fact they decided to store the quantized DC (which would lead
        to problems if Q could vary !) */
-#if defined ARCH_X86 && !defined PIC
+#if defined ARCH_X86 && !defined __PIC__
     asm volatile(
         "movl %3, %%eax		\n\t"
 	"shrl $1, %%eax		\n\t"
