diff -Naru ffmpeg-0.4.9-pre1.orig/libavcodec/libpostproc/Makefile ffmpeg-0.4.9-pre1/libavcodec/libpostproc/Makefile
--- ffmpeg-0.4.9-pre1.orig/libavcodec/libpostproc/Makefile	2003-05-11 14:26:35.000000000 +0300
+++ ffmpeg-0.4.9-pre1/libavcodec/libpostproc/Makefile	2004-07-29 14:27:29.471838988 +0300
@@ -50,7 +50,7 @@
 install: all
 ifeq ($(SHARED_PP),yes)
 	install -d $(prefix)/lib
-	install -s -m 755 $(SPPLIB) $(prefix)/lib/$(SPPLIB).$(SPPVERSION)
+	install -m 755 $(SPPLIB) $(prefix)/lib/$(SPPLIB).$(SPPVERSION)
 	ln -sf $(SPPLIB).$(SPPVERSION) $(prefix)/lib/$(SPPLIB)
 	ldconfig || true
 	mkdir -p $(prefix)/include/postproc
diff -Naru ffmpeg-0.4.9-pre1.orig/libavcodec/Makefile ffmpeg-0.4.9-pre1/libavcodec/Makefile
--- ffmpeg-0.4.9-pre1.orig/libavcodec/Makefile	2004-06-18 16:11:15.000000000 +0300
+++ ffmpeg-0.4.9-pre1/libavcodec/Makefile	2004-07-29 14:27:35.447923035 +0300
@@ -266,10 +266,10 @@
 ifeq ($(BUILD_SHARED),yes)
 install: all install-headers
 ifeq ($(CONFIG_WIN32),yes)
-	install -s -m 755 $(SLIB) "$(prefix)"
+	install -m 755 $(SLIB) "$(prefix)"
 else
 	install -d $(prefix)/lib
-	install -s -m 755 $(SLIB) $(prefix)/lib/libavcodec-$(VERSION).so
+	install -m 755 $(SLIB) $(prefix)/lib/libavcodec-$(VERSION).so
 	ln -sf libavcodec-$(VERSION).so $(prefix)/lib/libavcodec.so
 	ldconfig || true
 endif
diff -Naru ffmpeg-0.4.9-pre1.orig/libavformat/Makefile ffmpeg-0.4.9-pre1/libavformat/Makefile
--- ffmpeg-0.4.9-pre1.orig/libavformat/Makefile	2004-04-24 18:16:23.000000000 +0300
+++ ffmpeg-0.4.9-pre1/libavformat/Makefile	2004-07-29 14:27:24.069570995 +0300
@@ -102,10 +102,10 @@
 ifeq ($(BUILD_SHARED),yes)
 install: all install-headers
 ifeq ($(CONFIG_WIN32),yes)
-	install -s -m 755 $(SLIB) "$(prefix)"
+	install -m 755 $(SLIB) "$(prefix)"
 else
 	install -d $(prefix)/lib
-	install -s -m 755 $(SLIB) $(prefix)/lib/libavformat-$(VERSION).so
+	install -m 755 $(SLIB) $(prefix)/lib/libavformat-$(VERSION).so
 	ln -sf libavformat-$(VERSION).so $(prefix)/lib/libavformat.so
 	ldconfig || true
 endif
diff -Naru ffmpeg-0.4.9-pre1.orig/Makefile ffmpeg-0.4.9-pre1/Makefile
--- ffmpeg-0.4.9-pre1.orig/Makefile	2004-07-08 14:02:04.000000000 +0300
+++ ffmpeg-0.4.9-pre1/Makefile	2004-07-29 14:26:49.039802320 +0300
@@ -89,7 +89,6 @@
 
 ffmpeg$(EXESUF): ffmpeg_g$(EXESUF)
 	cp -p $< $@
-	$(STRIP) $@
 
 ffserver$(EXESUF): ffserver.o .libs
 	$(CC) $(LDFLAGS) $(FFSLDFLAGS) -o $@ ffserver.o $(FFLIBS) $(EXTRALIBS) 
@@ -99,7 +98,6 @@
 
 ffplay$(EXESUF): ffplay_g$(EXESUF)
 	cp -p $< $@
-	$(STRIP) $@
 
 output_example$(EXESUF): output_example.o .libs
 	$(CC) $(LDFLAGS) -o $@ output_example.o $(FFLIBS) $(EXTRALIBS)
