diff -Naupr gtkpod-0.88.2.orig/src/prefs_window.c gtkpod-0.88.2/src/prefs_window.c
--- gtkpod-0.88.2.orig/src/prefs_window.c	2004-12-29 11:00:32.000000000 +0100
+++ gtkpod-0.88.2/src/prefs_window.c	2005-04-20 16:08:39.000000000 +0200
@@ -59,7 +59,7 @@ static void prefs_window_set_path (PathT
    chooser. path_type[] specifies whether to browse for dirs or for
    files.
 */
-static const gchar *path_button_names[] =
+const gchar *path_button_names[] =
 {
     "play_now_path_button",
     "play_enqueue_path_button",
