diff -Naupr xine-lib-1.0.1.orig/configure xine-lib-1.0.1/configure
--- xine-lib-1.0.1.orig/configure	2005-04-26 10:17:56.000000000 +0200
+++ xine-lib-1.0.1/configure	2005-04-28 23:52:05.000000000 +0200
@@ -40979,7 +40979,6 @@ _ACEOF
 
 
     enable_w32dll="yes"
-    enable_ffmmx="yes"
     no_fpic="yes"
     ;;
 
@@ -41266,7 +41265,6 @@ cat >>confdefs.h <<\_ACEOF
 #define FPM_INTEL 1
 _ACEOF
 
-	enable_ffmmx="yes"
 	enable_w32dll="yes"
 
         case "$host_or_hostalias" in
diff -Naupr xine-lib-1.0.1.orig/configure.ac xine-lib-1.0.1/configure.ac
--- xine-lib-1.0.1.orig/configure.ac	2005-04-26 10:09:18.000000000 +0200
+++ xine-lib-1.0.1/configure.ac	2005-04-28 23:51:48.000000000 +0200
@@ -1664,7 +1664,6 @@ case "$host_or_hostalias" in
     AC_DEFINE_UNQUOTED(FPM_INTEL,,[Define to select libmad fixed point arithmetic implementation])
 
     enable_w32dll="yes"
-    enable_ffmmx="yes"
     no_fpic="yes"
     ;;
 
@@ -1770,7 +1769,6 @@ case "$host_or_hostalias" in
 	dnl
     	AC_DEFINE_UNQUOTED(ARCH_X86,,[Define this if you're running x86 architecture])
 	AC_DEFINE(FPM_INTEL,1,[Define to select libmad fixed point arithmetic implementation])
-	enable_ffmmx="yes"
 	enable_w32dll="yes"
 
         case "$host_or_hostalias" in
