--- SDL_Pango-0.1.2/src/SDL_Pango.c.warn	2004-12-10 18:06:33.000000000 +0900
+++ SDL_Pango-0.1.2/src/SDL_Pango.c	2006-09-23 22:49:42.000000000 +0900
@@ -437,6 +437,10 @@
     PangoRectangle ink_rect;
     int x_off = 0;
 
+    /* only suppress warning */
+    fg_color.red = fg_color.green = fg_color.blue = 0;
+    bg_color.red = bg_color.green = bg_color.blue = 0;
+
     while (tmp_list) {
 	SDLPango_Matrix color_matrix = context->color_matrix;
 	PangoUnderline uline = PANGO_UNDERLINE_NONE;
--- SDL_Pango-0.1.2/src/SDL_Pango.h.warn	2004-12-10 18:06:33.000000000 +0900
+++ SDL_Pango-0.1.2/src/SDL_Pango.h	2006-09-23 22:49:42.000000000 +0900
@@ -47,10 +47,10 @@
 } SDLPango_Matrix;
 
 const SDLPango_Matrix _MATRIX_WHITE_BACK
-    = {255, 0, 0, 0,
-       255, 0, 0, 0,
-       255, 0, 0, 0,
-       255, 255, 0, 0,};
+= {{{255, 0, 0, 0},
+    {255, 0, 0, 0},
+    {255, 0, 0, 0},
+    {255, 255, 0, 0}}};
 
 /*!
     Specifies white back and black letter.
@@ -58,40 +58,40 @@
 const SDLPango_Matrix *MATRIX_WHITE_BACK = &_MATRIX_WHITE_BACK;
 
 const SDLPango_Matrix _MATRIX_BLACK_BACK
-    = {0, 255, 0, 0,
-       0, 255, 0, 0,
-       0, 255, 0, 0,
-       255, 255, 0, 0,};
+= {{{0, 255, 0, 0},
+    {0, 255, 0, 0},
+    {0, 255, 0, 0},
+    {255, 255, 0, 0}}};
 /*!
     Specifies black back and white letter.
 */
 const SDLPango_Matrix *MATRIX_BLACK_BACK = &_MATRIX_BLACK_BACK;
 
 const SDLPango_Matrix _MATRIX_TRANSPARENT_BACK_BLACK_LETTER
-    = {0, 0, 0, 0,
-       0, 0, 0, 0,
-       0, 0, 0, 0,
-       0, 255, 0, 0,};
+= {{{0, 0, 0, 0},
+    {0, 0, 0, 0},
+    {0, 0, 0, 0},
+    {0, 255, 0, 0}}};
 /*!
     Specifies transparent back and black letter.
 */
 const SDLPango_Matrix *MATRIX_TRANSPARENT_BACK_BLACK_LETTER = &_MATRIX_TRANSPARENT_BACK_BLACK_LETTER;
 
 const SDLPango_Matrix _MATRIX_TRANSPARENT_BACK_WHITE_LETTER
-    = {255, 255, 0, 0,
-       255, 255, 0, 0,
-       255, 255, 0, 0,
-       0, 255, 0, 0,};
+= {{{255, 255, 0, 0},
+    {255, 255, 0, 0},
+    {255, 255, 0, 0},
+    {0, 255, 0, 0}}};
 /*!
     Specifies transparent back and white letter.
 */
 const SDLPango_Matrix *MATRIX_TRANSPARENT_BACK_WHITE_LETTER = &_MATRIX_TRANSPARENT_BACK_WHITE_LETTER;
 
 const SDLPango_Matrix _MATRIX_TRANSPARENT_BACK_TRANSPARENT_LETTER
-    = {255, 255, 0, 0,
-       255, 255, 0, 0,
-       255, 255, 0, 0,
-       0, 0, 0, 0,};
+= {{{255, 255, 0, 0},
+    {255, 255, 0, 0},
+    {255, 255, 0, 0},
+    {0, 0, 0, 0}}};
 /*!
     Specifies transparent back and transparent letter.
     This is useful for KARAOKE like rendering.
