diff -ur effectv-0.3.11/utils.c effectv-0.3.11.phh/utils.c
--- effectv-0.3.11/utils.c	2006-02-14 15:06:17.000000000 +0100
+++ effectv-0.3.11.phh/utils.c	2006-07-04 12:07:24.623118250 +0200
@@ -26,7 +26,7 @@
 /*
  * HSI color system utilities
  */
-static int trunc(double f)
+static int HSItrunc(double f)
 {
 	int i;
 
@@ -44,9 +44,9 @@
 	Gv=1+S*sin(H);
 	Bv=1+S*sin(H+2*M_PI/3);
 	T=255.999*I/2;
-	*r=trunc(Rv*T);
-	*g=trunc(Gv*T);
-	*b=trunc(Bv*T);
+	*r=HSItrunc(Rv*T);
+	*g=HSItrunc(Gv*T);
+	*b=HSItrunc(Bv*T);
 }
 
 /*
