diff -Naupr rhythmbox-0.8.6.orig/sources/rb-ipod-source.c rhythmbox-0.8.6/sources/rb-ipod-source.c
--- rhythmbox-0.8.6.orig/sources/rb-ipod-source.c	2004-09-17 00:03:09.000000000 +0200
+++ rhythmbox-0.8.6/sources/rb-ipod-source.c	2004-09-20 00:23:56.656986600 +0200
@@ -238,7 +238,7 @@ load_ipod_db_idle_cb (RBiPodSongAdderCtx
 			GValue value = {0, };
 			g_value_init (&value, G_TYPE_ULONG);
 			g_value_set_ulong (&value, song->cd_nr);
-			rhythmdb_entry_set (RHYTHMDB (db), entry, 
+			rhythmdb_entry_set (RHYTHMDB (ctxt->db), entry, 
 					    RHYTHMDB_PROP_DISC_NUMBER, 
 					    &value);
 			g_value_unset (&value);
