--- mozilla/xpinstall/src/nsSoftwareUpdate.cpp.orig	2004-05-22 10:02:23.000000000 +0200
+++ mozilla/xpinstall/src/nsSoftwareUpdate.cpp	2004-07-27 17:12:42.666752904 +0200
@@ -512,7 +512,7 @@ NS_GENERIC_FACTORY_SINGLETON_CONSTRUCTOR
                                          nsSoftwareUpdate::GetInstance)
 NS_GENERIC_FACTORY_CONSTRUCTOR(nsInstallTrigger)
 NS_GENERIC_FACTORY_CONSTRUCTOR(nsInstallVersion)
-NS_GENERIC_FACTORY_CONSTRUCTOR(nsXPInstallManager);
+NS_GENERIC_FACTORY_CONSTRUCTOR(nsXPInstallManager)
 NS_GENERIC_FACTORY_CONSTRUCTOR(nsSoftwareUpdateNameSet)
 
 //----------------------------------------------------------------------
