--- nagios-plugins-1.4.4/plugins/check_ide_smart.c.orig	2006-10-19 03:25:16.000000000 +0300
+++ nagios-plugins-1.4.4/plugins/check_ide_smart.c	2006-10-26 11:28:53.000000000 +0300
@@ -439,7 +439,7 @@ print_values (values_t * p, thresholds_t
 
 
 void
-print_thresholds (thresholds_t * p) 
+smart_print_thresholds (thresholds_t * p) 
 {
 	threshold_t * threshold = p->thresholds;
 	int i;
@@ -541,7 +541,7 @@ print_help (void)
 void
 print_usage (void)
 {
-	printf (_("Usage:");
+  printf (_("Usage:"));
   printf ("%s [-d <device>] [-i <immediate>] [-q quiet] [-1 <auto-on>]",progname);
-  pritnf (" [-O <auto-off>] [-n <nagios>]\n");
+  printf (" %s\n", "[-O <auto-off>] [-n <nagios>]");
 }
