--- modules-3.2.7/init/Makefile.in.bindir	2009-09-22 12:13:52.000000000 -0600
+++ modules-3.2.7/init/Makefile.in	2009-09-23 12:19:50.797470155 -0600
@@ -404,7 +404,7 @@
 	sed -e "/@$(if $(subst 0,,$(WANTS_VERSIONING)),NOT,)VERSIONING\@/d; \
 	s,@$(if $(subst 0,,$(WANTS_VERSIONING)),,NOT)VERSIONING\@,,g; \
 	s,@prefix\@,${prefix},g; \
-	s,@bindir\@,${exec_prefix}/bin,g; \
+	s,@bindir\@,${bindir},g; \
 	s,@VERSION\@,@VERSION@,g; \
 	s,@BASEPREFIX\@,@BASEPREFIX@,g;" < $< > $@
 
