--- thinkfinger-0.3/libthinkfinger/libthinkfinger.c.has-device	2007-03-30 10:57:24.000000000 +0200
+++ thinkfinger-0.3/libthinkfinger/libthinkfinger.c	2007-09-22 12:32:24.000000000 +0200
@@ -1,4 +1,5 @@
-/*
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
  *   ThinkFinger - A driver for the UPEK/SGS Thomson Microelectronics
  *   fingerprint reader.
  *
@@ -41,6 +42,14 @@
 #define DEFAULT_BULK_SIZE 0x40
 #define INITIAL_SEQUENCE  0x60
 
+#ifndef        FALSE
+#define        FALSE   (0)
+#endif
+
+#ifndef        TRUE
+#define        TRUE    (!FALSE)
+#endif
+
 static char init_a[17] = {
 	0x43, 0x69, 0x61, 0x6f, 0x04, 0x00, 0x08, 0x01,
 	0x00, 0xe8, 0x03, 0x00, 0x00, 0xff, 0x07, 0xdb,
@@ -405,6 +414,20 @@
 	return;
 }
 
+int libthinkfinger_has_device (libthinkfinger *tf)
+{
+	struct usb_device *usb_dev;
+	int retval;
+
+	retval = FALSE;
+	usb_dev = _libthinkfinger_usb_device_find ();
+	if (usb_dev != NULL) {
+		retval = TRUE;
+	}
+
+	return retval;
+}
+
 static libthinkfinger_init_status _libthinkfinger_usb_init (libthinkfinger *tf)
 {
 	libthinkfinger_init_status retval = TF_INIT_UNDEFINED;
--- thinkfinger-0.3/libthinkfinger/libthinkfinger.h.has-device	2007-03-30 10:22:02.000000000 +0200
+++ thinkfinger-0.3/libthinkfinger/libthinkfinger.h	2007-09-22 12:32:24.000000000 +0200
@@ -165,6 +165,16 @@
  */
 libthinkfinger *libthinkfinger_new(libthinkfinger_init_status* init_status);
 
+/** @brief check to see if a scanner device is present
+ *
+ * check to see if a scanner device is present
+ *
+ * @param tf struct libthinkfinger
+ *
+ * @return TRUE if a device is present
+ */
+int libthinkfinger_has_device(libthinkfinger *tf);
+
 /** @brief free an instance of libthinkfinger
  *
  * @param tf pointer to struct libthinkfinger
--- thinkfinger-0.3/pam/pam_thinkfinger.c.has-device	2007-03-29 17:00:45.000000000 +0200
+++ thinkfinger-0.3/pam/pam_thinkfinger.c	2007-09-22 12:32:24.000000000 +0200
@@ -1,4 +1,6 @@
-/*   ThinkFinger Pluggable Authentication Module
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ *   ThinkFinger Pluggable Authentication Module
  *
  *   PAM module for libthinkfinger which is a driver for the UPEK/SGS Thomson
  *   Microelectronics fingerprint reader.
@@ -158,14 +160,13 @@
 		pam_thinkfinger->swipe_retval = PAM_AUTH_ERR;
 		pam_thinkfinger_log (pam_thinkfinger, LOG_NOTICE,
 				     "User '%s' verification failed (0x%x).", pam_thinkfinger->user, tf_state);
-		goto out;
 	}
 
 	ret = uinput_cr (&pam_thinkfinger->uinput_fd);
 	if (ret != 0)
 		pam_thinkfinger_log (pam_thinkfinger, LOG_ERR,
 				     "Could not send carriage return via uinput: %s.", strerror (ret));
-out:
+
 	pam_thinkfinger_log (pam_thinkfinger, LOG_NOTICE,
 			     "%s returning '%d': %s.", __FUNCTION__, pam_thinkfinger->swipe_retval,
 			     pam_thinkfinger->swipe_retval ? pam_strerror (pam_thinkfinger->pamh, pam_thinkfinger->swipe_retval) : "success");
@@ -258,6 +259,15 @@
 		goto out;
 	}
 
+	ret = libthinkfinger_has_device (pam_thinkfinger.tf);
+	if (! ret) {
+		retval = PAM_AUTHINFO_UNAVAIL;
+
+		if (pam_thinkfinger.tf)
+			libthinkfinger_free (pam_thinkfinger.tf);
+		goto out;
+	}
+
 	ret = pthread_create (&pam_thinkfinger.t_pam_prompt, NULL, (void *) &pam_prompt_thread, &pam_thinkfinger);
 	if (ret != 0) {
 		pam_thinkfinger_log (&pam_thinkfinger, LOG_ERR, "Error calling pthread_create (%s).", strerror (ret));
