--- libsigc++-1.2.5/sigc++-1.2.pc.in.~1~	2002-10-09 02:05:34.000000000 -0700
+++ libsigc++-1.2.5/sigc++-1.2.pc.in	2004-03-25 12:27:27.454635054 -0800
@@ -7,4 +7,4 @@
 Description: Type-safe signal and callback system for C++
 Version: @VERSION@
 Libs: -L${libdir} -lsigc-1.2
-Cflags: -I${exec_prefix}/lib/sigc++-1.2/include -I${includedir}/sigc++-1.2
+Cflags: -I${libdir}/sigc++-1.2/include -I${includedir}/sigc++-1.2
