diff -Naupr zaptel-1.0.9.2.orig/Makefile zaptel-1.0.9.2/Makefile
--- zaptel-1.0.9.2.orig/Makefile	2005-08-08 20:06:02.000000000 +0200
+++ zaptel-1.0.9.2/Makefile	2005-09-15 18:30:16.000000000 +0200
@@ -13,7 +13,8 @@ BASEADDR=0xd0000
 # (assuming He's running Linux -- which we all know He must).
 #
 HOSTCC=gcc
-KINCLUDES=$(shell if [ -d /usr/src/linux-2.4/include ]; then echo /usr/src/linux-2.4/include ; else echo /usr/src/linux/include ; fi)
+[ -z "${KVERSION}" ] && KVERSION=`uname -r`
+KINCLUDES=$(shell if [ -d /usr/src/linux-2.4/include ]; then echo /usr/src/linux-2.4/include ; else echo /lib/modules/${KVERSION}/build ; fi)
 
 CFLAGS+=-I. -O4 -g -Wall -DBUILDING_TONEZONE #-DTONEZONE_DRIVER
 CFLAGS+=$(shell if uname -m | grep -q ppc; then echo "-fsigned-char"; fi)
@@ -38,7 +39,7 @@ INSTALL_PREFIX=
 CONFIG_FILE=$(INSTALL_PREFIX)/etc/zaptel.conf
 CFLAGS+=-DZAPTEL_CONFIG=\"$(CONFIG_FILE)\"
 
-BUILDVER=$(shell if uname -r | grep -q ^2.6; then echo "linux26"; else echo "linux24"; fi)
+BUILDVER=$(shell if echo ${KVERSION} | grep -q ^2.6; then echo "linux26"; else echo "linux24"; fi)
 MODCONF=$(shell if [ -d $(ROOT_PREFIX)/etc/modprobe.d ]; then echo "$(ROOT_PREFIX)/etc/modprobe.d/zaptel"; elif [ -d $(ROOT_PREFIX)/etc/modutils ]; then echo "$(ROOT_PREFIX)/etc/modutils/zaptel"; elif [ -f $(ROOT_PREFIX)/etc/modprobe.conf ]; then echo "$(ROOT_PREFIX)/etc/modprobe.conf"; elif [ -f $(ROOT_PREFIX)/etc/modules.conf ]; then echo "$(ROOT_PREFIX)/etc/modules.conf"; else echo $(ROOT_PREFIX)/etc/conf.modules ; fi)
 
 ifeq (${BUILDVER},linux24)
@@ -58,7 +59,7 @@ endif
 TZOBJS=zonedata.lo tonezone.lo
 LIBTONEZONE=libtonezone.so.1.0
 MODULES=zaptel tor2 torisa wcusb wcfxo wcfxs \
-	ztdynamic ztd-eth wct1xxp wct4xxp wcte11xp # ztdummy
+	ztdynamic ztd-eth wct1xxp wct4xxp wcte11xp ztdummy
 #MODULES+=wcfxsusb
 
 MODULESO=$(shell for x in $(MODULES); do echo "$$x.o "; done )
@@ -71,13 +72,12 @@ BINS=ztcfg torisatool makefw ztmonitor z
 PRIMARY=torisa
 #PRIMARY=wcfxo
 PWD=$(shell pwd)
-KERNEL_SOURCE?=/lib/modules/`uname -r`/build
+KERNEL_SOURCE?=/lib/modules/${KVERSION}/build
 
 all: $(BUILDVER) $(LIBTONEZONE)
 
 linux24: $(MODULESO) $(BINS)
 
-linux26: 
 linux26: prereq $(BINS)
 	@if ! [ -d $(KERNEL_SOURCE) ]; then echo "You do not appear to have the kernel sources for your current kernel installed."; exit 1 ; fi
 	make -C $(KERNEL_SOURCE) SUBDIRS=$(PWD) modules
@@ -262,7 +262,7 @@ else
 	@echo "**** If you are running udev, read README.udev"
 endif
 
-install:  all devices
+install:  all
 	install -D -m 755 ztcfg $(INSTALL_PREFIX)/sbin/ztcfg
 	if [ -f sethdlc-new ]; then \
 		install -D -m 755 sethdlc-new $(INSTALL_PREFIX)/sbin/sethdlc; \
@@ -272,17 +272,17 @@ install:  all devices
 	if [ -f zttool ]; then install -D -m 755 zttool $(INSTALL_PREFIX)/sbin/zttool; fi
 	if [ -f zaptel.ko ]; then \
 		for x in $(MODULESKO) ztdummy.ko; do \
-			install -D -m 644 $$x $(INSTALL_PREFIX)/lib/modules/`uname -r`/misc/$$x ; \
+			install -D -m 644 $$x $(INSTALL_PREFIX)/lib/modules/${KVERSION}/kernel/misc/$$x ; \
 		done; \
 		if ! [ -f wcfxsusb.ko ]; then \
-			rm -f $(INSTALL_PREFIX)/lib/modules/`uname -r`/misc/wcfxsusb.o; \
+			rm -f $(INSTALL_PREFIX)/lib/modules/${KVERSION}/kernel/misc/wcfxsusb.o; \
 		fi; \
 	else \
 		for x in $(MODULESO); do \
-			install -D -m 644 $$x $(INSTALL_PREFIX)/lib/modules/`uname -r`/misc/$$x ; \
+			install -D -m 644 $$x $(INSTALL_PREFIX)/lib/modules/${KVERSION}/kernel/misc/$$x ; \
 		done; \
 		if ! [ -f wcfxsusb.o ]; then \
-			rm -f $(INSTALL_PREFIX)/lib/modules/`uname -r`/misc/wcfxsusb.ko; \
+			rm -f $(INSTALL_PREFIX)/lib/modules/${KVERSION}/kernel/misc/wcfxsusb.ko; \
 		fi; \
 	fi
 	install -D -m 755 $(LIBTONEZONE) $(INSTALL_PREFIX)/usr/lib/$(LIBTONEZONE)
