--- nut-2.4.1/m4/nut_check_libhal.m4.halpath	2009-02-17 12:20:40.000000000 +0300
+++ nut-2.4.1/m4/nut_check_libhal.m4	2009-12-25 20:13:55.000000000 +0300
@@ -75,50 +75,12 @@
    dnl As per HAL spec, §5 Callouts and §2 Device Information Files
    dnl - addon install path: $libdir/hal
    AC_MSG_CHECKING(for libhal Callouts path)
-   HAL_CALLOUTS_PATH=`pkg-config --silence-errors  --variable=libexecdir hal`
-   if (test -z "$HAL_CALLOUTS_PATH")
-   then
-     # fallback to detecting the right path
-     if (test -d "${libdir}/hal")
-     then
-       # For Debian
-       HAL_CALLOUTS_PATH="${libdir}/hal"
-	     AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
-     else # For RedHat
-       if (test -d "/usr/libexec")
-       then
-         HAL_CALLOUTS_PATH="${libexecdir}"
-         AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
-       else
-         # FIXME
-         HAL_CALLOUTS_PATH="${libdir}/hal"
-	     AC_MSG_RESULT(using default (${HAL_CALLOUTS_PATH}))
-       fi
-     fi
-   else
-	   AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
-   fi
+   HAL_CALLOUTS_PATH="${libexecdir}"
+   AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
    
    dnl - fdi install path: $datarootdir/hal/fdi/information/20thirdparty
-   AC_MSG_CHECKING(for libhal Device Information path)
-   HAL_FDI_PATH=`pkg-config --silence-errors  --variable=hal_fdidir hal`
-   if (test -z "$HAL_FDI_PATH")
-   then
-     # fallback to detecting the right path
-     if (test -d "/usr/share/hal/fdi/information/20thirdparty")
-     then
-       # seems supported everywhere
-       HAL_FDI_PATH="${datarootdir}/hal/fdi/information/20thirdparty"
-       AC_MSG_RESULT(${HAL_FDI_PATH})
-     else
-       # FIXME
-       HAL_FDI_PATH=""
-       AC_MSG_RESULT(not found)
-     fi
-   else
-     HAL_FDI_PATH="${HAL_FDI_PATH}/information/20thirdparty"
-     AC_MSG_RESULT(${HAL_FDI_PATH})
-   fi
+   HAL_FDI_PATH="${datadir}/hal/fdi/information/20thirdparty"
+   AC_MSG_RESULT(${HAL_FDI_PATH})
 
    if test "${nut_have_libhal}" != "yes"; then
       dnl try again
