--- samba-2.0.7/source/script/installbin.sh.foo	Mon Oct 16 13:29:27 2000
+++ samba-2.0.7/source/script/installbin.sh	Mon Oct 16 13:30:22 2000
@@ -33,9 +33,9 @@
  chmod $INSTALLPERMS $BINDIR/$p2
 
  # this is a special case, mount needs this in a specific location
- if [ $p2 = smbmount ]; then
-   ln -sf $BINDIR/$p2 /sbin/mount.smbfs
- fi
+ #if [ $p2 = smbmount ]; then
+ #  ln -sf $BINDIR/$p2 /sbin/mount.smbfs
+ #fi
 done
 
 
