diff -Naupr faad2.orig/Makefile.am faad2/Makefile.am
--- faad2.orig/Makefile.am	2004-02-06 11:04:39.000000000 -0500
+++ faad2/Makefile.am	2004-02-26 07:47:09.367426792 -0500
@@ -2,7 +2,7 @@ SUBDIRS = libfaad common frontend plugin
 
 EXTRA_DIST = faad2.spec
 
-rpm:    Makefile
-       make dist
-       $(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
-       rm $(PACKAGE)-$(VERSION).tar.gz
+rpm:	Makefile
+	make dist
+	$(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
+	rm $(PACKAGE)-$(VERSION).tar.gz
