diff -uNr faad2-orig/common/mp4ff/Makefile.am faad2/common/mp4ff/Makefile.am
--- faad2-orig/common/mp4ff/Makefile.am	2004-02-06 14:12:28.000000000 +0100
+++ faad2/common/mp4ff/Makefile.am	2006-01-02 21:27:46.000000000 +0100
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libmp4ff.la
 
-include_HEADERS = mp4ff.h
+include_HEADERS = mp4ff.h mp4ff_int_types.h
 
 libmp4ff_la_SOURCES = mp4ff.c mp4atom.c mp4meta.c mp4sample.c mp4util.c \
    mp4tagupdate.c mp4ff.h mp4ffint.h mp4ff_int_types.h \
