diff -Naupr torcs-1.2.2.orig/Make-config torcs-1.2.2/Make-config
--- torcs-1.2.2.orig/Make-config	2004-02-12 19:12:42.000000000 +0100
+++ torcs-1.2.2/Make-config	2004-07-23 18:09:23.050675688 +0200
@@ -47,13 +47,13 @@ INSTALL_SCRIPT = ${INSTALL_PROGRAM}
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 
 CFLAGSD = -D_SVID_SOURCE -D_BSD_SOURCE -DSHM
-CXXFLAGS =  -march=i686 -Wall -Wstrict-prototypes -O3 -mieee-fp -Wall -Wstrict-prototypes -O3 -mieee-fp -march=i686 ${CFLAGSD}
-CFLAGS   =  -march=i686 -Wall -Wstrict-prototypes -O3 -mieee-fp -Wall -Wstrict-prototypes -O3 -mieee-fp ${CFLAGSD}
+CXXFLAGS = $(RPM_OPT_FLAGS) -fPIC -Wall -mieee-fp -Wall -Wstrict-prototypes ${CFLAGSD}
+CFLAGS   = $(RPM_OPT_FLAGS) -fPIC -Wall -mieee-fp -Wall -Wstrict-prototypes ${CFLAGSD}
 
 #EXT_LIBS = -lpthread -lplibssgaux -lplibssg -lplibsm -lplibsl -lplibsg -lplibul -lglut -lGLU -lGL -lpng -lz -ldl -lXxf86vm -lXmu -lXi -lXt -lSM -lICE -lXext -lX11 -lm 
 EXT_LIBS = -lplibssgaux -lplibssg -lplibsm -lplibsl -lplibsg -lplibul -lglut -lGLU -lGL -lpng -lz -ldl -lXxf86vm -lXmu -lXi -lXt -lSM -lICE -lXext -lX11 -lm 
 #LDFLAGS =   -L/usr/X11R6/lib -L${EXPORTBASE}/lib -L/usr/lib -L/usr/local/lib
-LDFLAGS =  -L${EXPORTBASE}/lib  -L/usr/X11R6/lib -L/usr/lib 
+LDFLAGS =  -fPIC -L${EXPORTBASE}/lib  -L/usr/X11R6/lib -L/usr/lib 
 
 #INCFLAGS =  -I${EXPORTBASE}/include -I/usr/include -I/usr/local/include
 INCFLAGS =  -I${EXPORTBASE}/include 
