--- SDL_Perl-2.1.3/t/mixerpm.t.no-mixertest	2005-10-05 11:25:49.000000000 +0200
+++ SDL_Perl-2.1.3/t/mixerpm.t	2006-08-15 08:42:41.000000000 +0200
@@ -16,7 +16,7 @@
 use Test::More;
 
 if ( SDL::Config->has('SDL_mixer') ) {
-	plan ( tests => 3 );
+	plan ( tests => 2 );
 } else {
 	plan ( skip_all => 'SDL_mixer support not compiled' );
 }
@@ -59,9 +59,3 @@
 	playing
 	playing_music
 	/);
-
-# these are exported by default, so main:: should know them:
-SDL::Init(SDL_INIT_AUDIO);
-my $mixer = SDL::Mixer->new();
-isa_ok($mixer, 'SDL::Mixer');
-
