--- GNUmakefile.orig	2008-09-18 12:15:13.000000000 +0200
+++ GNUmakefile	2008-09-18 12:18:32.000000000 +0200
@@ -5,6 +5,7 @@
 ARFLAGS = -cr	# ar needs the dash on OpenBSD
 RANLIB = ranlib
 CP = cp
+INSTALL = install
 MKDIR = mkdir
 EGREP = egrep
 UNAME = $(shell uname)
@@ -102,22 +103,32 @@
 TESTIMPORTOBJS = $(TESTOBJS:.o=.import.o)
 DLLTESTOBJS = dlltest.dllonly.o
 
-all: cryptest.exe
+all: libcryptopp.a cryptest.exe
+
+shared: libcryptopp.so cryptest.exe
 
 clean:
-	$(RM) cryptest.exe libcryptopp.a $(LIBOBJS) $(TESTOBJS) cryptopp.dll libcryptopp.dll.a libcryptopp.import.a cryptest.import.exe dlltest.exe $(DLLOBJS) $(LIBIMPORTOBJS) $(TESTIMPORTOBJS) $(DLLTESTOBJS)
+	$(RM) cryptest.exe libcryptopp.a libcryptopp.so.* $(LIBOBJS) $(TESTOBJS) cryptopp.dll libcryptopp.dll.a libcryptopp.import.a cryptest.import.exe dlltest.exe $(DLLOBJS) $(LIBIMPORTOBJS) $(TESTIMPORTOBJS) $(DLLTESTOBJS)
 
 install:
-	$(MKDIR) -p $(PREFIX)/include/cryptopp $(PREFIX)/lib $(PREFIX)/bin
-	$(CP) *.h $(PREFIX)/include/cryptopp
-	$(CP) *.a $(PREFIX)/lib
-	$(CP) *.exe $(PREFIX)/bin
+	$(INSTALL) -Dp -m0755 cryptest.exe $(DESTDIR)$(PREFIX)/bin/cryptest
+	$(INSTALL) -d -m0755 $(DESTDIR)$(LIBDIR)
+	$(INSTALL) -Dp -m0755 libcryptopp.so* $(DESTDIR)$(LIBDIR)
+	$(INSTALL) -d -m0755 $(DESTDIR)$(PREFIX)/include/cryptopp/
+	$(INSTALL) -Dp -m0644 *.h $(DESTDIR)$(PREFIX)/include/cryptopp/
+	$(INSTALL) -d -m0755 $(DESTDIR)$(PREFIX)/share/cryptopp/
+	$(INSTALL) -Dp -m0644 *.dat $(DESTDIR)$(PREFIX)/share/cryptopp/
 
 libcryptopp.a: $(LIBOBJS)
 	$(AR) $(ARFLAGS) $@ $(LIBOBJS)
 	$(RANLIB) $@
 
-cryptest.exe: libcryptopp.a $(TESTOBJS)
+libcryptopp.so: $(LIBOBJS)
+	$(CXX) -shared  -Wl,-soname,$@.1 $(LDFLAGS) $(LDLIBS) -o $@.1.0 $(LIBOBJS)
+	ln -fs libcryptopp.so.1.0 libcryptopp.so.1
+	ln -fs libcryptopp.so.1.0 libcryptopp.so
+
+cryptest.exe: $(TESTOBJS)
 	$(CXX) -o $@ $(CXXFLAGS) $(TESTOBJS) -L. -lcryptopp $(LDFLAGS) $(LDLIBS)
 
 nolib: $(OBJS)		# makes it faster to test changes
