diff -urNp fdupes-1.40.orig/Makefile fdupes-1.40/Makefile
--- fdupes-1.40.orig/Makefile	2001-03-15 08:19:11.000000000 +0530
+++ fdupes-1.40/Makefile	2007-12-27 22:49:29.000000000 +0530
@@ -1,3 +1,5 @@
+DESTDIR =
+
 #
 # INSTALLDIR indicates directory where program is to be installed. 
 # Suggested values are "/usr/local/bin" or "/usr/bin".
@@ -38,8 +40,8 @@ fdupes: fdupes.c md5/md5.c	
 	gcc fdupes.c md5/md5.c -Wall -o fdupes -DVERSION=\"$(VERSION)\" $(EXTERNAL_MD5) $(EXPERIMENTAL_RBTREE)
 
 install: fdupes
-	cp fdupes $(INSTALLDIR)
-	cp fdupes.1 $(MANPAGEDIR)/man1
+	cp -p fdupes $(DESTDIR)$(INSTALLDIR)
+	cp -p fdupes.1 $(DESTDIR)$(MANPAGEDIR)/man1
 
 tarball: clean
 	tar --directory=.. -c -z -v -f ../fdupes-$(VERSION).tar.gz fdupes-$(VERSION)
