diff -Naupr lighttpd-1.3.10.orig/doc/rc.lighttpd.redhat lighttpd-1.3.10/doc/rc.lighttpd.redhat
--- lighttpd-1.3.10.orig/doc/rc.lighttpd.redhat	2004-04-06 13:13:38.000000000 +0200
+++ lighttpd-1.3.10/doc/rc.lighttpd.redhat	2005-02-17 19:06:12.429157784 +0100
@@ -1,13 +1,11 @@
-#! /bin/sh
-
-# lighttpd init file for redhat
+#!/bin/sh
 #
-# Tue Apr 9 2004 Matthijs van der Klip
+# lighttpd     Startup script for the lighttpd server
 #
-# chkconfig: 2345 99 01
-# description: Jan Kneschke's lighttpd
+# chkconfig: - 85 15
+# description: Lightning fast webserver with light system requirements
 #
-# processname: /usr/sbin/lighttpd
+# processname: lighttpd
 # config: /etc/lighttpd/lighttpd.conf
 # config: /etc/sysconfig/lighttpd
 # pidfile: /var/run/lighttpd.pid
@@ -15,39 +13,43 @@
 # Note: pidfile is assumed to be created
 # by lighttpd (config: server.pid-file).
 # If not, uncomment 'pidof' line.
-#
-# source function library
-. /etc/init.d/functions
 
-RETVAL=0
+# Source function library
+. /etc/rc.d/init.d/functions
+
+if [ -f /etc/sysconfig/lighttpd ]; then
+	. /etc/sysconfig/lighttpd
+fi
+
+if [ -z "$LIGHTTPD_CONF_PATH" ]; then
+	LIGHTTPD_CONF_PATH="/etc/lighttpd/lighttpd.conf"
+fi
+
 prog="lighttpd"
-pidfile="/var/run/lighttpd.pid"
-LIGHTTPD_BIN=/usr/sbin/lighttpd
-LIGHTTPD_CONFIG=/etc/sysconfig/lighttpd
-. $LIGHTTPD_CONFIG
+lighttpd="/usr/sbin/lighttpd"
+RETVAL=0
 
 start() {
-	echo -n "Starting $prog: "
-	daemon $LIGHTTPD_BIN -f $LIGHTTPD_CONF_PATH
+	echo -n $"Starting $prog: "
+	daemon $lighttpd -f $LIGHTTPD_CONF_PATH
 	RETVAL=$?
 	echo
-	touch /var/lock/subsys/$prog
-	#pidof $prog > $pidfile
+	[ $RETVAL -eq 0 ] && touch /var/lock/subsys/$prog
 	return $RETVAL
 }
 
 stop() {
-	echo -n "Stopping $prog: "
-	killproc $LIGHTTPD_BIN
+	echo -n $"Stopping $prog: "
+	killproc $lighttpd
 	RETVAL=$?
 	echo
-	rm -f /var/lock/subsys/$prog
+	[ $RETVAL -eq 0 ] && rm -f /var/lock/subsys/$prog
 	return $RETVAL
 }
 
 reload() {
-	echo -n "Reloading $prog: "
-	killproc $LIGHTTPD_BIN -HUP
+	echo -n $"Reloading $prog: "
+	killproc $lighttpd -HUP
 	RETVAL=$?
 	echo
 	return $RETVAL
@@ -64,15 +66,21 @@ case "$1" in
 		stop
 		start
 		;;
+	condrestart)
+		if [ -f /var/lock/subsys/$prog ]; then
+			stop
+			start
+		fi
+		;;
 	reload)
 		reload
 		;;
 	status)
-		status $prog
+		status $lighttpd
 		RETVAL=$?
 		;;
 	*)
-		echo $"Usage: $0 {start|stop|restart|reload|status}"
+		echo $"Usage: $0 {start|stop|restart|condrestart|reload|status}"
 		RETVAL=1
 esac
 
