--- toolkit/components/satchel/src/nsFormHistory.cpp.~1~	2004-03-17 17:52:51.000000000 -0800
+++ toolkit/components/satchel/src/nsFormHistory.cpp	2004-07-29 22:06:01.488644861 -0700
@@ -745,7 +745,7 @@
     for (i = 0; i < count; ++i)
       items[i] = i;
 
-    NS_QuickSort(items, count, sizeof(nsIMdbRow*),
+    NS_QuickSort(items, count, sizeof(PRUint32),
                  SortComparison, &matchingValues);
 
     for (i = 0; i < count; ++i) {
