Index: toolkit/xre/nsAppRunner.cpp
===================================================================
RCS file: /cvsroot/mozilla/toolkit/xre/nsAppRunner.cpp,v
retrieving revision 1.27
diff -u -p -r1.27 nsAppRunner.cpp
--- toolkit/xre/nsAppRunner.cpp	8 Apr 2004 14:01:15 -0000	1.27
+++ toolkit/xre/nsAppRunner.cpp	11 Apr 2004 01:44:49 -0000
@@ -1472,6 +1472,9 @@ static int HandleRemoteArguments(int arg
     }
   }
 
+  if (!remote)
+    return 0; // No remote argument == success
+
   // try to get the X remote client
   nsCOMPtr<nsIXRemoteClient> client (do_CreateInstance(NS_XREMOTECLIENT_CONTRACTID));
   if (!client)
Index: xpfe/bootstrap/nsAppRunner.cpp
===================================================================
RCS file: /cvsroot/mozilla/xpfe/bootstrap/nsAppRunner.cpp,v
retrieving revision 1.413
diff -u -p -r1.413 nsAppRunner.cpp
--- xpfe/bootstrap/nsAppRunner.cpp	8 Apr 2004 14:01:17 -0000	1.413
+++ xpfe/bootstrap/nsAppRunner.cpp	11 Apr 2004 01:44:50 -0000
@@ -1469,6 +1469,9 @@ static int HandleRemoteArguments(int arg
     }
   }
 
+  if (!remote)
+    return 0; // No remote argument == success
+
   // try to get the X remote client
   nsCOMPtr<nsIXRemoteClient> client (do_CreateInstance(NS_XREMOTECLIENT_CONTRACTID));
   if (!client)
