--- Makefile.orig	2004-01-02 09:24:00.000000000 +0100
+++ Makefile	2004-01-14 10:32:16.000000000 +0100
@@ -9,11 +9,17 @@
 PROGRAMS = tcrontab tcrontab-ap tcrond idle-halt
 VERSION = 0.4.5
 SYS_CRON_SPOOL = /var/spool/tcron
-TCRONTAB_AP_LIB = /usr/local/lib
+
+DESTDIR =
+PREFIX = /usr/local
+TCRONTAB_AP_LIB = $(PREFIX)/lib
+LIBVERSION = .0
+
+
 
 #CC = /usr/local/bin/gcc
 
-all:    $(PROGRAMS) libtcrontab-api.so libtcrontab-api.a
+all:    $(PROGRAMS) libtcrontab-api.so$(LIBVERSION) libtcrontab-api.a
 CFLAGS = $(OFLAGS) -I. -DVERSION=\"$(VERSION)\" \
 	-DSYS_CRON_SPOOL=\"$(SYS_CRON_SPOOL)\" \
 	-DTCRONTAB_AP_LIB=\"$(TCRONTAB_AP_LIB)\" \
@@ -24,7 +30,7 @@
 tcrontab-ap:    $(OBJSTCRONTAB_AP)
 	$(CC) $(OBJSTCRONTAB_AP) -o $@
 
-libtcrontab-api.so:     $(TCRONTAB_API_SRC)
+libtcrontab-api.so$(LIBVERSION):     $(TCRONTAB_API_SRC)
 	$(CC) $(CFLAGS) -fpic -shared $(TCRONTAB_API_SRC) -o $@
 
 libtcrontab-api.a: $(TCRONTAB_API_SRC)
@@ -41,20 +47,28 @@
 #	$(CC) -o $@ $(OBJSTAT)
 
 install: $(PROGRAMS)
-	install -m 555 -s tcrond /usr/sbin
-	install -m 555 -s idle-halt /usr/bin
-	install -m 4555 -s tcrontab /usr/bin
-	install -m 644 tcrontab.conf /etc/tcrontab
-	install -m 4755 tcrontab-ap $(TCRONTAB_AP_LIB)
-	install tcrond.init /etc/init.d/tcrond
-	install -m 755 libtcrontab-api.so $(TCRONTAB_AP_LIB)
-	install -m 644 libtcrontab-api.a $(TCRONTAB_AP_LIB)
-	install -m 644 tcron.h /usr/local/include
-	chkconfig --add tcrond
-	if [ ! -d $(SYS_CRON_SPOOL) ]; then  mkdir $(SYS_CRON_SPOOL) ; fi
-	grep $(TCRONTAB_AP_LIB) /etc/ld.so.conf || echo -e "\n$(TCRONTAB_AP_LIB)" >> /etc/ld.so.conf
-	ldconfig
-	service tcrond restart
+	[ -d $(DESTDIR)$(PREFIX)/sbin ] || mkdir -p $(DESTDIR)$(PREFIX)/sbin
+	install -m 555 -s tcrond $(DESTDIR)$(PREFIX)/sbin
+	[ -d $(DESTDIR)$(PREFIX)/bin ] || mkdir -p $(DESTDIR)$(PREFIX)/bin
+	install -m 555 -s idle-halt $(DESTDIR)$(PREFIX)/bin
+	install -m 4555 -s tcrontab $(DESTDIR)$(PREFIX)/bin
+	[ -d $(DESTDIR)/etc/tcrontab ] || mkdir -p $(DESTDIR)/etc/tcrontab
+	install -m 644 tcrontab.conf $(DESTDIR)/etc/tcrontab
+	[ -d $(DESTDIR)$(TCRONTAB_AP_LIB) ] || mkdir -p $(DESTDIR)$(TCRONTAB_AP_LIB)
+	install -m 4755 tcrontab-ap $(DESTDIR)$(TCRONTAB_AP_LIB)
+	[ -d $(DESTDIR)/etc/init.d ] || mkdir -p $(DESTDIR)/etc/init.d
+	install tcrond.init $(DESTDIR)/etc/init.d/tcrond
+	install -m 755 libtcrontab-api.so$(LIBVERSION) $(DESTDIR)$(TCRONTAB_AP_LIB)
+	( cd $(DESTDIR)$(TCRONTAB_AP_LIB) ; ln -s libtcrontab-api.so$(LIBVERSION) libtcrontb-api.so)
+
+	install -m 644 libtcrontab-api.a $(DESTDIR)$(TCRONTAB_AP_LIB)
+	[ -d $(DESTDIR)$(PREFIX)/include ] || mkdir -p $(DESTDIR)$(PREFIX)/include
+	install -m 644 tcron.h $(DESTDIR)$(PREFIX)/include
+	[ `id -u` -ne 0 ] || echo chkconfig --add tcrond
+	if [ ! -d $(DESTDIR)$(SYS_CRON_SPOOL) ]; then  mkdir -p $(DESTDIR)$(SYS_CRON_SPOOL); fi
+
+	[ `id -u` -ne 0 ] || echo ldconfig
+	
 SRCDIR = tcron-$(VERSION)
 
 clean:
