diff -ru rfb-0.6.1.orig/Makefile rfb-0.6.1/Makefile
--- rfb-0.6.1.orig/Makefile	Mon Feb 25 17:45:39 2002
+++ rfb-0.6.1/Makefile	Tue Feb 26 16:12:35 2002
@@ -49,7 +49,7 @@
 rfbcat: librfb.a
 	cd rfbcat       &&  $(MAKE) -e $(ENVIRONMENT) $@
 
-xvncconnect:
+xvncconnect: librfb.a
 	cd xvncconnect  &&  $(MAKE) -e $(ENVIRONMENT) $@
 
 clean:
diff -ru rfb-0.6.1.orig/lib/Makefile rfb-0.6.1/lib/Makefile
--- rfb-0.6.1.orig/lib/Makefile	Sun Nov 26 08:39:04 2000
+++ rfb-0.6.1/lib/Makefile	Tue Feb 26 16:14:07 2002
@@ -23,7 +23,7 @@
 ZLIB=-lz
 
 CXXFLAGS = -D$(USE_ZLIB) $(XCLASS_INCLUDE_PATH) \
-           -I../include -finline-functions -funroll-loops -O3 -Wall
+           -I../include -finline-functions -funroll-loops $(RPM_OPT_FLAGS)
 
 OBJECTSlibrfb = Connection.o \
                 Framebuffer.o \
diff -ru rfb-0.6.1.orig/rfbcat/Makefile rfb-0.6.1/rfbcat/Makefile
--- rfb-0.6.1.orig/rfbcat/Makefile	Wed Feb  6 03:24:43 2002
+++ rfb-0.6.1/rfbcat/Makefile	Tue Feb 26 16:14:40 2002
@@ -22,7 +22,7 @@
 ZLIB=-lz
 
 CXXFLAGS = -D$(USE_ZLIB)\
-           -I../include -finline-functions -funroll-loops -O3 -Wall
+           -I../include -finline-functions -funroll-loops $(RPM_OPT_FLAGS)
 
 PROGRAM = rfbcat
 OBJECTS = rfbcat.o FBStreamRecorder.o
diff -ru rfb-0.6.1.orig/x0rfbserver/Makefile rfb-0.6.1/x0rfbserver/Makefile
--- rfb-0.6.1.orig/x0rfbserver/Makefile	Sat Nov 25 20:04:14 2000
+++ rfb-0.6.1/x0rfbserver/Makefile	Tue Feb 26 16:15:08 2002
@@ -22,7 +22,7 @@
 ZLIB=-lz
 
 CXXFLAGS = -D$(USE_ZLIB) `xc-config --cflags` \
-           -I../include -finline-functions -funroll-loops -O3 -Wall
+           -I../include -finline-functions -funroll-loops $(RPM_OPT_FLAGS)
 
 OBJECTSx0rfbserver = OXServiceApplet.o \
                      XUpdateScanner.o \
diff -ru rfb-0.6.1.orig/xrfbviewer/Makefile rfb-0.6.1/xrfbviewer/Makefile
--- rfb-0.6.1.orig/xrfbviewer/Makefile	Thu Aug 31 10:46:54 2000
+++ rfb-0.6.1/xrfbviewer/Makefile	Tue Feb 26 16:16:27 2002
@@ -22,7 +22,7 @@
 ZLIB=-lz
 
 CXXFLAGS = -D$(USE_ZLIB) `xc-config --cflags` \
-           -I../include -finline-functions -funroll-loops -O3 -Wall
+           -I../include -finline-functions -funroll-loops $(RPM_OPT_FLAGS)
 
 OBJECTSxrfbviewer = ConnectionInfo.o \
                     FBStreamRecorder.o \
diff -ru rfb-0.6.1.orig/xvncconnect/Makefile rfb-0.6.1/xvncconnect/Makefile
--- rfb-0.6.1.orig/xvncconnect/Makefile	Sun Jan 14 22:23:42 2001
+++ rfb-0.6.1/xvncconnect/Makefile	Tue Feb 26 16:16:46 2002
@@ -23,7 +23,7 @@
 SRCS = xvncconnect.cc
 
 CXX = g++
-CXXFLAGS = `xc-config --cflags` -Wall
+CXXFLAGS = `xc-config --cflags` -Wall $(RPM_OPT_FLAGS)
 
 LIBS = `xc-config --libs`
 
