--- pdns-2.9.20/pdns/pdns.in.fixinit	2006-12-17 18:14:14.000000000 +0100
+++ pdns-2.9.20/pdns/pdns.in	2006-12-17 18:15:42.000000000 +0100
@@ -47,6 +47,7 @@
 		if test "$NOTRUNNING" = "0" 
 		then 
 			doPC quit
+			rm -f /var/lock/subsys/pdns
 			echo $ret
 		else
 			echo "not running"
@@ -57,6 +58,7 @@
 	force-stop)
 		echo -n "Stopping PowerDNS authoritative nameserver: "
 		killall -v -9 pdns_server
+		rm -f /var/lock/subsys/pdns
 		echo "killed"
 	;;
 
@@ -69,6 +71,7 @@
 			$pdns_server --daemon --guardian=yes
 			if test "$?" = "0"
 			then
+				touch /var/lock/subsys/pdns
 				echo "started"	
 			fi
 		fi 
