diff -up libofx-0.9.0/lib/ofx_container_transaction.cpp.c++-fu libofx-0.9.0/lib/ofx_container_transaction.cpp
--- libofx-0.9.0/lib/ofx_container_transaction.cpp.c++-fu	2007-11-09 18:10:49.000000000 -0500
+++ libofx-0.9.0/lib/ofx_container_transaction.cpp	2008-09-09 11:14:40.000000000 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "messages.hh"
 #include "libofx.h"
diff -up libofx-0.9.0/lib/ofx_request_accountinfo.cpp.c++-fu libofx-0.9.0/lib/ofx_request_accountinfo.cpp
--- libofx-0.9.0/lib/ofx_request_accountinfo.cpp.c++-fu	2007-10-27 22:33:43.000000000 -0400
+++ libofx-0.9.0/lib/ofx_request_accountinfo.cpp	2008-09-09 11:14:40.000000000 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "libofx.h"
 #include "ofx_request_accountinfo.hh"
diff -up libofx-0.9.0/lib/ofx_request.cpp.c++-fu libofx-0.9.0/lib/ofx_request.cpp
--- libofx-0.9.0/lib/ofx_request.cpp.c++-fu	2007-10-27 22:37:52.000000000 -0400
+++ libofx-0.9.0/lib/ofx_request.cpp	2008-09-09 11:14:40.000000000 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstring>
 #include <string>
 #include "messages.hh"
 #include "libofx.h"
diff -up libofx-0.9.0/lib/ofx_request_statement.cpp.c++-fu libofx-0.9.0/lib/ofx_request_statement.cpp
--- libofx-0.9.0/lib/ofx_request_statement.cpp.c++-fu	2007-10-28 08:57:18.000000000 -0400
+++ libofx-0.9.0/lib/ofx_request_statement.cpp	2008-09-09 11:14:40.000000000 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "libofx.h"
 #include "ofx_utilities.hh"
diff -up libofx-0.9.0/lib/ofx_utilities.cpp.c++-fu libofx-0.9.0/lib/ofx_utilities.cpp
--- libofx-0.9.0/lib/ofx_utilities.cpp.c++-fu	2008-09-09 11:15:19.000000000 -0400
+++ libofx-0.9.0/lib/ofx_utilities.cpp	2008-09-09 11:15:54.000000000 -0400
@@ -21,7 +21,8 @@
 
 #include "ParserEventGeneratorKit.h"
 #include "SGMLApplication.h"
-#include <time.h>
+#include <ctime>
+#include <cstdlib>
 #include <string>
 #include <locale.h>
 #include "messages.hh"
diff -up libofx-0.9.0/ofxdump/ofxdump.cpp.c++-fu libofx-0.9.0/ofxdump/ofxdump.cpp
--- libofx-0.9.0/ofxdump/ofxdump.cpp.c++-fu	2006-08-02 00:19:22.000000000 -0400
+++ libofx-0.9.0/ofxdump/ofxdump.cpp	2008-09-09 11:14:40.000000000 -0400
@@ -29,6 +29,8 @@
  ***************************************************************************/
 #include <iostream>
 #include <iomanip>
+#include <cstdlib>
+#include <cstring>
 #include <string>
 #include "libofx.h"
 #include <stdio.h>		/* for printf() */
