diff -ur fbreader-0.7.4-orig/fbreader/common/formats/rtf/RtfReader.h fbreader-0.7.4/fbreader/common/formats/rtf/RtfReader.h
--- fbreader-0.7.4-orig/fbreader/common/formats/rtf/RtfReader.h	2006-05-26 14:42:01.000000000 +0200
+++ fbreader-0.7.4/fbreader/common/formats/rtf/RtfReader.h	2006-05-26 17:36:45.000000000 +0200
@@ -38,7 +38,7 @@
 class RtfReader {
 
 private:
-  static void RtfReader::fillKeywordMap();
+  static void fillKeywordMap();
 
 private:
   static std::map<std::string, RtfCommand*> ourKeywordMap;
Only in fbreader-0.7.4/fbreader/common/formats/rtf: RtfReader.h~
