diff -Naupr vobcopy-1.0.0.orig/Makefile vobcopy-1.0.0/Makefile
--- vobcopy-1.0.0.orig/Makefile	2006-04-02 18:50:41.000000000 +0200
+++ vobcopy-1.0.0/Makefile	2006-04-18 11:51:33.000000000 +0200
@@ -2,16 +2,15 @@
 #This is the makefile for vobcopy, mainly written by rosenauer. These things 
 #below here are variable definitions. They get substituted in the (CC) and 
 #stuff places.
-CC     = gcc-3.4
+CC     = gcc
 #PREFIX += /usr/local
 #BINDIR = ${PREFIX}/bin
 #MANDIR = ${PREFIX}/man
-PREFIX += /usr/local
-BINDIR = ${PREFIX}/bin
-MANDIR = ${PREFIX}/man
+PREFIX ?= /usr/local
+BINDIR ?= ${PREFIX}/bin
+MANDIR ?= ${PREFIX}/share/man
 LFS    = -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CFLAGS += -I/usr//include
-LDFLAGS += -ldvdread -L/usr//lib
+LDFLAGS += -ldvdread
 
 #This specifies the conversion from .c to .o 
 .c.o:
