diff -Nurip audacious-1.3.1-orig/src/audacious/ui_skinselector.c audacious-1.3.1/src/audacious/ui_skinselector.c
--- audacious-1.3.1-orig/src/audacious/ui_skinselector.c	2007-01-13 18:19:52.000000000 +0100
+++ audacious-1.3.1/src/audacious/ui_skinselector.c	2007-03-14 20:58:59.000000000 +0100
@@ -249,11 +249,15 @@ void
 skinlist_update(void)
 {
     gchar *skinsdir;
+    gchar xmmsskinsdir[] = "/usr/share/xmms/Skins";
 
     skinlist_clear();
 
     scan_skindir(bmp_paths[BMP_PATH_USER_SKIN_DIR]);
     scan_skindir(DATA_DIR G_DIR_SEPARATOR_S "Skins");
+    if (g_file_test(xmmsskinsdir, G_FILE_TEST_IS_DIR)) {
+        scan_skindir( xmmsskinsdir );
+    }
 
     skinsdir = getenv("SKINSDIR");
     if (skinsdir) {
