diff -Naupr audacious-0.2.2.orig/libaudacious/Makefile.in audacious-0.2.2/libaudacious/Makefile.in
--- audacious-0.2.2.orig/libaudacious/Makefile.in	2006-02-27 23:34:47.000000000 +0100
+++ audacious-0.2.2/libaudacious/Makefile.in	2006-03-13 13:04:04.000000000 +0100
@@ -49,7 +49,7 @@ install-posthook:
 	@echo [adding versioning information to libaudacious]
 	@mv -f ${DESTDIR}/${LIBDIR}/libaudacious.so \
 		${DESTDIR}/${LIBDIR}/libaudacious.so.2.0.0
-	@ln -sf ${DESTDIR}/${LIBDIR}/libaudacious.so.2.0.0 \
+	@ln -sf libaudacious.so.2.0.0 \
 		${DESTDIR}/${LIBDIR}/libaudacious.so.2
-	@ln -sf ${DESTDIR}/${LIBDIR}/libaudacious.so.2.0.0 \
+	@ln -sf libaudacious.so.2.0.0 \
 		${DESTDIR}/${LIBDIR}/libaudacious.so
