diff -Naupr audacious-1.1.0.orig/audacious/Makefile audacious-1.1.0/audacious/Makefile
--- audacious-1.1.0.orig/audacious/Makefile	2006-06-20 08:39:11.000000000 +0200
+++ audacious-1.1.0/audacious/Makefile	2006-07-19 12:47:59.000000000 +0200
@@ -102,5 +102,5 @@ build_stamp.c:
 	fi;
 
 audacious: $(OBJECTS)
-	$(CC) $(LDFLAGS) $(OBJECTS) $(LDADD) -o $@ -Wl,-rpath,${libdir}
+	$(CC) $(LDFLAGS) $(OBJECTS) $(LDADD) -o $@
 	@printf "%10s     %-20s\n" LINK $@
diff -Naupr audacious-1.1.0.orig/audtool/Makefile audacious-1.1.0/audtool/Makefile
--- audacious-1.1.0.orig/audtool/Makefile	2006-06-14 09:27:49.000000000 +0200
+++ audacious-1.1.0/audtool/Makefile	2006-07-19 12:48:07.000000000 +0200
@@ -25,5 +25,5 @@ SOURCES = audtool.c
 OBJECTS = ${SOURCES:.c=.o}
 
 audtool: $(OBJECTS)
-	$(CC) $(LDFLAGS) $(OBJECTS) $(LDADD) -o $@ -Wl,-rpath,${libdir}
+	$(CC) $(LDFLAGS) $(OBJECTS) $(LDADD) -o $@
 	@printf "%10s     %-20s\n" LINK $@
