--- gambas-0.94/Makefile.am.orig	2004-06-25 16:42:41.504044056 +0200
+++ gambas-0.94/Makefile.am	2004-06-25 16:43:56.250680840 +0200
@@ -4,33 +4,33 @@
 
 install-exec-local:
 	@echo
-	@if test "$(bindir)" != "/usr/bin" && test "$(bindir)" != "/usr/bin/"; then \
-	 echo "Making symbolic links in /usr/bin..."; \
-	 rm -f /usr/bin/gbx; \
-	 ln -s $(bindir)/gbx /usr/bin/gbx; \
-	 rm -f /usr/bin/gbc; \
-	 ln -s $(bindir)/gbc /usr/bin/gbc; \
-	 rm -f /usr/bin/gba; \
-	 ln -s $(bindir)/gba /usr/bin/gba; \
-	 rm -f /usr/bin/gbi; \
-	 ln -s $(bindir)/gbi /usr/bin/gbi; \
-	fi
+#	@if test "$(bindir)" != "/usr/bin" && test "$(bindir)" != "/usr/bin/"; then \
+#	 echo "Making symbolic links in /usr/bin..."; \
+#	 rm -f /usr/bin/gbx; \
+#	 ln -s $(bindir)/gbx /usr/bin/gbx; \
+#	 rm -f /usr/bin/gbc; \
+#	 ln -s $(bindir)/gbc /usr/bin/gbc; \
+#	 rm -f /usr/bin/gba; \
+#	 ln -s $(bindir)/gba /usr/bin/gba; \
+#	 rm -f /usr/bin/gbi; \
+#	 ln -s $(bindir)/gbi /usr/bin/gbi; \
+#	fi
 
 	@echo
 	@echo "Creating the library info files..."
-	@$(INSTALL) -d $(pkgdatadir)/info
-	@$(bindir)/gbi -a
-	@rm -f $(pkglibdir)/lib.gb.la
-	@rm -f $(pkglibdir)/lib.gb.so*
+#	@$(INSTALL) -d $(pkgdatadir)/info
+#	@$(bindir)/gbi -a
+#	@rm -f $(pkglibdir)/lib.gb.la
+#	@rm -f $(pkglibdir)/lib.gb.so*
 
 	@echo
 	@echo "Installing the development environment..."
 	@(cd $(srcdir)/app; d=`pwd`; for p in *; do echo "Compiling $$p..."; cd $$d/$$p; \
 	  gbc -ag; gba; rm -rf .gambas; $(INSTALL) $$p $(bindir); done)
-	@if test "$(bindir)" != "/usr/bin" && test "$(bindir)" != "/usr/bin/"; then \
-	 rm -f /usr/bin/gambas; \
-	 ln -s $(bindir)/gambas /usr/bin/gambas; \
-        fi
+#	@if test "$(bindir)" != "/usr/bin" && test "$(bindir)" != "/usr/bin/"; then \
+#	 rm -f /usr/bin/gambas; \
+#	 ln -s $(bindir)/gambas /usr/bin/gambas; \
+#        fi
 
 	@echo
 	@echo "Installing the gambas help files..."
