diff -Naupr mjpegtools-1.6.3-rc2.orig/mplex/Makefile.am mjpegtools-1.6.3-rc2/mplex/Makefile.am
--- mjpegtools-1.6.3-rc2.orig/mplex/Makefile.am	2004-08-06 17:34:36.000000000 +0200
+++ mjpegtools-1.6.3-rc2/mplex/Makefile.am	2005-08-15 19:25:14.000000000 +0200
@@ -65,7 +65,8 @@ libmplex2_la_LDFLAGS =  \
 	-release $(LT_RELEASE)
 
 libmplex2_la_LIBADD = \
-	$(top_builddir)/utils/libmjpegutils.la
+	$(top_builddir)/utils/libmjpegutils.la \
+	$(top_builddir)/mpeg2enc/libmpeg2encpp.la
 
 libmplex2_la_CXXFLAGS = $(ZALPHA_FLAGS)
 
diff -Naupr mjpegtools-1.6.3-rc2.orig/mplex/Makefile.in mjpegtools-1.6.3-rc2/mplex/Makefile.in
--- mjpegtools-1.6.3-rc2.orig/mplex/Makefile.in	2005-08-12 18:20:27.000000000 +0200
+++ mjpegtools-1.6.3-rc2/mplex/Makefile.in	2005-08-15 19:26:05.000000000 +0200
@@ -327,7 +327,8 @@ libmplex2_la_LDFLAGS = \
 	-release $(LT_RELEASE)
 
 libmplex2_la_LIBADD = \
-	$(top_builddir)/utils/libmjpegutils.la
+	$(top_builddir)/utils/libmjpegutils.la \
+	$(top_builddir)/mpeg2enc/libmpeg2encpp.la
 
 libmplex2_la_CXXFLAGS = $(ZALPHA_FLAGS)
 mplex_SOURCES = main.cpp 
