diff -udBbr mjpegtools-1.6.2/configure mjpegtools-1.6.2-r1-fix1/work/mjpegtools-1.6.2/configure
+++ mjpegtools-1.6.2/configure	2004-02-04 01:58:37.000000000 +0300
--- mjpegtools-1.6.2/configure	2004-04-24 11:44:34.961978760 +0400
@@ -25858,16 +25858,16 @@
   fi
   have_cmov=false
   case $arch in
-  i786) ARCHFLAGS="-mcpu=i786 -march=$arch"
+  i786) ARCHFLAGS="-march=$arch"
   have_cmov=true
   ;;
-  i686) ARCHFLAGS="-mcpu=i686 -march=$arch"
+  i686) ARCHFLAGS="-march=$arch"
   have_cmov=true
   ;;
-  i586) ARCHFLAGS="-mcpu=i586 -march=$arch"
+  i586) ARCHFLAGS="-march=$arch"
   have_cmov=false
   ;;
-  *) ARCHFLAGS="-mcpu=i686 -march=$arch"
+  *) ARCHFLAGS="-march=$arch"
   have_cmov=true
   ;;
   esac
diff -udBbr mjpegtools-1.6.2/lavtools/lavpipe.c mjpegtools-1.6.2/lavtools/lavpipe.c
--- mjpegtools-1.6.2/lavtools/lavpipe.c	2002-09-23 23:46:43.000000000 +0400
+++ mjpegtools-1.6.2/lavtools/lavpipe.c	2004-04-24 11:59:15.279150208 +0400
@@ -582,6 +582,7 @@
       mjpeg_info( "closing input %d (source %d)", i, current_index);
       decommission_pipe_source(source);
     KEEP_SOURCE:
+      ;
     }
   }
 }
diff -udBbr mjpegtools-1.6.2/mpeg2enc/picture.cc mjpegtools-1.6.2/mpeg2enc/picture.cc
--- mjpegtools-1.6.2/mpeg2enc/picture.cc	2003-10-25 21:36:32.000000000 +0400
+++ mjpegtools-1.6.2/mpeg2enc/picture.cc	2004-04-24 12:09:37.018631440 +0400
@@ -83,9 +83,9 @@
     }
 
 
-	curref = new (uint8_t *)[5];
-	curorg = new (uint8_t *)[5];
-	pred   = new (uint8_t *)[5];
+	curref = new uint8_t *[5];
+	curorg = new uint8_t *[5];
+	pred   = new uint8_t *[5];
 
 	for( i = 0 ; i<3; i++)
 	{