--- src/main.c.orig	2004-01-02 17:37:15.000000000 -0500
+++ src/main.c	2004-01-02 17:38:59.000000000 -0500
@@ -143,18 +143,18 @@
 
 void usage(const char *prog)
 {
-        fprintf(stderr, "usage: %s [options] <addr/mask | addr/offset | addr>
-        -C <class>      Network class (a, b, or c).  Must be used with -B
-	-B <bits>	Subnet bits (must be used with -C)
-        -i <if>		Reverse engineer an interface (e.g. eth0)
-	-a		Print all information available
-	-g		Print general information
-        -s 		Print all possible subnets
-	-h		Print host information
-	-c		Print CIDR information
-        -v		Print the program version
-        -?		Print this help message
-
-        \rReport bugs to %s\n",
+        fprintf(stderr, "usage: %s [options] <addr/mask | addr/offset | addr>"
+"	-C <class>      Network class (a, b, or c).  Must be used with -B"
+"	-B <bits>	Subnet bits (must be used with -C)"
+"	-i <if>		Reverse engineer an interface (e.g. eth0)"
+"	-a		Print all information available"
+"	-g		Print general information"
+"	-s 		Print all possible subnets"
+"	-h		Print host information"
+"	-c		Print CIDR information"
+"	-v		Print the program version"
+"	-?		Print this help message"
+""
+"	\rReport bugs to %s\n",
                         prog, MAINTAINER);
 }
