--- apt-0.5.15cnc5/apt-pkg/rpm/rpmversion.cc.epoch	2004-01-06 23:02:28.000000000 +0200
+++ apt-0.5.15cnc5/apt-pkg/rpm/rpmversion.cc	2004-01-06 23:02:48.000000000 +0200
@@ -201,8 +201,10 @@
 #if RPM_VERSION >= 0x040100
    rpmds pds = rpmdsSingle(RPMTAG_PROVIDENAME, "", PkgVer, PkgFlags);
    rpmds dds = rpmdsSingle(RPMTAG_REQUIRENAME, "", DepVer, DepFlags);
+#if RPM_VERSION >= 0x040201
    rpmdsSetNoPromote(pds, _rpmds_nopromote);
    rpmdsSetNoPromote(dds, _rpmds_nopromote);
+#endif
    rc = rpmdsCompare(pds, dds);
    rpmdsFree(pds);
    rpmdsFree(dds);
