--- qemu-0.8.0/ppc.ld~	2005-12-19 22:51:53.000000000 +0000
+++ qemu-0.8.0/ppc.ld	2006-03-21 16:46:58.000000000 +0000
@@ -90,7 +90,11 @@ SECTIONS
   /* We want the small data sections together, so single-instruction offsets
      can access them all, and initialized data all before uninitialized, so
      we can shorten the on-disk segment size.  */
-  .sdata     : { *(.sdata) }
+  .sdata     :
+  {
+    PROVIDE (_SDA_BASE_ = 32768);
+    *(.sdata .sdata.* .gnu.linkonce.s.*)
+  }
   _edata  =  .;
   PROVIDE (edata = .);
   __bss_start = .;
