diff -NurpP --minimal qemu-0.5.2/Makefile qemu-0.5.2-mdk/Makefile
--- qemu-0.5.2/Makefile	2004-01-25 16:50:02.000000000 +0100
+++ qemu-0.5.2-mdk/Makefile	2004-03-10 08:57:03.000000000 +0100
@@ -36,12 +36,12 @@ distclean: clean
         done
 
 install: all 
-	mkdir -p $(prefix)/bin
-	install -m 755 -s $(TOOLS) $(prefix)/bin
-	mkdir -p $(sharedir)
-	install -m 644 pc-bios/bios.bin pc-bios/vgabios.bin $(sharedir)
-	mkdir -p $(mandir)/man1
-	install qemu.1 $(mandir)/man1
+	mkdir -p $(DESTDIR)$(prefix)/bin
+	install -m 755 -s $(TOOLS) $(DESTDIR)$(prefix)/bin
+	mkdir -p $(DESTDIR)$(sharedir)
+	install -m 644 pc-bios/bios.bin pc-bios/vgabios.bin $(DESTDIR)$(sharedir)
+	mkdir -p $(DESTDIR)$(mandir)/man1
+	install qemu.1 $(DESTDIR)$(mandir)/man1
 	for d in $(TARGET_DIRS); do \
 	make -C $$d $@ || exit 1 ; \
         done
diff -NurpP --minimal qemu-0.5.2/Makefile.orig qemu-0.5.2-mdk/Makefile.orig
diff -NurpP --minimal qemu-0.5.2/Makefile.target qemu-0.5.2-mdk/Makefile.target
--- qemu-0.5.2/Makefile.target	2004-01-25 16:50:02.000000000 +0100
+++ qemu-0.5.2-mdk/Makefile.target	2004-03-10 08:59:21.000000000 +0100
@@ -275,7 +275,7 @@ clean:
 	rm -f *.o  *.a *~ $(PROGS) gen-op.h opc.h op.h
 
 install: all 
-	install -m 755 -s $(PROGS) $(prefix)/bin
+	install -m 755 -s $(PROGS) $(DESTDIR)$(prefix)/bin
 
 ifneq ($(wildcard .depend),)
 include .depend
diff -NurpP --minimal qemu-0.5.2/Makefile.target.orig qemu-0.5.2-mdk/Makefile.target.orig
