--- linux/MeterMaker.cc	Fri Nov 12 00:13:22 1999
+++ linux/MeterMaker.cc.ppc	Thu Jan 25 07:48:31 2001
@@ -56,7 +56,7 @@
       push(new DiskMeter(_xos, atof(_xos->getResource("diskBandwidth"))));
 
   // check for the serial meters.
-#ifndef __mc68000__
+#if !defined(__mc68000__) && !defined(__powerpc__)
   for (int i = 0 ; i < SerialMeter::numDevices() ; i++)
     if (_xos->isResourceTrue(SerialMeter::getResourceName(
       (SerialMeter::Device)i)))

