diff -u a/libmetrics/linux/metrics.c-orig b/libmetrics/linux/metrics.c
--- a/libmetrics/linux/metrics.c	2011-06-13 16:39:19.279753448 +0100
+++ b/libmetrics/linux/metrics.c	2011-06-13 16:46:36.936156216 +0100
@@ -1242,7 +1242,7 @@
       return max;
    }
    while ( fgets(procline, sizeof(procline), mounts) ) {
-      rc=sscanf(procline, "%s %s %s %s ", device, mount, type, mode);
+      rc=sscanf(procline, "%127s %127s %31s %127s ", device, mount, type, mode);
       if (!rc) continue;
       if (!strncmp(mode, "ro", 2)) continue;
       if (remote_mount(device, type)) continue;
