diff -uNr paragui-1.0.4-orig/include/pglog.h paragui-1.0.4/include/pglog.h
--- paragui-1.0.4-orig/include/pglog.h	2003-03-30 18:30:57.000000000 +0200
+++ paragui-1.0.4/include/pglog.h	2004-12-20 23:19:48.968364440 +0100
@@ -84,7 +84,7 @@
 	DECLSPEC void SetConsoleKey(SDLKey key);
 	DECLSPEC SDLKey GetConsoleKey();
 	DECLSPEC void SetTitle(const char* title, int alignment = PG_TA_CENTER);	
-};
+}
 
 #endif
 
