diff -Naupr scummvm-0.9.0.orig/Makefile scummvm-0.9.0/Makefile
--- scummvm-0.9.0.orig/Makefile	2006-06-21 23:19:07.000000000 +0200
+++ scummvm-0.9.0/Makefile	2006-10-29 18:56:40.000000000 +0100
@@ -45,7 +45,7 @@ config.mak: $(srcdir)/configure
 
 install: all
 	$(INSTALL) -d "$(DESTDIR)$(BINDIR)"
-	$(INSTALL) -c -s -m 755 "$(srcdir)/scummvm$(EXEEXT)" "$(DESTDIR)$(BINDIR)/scummvm$(EXEEXT)"
+	$(INSTALL) -c -m 755 "$(srcdir)/scummvm$(EXEEXT)" "$(DESTDIR)$(BINDIR)/scummvm$(EXEEXT)"
 	$(INSTALL) -d "$(DESTDIR)$(MANDIR)/man6/"
 	$(INSTALL) -c -m 644 "$(srcdir)/dists/scummvm.6" "$(DESTDIR)$(MANDIR)/man6/scummvm.6"
 	$(INSTALL) -d "$(DESTDIR)$(PREFIX)/share/pixmaps/"
