Index: configure
===================================================================
--- configure	(revision 23265)
+++ configure	(working copy)
@@ -292,6 +292,7 @@
   --prefix=DIR           use this prefix for installing ScummVM [/usr/local]
   --bindir=DIR           directory to install the scummvm binary in [PREFIX/bin]
   --mandir=DIR           directory to install the manpage in [PREFIX/man]
+  --datadir=DIR          directory to install the data files in [PREFIX/share]
 
 Special configuration feature:
   --host=HOST            cross-compile to target HOST (arm-linux, ...)
@@ -483,6 +484,9 @@
       --mandir=*)
         _mandir=`echo $ac_option | cut -d '=' -f 2`
         ;;
+      --datadir=*)
+        _datadir=`echo $ac_option | cut -d '=' -f 2`
+        ;;
       *)
         echo "error: unrecognised option: $ac_option
 Try \`$0 --help' for more information." >&2
@@ -1242,7 +1246,11 @@
 #
 test -z "$_bindir" && _bindir="$_prefix/bin"
 test -z "$_mandir" && _mandir="$_prefix/man"
+test -z "$_datadir" && _datadir="$_prefix/share"
 
+CXXFLAGS="$CXXFLAGS -DDATA_PATH=\\\"$_datadir/scummvm\\\""
+
+
 #
 # Check which engines ("frontends") are to be built
 #
Index: gui/ThemeNew.cpp
===================================================================
--- gui/ThemeNew.cpp	(revision 23265)
+++ gui/ThemeNew.cpp	(working copy)
@@ -73,6 +73,11 @@
 		clearAll();
 	}
 
+#ifdef DATA_PATH
+	Common::File::addDefaultDirectoryRecursive(DATA_PATH);
+#endif
+
+
 	if (ConfMan.hasKey("extrapath"))
 		Common::File::addDefaultDirectoryRecursive(ConfMan.get("extrapath"));
 
