--- imap/src/osdep/unix/Makefile.flock	Wed Aug 23 05:51:40 2000
+++ imap/src/osdep/unix/Makefile	Wed Aug 23 05:51:44 2000
@@ -103,7 +103,7 @@
 BINARIES=mail.o misc.o newsrc.o smanager.o osdep.o utf8.o siglocal.o \
  dummy.o pseudo.o netmsg.o flstring.o fdstring.o \
  rfc822.o nntp.o smtp.o imap4r1.o pop3.o \
- unix.o mbox.o mbx.o mmdf.o tenex.o mtx.o news.o phile.o mh.o mx.o
-CFLAGS=-g
+ unix.o mbox.o mbx.o mmdf.o tenex.o mtx.o news.o phile.o mh.o mx.o flock.o
+CFLAGS=$(BASECFLAGS) $(EXTRACFLAGS)
 
 CAT=cat
 MAKE=make
 MV=mv
