diff -urN pine4.31.orig/pine/osdep/lstcmpnt pine4.31/pine/osdep/lstcmpnt
--- pine4.31.orig/pine/osdep/lstcmpnt	Mon Oct 30 17:45:08 2000
+++ pine4.31/pine/osdep/lstcmpnt	Tue Dec 12 06:33:53 2000
@@ -9,10 +9,10 @@
 last_cmpnt(filename)
     char *filename;
 {
-    register char *p = NULL, *q = filename;
+    char *p = NULL, *q = filename;
 
-    if(!q)
-      return(q);
+    if(filename == 0)
+      return 0;
 
     while(q = strchr(q, '/'))
       if(*++q)
