--- pine/osdep/os-lnx.h.sendmail-options	Thu Aug 16 15:45:57 2001
+++ pine/osdep/os-lnx.h	Thu Aug 16 15:59:40 2001
@@ -212,7 +212,7 @@
 
 /*----- The usual sendmail configuration for sending mail on Unix ------*/
 #define SENDMAIL	"/usr/sbin/sendmail"
-#define SENDMAILFLAGS	"-bs -odb -oem"	/* send via smtp with backgroud
+#define SENDMAILFLAGS	"-t -oi -odb -oem"	/* send via smtp with backgroud
 					   delivery and mail back errors */
 
 
