--- pine4.30/build.ldap	Fri Oct 27 12:48:05 2000
+++ pine4.30/build	Fri Oct 27 12:49:00 2000
@@ -249,7 +249,7 @@
 	  case "$?" in
 	    1) if [ "$LLIBS" != "1" ]
 	       then
-		 L1="'LDAPLIBS=../ldap/libraries/libldap.a ../ldap/libraries/liblber.a'"
+		 L1="'LDAPLIBS=../ldap/libraries/libldap.so ../ldap/libraries/liblber.so ../ldap/libraries/libresolv.so'"
 	       fi
 	       if [ "$LFLAGS" != "1" ]
 	       then
