diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/APEDecompress.cpp mac-3.99-u4-b4/src/MACLib/APEDecompress.cpp
--- mac-3.99-u4-b4.orig/src/MACLib/APEDecompress.cpp	2005-07-03 16:31:35.000000000 +0200
+++ mac-3.99-u4-b4/src/MACLib/APEDecompress.cpp	2006-01-19 15:55:38.000000000 +0100
@@ -23,7 +23,7 @@ CAPEDecompress::CAPEDecompress(int * pEr
     }
 
     // get format information
-    GetInfo(APE_INFO_WAVEFORMATEX, (int) &m_wfeInput);
+    GetInfo(APE_INFO_WAVEFORMATEX, (long) &m_wfeInput);
     m_nBlockAlign = GetInfo(APE_INFO_BLOCK_ALIGN);
 
     // initialize other stuff
@@ -367,7 +367,7 @@ int CAPEDecompress::SeekToFrame(int nFra
 /*****************************************************************************************
 Get information from the decompressor
 *****************************************************************************************/
-int CAPEDecompress::GetInfo(APE_DECOMPRESS_FIELDS Field, int nParam1, int nParam2)
+int CAPEDecompress::GetInfo(APE_DECOMPRESS_FIELDS Field, long nParam1, long nParam2)
 {
     int nRetVal = 0;
     BOOL bHandled = TRUE;
@@ -452,7 +452,7 @@ int CAPEDecompress::GetInfo(APE_DECOMPRE
             }
             else
             {
-                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeFormat, 0);
+                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeFormat, 0);
                 WAVE_HEADER WAVHeader; FillWaveHeader(&WAVHeader, 
                     (m_nFinishBlock - m_nStartBlock) * GetInfo(APE_INFO_BLOCK_ALIGN), 
                     &wfeFormat,    0);
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/APEDecompress.h mac-3.99-u4-b4/src/MACLib/APEDecompress.h
--- mac-3.99-u4-b4.orig/src/MACLib/APEDecompress.h	2005-03-10 13:36:25.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/APEDecompress.h	2006-01-19 15:55:18.000000000 +0100
@@ -22,7 +22,7 @@ public:
     int GetData(char * pBuffer, int nBlocks, int * pBlocksRetrieved);
     int Seek(int nBlockOffset);
 
-    int GetInfo(APE_DECOMPRESS_FIELDS Field, int nParam1 = 0, int nParam2 = 0);
+    int GetInfo(APE_DECOMPRESS_FIELDS Field, long nParam1 = 0, long nParam2 = 0);
 
 protected:
 
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/APEInfo.cpp mac-3.99-u4-b4/src/MACLib/APEInfo.cpp
--- mac-3.99-u4-b4.orig/src/MACLib/APEInfo.cpp	2005-05-30 14:53:50.000000000 +0200
+++ mac-3.99-u4-b4/src/MACLib/APEInfo.cpp	2006-01-19 15:57:38.000000000 +0100
@@ -129,7 +129,7 @@ int CAPEInfo::GetFileInformation(BOOL bG
 /*****************************************************************************************
 Primary query function
 *****************************************************************************************/
-int CAPEInfo::GetInfo(APE_DECOMPRESS_FIELDS Field, int nParam1, int nParam2)
+int CAPEInfo::GetInfo(APE_DECOMPRESS_FIELDS Field, long nParam1, long nParam2)
 {
     int nRetVal = -1;
 
@@ -254,7 +254,7 @@ int CAPEInfo::GetInfo(APE_DECOMPRESS_FIE
             }
             else
             {
-                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeFormat, 0);
+                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeFormat, 0);
                 WAVE_HEADER WAVHeader; FillWaveHeader(&WAVHeader, m_APEFileInfo.nWAVDataBytes, &wfeFormat,
                     m_APEFileInfo.nWAVTerminatingBytes);
                 memcpy(pBuffer, &WAVHeader, sizeof(WAVE_HEADER));
@@ -311,7 +311,7 @@ int CAPEInfo::GetInfo(APE_DECOMPRESS_FIE
         break;
     }
     case APE_INFO_IO_SOURCE:
-        nRetVal = (int) m_spIO.GetPtr();
+        nRetVal = (long) m_spIO.GetPtr();
         break;
     case APE_INFO_FRAME_BYTES:
     {
@@ -350,10 +350,10 @@ int CAPEInfo::GetInfo(APE_DECOMPRESS_FIE
         break;
     }
     case APE_INFO_TAG:
-        nRetVal = (int) m_spAPETag.GetPtr();
+        nRetVal = (long) m_spAPETag.GetPtr();
         break;
     case APE_INTERNAL_INFO:
-        nRetVal = (int) &m_APEFileInfo;
+        nRetVal = (long) &m_APEFileInfo;
         break;
     default:
 	break;
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/APEInfo.h mac-3.99-u4-b4/src/MACLib/APEInfo.h
--- mac-3.99-u4-b4.orig/src/MACLib/APEInfo.h	2005-03-10 13:36:25.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/APEInfo.h	2006-01-19 15:57:57.000000000 +0100
@@ -82,7 +82,7 @@ public:
     virtual ~CAPEInfo();
 
     // query for information
-    int GetInfo(APE_DECOMPRESS_FIELDS Field, int nParam1 = 0, int nParam2 = 0);
+    int GetInfo(APE_DECOMPRESS_FIELDS Field, long nParam1 = 0, long nParam2 = 0);
     
 private:
 
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/APESimple.cpp mac-3.99-u4-b4/src/MACLib/APESimple.cpp
--- mac-3.99-u4-b4.orig/src/MACLib/APESimple.cpp	2005-03-10 13:36:25.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/APESimple.cpp	2006-01-19 15:59:07.000000000 +0100
@@ -291,14 +291,14 @@ int DecompressCore(const str_utf16 * pIn
         if (spAPEDecompress == NULL || nFunctionRetVal != ERROR_SUCCESS) throw(nFunctionRetVal);
 
         // get the input format
-        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeInput))
+        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeInput))
 
         // allocate space for the header
         spTempBuffer.Assign(new unsigned char [spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_BYTES)], TRUE);
         if (spTempBuffer == NULL) throw(ERROR_INSUFFICIENT_MEMORY);
 
         // get the header
-        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_DATA, (int) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_BYTES)));
+        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_DATA, (long) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_BYTES)));
 
         // initialize the output
         if (nOutputMode == UNMAC_DECODER_OUTPUT_WAV)
@@ -370,7 +370,7 @@ int DecompressCore(const str_utf16 * pIn
             {
                 spTempBuffer.Assign(new unsigned char[spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES)], TRUE);
                 if (spTempBuffer == NULL) throw(ERROR_INSUFFICIENT_MEMORY);
-                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (int) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES)))
+                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (long) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES)))
         
                 unsigned int nBytesToWrite = spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES);
                 unsigned int nBytesWritten = 0;
@@ -392,7 +392,7 @@ int DecompressCore(const str_utf16 * pIn
                 spTempBuffer.Assign(new unsigned char[nTerminatingBytes], TRUE);
                 if (spTempBuffer == NULL) throw(ERROR_INSUFFICIENT_MEMORY);
                 
-                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (int) spTempBuffer.GetPtr(), nTerminatingBytes))
+                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (long) spTempBuffer.GetPtr(), nTerminatingBytes))
 
                 if (bHasTag)
                 {
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/MACLib.h mac-3.99-u4-b4/src/MACLib/MACLib.h
--- mac-3.99-u4-b4.orig/src/MACLib/MACLib.h	2005-05-31 10:02:27.000000000 +0200
+++ mac-3.99-u4-b4/src/MACLib/MACLib.h	2006-01-19 15:59:40.000000000 +0100
@@ -259,7 +259,7 @@ public:
     //    int nParam2
     //        generic parameter... usage is listed in APE_DECOMPRESS_FIELDS
     //////////////////////////////////////////////////////////////////////////////////////////////
-    virtual int GetInfo(APE_DECOMPRESS_FIELDS Field, int nParam1 = 0, int nParam2 = 0) = 0;
+    virtual int GetInfo(APE_DECOMPRESS_FIELDS Field, long nParam1 = 0, long nParam2 = 0) = 0;
 };
 
 /*************************************************************************************************
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/MD5.cpp mac-3.99-u4-b4/src/MACLib/MD5.cpp
--- mac-3.99-u4-b4.orig/src/MACLib/MD5.cpp	2005-04-28 09:04:59.000000000 +0200
+++ mac-3.99-u4-b4/src/MACLib/MD5.cpp	2006-01-19 18:43:45.000000000 +0100
@@ -100,7 +100,7 @@ __MD5Transform ( uint32_t        state [
     CopyToLittleEndian (tempBuffer, in, 16);
     x = tempBuffer;
 #else
-    if ( (unsigned int)in & 3 ) {
+    if ( (unsigned long)in & 3 ) {
         memcpy ( tempBuffer, in, 64 );
         x = tempBuffer;
     } 
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/Old/APEDecompressOld.cpp mac-3.99-u4-b4/src/MACLib/Old/APEDecompressOld.cpp
--- mac-3.99-u4-b4.orig/src/MACLib/Old/APEDecompressOld.cpp	2005-03-11 13:23:37.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/Old/APEDecompressOld.cpp	2006-01-19 18:28:57.000000000 +0100
@@ -165,7 +165,7 @@ int CAPEDecompressOld::Seek(int nBlockOf
     return ERROR_SUCCESS;
 }
 
-int CAPEDecompressOld::GetInfo(APE_DECOMPRESS_FIELDS Field, int nParam1, int nParam2)
+int CAPEDecompressOld::GetInfo(APE_DECOMPRESS_FIELDS Field, long nParam1, long nParam2)
 {
     int nRetVal = 0;
     BOOL bHandled = TRUE;
@@ -250,7 +250,7 @@ int CAPEDecompressOld::GetInfo(APE_DECOM
             }
             else
             {
-                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeFormat, 0);
+                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeFormat, 0);
                 WAVE_HEADER WAVHeader; FillWaveHeader(&WAVHeader, 
                     (m_nFinishBlock - m_nStartBlock) * GetInfo(APE_INFO_BLOCK_ALIGN), 
                     &wfeFormat,    0);
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/Old/APEDecompressOld.h mac-3.99-u4-b4/src/MACLib/Old/APEDecompressOld.h
--- mac-3.99-u4-b4.orig/src/MACLib/Old/APEDecompressOld.h	2005-03-10 13:36:26.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/Old/APEDecompressOld.h	2006-01-19 18:32:47.000000000 +0100
@@ -13,7 +13,7 @@ public:
     int GetData(char * pBuffer, int nBlocks, int * pBlocksRetrieved);
     int Seek(int nBlockOffset);
 
-    int GetInfo(APE_DECOMPRESS_FIELDS Field, int nParam1 = 0, int nParam2 = 0);
+    int GetInfo(APE_DECOMPRESS_FIELDS Field, long nParam1 = 0, long nParam2 = 0);
     
 protected:
 
diff -Naupr mac-3.99-u4-b4.orig/src/MACLib/Old/UnMAC.cpp mac-3.99-u4-b4/src/MACLib/Old/UnMAC.cpp
--- mac-3.99-u4-b4.orig/src/MACLib/Old/UnMAC.cpp	2005-03-10 13:36:26.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/Old/UnMAC.cpp	2006-01-19 18:37:03.000000000 +0100
@@ -78,7 +78,7 @@ int CUnMAC::Initialize(IAPEDecompress *p
     // set the initialized flag to TRUE
     m_bInitialized = TRUE;
     
-    m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &m_wfeInput);
+    m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &m_wfeInput);
 
     // return a successful value
     return ERROR_SUCCESS;
@@ -197,7 +197,7 @@ int CUnMAC::DecompressFrameOld(unsigned 
     {
         m_pAPEDecompressCore->GenerateDecodedArrays(nBlocks, nSpecialCodes, FrameIndex, CPULoadBalancingFactor);
 
-        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &WaveFormatEx);
+        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &WaveFormatEx);
         m_pPrepare->UnprepareOld(m_pAPEDecompressCore->GetDataX(), m_pAPEDecompressCore->GetDataY(), nBlocks, &WaveFormatEx, 
             pOutputData, (unsigned int *) &CRC, (int *) &nSpecialCodes, m_pAPEDecompress->GetInfo(APE_INFO_FILE_VERSION));
     }
@@ -205,7 +205,7 @@ int CUnMAC::DecompressFrameOld(unsigned 
     {
         m_pAPEDecompressCore->GenerateDecodedArrays(nBlocks, nSpecialCodes, FrameIndex, CPULoadBalancingFactor);
         
-        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &WaveFormatEx);
+        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &WaveFormatEx);
         m_pPrepare->UnprepareOld(m_pAPEDecompressCore->GetDataX(), NULL, nBlocks, &WaveFormatEx, 
             pOutputData, (unsigned int *) &CRC, (int *) &nSpecialCodes, m_pAPEDecompress->GetInfo(APE_INFO_FILE_VERSION));
     }
