--- Makefile.orig	2003-04-11 19:18:58.000000000 +0200
+++ Makefile	2003-04-11 19:19:45.000000000 +0200
@@ -4,7 +4,7 @@
 
 VER_RELEASE:=4.8
 
-DIR_MOD:=/lib/modules
+DIR_MOD:=$(RPM_BUILD_ROOT)/lib/modules
 DIR_MOD_VER:=$(DIR_MOD)/$(shell uname -r)
 DIR_MODUTILS:=/sbin
 DIR_TMP:=/tmp
@@ -19,12 +19,11 @@
 
 install_modules:
 	if [ ! -d $(DIR_MOD_VER)/thinkpad ]; then mkdir -p -m 755 $(DIR_MOD_VER)/thinkpad ; fi
-	install -m 444 -o 0 -g 0 drivers/thinkpad.o $(DIR_MOD_VER)/thinkpad/thinkpad.o
-	install -m 444 -o 0 -g 0 drivers/smapi.o $(DIR_MOD_VER)/thinkpad/smapi.o
-	install -m 444 -o 0 -g 0 drivers/superio.o $(DIR_MOD_VER)/thinkpad/superio.o
-	install -m 444 -o 0 -g 0 drivers/rtcmosram.o $(DIR_MOD_VER)/thinkpad/rtcmosram.o
-	install -m 444 -o 0 -g 0 drivers/thinkpadpm.o $(DIR_MOD_VER)/thinkpad/thinkpadpm.o
-	depmod -a
+	install -m 444 drivers/thinkpad.o $(DIR_MOD_VER)/thinkpad/thinkpad.o
+	install -m 444 drivers/smapi.o $(DIR_MOD_VER)/thinkpad/smapi.o
+	install -m 444 drivers/superio.o $(DIR_MOD_VER)/thinkpad/superio.o
+	install -m 444 drivers/rtcmosram.o $(DIR_MOD_VER)/thinkpad/rtcmosram.o
+	install -m 444 drivers/thinkpadpm.o $(DIR_MOD_VER)/thinkpad/thinkpadpm.o
 
 uninstall_modules:
 	$(DIR_MODUTILS)/modprobe -r smapi
