--- s/sendmail/parseaddr.c
+++ s/sendmail/parseaddr.c
@@ -1000,6 +1008,8 @@
 	}
 	if (pvp == NULL)
 		return EX_USAGE;
+	if (maxatom <= 0)
+		return EX_USAGE;
 
 	/*
 	**  Run through the list of rewrite rules, applying
@@ -1866,6 +1880,7 @@
 	register ENVELOPE *e;
 {
 	bool tempfail = false;
+	int maxatom;
 	struct mailer **mp;
 	register struct mailer *m;
 	register char *p;
@@ -1880,6 +1895,7 @@
 		printav(tv);
 	}
 
+	maxatom = MAXATOM;
 	if (a == NULL)
 		a = (ADDRESS *) sm_rpool_malloc_x(e->e_rpool, sizeof *a);
 	memset((char *) a, '\0', sizeof *a);
@@ -1919,14 +1935,22 @@
 		return a;
 	}
 	mname = *++tv;
+	--maxatom;
 
 	/* extract host and user portions */
 	if (*++tv != NULL && (**tv & 0377) == CANONHOST)
+	{
 		hostp = ++tv;
+		--maxatom;
+	}
 	else
 		hostp = NULL;
+	--maxatom;
 	while (*tv != NULL && (**tv & 0377) != CANONUSER)
+	{
 		tv++;
+		--maxatom;
+	}
 	if (*tv == NULL)
 	{
 		syserr("554 5.3.5 buildaddr: no user");
@@ -1937,6 +1961,7 @@
 	else if (hostp != NULL)
 		cataddr(hostp, tv - 1, hbuf, sizeof hbuf, '\0');
 	cataddr(++tv, NULL, ubuf, sizeof ubuf, ' ');
+	--maxatom;
 
 	/* save away the host name */
 	if (sm_strcasecmp(mname, "error") == 0)
@@ -2041,6 +2066,7 @@
 	{
 		p++;
 		tv++;
+		--maxatom;
 		a->q_flags |= QNOTREMOTE;
 	}
 
@@ -2071,11 +2097,11 @@
 	    !bitset(RF_SENDERADDR|RF_HEADERADDR, flags))
 	{
 		/* sender addresses done later */
-		(void) REWRITE(tv, 2, e);
+		(void) rewrite(tv, 2, 0, e, maxatom);
 		if (m->m_re_rwset > 0)
-		       (void) REWRITE(tv, m->m_re_rwset, e);
+		       (void) rewrite(tv, m->m_re_rwset, 0, e, maxatom);
 	}
-	(void) REWRITE(tv, 4, e);
+	(void) rewrite(tv, 4, 0, e, maxatom);
 
 	/* save the result for the command line/RCPT argument */
 	cataddr(tv, NULL, ubuf, sizeof ubuf, '\0');

