--- sendmail-8.12.2/redhat.config.m4.in	Thu Aug 23 13:44:00 2001
+++ sendmail-8.12.2/redhat.config.m4.in	Thu Aug 23 13:44:00 2001
@@ -0,0 +1,10 @@
+define(`confMAPDEF', `-DNEWDB -DNIS -DHESIOD -DMAP_REGEX -DLDAPMAP')
+define(`confENVDEF', `$(RPM_OPT_FLAGS) -I/usr/include/db4 -I/usr/kerberos/include -Wall -DXDEBUG=0 -DTCPWRAPPERS -DSASL -DSTARTTLS -DMILTER -DNETINET6 -DHES_GETMAILHOST -DUSE_VENDOR_CF_PATH=1 -D_FFR_WORKAROUND_BROKEN_NAMESERVERS -D_FFR_SMTP_SSL -D_FFR_MILTER_ROOT_UNSAFE')
+define(`confLIBDIRS', `-L/usr/kerberos/%{_lib}')
+define(`confLIBS', `-lnsl -lwrap -lldap -llber -lsasl -lssl -lcrypto -lhesiod -lgdbm -lcrypt -ldb-4.0')
+define(`confMANOWN', `root')
+define(`confMANGRP', `root')
+define(`confMANMODE', `644')
+define(`confMAN1SRC', `1')
+define(`confMAN5SRC', `5')
+define(`confMAN8SRC', `8')
