--- c_stuff/lib/FBLE.pm	2003-02-16 18:14:28.000000000 -0500
+++ FBLE.pm	2004-02-06 12:32:57.000000000 -0500
@@ -1094,11 +1094,11 @@
     $rect{screenshot} = SDL::Rect->new(-x => $POS_1P{p1}{left_limit} - 40, '-y' => 0, 
 				       -width => $POS_1P{p1}{right_limit} - $POS_1P{p1}{left_limit} + 80,
 				       -height => $POS_1P{bottom_limit} - $POS_1P{init_top_limit} + 190);
-    fb_c_stuff::shrink($app->{-surface},
-		       $file_browser_screenshots[$list_browser_highlight_offset]->display_format()->{-surface},
+    fb_c_stuff::shrink($$app,
+		       ${$file_browser_screenshots[$list_browser_highlight_offset]->display_format()},
 		       $rect{middle}->x + $rect{middle}->width - $rect{screenshot}->width/4 - 12,
 		       $rect{middle}->y + $rect{middle}->height/2 - $rect{screenshot}->height/8 - 3,
-		       $rect{screenshot}{-rect}, 4);
+		       ${$rect{screenshot}}, 4);
 }
 
 
@@ -1371,7 +1371,7 @@
     $rect{ls_name_erase} = SDL::Rect->new(-x => 195, '-y' => 0, -width => 445-195, -height => 35);
     $background->blit($rect{ls_name_erase}, $app, $rect{ls_name_erase});
     $app->update;
-    $app->print(($background->width - SDL_TEXTWIDTH(uc($levelset_name)))/2 - 6, 7, uc($levelset_name));
+    $app->print(($background->width - SDL::SFont::SDL_TEXTWIDTH(uc($levelset_name)))/2 - 6, 7, uc($levelset_name));
 }
 
 sub print_text_generic {
