--- linphone-0.12.2/oRTP/configure.orig	2005-01-21 11:24:48.993986731 -0500
+++ linphone-0.12.2/oRTP/configure	2005-01-21 11:25:08.144473681 -0500
@@ -20640,7 +20640,7 @@
 
 
 
-                                                  ac_config_files="$ac_config_files Makefile src/Makefile docs/Makefile build/Makefile build/win32/Makefile"
+                                                  ac_config_files="$ac_config_files Makefile src/Makefile build/Makefile build/win32/Makefile"
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
 # tests run on this system so they can be shared between configure
@@ -21206,7 +21206,6 @@
   # Handling of arguments.
   "Makefile" ) CONFIG_FILES="$CONFIG_FILES Makefile" ;;
   "src/Makefile" ) CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
-  "docs/Makefile" ) CONFIG_FILES="$CONFIG_FILES docs/Makefile" ;;
   "build/Makefile" ) CONFIG_FILES="$CONFIG_FILES build/Makefile" ;;
   "build/win32/Makefile" ) CONFIG_FILES="$CONFIG_FILES build/win32/Makefile" ;;
   "depfiles" ) CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
--- linphone-0.12.2/oRTP/Makefile.am.orig	2005-01-21 11:28:40.834457361 -0500
+++ linphone-0.12.2/oRTP/Makefile.am	2005-01-21 11:28:46.102490995 -0500
@@ -6,7 +6,7 @@
 EXTRA_DIST=oRTP.prj   ortp-config.h.in	TODO
 
 
-SUBDIRS=src docs build
+SUBDIRS=src build
 
 mod-dist:
 	mkdir oRTP-bin
--- linphone-0.12.2/osipua/configure.orig	2005-01-21 11:33:56.904478419 -0500
+++ linphone-0.12.2/osipua/configure	2005-01-21 11:34:22.628759708 -0500
@@ -22300,7 +22300,7 @@
 
 
 
-                              ac_config_files="$ac_config_files Makefile src/Makefile docs/Makefile"
+                              ac_config_files="$ac_config_files Makefile src/Makefile"
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
 # tests run on this system so they can be shared between configure
@@ -22852,7 +22852,6 @@
   # Handling of arguments.
   "Makefile" ) CONFIG_FILES="$CONFIG_FILES Makefile" ;;
   "src/Makefile" ) CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
-  "docs/Makefile" ) CONFIG_FILES="$CONFIG_FILES docs/Makefile" ;;
   "depfiles" ) CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
   "osipua-config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS osipua-config.h" ;;
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
--- linphone-0.12.2/osipua/Makefile.am.orig	2005-01-21 11:34:02.887380950 -0500
+++ linphone-0.12.2/osipua/Makefile.am	2005-01-21 11:34:29.713460135 -0500
@@ -1,3 +1,3 @@
 EXTRA_DIST=	osipua-config.h.in
 
-SUBDIRS=src docs
+SUBDIRS=src
