diff -ru openal.orig/linux/src/arch/sdl/sdl.c openal/linux/src/arch/sdl/sdl.c
--- openal.orig/linux/src/arch/sdl/sdl.c	2002-09-09 06:45:39.000000000 +0200
+++ openal/linux/src/arch/sdl/sdl.c	2002-10-30 13:55:45.000000000 +0100
@@ -11,6 +11,7 @@
 
 #include <fcntl.h>
 #include <stdio.h>
+#include <string.h>
 #include <stdlib.h>
 #include <sys/stat.h>
 #include <sys/time.h>
diff -ru openal.orig/linux/src/extensions/al_ext_mp3.c openal/linux/src/extensions/al_ext_mp3.c
--- openal.orig/linux/src/extensions/al_ext_mp3.c	2002-03-18 15:41:44.000000000 +0100
+++ openal/linux/src/extensions/al_ext_mp3.c	2002-10-30 13:56:21.000000000 +0100
@@ -7,6 +7,9 @@
  */
 #include "al_siteconfig.h"
 #include <stdio.h>
+#include <string.h>
+
+#include "AL/al.h"
 
 #include "al_ext_needed.h"
 #include "al_ext_mp3.h"
diff -ru openal.orig/linux/src/extensions/al_ext_vorbis.c openal/linux/src/extensions/al_ext_vorbis.c
--- openal.orig/linux/src/extensions/al_ext_vorbis.c	2002-09-26 02:51:23.000000000 +0200
+++ openal/linux/src/extensions/al_ext_vorbis.c	2002-10-30 13:56:40.000000000 +0100
@@ -12,6 +12,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include "al_ext_needed.h"
 #include "al_ext_vorbis.h"
