--- linux/src/al_config.c
+++ linux/src/al_config.c
@@ -356,7 +356,18 @@
 
 		/* for later malloc, get size */
 		filelen = buf.st_size;
-	}
+	} else {
+                /*
+                 * try system wide OpenAL config file
+                 */
+                sprintf(pathname, "/etc/%s", _AL_FNAME);
+                if(stat(pathname, &buf) != -1) {
+                        fh = fopen(pathname, "rb");
+
+                        /* for later malloc, get size */
+                        filelen = buf.st_size;
+                }
+        }
 
 	if( fh == NULL ) {
 		return NULL;
