--- bluez-utils-2.3/pcmcia/bluetooth~	Fri Apr 25 16:45:30 2003
+++ bluez-utils-2.3/pcmcia/bluetooth	Tue May  6 10:54:17 2003
@@ -25,7 +25,7 @@
     IRQ=`setserial /dev/$DEVICE | sed -e 's/.*IRQ: //'`
     setserial /dev/$DEVICE irq 0 ; setserial /dev/$DEVICE irq $IRQ
  
-    /usr/sbin/hciattach $DEVICE $MANFID > /tmp/pcmcia
+    /usr/sbin/hciattach $DEVICE $MANFID
 }
 stop_serial() {
     do_fuser -k -HUP /dev/$DEVICE > /dev/null
