--- bluez-utils-2.2/pcmcia/bluetooth.conf.foo	Mon Feb 24 17:50:39 2003
+++ bluez-utils-2.2/pcmcia/bluetooth.conf	Mon Feb 24 17:50:54 2003
@@ -16,31 +16,31 @@
 
 card "Xircom CreditCard Bluetooth Adapter"
   version "Xircom", "*", "CBT"
-  bind "serial_cs" class "bluetooth"
+  bind "serial_cs"
 
 card "Xircom RealPort2 Bluetooth Adapter"
   version "Xircom", "*", "R2BT"
-  bind "serial_cs" class "bluetooth"
+  bind "serial_cs"
 
 card "Brain Boxes BL-620 Bluetooth Adapter"
   version "Brain Boxes", "Bluetooth PC Card"
-  bind "serial_cs" class "bluetooth"
+  bind "serial_cs"
 
 card "COM One Platinium Bluetooth PC Card"
   version "COM1 SA", "MC310 CARD"
-  bind "serial_cs" class "bluetooth"
+  bind "serial_cs"
 
 card "Sphinx PICO Card"
   version "SPHINX", "BT-CARD"
-  bind "serial_cs" class "bluetooth"
+  bind "serial_cs"
 
 card "H-Soft blue+Card"
   version "H-Soft", "Blue+CARD"
-  bind "serial_cs" class "bluetooth"
+  bind "serial_cs"
 
 card "Compaq iPAQ Bluetooth Sleeve"
   version "CF CARD", "GENERIC"
-  bind "serial_cs" class "bluetooth"
+  bind "serial_cs"
 
 
 card "Nokia Bluetooth Card"
