--- as10k1/as10k1.c.orig	2003-12-01 01:31:48.000000000 +0100
+++ as10k1/as10k1.c	2003-12-01 01:34:05.000000000 +0100
@@ -39,33 +39,30 @@
 
 char version[]="As10k1 assembler version 1.0";
 
-char help[]="
-Usage: as10k1 [option...] asmfile
-
-The as10k1 assembler is for the emu10k1 dsp processor
-found in Creative Lab's sblive series of sound cards.
-
-Author: Daniel Bertrand <d.bertrand@ieee.org>
-BUGS:   emu10k1-devel mailing list <emu10k1-devel@opensource.creative.com>
-URLs:   members.home.com/d-bertrand, opensource.creative.com
-
-Options:\n
-     -l [listfile]         Specify a listing file, default is none.
-     -o [binfile]          Specify an output file, default is based input,
-                           Substituting .asm fo .bin
-     -d [dbg_options...]   Turn on debug messages.
-            s              prints all symbols                
-            g              prints defined gprs
-            t              prints tram usage
-            i              prints instructions
-
-     -n                    no gpr declaration check 
-     -h                    Prints this message  
-     -v                    Prints version number. 
-
-This program is distributed under the GPL.
-
-";
+char help[]="\n\
+Usage: as10k1 [option...] asmfile \n\
+\n\
+The as10k1 assembler is for the emu10k1 dsp processor\n\
+found in Creative Lab's sblive series of sound cards.\n\
+\n\
+Author: Daniel Bertrand <d.bertrand@ieee.org>\n\
+BUGS:   emu10k1-devel mailing list <emu10k1-devel@opensource.creative.com>\n\
+URLs:   members.home.com/d-bertrand, opensource.creative.com\n\
+\n\
+Options:\n\
+     -l [listfile]         Specify a listing file, default is none.\n\
+     -o [binfile]          Specify an output file, default is based input,\n\
+                           Substituting .asm fo .bin\n\
+     -d [dbg_options...]   Turn on debug messages.\n\
+            s                      prints all symbols\n\
+            g                      prints defined gprs\n\
+            t                      prints tram usage\n\
+            i                      prints instructions\n\
+     -n                    no gpr declaration check\n\
+     -h                    Prints this message\n\
+     -v                    Prints version number.\n\
+\n\
+This program is distributed under the GPL.\n\n";
 
 //  cli==Command-Line Interface && !Creative Labs Inc.
 
--- mixer/lib/debug.c.orig	2003-12-01 01:36:03.000000000 +0100
+++ mixer/lib/debug.c	2003-12-01 01:36:47.000000000 +0100
@@ -325,18 +325,18 @@
 			k=i;
 	}
 	printf("\
-Usage Summary:
---------
-Instructions : %d
-GPRS:          %d (last used is %d)
-
-Tram
- Internal
-   buffers:    %d
-   space  :    0x%06x
- External
-   buffers:    %d
-   space  :    0x%06x
+Usage Summary:\n\
+--------\n\
+Instructions : %d\n\
+GPRS:          %d (last used is %d)\n\
+\n\
+Tram\n\
+ Internal\n\
+   buffers:    %d\n\
+   space  :    0x%06x\n\
+ External\n\
+   buffers:    %d\n\
+   space  :    0x%06x\n\
 ",
 	       mgr->code_free_start/2, j,k,
 	       mgr->traml_ifree_start,  mgr->tramb_ifree_start,
