diff -Naupr armagetronad-0.2.7.1.orig/src/engine/eCamera.cpp armagetronad-0.2.7.1/src/engine/eCamera.cpp
--- armagetronad-0.2.7.1.orig/src/engine/eCamera.cpp	2005-02-28 19:22:53.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eCamera.cpp	2005-04-22 10:50:54.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "eSensor.h"
 #include "eCamera.h"
 #include "rScreen.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/engine/eDisplay.cpp armagetronad-0.2.7.1/src/engine/eDisplay.cpp
--- armagetronad-0.2.7.1.orig/src/engine/eDisplay.cpp	2005-02-22 23:29:00.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eDisplay.cpp	2005-04-22 10:51:40.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 //void se_FetchAndStoreSDLInput();
 
 #include "tConfiguration.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/engine/eNetGameObject.cpp armagetronad-0.2.7.1/src/engine/eNetGameObject.cpp
--- armagetronad-0.2.7.1.orig/src/engine/eNetGameObject.cpp	2004-11-29 11:26:42.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eNetGameObject.cpp	2005-04-22 10:52:10.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "eNetGameObject.h"
 #include "ePlayer.h"
 #include "eTimer.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/engine/eSound.cpp armagetronad-0.2.7.1/src/engine/eSound.cpp
--- armagetronad-0.2.7.1.orig/src/engine/eSound.cpp	2005-02-26 19:26:52.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eSound.cpp	2005-04-22 10:52:50.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "config.h"
 #include "tMemManager.h"
 #include "tDirectories.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/engine/eVoter.cpp armagetronad-0.2.7.1/src/engine/eVoter.cpp
--- armagetronad-0.2.7.1.orig/src/engine/eVoter.cpp	2005-03-02 23:25:18.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eVoter.cpp	2005-04-22 10:53:24.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "eVoter.h"
 
 #include "tMemManager.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/render/rScreen.cpp armagetronad-0.2.7.1/src/render/rScreen.cpp
--- armagetronad-0.2.7.1.orig/src/render/rScreen.cpp	2005-02-28 00:10:51.000000000 +0100
+++ armagetronad-0.2.7.1/src/render/rScreen.cpp	2005-04-22 10:56:56.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "defs.h"
 
 
diff -Naupr armagetronad-0.2.7.1.orig/src/render/rTexture.cpp armagetronad-0.2.7.1/src/render/rTexture.cpp
--- armagetronad-0.2.7.1.orig/src/render/rTexture.cpp	2004-11-29 11:26:42.000000000 +0100
+++ armagetronad-0.2.7.1/src/render/rTexture.cpp	2005-04-22 10:56:24.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "config.h"
 
 #include "rTexture.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gAIBase.cpp armagetronad-0.2.7.1/src/tron/gAIBase.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gAIBase.cpp	2005-01-26 16:06:33.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gAIBase.cpp	2005-04-22 10:45:28.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gAIBase.h"
 #include "gArena.h"
 #include "eGrid.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gArmagetron.cpp armagetronad-0.2.7.1/src/tron/gArmagetron.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gArmagetron.cpp	2005-03-02 23:25:18.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gArmagetron.cpp	2005-04-22 10:46:00.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gStuff.h"
 #include "tSysTime.h"
 #include "tDirectories.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gCamera.cpp armagetronad-0.2.7.1/src/tron/gCamera.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gCamera.cpp	2004-12-07 18:39:00.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gCamera.cpp	2005-04-22 10:47:44.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "tMemManager.h"
 #include "gCycle.h"
 #include "gCamera.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gCycle.cpp armagetronad-0.2.7.1/src/tron/gCycle.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gCycle.cpp	2005-03-06 09:51:22.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gCycle.cpp	2005-04-22 10:46:26.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gCycle.h"
 #include "nConfig.h"
 #include "rModel.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gFloor.cpp armagetronad-0.2.7.1/src/tron/gFloor.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gFloor.cpp	2005-02-20 11:48:15.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gFloor.cpp	2005-04-22 10:41:40.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "defs.h"
 #include "gStuff.h"
 #include "gLogo.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gLanguageMenu.cpp armagetronad-0.2.7.1/src/tron/gLanguageMenu.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gLanguageMenu.cpp	2004-11-29 11:26:43.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gLanguageMenu.cpp	2005-04-22 10:48:44.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gLanguageMenu.h"
 #include "uMenu.h"
 #include "tLocale.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gLogo.cpp armagetronad-0.2.7.1/src/tron/gLogo.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gLogo.cpp	2004-11-29 11:26:43.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gLogo.cpp	2005-04-22 10:49:10.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gLogo.h"
 
 #include "gStuff.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gMenus.cpp armagetronad-0.2.7.1/src/tron/gMenus.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gMenus.cpp	2005-02-26 22:25:36.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gMenus.cpp	2005-04-22 10:48:14.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gMenus.h"
 #include "ePlayer.h"
 #include "rScreen.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gServerBrowser.cpp armagetronad-0.2.7.1/src/tron/gServerBrowser.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gServerBrowser.cpp	2005-01-25 18:21:53.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gServerBrowser.cpp	2005-04-22 10:39:33.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gServerBrowser.h"
 #include "gGame.h"
 #include "gLogo.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gStuff.cpp armagetronad-0.2.7.1/src/tron/gStuff.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gStuff.cpp	2004-11-29 11:26:43.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gStuff.cpp	2005-04-22 10:47:15.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gStuff.h"
 #include "eCoord.h"
 #include "tConfiguration.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gTeam.cpp armagetronad-0.2.7.1/src/tron/gTeam.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gTeam.cpp	2004-11-29 11:26:43.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gTeam.cpp	2005-04-22 10:49:41.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gTeam.h"
 #include "uMenu.h"
 #include "ePlayer.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gWall.cpp armagetronad-0.2.7.1/src/tron/gWall.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gWall.cpp	2005-03-04 21:20:45.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gWall.cpp	2005-04-22 10:42:47.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gStuff.h"
 #include "eGrid.h"
 #include "eWall.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/tron/gWinZone.cpp armagetronad-0.2.7.1/src/tron/gWinZone.cpp
--- armagetronad-0.2.7.1.orig/src/tron/gWinZone.cpp	2004-11-29 11:26:43.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gWinZone.cpp	2005-04-22 10:50:09.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "gWinZone.h"
 #include "eTimer.h"
 #include "gCycle.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/ui/uInput.cpp armagetronad-0.2.7.1/src/ui/uInput.cpp
--- armagetronad-0.2.7.1.orig/src/ui/uInput.cpp	2004-11-29 11:26:44.000000000 +0100
+++ armagetronad-0.2.7.1/src/ui/uInput.cpp	2005-04-22 10:55:48.000000000 +0200
@@ -25,6 +25,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "tMemManager.h"
 #include "rScreen.h"
 #include "tInitExit.h"
diff -Naupr armagetronad-0.2.7.1.orig/src/ui/uMenu.cpp armagetronad-0.2.7.1/src/ui/uMenu.cpp
--- armagetronad-0.2.7.1.orig/src/ui/uMenu.cpp	2005-03-06 18:17:52.000000000 +0100
+++ armagetronad-0.2.7.1/src/ui/uMenu.cpp	2005-04-22 10:55:16.000000000 +0200
@@ -26,6 +26,8 @@ Foundation, Inc., 59 Temple Place - Suit
 
 */
 
+#include "rSDL.h"
+
 #include "tSysTime.h"
 #include "uMenu.h"
 #include "rSysdep.h"
