diff -Naupr Coro-1.9.orig/Coro/Makefile.PL Coro-1.9/Coro/Makefile.PL
--- Coro-1.9.orig/Coro/Makefile.PL	2005-12-26 19:47:20.000000000 +0100
+++ Coro-1.9/Coro/Makefile.PL	2006-04-25 13:33:44.000000000 +0200
@@ -16,7 +16,7 @@ if ($^O =~ /win32/i or $^O =~ /cygwin/ o
    $iface = "i";
 } elsif ($^O =~ /linux/) {
    # default to setjmp/longjmp on non-x86...
-   $iface = $Config{archname} =~ /^(i[3456]86|amd64|x86_64)-/ ? "l" : "s";
+   $iface = $Config{archname} =~ /^(i[3456]86|amd64|x86_64)-/ ? "l" : "u";
 } elsif ($^O =~ /(free|net|open)bsd/) {
    # FreeBSD 4.x has ucontext.h but no makecontext et al. (see BUGS section of
    # man context). Assume the same problem for all other BSDs.
@@ -52,10 +52,8 @@ default.
 
 EOF
 
-if (prompt ("Do you want to enable C context sharing (y/n)", "y") !~ /^\s*n/i) {
-   print "\nC context sharing enabled.\n\n";
-   $DEFINE .= " -DCORO_LAZY_STACK";
-}
+print "\nC context sharing enabled.\n\n";
+$DEFINE .= " -DCORO_LAZY_STACK";
 
 if ($iface) {
    print <<EOF;
@@ -98,10 +96,7 @@ EOF
 
 retry:
 
-   my $r = prompt "Use which implementation,\n" .
-                  "<s>etjmp/longjump, <u>context, <i>rix, <l>inux or <w>indows?",
-                  $iface;
-   $iface = lc $1 if $r =~ /(\S)/;
+   $iface = "u";
 
    if ($iface eq "u") {
       $DEFINE .= " -DCORO_UCONTEXT";
@@ -152,7 +147,7 @@ function prologue) or in t/10_bugs.t, yo
 
 EOF
 
-my $stacksize = prompt ("C stack size factor", "16384");
+my $stacksize = "16384";
 $DEFINE .= " -DSTACKSIZE=$stacksize";
 
 print "using a stacksize of $stacksize * sizeof(long)\n";
