--- rpcmgr11.c.orig	2003-12-01 04:56:02.000000000 +0100
+++ rpcmgr11.c	2003-12-01 04:57:21.000000000 +0100
@@ -180,7 +180,8 @@
   { "HITACHI ", "DVD-ROM GD-3000 ", rpc2_hitachi },
   { "HITACHI ", "DVD-ROM GD-5000 ", rpc2_hitachi },
   { "LG      ", "DVD-ROM DRD8080B", rpc2_lg },
-  { "LG      ", "DVD-ROM DRN8080B", rpc2_lg }
+  { "LG      ", "DVD-ROM DRN8080B", rpc2_lg },
+  { "HL-DT-ST", "RW/DVD GCC-4320B", rpc2_lg },
 };
 
 
@@ -207,10 +208,10 @@
 	  "  -u, --reset-user    reset RPC2 user counter\n"
 	  "                      warning: decreases RPC2 vendor counter!\n"
           "                      COMPAQ  DRD8080B, DRN8080B\n"
-	  "                      LG      DRD8080B, DRN8080B\n"
+	  "                      LG      DRD8080B, DRN8080B, GCC4320B\n"
 	  "  -v, --reset-vendor  reset RPC2 vendor counter\n"
           "                      COMPAQ  DRD8080B, DRN8080B\n"
-	  "                      LG      DRD8080B, DRN8080B\n"
+	  "                      LG      DRD8080B, DRN8080B, GCC4320B\n"
 	  "  -V, --version       print version information and exit\n", PROG);
   exit(1);
 }
