diff -Naupr xmule-1.7.4b.orig/src/DownloadClient.cpp xmule-1.7.4b/src/DownloadClient.cpp
--- xmule-1.7.4b.orig/src/DownloadClient.cpp	2004-02-26 20:23:39.000000000 +0100
+++ xmule-1.7.4b/src/DownloadClient.cpp	2004-03-04 20:41:47.816133024 +0100
@@ -393,6 +393,7 @@ void CUpDownClient::SendBlockRequests(){
 		pblock->block = m_DownloadBlocks_list.RemoveHead();
 		pblock->zStream = NULL;
 		pblock->totalUnzipped = 0;
+		pblock->bZStreamError = false;
 		m_PendingBlocks_list.AddTail(pblock);
 	}
 	if (m_PendingBlocks_list.IsEmpty()){
