--- linux-wlan-ng-0.1.15/config.in.orig	Sat Apr 20 15:32:41 2002
+++ linux-wlan-ng-0.1.15/config.in	Sat Apr 20 15:46:59 2002
@@ -4,9 +4,9 @@
 WLAN_EXTRAVERSION=
 LINUX_SRC=/usr/src/linux
-PCMCIA_SRC=
+PCMCIA_SRC=/usr/src/pcmcia-cs
 PREFIX=
 INST_EXEDIR=/sbin
-TARGET_ROOT_ON_HOST=
+TARGET_ROOT_ON_HOST=/usr/src/redhat/BUILD/wlan-install
 RC_DIR=/etc/rc.d
 PCMCIA_DIR=/etc/pcmcia
 SYSV_INIT=y
@@ -16,7 +16,8 @@
 CROSS_COMPILE_ENABLED=n
 CROSS_COMPILE=
 HOST_COMPILE=
-PRISM2_PLX=n
+PRISM2_PLX=y
 PRISM2_PCMCIA=y
-PRISM2_PCI=n
-PRISM2_USB=n
+PRISM2_PCI=y
+PRISM2_USB=y
+WLAN_KERN_PCMCIA=y
