diff -Naupr js.orig/src/Makefile.ref js/src/Makefile.ref
--- js.orig/src/Makefile.ref	2003-11-15 01:10:55.000000000 +0100
+++ js/src/Makefile.ref	2006-07-24 17:06:59.000000000 +0200
@@ -61,11 +61,11 @@ endif
 
 ifdef JS_THREADSAFE
 DEFINES += -DJS_THREADSAFE
-INCLUDES += -I../../dist/$(OBJDIR)/include
+INCLUDES += $(shell nspr-config --cflags)
 ifdef USE_MSVC
 OTHER_LIBS += ../../dist/$(OBJDIR)/lib/libnspr${NSPR_LIBSUFFIX}.lib
 else
-OTHER_LIBS += -L../../dist/$(OBJDIR)/lib -lnspr${NSPR_LIBSUFFIX}
+OTHER_LIBS += $(shell nspr-config --libs)
 endif
 endif
 
